{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Enrichment of regulons from Regulons_by_strain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from importlib import reload\n",
    "\n",
    "module_path = os.path.abspath(os.path.join('..'))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.insert(0,module_path)\n",
    "\n",
    "import pandas as pd\n",
    "from IPython.display import display,HTML\n",
    "\n",
    "from orangecontrib.bioinformatics.geneset.__init__ import list_all\n",
    "\n",
    "import jupyter_functions as jf\n",
    "reload(jf)\n",
    "import jupyter_functions as jf\n",
    "from functionsDENet import loadPickle, savePickle\n",
    "import stages_DE.enrichment_library as enr\n",
    "reload(enr)\n",
    "import stages_DE.enrichment_library as enr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataPath='/home/karin/Documents/timeTrajectories/data/RPKUM/combined/'\n",
    "pathRegulons='/home/karin/Documents/timeTrajectories/data/regulons/by_strain/kN300_mean0std1_log/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Used gene sets\n",
    "Used gene sets that do not have less than 5 or more than 500 genes. Used all Orange ontologies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all gene sets\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=True,set_sizes=(5,500))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For conversion of gene names to EID\n",
    "genes = pd.read_csv(dataPath + 'mergedGenes_RPKUM.tsv', sep='\\t', index_col=0)\n",
    "all_gene_names= genes[(genes != 0).any(axis=1)].index\n",
    "NAME_EID=enr.name_genes_entrez(gene_names=all_gene_names, key_entrez=False)\n",
    "ALL_GENE_NAMES_EID=enr.convert_EID(genes=all_gene_names, name_EID=NAME_EID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('max_colwidth', 55)\n",
    "def group_diff_enrichment(data:pd.DataFrame,group:str,padj:float=0.25,min_overlap:int=None,\n",
    "                          use_annotated_genes:bool=False):\n",
    "    #Displays only gene sets that have overlap with query greater or equal to min_overlap \n",
    "    #For p value and padj calculation uses alll that have overlap >=1 } from gene_set_enrichment\n",
    "    \"\"\"\n",
    "    :param use_annotated_genes: if True use for reference and query  only genes that have at \n",
    "    least one gene set annotation\n",
    "    \"\"\"\n",
    "    selected=list(data[data['Cluster']==group].index)\n",
    "    query_EID=enr.convert_EID(genes=selected, name_EID=NAME_EID)\n",
    "    print('***  '+group+' selected:',len(selected),'with EID:',len(query_EID))\n",
    "    \n",
    "    reference_gene_eids=ALL_GENE_NAMES_EID.copy()\n",
    "    query_eids=query_EID.copy()\n",
    "    \n",
    "    if use_annotated_genes:\n",
    "        gene_sets_genes=set()\n",
    "        for gene_set_name, gene_sets in GENE_SETS_ONTOLOGY.items():\n",
    "            for gene_set in gene_sets:\n",
    "                gene_sets_genes.update(gene_set.genes)\n",
    "        reference_gene_eids=set(reference_gene_eids) & gene_sets_genes\n",
    "        query_eids=set(query_eids) & gene_sets_genes\n",
    "        print('Ratio of genes annotated with a gene set in reference',\n",
    "              round(len(reference_gene_eids)/len(ALL_GENE_NAMES_EID),2),\n",
    "             'and query',round(len(query_eids)/len(query_EID),2))\n",
    "    \n",
    "    \n",
    "    result=None\n",
    "    query_in_enriched=set()\n",
    "    if len(query_eids) > 0:\n",
    "        enrichment=enr.gene_set_enrichment(query_eids, reference_EID=reference_gene_eids, \n",
    "                                                padj_threshold=padj,min_overlap=min_overlap,\n",
    "                                                gene_sets_ontology=GENE_SETS_ONTOLOGY)\n",
    "\n",
    "        if len(enrichment)>0:\n",
    "            enrichment_display=list()\n",
    "            enrichment= sorted(enrichment, key=lambda data: data.padj)\n",
    "            for enriched in enrichment:\n",
    "                query_in_enriched.update(enriched.gene_set.genes & query_eids)\n",
    "                enrichment_display.append({'Gene set':enriched.gene_set.name,\n",
    "                                           'Ontology':enriched.ontology[0]+' '+enriched.ontology[1],\n",
    "                                           'FDR':\"{:.2e}\".format(enriched.padj),'N in query':enriched.in_query})\n",
    "            result=pd.DataFrame(enrichment_display)\n",
    "    print('Enrichment at FDR: '+str(padj)+' and min query - gene set overlap',str(min_overlap))\n",
    "    print('N query genes in displayed gene sets:',len(query_in_enriched),'out of', len(query_eids),\n",
    "          'query genes used for enrichment calculation.')\n",
    "    if result is not None:\n",
    "        display(HTML(result.to_html(index=False)))\n",
    "    else:\n",
    "        display(result)\n",
    "    print('\\n')\n",
    "    return result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Enrichment for mergedGenes_min18_clustersfilledDiagonal11.tab and expression_clustersfilledDiagonal11.png"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 36 with EID: 36\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 36 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.31e-14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.37e-10</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.90e-09</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.74e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.72e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.92e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.92e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.17e-07</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.71e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.01e-04</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.43e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.41e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.32e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C2 selected: 42 with EID: 42\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 42 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.07e-04</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>6.31e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.72e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.65e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>phosphatase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>7.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.60e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.91e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.91e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.91e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.25e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.31e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.65e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.65e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.22e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C3 selected: 30 with EID: 30\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 30 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.61e-19</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.81e-17</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.97e-16</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.35e-07</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.44e-06</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.00e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.00e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.00e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.30e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>7.78e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C4 selected: 17 with EID: 17\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 5 out of 17 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.28e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.28e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.53e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.20e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.36e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C5 selected: 5 with EID: 5\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 5 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.45e-04</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.73e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C6 selected: 4 with EID: 4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 4 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C7 selected: 15 with EID: 14\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 7 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>aberrant cellular response to stress</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.50e-04</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>increased cell size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.24e-04</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.21e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>homeostatic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.48e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.54e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.60e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.26e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.38e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C8 selected: 18 with EID: 18\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 16 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>7.68e-29</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>7.68e-29</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.84e-24</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.14e-22</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.79e-18</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.71e-06</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.03e-05</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.16e-05</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.18e-04</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.52e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C9 selected: 29 with EID: 29\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 29 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>translucent sorus</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.12e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.81e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.81e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.81e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.81e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.87e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.87e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.48e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.81e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.76e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.76e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C10 selected: 47 with EID: 47\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 43 out of 47 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.15e-66</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.54e-32</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.19e-29</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.44e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.70e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.80e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>7.77e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.32e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.82e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C11 selected: 6 with EID: 6\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 6 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C12 selected: 51 with EID: 51\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 45 out of 51 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.01e-92</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.89e-90</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.45e-80</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.23e-73</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.31e-66</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.81e-10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.37e-07</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C13 selected: 14 with EID: 14\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>7.60e-11</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.19e-10</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.37e-10</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.13e-10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.49e-09</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.09e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.75e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.75e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.56e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.49e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.33e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>9.40e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.40e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.40e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.45e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.49e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.77e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.36e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.72e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.28e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.49e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.90e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C14 selected: 51 with EID: 51\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 28 out of 51 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.07e-09</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.12e-06</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.12e-06</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.07e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.59e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.82e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.82e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.09e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.73e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.58e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.10e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>vacuole</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.48e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_min18_clusters_larger.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2)\n",
    "    if enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/filledDiagonal_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Enrichment for expression_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 91 with EID: 91\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 91 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>8.23e-11</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>8.23e-11</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.29e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.53e-08</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.17e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.85e-06</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.58e-04</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.13e-04</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>8.81e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.06e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.69e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.02e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.21e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.02e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.21e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>8.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.29e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C2 selected: 77 with EID: 77\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 77 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.82e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.79e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C3 selected: 74 with EID: 74\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 24 out of 74 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.02e-06</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.99e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.56e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.06e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.23e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.90e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.90e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.90e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.90e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.00e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.08e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.33e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.95e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.18e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.18e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.18e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.94e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C4 selected: 74 with EID: 74\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 74 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.60e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C5 selected: 68 with EID: 68\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 65 out of 68 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>6.41e-126</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.47e-124</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.17e-109</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.72e-98</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.92e-86</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.97e-15</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.86e-13</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.69e-05</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C6 selected: 66 with EID: 65\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 65 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>6.98e-10</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.55e-08</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>8.07e-05</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.90e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.28e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.92e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>8.95e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.24e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.03e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cellular response to stress</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.38e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.42e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.46e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.06e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.06e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.72e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.80e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.80e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>6.77e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.68e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>7.70e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.53e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>vacuole</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>9.38e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.50e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ligase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.46e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C7 selected: 64 with EID: 64\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 64 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.46e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.95e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C8 selected: 61 with EID: 59\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 59 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C9 selected: 58 with EID: 58\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 58 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>8.13e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.13e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C10 selected: 57 with EID: 57\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 57 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.59e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.72e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C11 selected: 55 with EID: 55\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 55 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.97e-08</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.98e-04</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.93e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.03e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.40e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.74e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>7.78e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.33e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.33e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.06e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C12 selected: 54 with EID: 54\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 54 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.50e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.50e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C13 selected: 46 with EID: 45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 45 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.97e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.82e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.21e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C14 selected: 45 with EID: 45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 45 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C15 selected: 41 with EID: 41\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 38 out of 41 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.80e-69</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>7.55e-34</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.87e-32</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.73e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.73e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.27e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C16 selected: 40 with EID: 40\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 40 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.48e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translucent sorus</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell death</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.14e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.14e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.14e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.44e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.88e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.88e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.93e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.82e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.02e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C17 selected: 36 with EID: 36\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 36 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.31e-14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.37e-10</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.90e-09</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.74e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.72e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.92e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.92e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.17e-07</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.71e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.01e-04</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.43e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.41e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.32e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C18 selected: 30 with EID: 30\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 30 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.61e-19</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.81e-17</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.97e-16</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.35e-07</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.44e-06</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.00e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.00e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.00e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.30e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>7.78e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C19 selected: 23 with EID: 23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 23 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C20 selected: 20 with EID: 20\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 20 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C21 selected: 19 with EID: 19\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 19 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2)\n",
    "    if enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Using only genes annotated with at least one gene set \n",
    "To find genes that have gene set annotations only gene sets used in enrichment are used (see above). Genes that are in at least one of these gene sets are used for enrichment calculation, both for reference and gene group of interest (query)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 91 with EID: 91\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 41 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.81e-10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.81e-10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.74e-08</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.03e-06</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.66e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.53e-05</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.61e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.46e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.50e-03</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.28e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.16e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C2 selected: 77 with EID: 77\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.29\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 15 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.48e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.08e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.24e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.83e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>9.08e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.38e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.72e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.18e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.18e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C3 selected: 74 with EID: 74\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.42\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 24 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.37e-05</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.80e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.80e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.59e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.86e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.15e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.15e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.80e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.64e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>8.57e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.57e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.57e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.57e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.32e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C4 selected: 74 with EID: 74\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 15 out of 17 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.67e-06</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.42e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C5 selected: 68 with EID: 68\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.97\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 62 out of 66 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>6.55e-98</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.07e-97</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>9.42e-84</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.86e-73</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.02e-61</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.37e-09</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.80e-08</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>7.48e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C6 selected: 66 with EID: 65\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.58\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 38 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>9.41e-08</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>8.64e-05</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.09e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.11e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.99e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>6.94e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.94e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.94e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cellular response to stress</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.94e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.41e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.41e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C7 selected: 64 with EID: 64\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.16\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 10 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.86e-04</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.72e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>5.90e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>9.93e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C8 selected: 61 with EID: 59\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.19\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 11 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.24e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.24e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C9 selected: 58 with EID: 58\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 12 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.40e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.40e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.85e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C10 selected: 57 with EID: 57\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 11 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.89e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.11e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.42e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C11 selected: 55 with EID: 55\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.44\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 24 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.76e-07</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.15e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.15e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.15e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.49e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.55e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.32e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.32e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C12 selected: 54 with EID: 54\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 11 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C13 selected: 46 with EID: 45\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.33\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.26e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.85e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.24e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C14 selected: 45 with EID: 45\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.13\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 6 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.35e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C15 selected: 41 with EID: 41\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.98\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 35 out of 40 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.20e-55</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.26e-24</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.25e-20</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C16 selected: 40 with EID: 40\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.96e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translucent sorus</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.13e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.13e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.13e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell death</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.76e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.76e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.76e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.08e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.74e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.74e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.59e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.59e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.57e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.98e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C17 selected: 36 with EID: 36\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.53\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 19 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.25e-13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.81e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.50e-08</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.03e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.66e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.06e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.75e-06</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.75e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.99e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.17e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.47e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C18 selected: 30 with EID: 30\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.7\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 21 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.08e-15</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>8.61e-14</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.18e-13</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.10e-05</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.66e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.45e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.42e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.41e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.26e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.63e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C19 selected: 23 with EID: 23\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 6 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.71e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C20 selected: 20 with EID: 20\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.0\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 0 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C21 selected: 19 with EID: 19\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.05\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 1 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2,use_annotated_genes=True)\n",
    "    if enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30_annotatedOnly_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Enrichment for mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 96 with EID: 96\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 30 out of 96 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.66e-05</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.27e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.47e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.47e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.18e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.52e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.52e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.52e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.65e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.40e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.77e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.77e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.77e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.81e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>phosphatase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.20e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.35e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.97e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.30e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.39e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>9.56e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.56e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.42e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.01e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>endosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C2 selected: 89 with EID: 88\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 88 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.63e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.90e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C3 selected: 87 with EID: 87\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 36 out of 87 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>7.49e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>7.49e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.47e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.70e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.51e-04</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.89e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>5.41e-04</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.98e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.27e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.25e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>9.82e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.40e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.80e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.02e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.62e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.62e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.90e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>6.85e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>6.85e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>7.82e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glyoxylate and dicarboxylate metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>8.23e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.23e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>9.13e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>9.45e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.24e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>lyase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.83e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C4 selected: 85 with EID: 85\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 7 out of 85 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.97e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C5 selected: 73 with EID: 73\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 73 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C6 selected: 72 with EID: 72\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 72 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.28e-07</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C7 selected: 70 with EID: 70\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 70 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C8 selected: 65 with EID: 63\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 63 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.74e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.74e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.81e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>9.11e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>lipid binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.14e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C9 selected: 57 with EID: 56\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 56 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.23e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C10 selected: 54 with EID: 54\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 26 out of 54 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>7.61e-08</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>7.61e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>9.01e-06</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.52e-05</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.12e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.76e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.07e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.03e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.42e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.03e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.06e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.06e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.28e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.51e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ligase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.73e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>vacuole</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C11 selected: 54 with EID: 54\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 44 out of 54 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.13e-86</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.68e-85</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.59e-76</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.25e-69</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.73e-62</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.33e-10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>6.82e-07</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C12 selected: 49 with EID: 49\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 49 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.88e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.85e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.29e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.07e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.65e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.74e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.78e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>6.00e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.25e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.55e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.55e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.55e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.55e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C13 selected: 49 with EID: 49\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 42 out of 49 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.18e-65</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.24e-31</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.33e-29</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.25e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.25e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.13e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>7.68e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.78e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C14 selected: 43 with EID: 43\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 43 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.38e-10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.13e-06</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.13e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.13e-06</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.66e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.73e-05</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.54e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.69e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.37e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.54e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.66e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.84e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.58e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.39e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.29e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.29e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.31e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.05e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.78e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C15 selected: 38 with EID: 38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 38 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.30e-14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.94e-10</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.84e-09</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.61e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.45e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.74e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.74e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.05e-07</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.47e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.17e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.65e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.21e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.99e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.51e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C16 selected: 35 with EID: 35\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 35 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.12e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.43e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.78e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell death</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.51e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.94e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.84e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.79e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C17 selected: 31 with EID: 31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C18 selected: 31 with EID: 31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>7.70e-19</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>9.74e-17</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.15e-15</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.06e-07</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.36e-05</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.46e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.82e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>9.37e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C19 selected: 20 with EID: 20\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 20 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.29e-33</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.70e-33</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.12e-28</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.46e-26</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.80e-22</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.78e-06</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.82e-05</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.81e-05</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.02e-04</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.17e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C20 selected: 1 with EID: 1\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 1 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2)\n",
    "    if enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Using only genes annotated with at least one gene set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 96 with EID: 96\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.39\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 30 out of 37 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.78e-05</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.02e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.13e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.13e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.13e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.18e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.18e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.36e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.36e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.04e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.04e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>phosphatase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.76e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.76e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.88e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>7.07e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.33e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.33e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.95e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.96e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.97e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.97e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C2 selected: 89 with EID: 88\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.50e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.90e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.42e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.25e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C3 selected: 87 with EID: 87\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 39 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.58e-08</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.58e-08</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.63e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.53e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.05e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.52e-03</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.85e-03</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.48e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.28e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.90e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.90e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Glyoxylate and dicarboxylate metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.82e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.22e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C4 selected: 85 with EID: 85\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.24\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 7 out of 20 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.88e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C5 selected: 73 with EID: 73\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.11\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 5 out of 8 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.91e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.03e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C6 selected: 72 with EID: 72\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.24\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 17 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.29e-09</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C7 selected: 70 with EID: 70\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.17e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>6.17e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C8 selected: 65 with EID: 63\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.14\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 9 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.97e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.97e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.97e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.34e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.34e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.34e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>lipid binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.79e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.66e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C9 selected: 57 with EID: 56\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 11 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.25e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C10 selected: 54 with EID: 54\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.57\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 26 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>6.18e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.39e-05</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.71e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.90e-03</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.90e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>9.33e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.62e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.14e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.24e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.36e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C11 selected: 54 with EID: 54\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.89\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 44 out of 48 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>7.62e-70</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.18e-68</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.22e-60</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.02e-53</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.23e-46</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.25e-05</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>3.87e-05</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C12 selected: 49 with EID: 49\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>7.42e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.80e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.00e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.11e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.11e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.98e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.25e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>5.28e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.97e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.64e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.64e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.99e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>9.10e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C13 selected: 49 with EID: 49\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.94\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 35 out of 46 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>1.94e-52</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>4.64e-22</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.21e-18</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C14 selected: 43 with EID: 43\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.51\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.28e-09</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.28e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.28e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.28e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.44e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>4.44e-04</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.82e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>3.16e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.62e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>4.84e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.98e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.19e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>8.55e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C15 selected: 38 with EID: 38\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.5\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 19 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>8.25e-13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.81e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.50e-08</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.03e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.66e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.06e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.75e-06</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell division</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.75e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>5.99e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>6.17e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.47e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C16 selected: 35 with EID: 35\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>3.66e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell death</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.88e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>4.88e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>6.60e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.93e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>6.93e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>9.01e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>1.72e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.44e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C17 selected: 31 with EID: 31\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 8 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C18 selected: 31 with EID: 31\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.71\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>3.09e-15</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>2.23e-13</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.47e-12</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>1.76e-05</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>9.85e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>7.86e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>2.21e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.78e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>7.54e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C19 selected: 20 with EID: 20\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.9\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>1.92e-26</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG Pathways</td>\n",
       "      <td>2.10e-26</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>1.65e-22</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.13e-20</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>translation</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>3.87e-16</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>8.21e-05</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO molecular_function</td>\n",
       "      <td>2.30e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO cellular_component</td>\n",
       "      <td>4.53e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO biological_process</td>\n",
       "      <td>6.00e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase Phenotypes</td>\n",
       "      <td>8.30e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C20 selected: 1 with EID: 1\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.0\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 0 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2,use_annotated_genes=True)\n",
    "    if enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
