{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Enrichment of regulons from Regulons_by_strain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from importlib import reload\n",
    "\n",
    "module_path = os.path.abspath(os.path.join('..'))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.insert(0,module_path)\n",
    "\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "\n",
    "from orangecontrib.bioinformatics.geneset.__init__ import list_all\n",
    "\n",
    "import jupyter_functions as jf\n",
    "reload(jf)\n",
    "import jupyter_functions as jf\n",
    "from functionsDENet import loadPickle, savePickle\n",
    "import stages_DE.enrichment_library as enr\n",
    "reload(enr)\n",
    "import stages_DE.enrichment_library as enr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Path for mergedGenes_RPKUM.tsv\n",
    "dataPath='/home/karin/Documents/timeTrajectories/data/RPKUM/combined/'\n",
    "#Path for \n",
    "pathRegulons='/home/karin/Documents/timeTrajectories/data/regulons/by_strain/kN300_mean0std1_log/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Used gene sets\n",
    "Use gene sets that do not have less than 5 or more than 500 genes. Use all ontologies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all gene sets\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=True,set_sizes=(5,500))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For conversion of gene names to EID\n",
    "genes = pd.read_csv(dataPath + 'mergedGenes_RPKUM.tsv', sep='\\t', index_col=0)\n",
    "all_gene_names= genes[(genes != 0).any(axis=1)].index\n",
    "NAME_EID=enr.name_genes_entrez(gene_names=all_gene_names, key_entrez=False)\n",
    "ALL_GENE_NAMES_EID=enr.convert_EID(genes=all_gene_names, name_EID=NAME_EID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def group_diff_enrichment(data:pd.DataFrame,group:str,padj:float=0.25,min_overlap:int=None,\n",
    "                          use_annotated_genes:bool=False):\n",
    "    #Displays only gene sets that have overlap with query greater or equal to min_overlap \n",
    "    #For p value and padj calculation uses alll that have overlap >=1 } from gene_set_enrichment\n",
    "    \"\"\"\n",
    "    :param use_annotated_genes: if True use for reference and query  only genes that have at \n",
    "    least one gene set annotation\n",
    "    \"\"\"\n",
    "    selected=list(data[data['Cluster']==group].index)\n",
    "    query_EID=enr.convert_EID(genes=selected, name_EID=NAME_EID)\n",
    "    print('***  '+group+' selected:',len(selected),'with EID:',len(query_EID))\n",
    "    \n",
    "    reference_gene_eids=ALL_GENE_NAMES_EID.copy()\n",
    "    query_eids=query_EID.copy()\n",
    "    \n",
    "    if use_annotated_genes:\n",
    "        gene_sets_genes=set()\n",
    "        for gene_set_name, gene_sets in GENE_SETS_ONTOLOGY.items():\n",
    "            for gene_set in gene_sets:\n",
    "                gene_sets_genes.update(gene_set.genes)\n",
    "        reference_gene_eids=set(reference_gene_eids) & gene_sets_genes\n",
    "        query_eids=set(query_eids) & gene_sets_genes\n",
    "        print('Ratio of genes annotated with a gene set in reference',\n",
    "              round(len(reference_gene_eids)/len(ALL_GENE_NAMES_EID),2),\n",
    "             'and query',round(len(query_eids)/len(query_EID),2))\n",
    "    \n",
    "    result=None\n",
    "    if len(query_eids) > 0:\n",
    "        enrichment=enr.gene_set_enrichment(query_eids, reference_EID=reference_gene_eids, \n",
    "                                                padj_threshold=padj,min_overlap=min_overlap,\n",
    "                                                gene_sets_ontology=GENE_SETS_ONTOLOGY)\n",
    "        query_in_enriched=set()\n",
    "        if len(enrichment)>0:\n",
    "            enrichment_display=list()\n",
    "            enrichment= sorted(enrichment, key=lambda data: data.padj)\n",
    "            for enriched in enrichment:\n",
    "                query_in_enriched.update(enriched.gene_set.genes & query_eids)\n",
    "                enrichment_display.append({'Gene set':enriched.gene_set.name,'Ontology':enriched.ontology,\n",
    "                                           'FDR':\"{:.2e}\".format(enriched.padj),'N in query':enriched.in_query})\n",
    "            result=pd.DataFrame(enrichment_display)\n",
    "    print('Enrichment at FDR: '+str(padj)+' and min query - gene set overlap',str(min_overlap))\n",
    "    print('N query genes in displayed gene sets:',len(query_in_enriched),'out of', len(query_eids),\n",
    "          'query genes used for enrichment calculation.')\n",
    "    display(result)\n",
    "    return result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enrichment reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_enrichment=False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Enrichment for expression_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 91 with EID: 91\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 91 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.23e-11</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.23e-11</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.29e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.53e-08</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.17e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.85e-06</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.58e-04</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.13e-04</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.81e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.06e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.69e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.02e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.21e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.02e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.21e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.29e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                           cell wall   \n",
       "1                    external encapsulating structure   \n",
       "2                      aberrant spore coat morphology   \n",
       "3   anatomical structure formation involved in mor...   \n",
       "4                cell wall organization or biogenesis   \n",
       "5                                cell differentiation   \n",
       "6                    anatomical structure development   \n",
       "7                         abolished cellulose binding   \n",
       "8                       Starch and sucrose metabolism   \n",
       "9                abolished stalk cell differentiation   \n",
       "10                               extracellular region   \n",
       "11                                 Metabolic pathways   \n",
       "12                     carbohydrate metabolic process   \n",
       "13                        cellular component assembly   \n",
       "14                                        DNA binding   \n",
       "15                          decreased spore viability   \n",
       "16                             Glutathione metabolism   \n",
       "17                              abolished sporulation   \n",
       "18                               aberrant culmination   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  8.23e-11           7  \n",
       "1   (GO, cellular_component)  8.23e-11           7  \n",
       "2    (Dictybase, Phenotypes)  4.29e-09           6  \n",
       "3   (GO, biological_process)  8.53e-08           9  \n",
       "4   (GO, biological_process)  1.17e-06           5  \n",
       "5   (GO, biological_process)  1.85e-06          10  \n",
       "6   (GO, biological_process)  2.58e-04          13  \n",
       "7    (Dictybase, Phenotypes)  7.13e-04           2  \n",
       "8           (KEGG, Pathways)  8.81e-04           4  \n",
       "9    (Dictybase, Phenotypes)  1.06e-02           2  \n",
       "10  (GO, cellular_component)  1.06e-02           6  \n",
       "11          (KEGG, Pathways)  2.69e-02          12  \n",
       "12  (GO, biological_process)  3.02e-02           4  \n",
       "13  (GO, biological_process)  3.21e-02           6  \n",
       "14  (GO, molecular_function)  4.02e-02           6  \n",
       "15   (Dictybase, Phenotypes)  6.21e-02           3  \n",
       "16          (KEGG, Pathways)  8.20e-02           2  \n",
       "17   (Dictybase, Phenotypes)  8.20e-02           2  \n",
       "18   (Dictybase, Phenotypes)  1.29e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C2 selected: 77 with EID: 77\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 77 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.82e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.79e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                               cell differentiation   \n",
       "1                                       reproduction   \n",
       "2                      Starch and sucrose metabolism   \n",
       "3                   anatomical structure development   \n",
       "4                                      cell adhesion   \n",
       "5                              decreased sporulation   \n",
       "6                 development arrests at mound stage   \n",
       "7                              abolished culmination   \n",
       "8                                delayed aggregation   \n",
       "9  anatomical structure formation involved in mor...   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, biological_process)  2.82e-03           7  \n",
       "1  (GO, biological_process)  6.79e-02           4  \n",
       "2          (KEGG, Pathways)  1.01e-01           2  \n",
       "3  (GO, biological_process)  1.01e-01           7  \n",
       "4  (GO, biological_process)  1.01e-01           3  \n",
       "5   (Dictybase, Phenotypes)  1.44e-01           2  \n",
       "6   (Dictybase, Phenotypes)  1.44e-01           2  \n",
       "7   (Dictybase, Phenotypes)  1.66e-01           2  \n",
       "8   (Dictybase, Phenotypes)  2.00e-01           2  \n",
       "9  (GO, biological_process)  2.00e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C3 selected: 74 with EID: 74\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 24 out of 74 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.02e-06</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.99e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.56e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.06e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.23e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.90e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.90e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.90e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.90e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.00e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.08e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.33e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.95e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.18e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.18e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.18e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.94e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                  Ontology  \\\n",
       "0           anatomical structure development  (GO, biological_process)   \n",
       "1                              cell adhesion  (GO, biological_process)   \n",
       "2   development arrests at loose mound stage   (Dictybase, Phenotypes)   \n",
       "3                       extracellular region  (GO, cellular_component)   \n",
       "4              Glycosaminoglycan degradation          (KEGG, Pathways)   \n",
       "5               decreased fruiting body size   (Dictybase, Phenotypes)   \n",
       "6                   decreased aggregate size   (Dictybase, Phenotypes)   \n",
       "7                   increased slug migration   (Dictybase, Phenotypes)   \n",
       "8                         response to stress  (GO, biological_process)   \n",
       "9                        signal transduction  (GO, biological_process)   \n",
       "10                    multiple tips in mound   (Dictybase, Phenotypes)   \n",
       "11                      aberrant aggregation   (Dictybase, Phenotypes)   \n",
       "12                           plasma membrane  (GO, cellular_component)   \n",
       "13                                locomotion  (GO, biological_process)   \n",
       "14                 abolished gene expression   (Dictybase, Phenotypes)   \n",
       "15                      cell differentiation  (GO, biological_process)   \n",
       "16                    precocious development   (Dictybase, Phenotypes)   \n",
       "17                 decreased gene expression   (Dictybase, Phenotypes)   \n",
       "18                 decreased chemotaxis rate   (Dictybase, Phenotypes)   \n",
       "19         aberrant fruiting body morphology   (Dictybase, Phenotypes)   \n",
       "20             aberrant F-actin organization   (Dictybase, Phenotypes)   \n",
       "21                             cell motility  (GO, biological_process)   \n",
       "22                   decreased cell motility   (Dictybase, Phenotypes)   \n",
       "23                     abolished aggregation   (Dictybase, Phenotypes)   \n",
       "24                              cytoskeleton  (GO, cellular_component)   \n",
       "25                 cytoskeleton organization  (GO, biological_process)   \n",
       "26                                 wild type   (Dictybase, Phenotypes)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   4.02e-06          15  \n",
       "1   7.99e-03           5  \n",
       "2   8.56e-03           3  \n",
       "3   1.06e-02           6  \n",
       "4   1.23e-02           2  \n",
       "5   1.36e-02           4  \n",
       "6   1.36e-02           3  \n",
       "7   1.36e-02           2  \n",
       "8   1.36e-02           8  \n",
       "9   1.36e-02           8  \n",
       "10  3.90e-02           2  \n",
       "11  3.90e-02           3  \n",
       "12  3.90e-02           6  \n",
       "13  3.90e-02           5  \n",
       "14  6.00e-02           2  \n",
       "15  6.08e-02           4  \n",
       "16  6.33e-02           2  \n",
       "17  6.95e-02           3  \n",
       "18  8.18e-02           2  \n",
       "19  8.18e-02           3  \n",
       "20  8.18e-02           2  \n",
       "21  1.55e-01           2  \n",
       "22  1.62e-01           2  \n",
       "23  1.71e-01           2  \n",
       "24  1.94e-01           3  \n",
       "25  2.07e-01           3  \n",
       "26  2.42e-01           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C4 selected: 74 with EID: 74\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 74 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.60e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                  Ontology       FDR  N in query\n",
       "0  extracellular region  (GO, cellular_component)  1.60e-04           8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C5 selected: 68 with EID: 68\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 65 out of 68 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.41e-126</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.47e-124</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.17e-109</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.72e-98</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.92e-86</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.97e-15</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.86e-13</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.69e-05</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Gene set                  Ontology        FDR  \\\n",
       "0                             Ribosome          (KEGG, Pathways)  6.41e-126   \n",
       "1                             ribosome  (GO, cellular_component)  1.47e-124   \n",
       "2   structural constituent of ribosome  (GO, molecular_function)  8.17e-109   \n",
       "3         structural molecule activity  (GO, molecular_function)   1.72e-98   \n",
       "4                          translation  (GO, biological_process)   5.92e-86   \n",
       "5                          RNA binding  (GO, molecular_function)   2.97e-15   \n",
       "6                         rRNA binding  (GO, molecular_function)   8.86e-13   \n",
       "7                              cytosol  (GO, cellular_component)   5.69e-05   \n",
       "8                  ribosome biogenesis  (GO, biological_process)   1.68e-02   \n",
       "9                  delayed development   (Dictybase, Phenotypes)   1.62e-01   \n",
       "10  transmembrane transporter activity  (GO, molecular_function)   1.62e-01   \n",
       "11             transmembrane transport  (GO, biological_process)   1.80e-01   \n",
       "\n",
       "    N in query  \n",
       "0           60  \n",
       "1           59  \n",
       "2           53  \n",
       "3           53  \n",
       "4           52  \n",
       "5           16  \n",
       "6            7  \n",
       "7            8  \n",
       "8            3  \n",
       "9            2  \n",
       "10           3  \n",
       "11           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C6 selected: 66 with EID: 65\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 65 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.98e-10</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.55e-08</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.07e-05</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.90e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.28e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.92e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.95e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.24e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.03e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant cellular response to stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.38e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.42e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.46e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.06e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.06e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.72e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.80e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.80e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.77e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.68e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.70e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.53e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>vacuole</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.38e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.50e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.46e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                  Ontology  \\\n",
       "0                        Oxidative phosphorylation          (KEGG, Pathways)   \n",
       "1                               Metabolic pathways          (KEGG, Pathways)   \n",
       "2                                    mitochondrion  (GO, cellular_component)   \n",
       "3                        Citrate cycle (TCA cycle)          (KEGG, Pathways)   \n",
       "4                                        Phagosome          (KEGG, Pathways)   \n",
       "5                      Biosynthesis of antibiotics          (KEGG, Pathways)   \n",
       "6                              cytoplasmic vesicle  (GO, cellular_component)   \n",
       "7   generation of precursor metabolites and energy  (GO, biological_process)   \n",
       "8                                  protein folding  (GO, biological_process)   \n",
       "9             aberrant cellular response to stress   (Dictybase, Phenotypes)   \n",
       "10                                        Ribosome          (KEGG, Pathways)   \n",
       "11          Various types of N-glycan biosynthesis          (KEGG, Pathways)   \n",
       "12                              cell morphogenesis  (GO, biological_process)   \n",
       "13           Biosynthesis of secondary metabolites          (KEGG, Pathways)   \n",
       "14              transmembrane transporter activity  (GO, molecular_function)   \n",
       "15                        unfolded protein binding  (GO, molecular_function)   \n",
       "16                           N-Glycan biosynthesis          (KEGG, Pathways)   \n",
       "17                           Propanoate metabolism          (KEGG, Pathways)   \n",
       "18                         transmembrane transport  (GO, biological_process)   \n",
       "19                               Carbon metabolism          (KEGG, Pathways)   \n",
       "20                        aberrant cell morphology   (Dictybase, Phenotypes)   \n",
       "21                                         cytosol  (GO, cellular_component)   \n",
       "22                         oxidoreductase activity  (GO, molecular_function)   \n",
       "23                                         vacuole  (GO, cellular_component)   \n",
       "24                              aberrant streaming   (Dictybase, Phenotypes)   \n",
       "25                                 ligase activity  (GO, molecular_function)   \n",
       "26     Protein processing in endoplasmic reticulum          (KEGG, Pathways)   \n",
       "27                            aberrant cytokinesis   (Dictybase, Phenotypes)   \n",
       "28                              mitotic cell cycle  (GO, biological_process)   \n",
       "29                small molecule metabolic process  (GO, biological_process)   \n",
       "30                                   cell division  (GO, biological_process)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   6.98e-10           9  \n",
       "1   2.55e-08          19  \n",
       "2   8.07e-05           9  \n",
       "3   3.90e-04           4  \n",
       "4   3.28e-03           4  \n",
       "5   4.92e-03           6  \n",
       "6   8.95e-03           6  \n",
       "7   9.24e-03           4  \n",
       "8   1.03e-02           3  \n",
       "9   1.31e-02           2  \n",
       "10  1.38e-02           4  \n",
       "11  4.42e-02           2  \n",
       "12  4.46e-02           2  \n",
       "13  5.06e-02           5  \n",
       "14  5.06e-02           4  \n",
       "15  5.06e-02           2  \n",
       "16  5.72e-02           2  \n",
       "17  5.80e-02           2  \n",
       "18  5.80e-02           4  \n",
       "19  6.77e-02           3  \n",
       "20  7.68e-02           2  \n",
       "21  7.70e-02           4  \n",
       "22  8.53e-02           4  \n",
       "23  9.38e-02           3  \n",
       "24  1.50e-01           2  \n",
       "25  1.62e-01           2  \n",
       "26  1.86e-01           2  \n",
       "27  2.07e-01           2  \n",
       "28  2.09e-01           2  \n",
       "29  2.30e-01           4  \n",
       "30  2.46e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C7 selected: 64 with EID: 64\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 64 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.46e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.95e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Gene set                 Ontology       FDR  \\\n",
       "0        aberrant cell-cell adhesion  (Dictybase, Phenotypes)  4.46e-03   \n",
       "1  aberrant fruiting body morphology  (Dictybase, Phenotypes)  1.95e-01   \n",
       "\n",
       "   N in query  \n",
       "0           2  \n",
       "1           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C8 selected: 61 with EID: 59\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 59 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C9 selected: 58 with EID: 58\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 58 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.13e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.13e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                               extracellular region   \n",
       "1                                      cell adhesion   \n",
       "2                                 hydrolase activity   \n",
       "3                               cell differentiation   \n",
       "4  anatomical structure formation involved in mor...   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  8.13e-02           4  \n",
       "1  (GO, biological_process)  8.13e-02           3  \n",
       "2  (GO, molecular_function)  1.13e-01           2  \n",
       "3  (GO, biological_process)  1.59e-01           3  \n",
       "4  (GO, biological_process)  1.76e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C10 selected: 57 with EID: 57\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 57 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.59e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.72e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                  Ontology       FDR  N in query\n",
       "0       extracellular region  (GO, cellular_component)  3.59e-04           7\n",
       "1        cell-cell signaling  (GO, biological_process)  3.72e-02           2\n",
       "2  decreased spore viability   (Dictybase, Phenotypes)  1.87e-01           2\n",
       "3         peptidase activity  (GO, molecular_function)  2.07e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C11 selected: 55 with EID: 55\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 55 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.97e-08</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.98e-04</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.93e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.03e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.40e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.74e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.78e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.33e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.33e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.06e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                  Ontology       FDR  \\\n",
       "0          cytoskeletal protein binding  (GO, molecular_function)  5.97e-08   \n",
       "1                           Endocytosis          (KEGG, Pathways)  3.98e-04   \n",
       "2             cytoskeleton organization  (GO, biological_process)  2.93e-03   \n",
       "3           cellular component assembly  (GO, biological_process)  3.03e-03   \n",
       "4   protein-containing complex assembly  (GO, biological_process)  3.40e-03   \n",
       "5                    response to stress  (GO, biological_process)  5.74e-03   \n",
       "6                          cytoskeleton  (GO, cellular_component)  7.78e-03   \n",
       "7                   cytoplasmic vesicle  (GO, cellular_component)  5.33e-02   \n",
       "8                               cytosol  (GO, cellular_component)  5.33e-02   \n",
       "9                       plasma membrane  (GO, cellular_component)  1.06e-01   \n",
       "10              decreased cell motility   (Dictybase, Phenotypes)  1.48e-01   \n",
       "11                            wild type   (Dictybase, Phenotypes)  1.62e-01   \n",
       "12           vesicle-mediated transport  (GO, biological_process)  1.71e-01   \n",
       "\n",
       "    N in query  \n",
       "0           10  \n",
       "1            5  \n",
       "2            6  \n",
       "3            6  \n",
       "4            5  \n",
       "5            7  \n",
       "6            5  \n",
       "7            4  \n",
       "8            4  \n",
       "9            4  \n",
       "10           2  \n",
       "11           4  \n",
       "12           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C12 selected: 54 with EID: 54\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 54 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.50e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.50e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                 Ontology       FDR  N in query\n",
       "0                  Phagosome         (KEGG, Pathways)  2.50e-01           2\n",
       "1  decreased spore viability  (Dictybase, Phenotypes)  2.50e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C13 selected: 46 with EID: 45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 45 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.97e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.82e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.21e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                               extracellular region   \n",
       "1  anatomical structure formation involved in mor...   \n",
       "2                               cell differentiation   \n",
       "3                            oxidoreductase activity   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  4.97e-04           6  \n",
       "1  (GO, biological_process)  2.82e-02           3  \n",
       "2  (GO, biological_process)  1.21e-01           3  \n",
       "3  (GO, molecular_function)  1.52e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C14 selected: 45 with EID: 45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 45 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C15 selected: 41 with EID: 41\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 38 out of 41 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.80e-69</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.55e-34</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.87e-32</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.73e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.73e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.27e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Gene set                  Ontology  \\\n",
       "0                                   Proteasome          (KEGG, Pathways)   \n",
       "1                           peptidase activity  (GO, molecular_function)   \n",
       "2                            catabolic process  (GO, biological_process)   \n",
       "3                           response to stress  (GO, biological_process)   \n",
       "4                    enzyme regulator activity  (GO, molecular_function)   \n",
       "5                              ATPase activity  (GO, molecular_function)   \n",
       "6  Protein processing in endoplasmic reticulum          (KEGG, Pathways)   \n",
       "\n",
       "        FDR  N in query  \n",
       "0  2.80e-69          29  \n",
       "1  7.55e-34          21  \n",
       "2  6.87e-32          25  \n",
       "3  3.73e-02           5  \n",
       "4  3.73e-02           3  \n",
       "5  8.27e-02           3  \n",
       "6  1.35e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C16 selected: 40 with EID: 40\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 40 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>translucent sorus</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.14e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell death</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.14e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.14e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.44e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.88e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.88e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.93e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.82e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.02e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0   aberrant CRAC localization to the plasma membrane   \n",
       "1           development arrests at tipped mound stage   \n",
       "2                                   translucent sorus   \n",
       "3                  decreased intracellular cAMP level   \n",
       "4                    anatomical structure development   \n",
       "5                                          cell death   \n",
       "6   anatomical structure formation involved in mor...   \n",
       "7                                     kinase activity   \n",
       "8                                     RNA degradation   \n",
       "9                                cell differentiation   \n",
       "10                               aberrant aggregation   \n",
       "11                       decreased chemotaxis to cAMP   \n",
       "12                                         locomotion   \n",
       "13                       decreased fruiting body size   \n",
       "14                  aberrant fruiting body morphology   \n",
       "15                                          wild type   \n",
       "16                                       cytoskeleton   \n",
       "17                                        DNA binding   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0    (Dictybase, Phenotypes)  1.48e-02           2  \n",
       "1    (Dictybase, Phenotypes)  1.61e-02           2  \n",
       "2    (Dictybase, Phenotypes)  1.61e-02           2  \n",
       "3    (Dictybase, Phenotypes)  1.61e-02           2  \n",
       "4   (GO, biological_process)  3.14e-02           6  \n",
       "5   (GO, biological_process)  3.14e-02           2  \n",
       "6   (GO, biological_process)  3.14e-02           3  \n",
       "7   (GO, molecular_function)  3.44e-02           5  \n",
       "8           (KEGG, Pathways)  4.88e-02           2  \n",
       "9   (GO, biological_process)  4.88e-02           3  \n",
       "10   (Dictybase, Phenotypes)  4.89e-02           2  \n",
       "11   (Dictybase, Phenotypes)  4.93e-02           2  \n",
       "12  (GO, biological_process)  5.82e-02           3  \n",
       "13   (Dictybase, Phenotypes)  5.92e-02           2  \n",
       "14   (Dictybase, Phenotypes)  8.02e-02           2  \n",
       "15   (Dictybase, Phenotypes)  1.48e-01           3  \n",
       "16  (GO, cellular_component)  1.75e-01           2  \n",
       "17  (GO, molecular_function)  2.14e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C17 selected: 36 with EID: 36\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 36 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.31e-14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.37e-10</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.90e-09</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.74e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.72e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.92e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.92e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.17e-07</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.71e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.01e-04</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.43e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.41e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.32e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Gene set                  Ontology       FDR  \\\n",
       "0                      cell cycle  (GO, biological_process)  2.31e-14   \n",
       "1          chromosome segregation  (GO, biological_process)  1.37e-10   \n",
       "2        mitotic nuclear division  (GO, biological_process)  5.90e-09   \n",
       "3         chromosome organization  (GO, biological_process)  1.74e-08   \n",
       "4              mitotic cell cycle  (GO, biological_process)  9.72e-08   \n",
       "5                 DNA replication          (KEGG, Pathways)  1.92e-07   \n",
       "6                   cell division  (GO, biological_process)  1.92e-07   \n",
       "7                      chromosome  (GO, cellular_component)  2.17e-07   \n",
       "8                    cytoskeleton  (GO, cellular_component)  3.71e-06   \n",
       "9       cytoskeleton organization  (GO, biological_process)  9.01e-04   \n",
       "10  microtubule organizing center  (GO, cellular_component)  1.41e-03   \n",
       "11   cytoskeletal protein binding  (GO, molecular_function)  3.43e-03   \n",
       "12          DNA metabolic process  (GO, biological_process)  7.24e-02   \n",
       "13                ATPase activity  (GO, molecular_function)  1.41e-01   \n",
       "14                plasma membrane  (GO, cellular_component)  2.32e-01   \n",
       "\n",
       "    N in query  \n",
       "0           13  \n",
       "1            6  \n",
       "2            5  \n",
       "3            7  \n",
       "4            7  \n",
       "5            5  \n",
       "6            7  \n",
       "7            6  \n",
       "8            7  \n",
       "9            5  \n",
       "10           3  \n",
       "11           4  \n",
       "12           2  \n",
       "13           2  \n",
       "14           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C18 selected: 30 with EID: 30\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 30 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.61e-19</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.81e-17</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.97e-16</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.35e-07</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.44e-06</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.00e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.00e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.00e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.30e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.78e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                  Ontology       FDR  \\\n",
       "0     Ribosome biogenesis in eukaryotes          (KEGG, Pathways)  3.61e-19   \n",
       "1                             nucleolus  (GO, cellular_component)  4.81e-17   \n",
       "2                   ribosome biogenesis  (GO, biological_process)  5.97e-16   \n",
       "3    ribonucleoprotein complex assembly  (GO, biological_process)  5.35e-07   \n",
       "4                           RNA binding  (GO, molecular_function)  8.44e-06   \n",
       "5   protein-containing complex assembly  (GO, biological_process)  1.00e-03   \n",
       "6                          rRNA binding  (GO, molecular_function)  1.00e-03   \n",
       "7           cellular component assembly  (GO, biological_process)  3.00e-03   \n",
       "8                     helicase activity  (GO, molecular_function)  1.20e-02   \n",
       "9            methyltransferase activity  (GO, molecular_function)  1.30e-02   \n",
       "10                      ATPase activity  (GO, molecular_function)  7.78e-02   \n",
       "\n",
       "    N in query  \n",
       "0           12  \n",
       "1           11  \n",
       "2           10  \n",
       "3            4  \n",
       "4            6  \n",
       "5            4  \n",
       "6            2  \n",
       "7            4  \n",
       "8            2  \n",
       "9            2  \n",
       "10           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C19 selected: 23 with EID: 23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 23 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene set                  Ontology       FDR  N in query\n",
       "0  kinase activity  (GO, molecular_function)  1.10e-01           3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C20 selected: 20 with EID: 20\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 20 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C21 selected: 19 with EID: 19\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 19 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2)\n",
    "    if save_enrichment and enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Using only genes annotated with at least one gene set \n",
    "To find genes that have gene set annotations only gene sets used in enrichment are used (see above). Genes that are in at least one of these gene sets are used for enrichment calculation, both for reference and gene group of interest (query)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 91 with EID: 91\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 41 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.81e-10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.81e-10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.74e-08</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.03e-06</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.66e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.53e-05</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.61e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.46e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.50e-03</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.28e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.16e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                           cell wall   \n",
       "1                    external encapsulating structure   \n",
       "2                      aberrant spore coat morphology   \n",
       "3   anatomical structure formation involved in mor...   \n",
       "4                cell wall organization or biogenesis   \n",
       "5                                cell differentiation   \n",
       "6                         abolished cellulose binding   \n",
       "7                       Starch and sucrose metabolism   \n",
       "8                    anatomical structure development   \n",
       "9                abolished stalk cell differentiation   \n",
       "10                               extracellular region   \n",
       "11                     carbohydrate metabolic process   \n",
       "12                        cellular component assembly   \n",
       "13                             Glutathione metabolism   \n",
       "14                              abolished sporulation   \n",
       "15                          decreased spore viability   \n",
       "16                                        DNA binding   \n",
       "17                                 Metabolic pathways   \n",
       "18                               aberrant culmination   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  6.81e-10           7  \n",
       "1   (GO, cellular_component)  6.81e-10           7  \n",
       "2    (Dictybase, Phenotypes)  2.74e-08           6  \n",
       "3   (GO, biological_process)  1.03e-06           9  \n",
       "4   (GO, biological_process)  5.66e-06           5  \n",
       "5   (GO, biological_process)  2.53e-05          10  \n",
       "6    (Dictybase, Phenotypes)  1.61e-03           2  \n",
       "7           (KEGG, Pathways)  3.46e-03           4  \n",
       "8   (GO, biological_process)  3.50e-03          13  \n",
       "9    (Dictybase, Phenotypes)  2.28e-02           2  \n",
       "10  (GO, cellular_component)  5.16e-02           6  \n",
       "11  (GO, biological_process)  1.03e-01           4  \n",
       "12  (GO, biological_process)  1.55e-01           6  \n",
       "13          (KEGG, Pathways)  1.61e-01           2  \n",
       "14   (Dictybase, Phenotypes)  1.61e-01           2  \n",
       "15   (Dictybase, Phenotypes)  1.61e-01           3  \n",
       "16  (GO, molecular_function)  1.61e-01           6  \n",
       "17          (KEGG, Pathways)  1.64e-01          12  \n",
       "18   (Dictybase, Phenotypes)  2.29e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C2 selected: 77 with EID: 77\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.29\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 15 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.48e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.08e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.24e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.83e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.08e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.38e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.72e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.18e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.18e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                cell differentiation   \n",
       "1                                        reproduction   \n",
       "2                    anatomical structure development   \n",
       "3                                       cell adhesion   \n",
       "4                       Starch and sucrose metabolism   \n",
       "5                               decreased sporulation   \n",
       "6                  development arrests at mound stage   \n",
       "7                               abolished culmination   \n",
       "8                                 delayed aggregation   \n",
       "9   anatomical structure formation involved in mor...   \n",
       "10                          cytoskeleton organization   \n",
       "11                                 peptidase activity   \n",
       "12                        cellular component assembly   \n",
       "13                                          wild type   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  7.48e-04           7  \n",
       "1   (GO, biological_process)  6.08e-02           4  \n",
       "2   (GO, biological_process)  7.24e-02           7  \n",
       "3   (GO, biological_process)  8.83e-02           3  \n",
       "4           (KEGG, Pathways)  9.08e-02           2  \n",
       "5    (Dictybase, Phenotypes)  1.20e-01           2  \n",
       "6    (Dictybase, Phenotypes)  1.20e-01           2  \n",
       "7    (Dictybase, Phenotypes)  1.38e-01           2  \n",
       "8    (Dictybase, Phenotypes)  1.69e-01           2  \n",
       "9   (GO, biological_process)  1.72e-01           2  \n",
       "10  (GO, biological_process)  2.18e-01           3  \n",
       "11  (GO, molecular_function)  2.18e-01           2  \n",
       "12  (GO, biological_process)  2.31e-01           3  \n",
       "13   (Dictybase, Phenotypes)  2.43e-01           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C3 selected: 74 with EID: 74\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.42\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 24 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.37e-05</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.80e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.80e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.59e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.86e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.15e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.15e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.80e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.64e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.57e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.57e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.57e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.57e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.32e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                  Ontology  \\\n",
       "0           anatomical structure development  (GO, biological_process)   \n",
       "1   development arrests at loose mound stage   (Dictybase, Phenotypes)   \n",
       "2                              cell adhesion  (GO, biological_process)   \n",
       "3              Glycosaminoglycan degradation          (KEGG, Pathways)   \n",
       "4                       extracellular region  (GO, cellular_component)   \n",
       "5                   decreased aggregate size   (Dictybase, Phenotypes)   \n",
       "6                   increased slug migration   (Dictybase, Phenotypes)   \n",
       "7               decreased fruiting body size   (Dictybase, Phenotypes)   \n",
       "8                         response to stress  (GO, biological_process)   \n",
       "9                        signal transduction  (GO, biological_process)   \n",
       "10                    multiple tips in mound   (Dictybase, Phenotypes)   \n",
       "11                 abolished gene expression   (Dictybase, Phenotypes)   \n",
       "12                      aberrant aggregation   (Dictybase, Phenotypes)   \n",
       "13                           plasma membrane  (GO, cellular_component)   \n",
       "14                                locomotion  (GO, biological_process)   \n",
       "15                    precocious development   (Dictybase, Phenotypes)   \n",
       "16                      cell differentiation  (GO, biological_process)   \n",
       "17                 decreased gene expression   (Dictybase, Phenotypes)   \n",
       "18                 decreased chemotaxis rate   (Dictybase, Phenotypes)   \n",
       "19             aberrant F-actin organization   (Dictybase, Phenotypes)   \n",
       "20         aberrant fruiting body morphology   (Dictybase, Phenotypes)   \n",
       "21                             cell motility  (GO, biological_process)   \n",
       "22                   decreased cell motility   (Dictybase, Phenotypes)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   2.37e-05          15  \n",
       "1   1.80e-02           3  \n",
       "2   1.80e-02           5  \n",
       "3   2.59e-02           2  \n",
       "4   2.86e-02           6  \n",
       "5   3.15e-02           3  \n",
       "6   3.15e-02           2  \n",
       "7   3.80e-02           4  \n",
       "8   5.49e-02           8  \n",
       "9   5.49e-02           8  \n",
       "10  6.64e-02           2  \n",
       "11  8.57e-02           2  \n",
       "12  8.57e-02           3  \n",
       "13  8.57e-02           6  \n",
       "14  8.57e-02           5  \n",
       "15  1.09e-01           2  \n",
       "16  1.16e-01           4  \n",
       "17  1.17e-01           3  \n",
       "18  1.17e-01           2  \n",
       "19  1.20e-01           2  \n",
       "20  1.32e-01           3  \n",
       "21  2.12e-01           2  \n",
       "22  2.42e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C4 selected: 74 with EID: 74\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 15 out of 17 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.67e-06</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.42e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                  Ontology       FDR  N in query\n",
       "0       extracellular region  (GO, cellular_component)  3.67e-06           8\n",
       "1                  wild type   (Dictybase, Phenotypes)  1.42e-01           5\n",
       "2  decreased gene expression   (Dictybase, Phenotypes)  2.08e-01           2\n",
       "3  decreased spore viability   (Dictybase, Phenotypes)  2.08e-01           2\n",
       "4                DNA binding  (GO, molecular_function)  2.15e-01           3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C5 selected: 68 with EID: 68\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.97\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 62 out of 66 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.55e-98</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.07e-97</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.42e-84</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.86e-73</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.02e-61</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.37e-09</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.80e-08</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.48e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                  Ontology       FDR  \\\n",
       "0                            Ribosome          (KEGG, Pathways)  6.55e-98   \n",
       "1                            ribosome  (GO, cellular_component)  6.07e-97   \n",
       "2  structural constituent of ribosome  (GO, molecular_function)  9.42e-84   \n",
       "3        structural molecule activity  (GO, molecular_function)  1.86e-73   \n",
       "4                         translation  (GO, biological_process)  2.02e-61   \n",
       "5                        rRNA binding  (GO, molecular_function)  2.37e-09   \n",
       "6                         RNA binding  (GO, molecular_function)  3.80e-08   \n",
       "7                             cytosol  (GO, cellular_component)  7.48e-02   \n",
       "\n",
       "   N in query  \n",
       "0          60  \n",
       "1          59  \n",
       "2          53  \n",
       "3          53  \n",
       "4          52  \n",
       "5           7  \n",
       "6          16  \n",
       "7           8  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C6 selected: 66 with EID: 65\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.58\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 38 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.41e-08</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.64e-05</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.09e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.11e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.99e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.94e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant cellular response to stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.94e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.94e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.94e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.41e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.41e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                  Ontology  \\\n",
       "0                        Oxidative phosphorylation          (KEGG, Pathways)   \n",
       "1                               Metabolic pathways          (KEGG, Pathways)   \n",
       "2                        Citrate cycle (TCA cycle)          (KEGG, Pathways)   \n",
       "3                                    mitochondrion  (GO, cellular_component)   \n",
       "4                                        Phagosome          (KEGG, Pathways)   \n",
       "5                      Biosynthesis of antibiotics          (KEGG, Pathways)   \n",
       "6             aberrant cellular response to stress   (Dictybase, Phenotypes)   \n",
       "7   generation of precursor metabolites and energy  (GO, biological_process)   \n",
       "8                                  protein folding  (GO, biological_process)   \n",
       "9                                         Ribosome          (KEGG, Pathways)   \n",
       "10                             cytoplasmic vesicle  (GO, cellular_component)   \n",
       "11          Various types of N-glycan biosynthesis          (KEGG, Pathways)   \n",
       "12                              cell morphogenesis  (GO, biological_process)   \n",
       "13                        unfolded protein binding  (GO, molecular_function)   \n",
       "14                           N-Glycan biosynthesis          (KEGG, Pathways)   \n",
       "15                           Propanoate metabolism          (KEGG, Pathways)   \n",
       "16                        aberrant cell morphology   (Dictybase, Phenotypes)   \n",
       "17              transmembrane transporter activity  (GO, molecular_function)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   9.41e-08           9  \n",
       "1   8.64e-05          19  \n",
       "2   5.09e-03           4  \n",
       "3   5.11e-03           9  \n",
       "4   2.99e-02           4  \n",
       "5   6.94e-02           6  \n",
       "6   6.94e-02           2  \n",
       "7   6.94e-02           4  \n",
       "8   6.94e-02           3  \n",
       "9   1.11e-01           4  \n",
       "10  1.11e-01           6  \n",
       "11  1.40e-01           2  \n",
       "12  1.40e-01           2  \n",
       "13  1.62e-01           2  \n",
       "14  1.77e-01           2  \n",
       "15  1.81e-01           2  \n",
       "16  2.41e-01           2  \n",
       "17  2.41e-01           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C7 selected: 64 with EID: 64\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.16\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 10 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.86e-04</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.72e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.90e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.93e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Gene set                  Ontology       FDR  \\\n",
       "0        aberrant cell-cell adhesion   (Dictybase, Phenotypes)  9.86e-04   \n",
       "1   anatomical structure development  (GO, biological_process)  5.72e-02   \n",
       "2  aberrant fruiting body morphology   (Dictybase, Phenotypes)  5.90e-02   \n",
       "3               extracellular region  (GO, cellular_component)  9.93e-02   \n",
       "\n",
       "   N in query  \n",
       "0           2  \n",
       "1           4  \n",
       "2           2  \n",
       "3           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C8 selected: 61 with EID: 59\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.19\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 11 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.24e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.24e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                  Ontology       FDR  \\\n",
       "0  anatomical structure development  (GO, biological_process)  2.08e-01   \n",
       "1                         wild type   (Dictybase, Phenotypes)  2.24e-01   \n",
       "2              extracellular region  (GO, cellular_component)  2.24e-01   \n",
       "\n",
       "   N in query  \n",
       "0           4  \n",
       "1           3  \n",
       "2           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C9 selected: 58 with EID: 58\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 12 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.40e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.40e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.85e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                               extracellular region   \n",
       "1                                      cell adhesion   \n",
       "2                               cell differentiation   \n",
       "3                                 hydrolase activity   \n",
       "4  anatomical structure formation involved in mor...   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  2.26e-02           4  \n",
       "1  (GO, biological_process)  2.26e-02           3  \n",
       "2  (GO, biological_process)  4.40e-02           3  \n",
       "3  (GO, molecular_function)  4.40e-02           2  \n",
       "4  (GO, biological_process)  8.85e-02           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C10 selected: 57 with EID: 57\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 11 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.89e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.11e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.42e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                  Ontology       FDR  \\\n",
       "0              extracellular region  (GO, cellular_component)  1.89e-05   \n",
       "1               cell-cell signaling  (GO, biological_process)  2.11e-02   \n",
       "2         decreased spore viability   (Dictybase, Phenotypes)  1.20e-01   \n",
       "3                peptidase activity  (GO, molecular_function)  1.42e-01   \n",
       "4                   plasma membrane  (GO, cellular_component)  1.70e-01   \n",
       "5  anatomical structure development  (GO, biological_process)  1.70e-01   \n",
       "\n",
       "   N in query  \n",
       "0           7  \n",
       "1           2  \n",
       "2           2  \n",
       "3           2  \n",
       "4           3  \n",
       "5           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C11 selected: 55 with EID: 55\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.44\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 24 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.76e-07</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.15e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.15e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.15e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.49e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.55e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.32e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.32e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                  Ontology       FDR  \\\n",
       "0          cytoskeletal protein binding  (GO, molecular_function)  3.76e-07   \n",
       "1                           Endocytosis          (KEGG, Pathways)  1.41e-03   \n",
       "2           cellular component assembly  (GO, biological_process)  1.15e-02   \n",
       "3   protein-containing complex assembly  (GO, biological_process)  1.15e-02   \n",
       "4             cytoskeleton organization  (GO, biological_process)  1.15e-02   \n",
       "5                    response to stress  (GO, biological_process)  2.49e-02   \n",
       "6                          cytoskeleton  (GO, cellular_component)  2.55e-02   \n",
       "7                   cytoplasmic vesicle  (GO, cellular_component)  1.32e-01   \n",
       "8                               cytosol  (GO, cellular_component)  1.32e-01   \n",
       "9               decreased cell motility   (Dictybase, Phenotypes)  2.33e-01   \n",
       "10                      plasma membrane  (GO, cellular_component)  2.33e-01   \n",
       "\n",
       "    N in query  \n",
       "0           10  \n",
       "1            5  \n",
       "2            6  \n",
       "3            5  \n",
       "4            6  \n",
       "5            7  \n",
       "6            5  \n",
       "7            4  \n",
       "8            4  \n",
       "9            2  \n",
       "10           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C12 selected: 54 with EID: 54\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 11 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                 Ontology       FDR  N in query\n",
       "0                  Phagosome         (KEGG, Pathways)  1.60e-01           2\n",
       "1  decreased spore viability  (Dictybase, Phenotypes)  1.60e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C13 selected: 46 with EID: 45\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.33\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.26e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.85e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.24e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                               extracellular region   \n",
       "1  anatomical structure formation involved in mor...   \n",
       "2                               cell differentiation   \n",
       "3                            oxidoreductase activity   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  3.26e-04           6  \n",
       "1  (GO, biological_process)  2.85e-02           3  \n",
       "2  (GO, biological_process)  1.24e-01           3  \n",
       "3  (GO, molecular_function)  1.60e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C14 selected: 45 with EID: 45\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.13\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 6 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.35e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene set                  Ontology       FDR  N in query\n",
       "0             cytosol  (GO, cellular_component)  2.35e-01           2\n",
       "1  response to stress  (GO, biological_process)  2.43e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C15 selected: 41 with EID: 41\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.98\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 35 out of 40 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.20e-55</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.26e-24</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.25e-20</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene set                  Ontology       FDR  N in query\n",
       "0          Proteasome          (KEGG, Pathways)  2.20e-55          29\n",
       "1  peptidase activity  (GO, molecular_function)  8.26e-24          21\n",
       "2   catabolic process  (GO, biological_process)  4.25e-20          25"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C16 selected: 40 with EID: 40\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.96e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>translucent sorus</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.76e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell death</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.76e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.76e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.08e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.74e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.74e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.59e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.59e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.57e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.98e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0   aberrant CRAC localization to the plasma membrane   \n",
       "1           development arrests at tipped mound stage   \n",
       "2                                   translucent sorus   \n",
       "3                  decreased intracellular cAMP level   \n",
       "4                    anatomical structure development   \n",
       "5                                          cell death   \n",
       "6   anatomical structure formation involved in mor...   \n",
       "7                                     kinase activity   \n",
       "8                                     RNA degradation   \n",
       "9                                cell differentiation   \n",
       "10                               aberrant aggregation   \n",
       "11                       decreased chemotaxis to cAMP   \n",
       "12                       decreased fruiting body size   \n",
       "13                                         locomotion   \n",
       "14                  aberrant fruiting body morphology   \n",
       "15                                          wild type   \n",
       "16                                       cytoskeleton   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0    (Dictybase, Phenotypes)  1.96e-02           2  \n",
       "1    (Dictybase, Phenotypes)  2.13e-02           2  \n",
       "2    (Dictybase, Phenotypes)  2.13e-02           2  \n",
       "3    (Dictybase, Phenotypes)  2.13e-02           2  \n",
       "4   (GO, biological_process)  3.76e-02           6  \n",
       "5   (GO, biological_process)  3.76e-02           2  \n",
       "6   (GO, biological_process)  3.76e-02           3  \n",
       "7   (GO, molecular_function)  5.08e-02           5  \n",
       "8           (KEGG, Pathways)  5.74e-02           2  \n",
       "9   (GO, biological_process)  5.74e-02           3  \n",
       "10   (Dictybase, Phenotypes)  5.92e-02           2  \n",
       "11   (Dictybase, Phenotypes)  5.92e-02           2  \n",
       "12   (Dictybase, Phenotypes)  7.59e-02           2  \n",
       "13  (GO, biological_process)  7.59e-02           3  \n",
       "14   (Dictybase, Phenotypes)  9.57e-02           2  \n",
       "15   (Dictybase, Phenotypes)  1.98e-01           3  \n",
       "16  (GO, cellular_component)  2.21e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C17 selected: 36 with EID: 36\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.53\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 19 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.25e-13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.81e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.50e-08</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.03e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.66e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.06e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.75e-06</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.75e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.99e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.17e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.47e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Gene set                  Ontology       FDR  \\\n",
       "0                      cell cycle  (GO, biological_process)  8.25e-13   \n",
       "1          chromosome segregation  (GO, biological_process)  1.81e-09   \n",
       "2        mitotic nuclear division  (GO, biological_process)  5.50e-08   \n",
       "3         chromosome organization  (GO, biological_process)  3.03e-07   \n",
       "4              mitotic cell cycle  (GO, biological_process)  1.66e-06   \n",
       "5                 DNA replication          (KEGG, Pathways)  2.06e-06   \n",
       "6                      chromosome  (GO, cellular_component)  2.75e-06   \n",
       "7                   cell division  (GO, biological_process)  2.75e-06   \n",
       "8                    cytoskeleton  (GO, cellular_component)  5.99e-05   \n",
       "9   microtubule organizing center  (GO, cellular_component)  6.17e-03   \n",
       "10      cytoskeleton organization  (GO, biological_process)  6.47e-03   \n",
       "11   cytoskeletal protein binding  (GO, molecular_function)  1.61e-02   \n",
       "12          DNA metabolic process  (GO, biological_process)  1.74e-01   \n",
       "\n",
       "    N in query  \n",
       "0           13  \n",
       "1            6  \n",
       "2            5  \n",
       "3            7  \n",
       "4            7  \n",
       "5            5  \n",
       "6            6  \n",
       "7            7  \n",
       "8            7  \n",
       "9            3  \n",
       "10           5  \n",
       "11           4  \n",
       "12           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C18 selected: 30 with EID: 30\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.7\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 21 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.08e-15</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.61e-14</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.18e-13</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.10e-05</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.66e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.45e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.42e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.41e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.26e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.63e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Gene set                  Ontology       FDR  \\\n",
       "0    Ribosome biogenesis in eukaryotes          (KEGG, Pathways)  1.08e-15   \n",
       "1                            nucleolus  (GO, cellular_component)  8.61e-14   \n",
       "2                  ribosome biogenesis  (GO, biological_process)  6.18e-13   \n",
       "3   ribonucleoprotein complex assembly  (GO, biological_process)  1.10e-05   \n",
       "4                          RNA binding  (GO, molecular_function)  5.66e-04   \n",
       "5                         rRNA binding  (GO, molecular_function)  5.45e-03   \n",
       "6  protein-containing complex assembly  (GO, biological_process)  1.42e-02   \n",
       "7          cellular component assembly  (GO, biological_process)  4.41e-02   \n",
       "8                    helicase activity  (GO, molecular_function)  5.26e-02   \n",
       "9           methyltransferase activity  (GO, molecular_function)  5.63e-02   \n",
       "\n",
       "   N in query  \n",
       "0          12  \n",
       "1          11  \n",
       "2          10  \n",
       "3           4  \n",
       "4           6  \n",
       "5           2  \n",
       "6           4  \n",
       "7           4  \n",
       "8           2  \n",
       "9           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C19 selected: 23 with EID: 23\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 6 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.71e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene set                  Ontology       FDR  N in query\n",
       "0  kinase activity  (GO, molecular_function)  4.71e-02           3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C20 selected: 20 with EID: 20\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.0\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n"
     ]
    },
    {
     "ename": "UnboundLocalError",
     "evalue": "local variable 'query_in_enriched' referenced before assignment",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mUnboundLocalError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-3723d0592c9a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mregulons\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_table\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpathRegulons\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.tab'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mindex_col\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mgroup\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mregulons\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Cluster'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munique\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m     \u001b[0menriched\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgroup_diff_enrichment\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mregulons\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mgroup\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'C'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgroup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmin_overlap\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0muse_annotated_genes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0menriched\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m         \u001b[0menriched\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpathRegulons\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30_annotatedOnly_enrichment_cluster'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgroup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'.tsv'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-5-9c765ea4997f>\u001b[0m in \u001b[0;36mgroup_diff_enrichment\u001b[0;34m(data, group, padj, min_overlap, use_annotated_genes)\u001b[0m\n\u001b[1;32m     40\u001b[0m             \u001b[0mresult\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0menrichment_display\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     41\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Enrichment at FDR: '\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpadj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m' and min query - gene set overlap'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmin_overlap\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 42\u001b[0;31m     print('N query genes in displayed gene sets:',len(query_in_enriched),'out of', len(query_eids),\n\u001b[0m\u001b[1;32m     43\u001b[0m           'query genes used for enrichment calculation.')\n\u001b[1;32m     44\u001b[0m     \u001b[0mdisplay\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mUnboundLocalError\u001b[0m: local variable 'query_in_enriched' referenced before assignment"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2,use_annotated_genes=True)\n",
    "    if  save_enrichment and enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30_annotatedOnly_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Enrichment for mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 96 with EID: 96\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 30 out of 96 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.66e-05</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.27e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.47e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.18e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.52e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.52e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.52e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.65e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.40e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.77e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.77e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.77e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.81e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>phosphatase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.20e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.35e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.97e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.30e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.39e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.56e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.56e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.42e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.01e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>endosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Gene set                  Ontology  \\\n",
       "0            anatomical structure development  (GO, biological_process)   \n",
       "1                        cell differentiation  (GO, biological_process)   \n",
       "2    development arrests at loose mound stage   (Dictybase, Phenotypes)   \n",
       "3                         signal transduction  (GO, biological_process)   \n",
       "4               Glycosaminoglycan degradation          (KEGG, Pathways)   \n",
       "5                    increased slug migration   (Dictybase, Phenotypes)   \n",
       "6                        aberrant aggregation   (Dictybase, Phenotypes)   \n",
       "7                        extracellular region  (GO, cellular_component)   \n",
       "8                    decreased aggregate size   (Dictybase, Phenotypes)   \n",
       "9                decreased fruiting body size   (Dictybase, Phenotypes)   \n",
       "10  development arrests at tipped mound stage   (Dictybase, Phenotypes)   \n",
       "11                                 locomotion  (GO, biological_process)   \n",
       "12                              cell adhesion  (GO, biological_process)   \n",
       "13                            plasma membrane  (GO, cellular_component)   \n",
       "14         development arrests at mound stage   (Dictybase, Phenotypes)   \n",
       "15                     multiple tips in mound   (Dictybase, Phenotypes)   \n",
       "16          aberrant fruiting body morphology   (Dictybase, Phenotypes)   \n",
       "17                    delayed gene expression   (Dictybase, Phenotypes)   \n",
       "18                       phosphatase activity  (GO, molecular_function)   \n",
       "19                  abolished gene expression   (Dictybase, Phenotypes)   \n",
       "20                      abolished sporulation   (Dictybase, Phenotypes)   \n",
       "21                  cytoskeleton organization  (GO, biological_process)   \n",
       "22                         response to stress  (GO, biological_process)   \n",
       "23                     precocious development   (Dictybase, Phenotypes)   \n",
       "24                  decreased gene expression   (Dictybase, Phenotypes)   \n",
       "25                  decreased chemotaxis rate   (Dictybase, Phenotypes)   \n",
       "26              aberrant F-actin organization   (Dictybase, Phenotypes)   \n",
       "27                         peptidase activity  (GO, molecular_function)   \n",
       "28                   abolished slug migration   (Dictybase, Phenotypes)   \n",
       "29                                  wild type   (Dictybase, Phenotypes)   \n",
       "30                               cytoskeleton  (GO, cellular_component)   \n",
       "31                               reproduction  (GO, biological_process)   \n",
       "32                       aberrant development   (Dictybase, Phenotypes)   \n",
       "33                      decreased sporulation   (Dictybase, Phenotypes)   \n",
       "34                cellular component assembly  (GO, biological_process)   \n",
       "35                              cell motility  (GO, biological_process)   \n",
       "36                        delayed aggregation   (Dictybase, Phenotypes)   \n",
       "37                    decreased cell motility   (Dictybase, Phenotypes)   \n",
       "38                                   endosome  (GO, cellular_component)   \n",
       "39                      abolished aggregation   (Dictybase, Phenotypes)   \n",
       "40                       aberrant culmination   (Dictybase, Phenotypes)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   2.66e-05          16  \n",
       "1   8.27e-03           7  \n",
       "2   1.47e-02           3  \n",
       "3   1.47e-02          10  \n",
       "4   2.18e-02           2  \n",
       "5   2.52e-02           2  \n",
       "6   2.52e-02           4  \n",
       "7   2.52e-02           6  \n",
       "8   2.65e-02           3  \n",
       "9   3.40e-02           4  \n",
       "10  3.77e-02           2  \n",
       "11  3.77e-02           6  \n",
       "12  3.77e-02           4  \n",
       "13  3.81e-02           7  \n",
       "14  4.39e-02           3  \n",
       "15  4.39e-02           2  \n",
       "16  4.39e-02           4  \n",
       "17  4.39e-02           2  \n",
       "18  4.39e-02           3  \n",
       "19  6.61e-02           2  \n",
       "20  6.61e-02           2  \n",
       "21  7.20e-02           5  \n",
       "22  7.35e-02           7  \n",
       "23  7.97e-02           2  \n",
       "24  9.30e-02           3  \n",
       "25  9.39e-02           2  \n",
       "26  9.56e-02           2  \n",
       "27  9.56e-02           3  \n",
       "28  1.14e-01           2  \n",
       "29  1.22e-01           6  \n",
       "30  1.34e-01           4  \n",
       "31  1.34e-01           3  \n",
       "32  1.37e-01           2  \n",
       "33  1.42e-01           2  \n",
       "34  1.61e-01           4  \n",
       "35  1.74e-01           2  \n",
       "36  1.89e-01           2  \n",
       "37  2.01e-01           2  \n",
       "38  2.06e-01           2  \n",
       "39  2.12e-01           2  \n",
       "40  2.38e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C2 selected: 89 with EID: 88\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 88 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.63e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.90e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                               extracellular region   \n",
       "1  anatomical structure formation involved in mor...   \n",
       "2                      Starch and sucrose metabolism   \n",
       "3                               cell differentiation   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  2.63e-03           7  \n",
       "1  (GO, biological_process)  1.90e-02           4  \n",
       "2          (KEGG, Pathways)  1.02e-01           2  \n",
       "3  (GO, biological_process)  1.02e-01           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C3 selected: 87 with EID: 87\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 36 out of 87 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.49e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.49e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.47e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.70e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.51e-04</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.89e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.41e-04</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.98e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.27e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.25e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.82e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.40e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.80e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.02e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.62e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.62e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.90e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.85e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.85e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.82e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Glyoxylate and dicarboxylate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.23e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.23e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.13e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.45e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.24e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>lyase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.83e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                           cell wall   \n",
       "1                    external encapsulating structure   \n",
       "2                      aberrant spore coat morphology   \n",
       "3                cell wall organization or biogenesis   \n",
       "4   anatomical structure formation involved in mor...   \n",
       "5                    anatomical structure development   \n",
       "6                                extracellular region   \n",
       "7                         abolished cellulose binding   \n",
       "8                                cell differentiation   \n",
       "9                         aberrant cell-cell adhesion   \n",
       "10                          decreased spore viability   \n",
       "11                               extracellular matrix   \n",
       "12                                 Metabolic pathways   \n",
       "13              Biosynthesis of secondary metabolites   \n",
       "14                        Biosynthesis of antibiotics   \n",
       "15                             Glutathione metabolism   \n",
       "16                                 hydrolase activity   \n",
       "17                        cellular component assembly   \n",
       "18                      Starch and sucrose metabolism   \n",
       "19                          Citrate cycle (TCA cycle)   \n",
       "20                       cytoskeletal protein binding   \n",
       "21            Glyoxylate and dicarboxylate metabolism   \n",
       "22                                  catabolic process   \n",
       "23                                  Carbon metabolism   \n",
       "24                                    plasma membrane   \n",
       "25                       structural molecule activity   \n",
       "26                                            cytosol   \n",
       "27                              abolished culmination   \n",
       "28                                        DNA binding   \n",
       "29                                     lyase activity   \n",
       "30                 transmembrane transporter activity   \n",
       "31                     carbohydrate metabolic process   \n",
       "32                               aberrant culmination   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  7.49e-09           6  \n",
       "1   (GO, cellular_component)  7.49e-09           6  \n",
       "2    (Dictybase, Phenotypes)  3.47e-07           5  \n",
       "3   (GO, biological_process)  9.70e-07           5  \n",
       "4   (GO, biological_process)  1.47e-05           7  \n",
       "5   (GO, biological_process)  1.51e-04          13  \n",
       "6   (GO, cellular_component)  1.89e-04           8  \n",
       "7    (Dictybase, Phenotypes)  5.41e-04           2  \n",
       "8   (GO, biological_process)  6.98e-04           7  \n",
       "9    (Dictybase, Phenotypes)  4.27e-03           2  \n",
       "10   (Dictybase, Phenotypes)  6.25e-03           4  \n",
       "11  (GO, cellular_component)  9.82e-03           2  \n",
       "12          (KEGG, Pathways)  1.40e-02          12  \n",
       "13          (KEGG, Pathways)  3.80e-02           6  \n",
       "14          (KEGG, Pathways)  4.02e-02           5  \n",
       "15          (KEGG, Pathways)  5.62e-02           2  \n",
       "16  (GO, molecular_function)  5.62e-02           2  \n",
       "17  (GO, biological_process)  5.90e-02           5  \n",
       "18          (KEGG, Pathways)  6.85e-02           2  \n",
       "19          (KEGG, Pathways)  6.85e-02           2  \n",
       "20  (GO, molecular_function)  7.82e-02           4  \n",
       "21          (KEGG, Pathways)  8.23e-02           2  \n",
       "22  (GO, biological_process)  8.23e-02           5  \n",
       "23          (KEGG, Pathways)  9.13e-02           3  \n",
       "24  (GO, cellular_component)  9.45e-02           5  \n",
       "25  (GO, molecular_function)  1.18e-01           3  \n",
       "26  (GO, cellular_component)  1.24e-01           4  \n",
       "27   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "28  (GO, molecular_function)  1.56e-01           4  \n",
       "29  (GO, molecular_function)  1.69e-01           2  \n",
       "30  (GO, molecular_function)  1.83e-01           3  \n",
       "31  (GO, biological_process)  2.30e-01           2  \n",
       "32   (Dictybase, Phenotypes)  2.34e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C4 selected: 85 with EID: 85\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 7 out of 85 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.97e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                  Ontology       FDR  N in query\n",
       "0  extracellular region  (GO, cellular_component)  6.97e-03           7"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C5 selected: 73 with EID: 73\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 73 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene set                  Ontology       FDR  N in query\n",
       "0  cell-cell signaling  (GO, biological_process)  3.61e-02           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C6 selected: 72 with EID: 72\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 72 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.28e-07</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                  Ontology       FDR  N in query\n",
       "0  extracellular region  (GO, cellular_component)  6.28e-07          10"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C7 selected: 70 with EID: 70\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 70 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                  Ontology       FDR  N in query\n",
       "0  extracellular region  (GO, cellular_component)  1.54e-01           4\n",
       "1       kinase activity  (GO, molecular_function)  1.54e-01           6\n",
       "2         cell adhesion  (GO, biological_process)  2.27e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C8 selected: 65 with EID: 63\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 63 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.74e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.74e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.81e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.11e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.14e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>lipid binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                                          cell wall   \n",
       "1                   external encapsulating structure   \n",
       "2  anatomical structure formation involved in mor...   \n",
       "3                                 hydrolase activity   \n",
       "4                               cell differentiation   \n",
       "5                                      lipid binding   \n",
       "6                               extracellular region   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  1.74e-02           2  \n",
       "1  (GO, cellular_component)  1.74e-02           2  \n",
       "2  (GO, biological_process)  3.81e-02           3  \n",
       "3  (GO, molecular_function)  9.11e-02           2  \n",
       "4  (GO, biological_process)  1.14e-01           3  \n",
       "5  (GO, molecular_function)  1.14e-01           2  \n",
       "6  (GO, cellular_component)  1.20e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C9 selected: 57 with EID: 56\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 56 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.23e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                  Ontology       FDR  N in query\n",
       "0              cell adhesion  (GO, biological_process)  1.23e-02           4\n",
       "1  decreased spore viability   (Dictybase, Phenotypes)  1.80e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C10 selected: 54 with EID: 54\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 26 out of 54 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.61e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.61e-08</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.01e-06</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.52e-05</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.12e-04</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.76e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.07e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.03e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.42e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.03e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.06e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.06e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.28e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.51e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.73e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>vacuole</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                  Ontology  \\\n",
       "0                        Oxidative phosphorylation          (KEGG, Pathways)   \n",
       "1                                    mitochondrion  (GO, cellular_component)   \n",
       "2                               Metabolic pathways          (KEGG, Pathways)   \n",
       "3   generation of precursor metabolites and energy  (GO, biological_process)   \n",
       "4                                  protein folding  (GO, biological_process)   \n",
       "5                         unfolded protein binding  (GO, molecular_function)   \n",
       "6                          oxidoreductase activity  (GO, molecular_function)   \n",
       "7                                        Phagosome          (KEGG, Pathways)   \n",
       "8               transmembrane transporter activity  (GO, molecular_function)   \n",
       "9                          transmembrane transport  (GO, biological_process)   \n",
       "10                           Propanoate metabolism          (KEGG, Pathways)   \n",
       "11                                        Ribosome          (KEGG, Pathways)   \n",
       "12                       Citrate cycle (TCA cycle)          (KEGG, Pathways)   \n",
       "13                     Biosynthesis of antibiotics          (KEGG, Pathways)   \n",
       "14                               Carbon metabolism          (KEGG, Pathways)   \n",
       "15                                         cytosol  (GO, cellular_component)   \n",
       "16                                 ligase activity  (GO, molecular_function)   \n",
       "17                small molecule metabolic process  (GO, biological_process)   \n",
       "18           Biosynthesis of secondary metabolites          (KEGG, Pathways)   \n",
       "19                                         vacuole  (GO, cellular_component)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   7.61e-08           7  \n",
       "1   7.61e-08          11  \n",
       "2   9.01e-06          14  \n",
       "3   1.52e-05           6  \n",
       "4   2.12e-04           4  \n",
       "5   1.76e-03           3  \n",
       "6   2.07e-03           6  \n",
       "7   1.03e-02           3  \n",
       "8   2.42e-02           4  \n",
       "9   3.03e-02           4  \n",
       "10  3.06e-02           2  \n",
       "11  3.06e-02           3  \n",
       "12  3.06e-02           2  \n",
       "13  3.06e-02           4  \n",
       "14  3.28e-02           3  \n",
       "15  3.51e-02           4  \n",
       "16  1.23e-01           2  \n",
       "17  1.44e-01           4  \n",
       "18  1.73e-01           3  \n",
       "19  1.88e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C11 selected: 54 with EID: 54\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 44 out of 54 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.13e-86</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.68e-85</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.59e-76</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.25e-69</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.73e-62</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.33e-10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.82e-07</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                  Ontology       FDR  \\\n",
       "0                            Ribosome          (KEGG, Pathways)  1.13e-86   \n",
       "1                            ribosome  (GO, cellular_component)  4.68e-85   \n",
       "2  structural constituent of ribosome  (GO, molecular_function)  8.59e-76   \n",
       "3        structural molecule activity  (GO, molecular_function)  4.25e-69   \n",
       "4                         translation  (GO, biological_process)  2.73e-62   \n",
       "5                         RNA binding  (GO, molecular_function)  4.33e-10   \n",
       "6                        rRNA binding  (GO, molecular_function)  6.82e-07   \n",
       "\n",
       "   N in query  \n",
       "0          44  \n",
       "1          43  \n",
       "2          39  \n",
       "3          39  \n",
       "4          39  \n",
       "5          11  \n",
       "6           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C12 selected: 49 with EID: 49\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 49 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.88e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.85e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.29e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.07e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.65e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.74e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.78e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.00e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.25e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.55e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.55e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.55e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.55e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                cell differentiation   \n",
       "1   anatomical structure formation involved in mor...   \n",
       "2           DNA-binding transcription factor activity   \n",
       "3                                        reproduction   \n",
       "4                    anatomical structure development   \n",
       "5                           decreased gene expression   \n",
       "6                       Starch and sucrose metabolism   \n",
       "7                                         DNA binding   \n",
       "8                           decreased spore viability   \n",
       "9      generation of precursor metabolites and energy   \n",
       "10                     carbohydrate metabolic process   \n",
       "11                          cytoskeleton organization   \n",
       "12                                      cell adhesion   \n",
       "13                                    kinase activity   \n",
       "14                                signal transduction   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  1.88e-03           6  \n",
       "1   (GO, biological_process)  8.85e-03           4  \n",
       "2   (GO, molecular_function)  1.29e-02           3  \n",
       "3   (GO, biological_process)  2.07e-02           4  \n",
       "4   (GO, biological_process)  2.65e-02           7  \n",
       "5    (Dictybase, Phenotypes)  3.74e-02           3  \n",
       "6           (KEGG, Pathways)  4.78e-02           2  \n",
       "7   (GO, molecular_function)  6.00e-02           4  \n",
       "8    (Dictybase, Phenotypes)  7.25e-02           2  \n",
       "9   (GO, biological_process)  9.55e-02           2  \n",
       "10  (GO, biological_process)  9.55e-02           2  \n",
       "11  (GO, biological_process)  9.55e-02           3  \n",
       "12  (GO, biological_process)  9.55e-02           2  \n",
       "13  (GO, molecular_function)  2.15e-01           3  \n",
       "14  (GO, biological_process)  2.45e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C13 selected: 49 with EID: 49\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 42 out of 49 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.18e-65</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.24e-31</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.33e-29</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.25e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.25e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.13e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.68e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.78e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Gene set                  Ontology  \\\n",
       "0                                   Proteasome          (KEGG, Pathways)   \n",
       "1                           peptidase activity  (GO, molecular_function)   \n",
       "2                            catabolic process  (GO, biological_process)   \n",
       "3  Protein processing in endoplasmic reticulum          (KEGG, Pathways)   \n",
       "4                    enzyme regulator activity  (GO, molecular_function)   \n",
       "5                           response to stress  (GO, biological_process)   \n",
       "6                        endoplasmic reticulum  (GO, cellular_component)   \n",
       "7                              ATPase activity  (GO, molecular_function)   \n",
       "8                          cytoplasmic vesicle  (GO, cellular_component)   \n",
       "\n",
       "        FDR  N in query  \n",
       "0  1.18e-65          29  \n",
       "1  1.24e-31          21  \n",
       "2  4.33e-29          25  \n",
       "3  5.25e-02           3  \n",
       "4  5.25e-02           3  \n",
       "5  7.13e-02           5  \n",
       "6  7.68e-02           3  \n",
       "7  1.30e-01           3  \n",
       "8  1.78e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C14 selected: 43 with EID: 43\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 43 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.38e-10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.13e-06</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.13e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.13e-06</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.66e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.73e-05</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.54e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.69e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.37e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.54e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.66e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.84e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.58e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.39e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.29e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.29e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.31e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.05e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.78e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                  Ontology       FDR  \\\n",
       "0          cytoskeletal protein binding  (GO, molecular_function)  2.38e-10   \n",
       "1           cellular component assembly  (GO, biological_process)  8.13e-06   \n",
       "2   protein-containing complex assembly  (GO, biological_process)  8.13e-06   \n",
       "3             cytoskeleton organization  (GO, biological_process)  8.13e-06   \n",
       "4                          cytoskeleton  (GO, cellular_component)  3.66e-05   \n",
       "5                           Endocytosis          (KEGG, Pathways)  5.73e-05   \n",
       "6                    response to stress  (GO, biological_process)  1.54e-03   \n",
       "7                               cytosol  (GO, cellular_component)  4.69e-03   \n",
       "8                       plasma membrane  (GO, cellular_component)  1.36e-02   \n",
       "9               decreased cell motility   (Dictybase, Phenotypes)  1.37e-02   \n",
       "10                   cell morphogenesis  (GO, biological_process)  1.54e-02   \n",
       "11                  cytoplasmic vesicle  (GO, cellular_component)  2.66e-02   \n",
       "12             aberrant cell morphology   (Dictybase, Phenotypes)  3.84e-02   \n",
       "13     anatomical structure development  (GO, biological_process)  5.58e-02   \n",
       "14                   aberrant streaming   (Dictybase, Phenotypes)  7.39e-02   \n",
       "15                            wild type   (Dictybase, Phenotypes)  7.45e-02   \n",
       "16                abolished aggregation   (Dictybase, Phenotypes)  7.45e-02   \n",
       "17                 aberrant cytokinesis   (Dictybase, Phenotypes)  9.29e-02   \n",
       "18           vesicle-mediated transport  (GO, biological_process)  9.29e-02   \n",
       "19                   mitotic cell cycle  (GO, biological_process)  9.31e-02   \n",
       "20                         reproduction  (GO, biological_process)  1.03e-01   \n",
       "21                        cell division  (GO, biological_process)  1.05e-01   \n",
       "22                decreased growth rate   (Dictybase, Phenotypes)  1.33e-01   \n",
       "23              transmembrane transport  (GO, biological_process)  1.52e-01   \n",
       "24                           cell cycle  (GO, biological_process)  1.52e-01   \n",
       "25                      ATPase activity  (GO, molecular_function)  1.52e-01   \n",
       "26   transmembrane transporter activity  (GO, molecular_function)  1.52e-01   \n",
       "27                           locomotion  (GO, biological_process)  1.78e-01   \n",
       "\n",
       "    N in query  \n",
       "0           11  \n",
       "1            8  \n",
       "2            7  \n",
       "3            8  \n",
       "4            7  \n",
       "5            5  \n",
       "6            7  \n",
       "7            5  \n",
       "8            5  \n",
       "9            3  \n",
       "10           2  \n",
       "11           4  \n",
       "12           2  \n",
       "13           5  \n",
       "14           2  \n",
       "15           4  \n",
       "16           2  \n",
       "17           2  \n",
       "18           3  \n",
       "19           2  \n",
       "20           2  \n",
       "21           2  \n",
       "22           2  \n",
       "23           2  \n",
       "24           2  \n",
       "25           2  \n",
       "26           2  \n",
       "27           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C15 selected: 38 with EID: 38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 38 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.30e-14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.94e-10</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.84e-09</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.61e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.45e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.74e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.74e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.05e-07</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.47e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.17e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.65e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.21e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.99e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.51e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Gene set                  Ontology       FDR  \\\n",
       "0                      cell cycle  (GO, biological_process)  5.30e-14   \n",
       "1          chromosome segregation  (GO, biological_process)  1.94e-10   \n",
       "2        mitotic nuclear division  (GO, biological_process)  7.84e-09   \n",
       "3         chromosome organization  (GO, biological_process)  2.61e-08   \n",
       "4              mitotic cell cycle  (GO, biological_process)  1.45e-07   \n",
       "5                 DNA replication          (KEGG, Pathways)  2.74e-07   \n",
       "6                   cell division  (GO, biological_process)  2.74e-07   \n",
       "7                      chromosome  (GO, cellular_component)  3.05e-07   \n",
       "8                    cytoskeleton  (GO, cellular_component)  5.47e-06   \n",
       "9       cytoskeleton organization  (GO, biological_process)  1.17e-03   \n",
       "10  microtubule organizing center  (GO, cellular_component)  1.65e-03   \n",
       "11   cytoskeletal protein binding  (GO, molecular_function)  4.21e-03   \n",
       "12          DNA metabolic process  (GO, biological_process)  7.99e-02   \n",
       "13                ATPase activity  (GO, molecular_function)  1.51e-01   \n",
       "\n",
       "    N in query  \n",
       "0           13  \n",
       "1            6  \n",
       "2            5  \n",
       "3            7  \n",
       "4            7  \n",
       "5            5  \n",
       "6            7  \n",
       "7            6  \n",
       "8            7  \n",
       "9            5  \n",
       "10           3  \n",
       "11           4  \n",
       "12           2  \n",
       "13           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C16 selected: 35 with EID: 35\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 35 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.12e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.43e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.78e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell death</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.37e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.51e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.94e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.84e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.79e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0   aberrant CRAC localization to the plasma membrane   \n",
       "1           development arrests at tipped mound stage   \n",
       "2   anatomical structure formation involved in mor...   \n",
       "3                                          cell death   \n",
       "4                                     RNA degradation   \n",
       "5                                aberrant aggregation   \n",
       "6                        decreased chemotaxis to cAMP   \n",
       "7                    anatomical structure development   \n",
       "8                                cell differentiation   \n",
       "9                                     kinase activity   \n",
       "10                                         locomotion   \n",
       "11                       decreased fruiting body size   \n",
       "12                  aberrant fruiting body morphology   \n",
       "13                                          wild type   \n",
       "14                                       cytoskeleton   \n",
       "15                                        DNA binding   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0    (Dictybase, Phenotypes)  1.12e-02           2  \n",
       "1    (Dictybase, Phenotypes)  2.43e-02           2  \n",
       "2   (GO, biological_process)  3.78e-02           3  \n",
       "3   (GO, biological_process)  3.89e-02           2  \n",
       "4           (KEGG, Pathways)  4.37e-02           2  \n",
       "5    (Dictybase, Phenotypes)  4.37e-02           2  \n",
       "6    (Dictybase, Phenotypes)  4.37e-02           2  \n",
       "7   (GO, biological_process)  4.37e-02           5  \n",
       "8   (GO, biological_process)  4.37e-02           3  \n",
       "9   (GO, molecular_function)  4.37e-02           4  \n",
       "10  (GO, biological_process)  4.51e-02           3  \n",
       "11   (Dictybase, Phenotypes)  4.94e-02           2  \n",
       "12   (Dictybase, Phenotypes)  6.84e-02           2  \n",
       "13   (Dictybase, Phenotypes)  1.12e-01           3  \n",
       "14  (GO, cellular_component)  1.45e-01           2  \n",
       "15  (GO, molecular_function)  1.79e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C17 selected: 31 with EID: 31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C18 selected: 31 with EID: 31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.70e-19</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.74e-17</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.15e-15</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.06e-07</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.36e-05</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.46e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.82e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.37e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                  Ontology       FDR  \\\n",
       "0     Ribosome biogenesis in eukaryotes          (KEGG, Pathways)  7.70e-19   \n",
       "1                             nucleolus  (GO, cellular_component)  9.74e-17   \n",
       "2                   ribosome biogenesis  (GO, biological_process)  1.15e-15   \n",
       "3    ribonucleoprotein complex assembly  (GO, biological_process)  8.06e-07   \n",
       "4                           RNA binding  (GO, molecular_function)  1.36e-05   \n",
       "5   protein-containing complex assembly  (GO, biological_process)  1.41e-03   \n",
       "6                          rRNA binding  (GO, molecular_function)  1.41e-03   \n",
       "7           cellular component assembly  (GO, biological_process)  4.46e-03   \n",
       "8                     helicase activity  (GO, molecular_function)  1.68e-02   \n",
       "9            methyltransferase activity  (GO, molecular_function)  1.82e-02   \n",
       "10                      ATPase activity  (GO, molecular_function)  9.37e-02   \n",
       "\n",
       "    N in query  \n",
       "0           12  \n",
       "1           11  \n",
       "2           10  \n",
       "3            4  \n",
       "4            6  \n",
       "5            4  \n",
       "6            2  \n",
       "7            4  \n",
       "8            2  \n",
       "9            2  \n",
       "10           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C19 selected: 20 with EID: 20\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 20 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.29e-33</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.70e-33</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.12e-28</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.46e-26</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.80e-22</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.78e-06</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.82e-05</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.81e-05</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.02e-04</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.17e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Gene set                  Ontology       FDR  \\\n",
       "0                             ribosome  (GO, cellular_component)  4.29e-33   \n",
       "1                             Ribosome          (KEGG, Pathways)  4.70e-33   \n",
       "2   structural constituent of ribosome  (GO, molecular_function)  1.12e-28   \n",
       "3         structural molecule activity  (GO, molecular_function)  1.46e-26   \n",
       "4                          translation  (GO, biological_process)  5.80e-22   \n",
       "5                         rRNA binding  (GO, molecular_function)  3.78e-06   \n",
       "6                          RNA binding  (GO, molecular_function)  1.82e-05   \n",
       "7                              cytosol  (GO, cellular_component)  3.81e-05   \n",
       "8                  ribosome biogenesis  (GO, biological_process)  3.02e-04   \n",
       "9                  delayed development   (Dictybase, Phenotypes)  1.17e-02   \n",
       "10    anatomical structure development  (GO, biological_process)  2.14e-01   \n",
       "\n",
       "    N in query  \n",
       "0           17  \n",
       "1           17  \n",
       "2           15  \n",
       "3           15  \n",
       "4           14  \n",
       "5            3  \n",
       "6            5  \n",
       "7            5  \n",
       "8            3  \n",
       "9            2  \n",
       "10           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C20 selected: 1 with EID: 1\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 1 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2)\n",
    "    if  save_enrichment and enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Using only genes annotated with at least one gene set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 96 with EID: 96\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.39\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 30 out of 37 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.78e-05</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.02e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.13e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.13e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.13e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.18e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.18e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.36e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.36e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.04e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.04e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>phosphatase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.76e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.76e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.88e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.07e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.33e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.33e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.95e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.96e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.97e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.97e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Gene set                  Ontology  \\\n",
       "0            anatomical structure development  (GO, biological_process)   \n",
       "1                        cell differentiation  (GO, biological_process)   \n",
       "2               Glycosaminoglycan degradation          (KEGG, Pathways)   \n",
       "3    development arrests at loose mound stage   (Dictybase, Phenotypes)   \n",
       "4                         signal transduction  (GO, biological_process)   \n",
       "5                    increased slug migration   (Dictybase, Phenotypes)   \n",
       "6                        aberrant aggregation   (Dictybase, Phenotypes)   \n",
       "7                    decreased aggregate size   (Dictybase, Phenotypes)   \n",
       "8                        extracellular region  (GO, cellular_component)   \n",
       "9                decreased fruiting body size   (Dictybase, Phenotypes)   \n",
       "10  development arrests at tipped mound stage   (Dictybase, Phenotypes)   \n",
       "11                            plasma membrane  (GO, cellular_component)   \n",
       "12                                 locomotion  (GO, biological_process)   \n",
       "13                              cell adhesion  (GO, biological_process)   \n",
       "14                       phosphatase activity  (GO, molecular_function)   \n",
       "15                     multiple tips in mound   (Dictybase, Phenotypes)   \n",
       "16                    delayed gene expression   (Dictybase, Phenotypes)   \n",
       "17          aberrant fruiting body morphology   (Dictybase, Phenotypes)   \n",
       "18         development arrests at mound stage   (Dictybase, Phenotypes)   \n",
       "19                  abolished gene expression   (Dictybase, Phenotypes)   \n",
       "20                      abolished sporulation   (Dictybase, Phenotypes)   \n",
       "21                     precocious development   (Dictybase, Phenotypes)   \n",
       "22                  cytoskeleton organization  (GO, biological_process)   \n",
       "23                  decreased gene expression   (Dictybase, Phenotypes)   \n",
       "24                  decreased chemotaxis rate   (Dictybase, Phenotypes)   \n",
       "25              aberrant F-actin organization   (Dictybase, Phenotypes)   \n",
       "26                         response to stress  (GO, biological_process)   \n",
       "27                         peptidase activity  (GO, molecular_function)   \n",
       "28                   abolished slug migration   (Dictybase, Phenotypes)   \n",
       "29                       aberrant development   (Dictybase, Phenotypes)   \n",
       "30                               reproduction  (GO, biological_process)   \n",
       "31                      decreased sporulation   (Dictybase, Phenotypes)   \n",
       "32                                  wild type   (Dictybase, Phenotypes)   \n",
       "33                               cytoskeleton  (GO, cellular_component)   \n",
       "34                cellular component assembly  (GO, biological_process)   \n",
       "35                              cell motility  (GO, biological_process)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   6.78e-05          16  \n",
       "1   2.02e-02           7  \n",
       "2   3.13e-02           2  \n",
       "3   3.13e-02           3  \n",
       "4   3.13e-02          10  \n",
       "5   4.18e-02           2  \n",
       "6   4.18e-02           4  \n",
       "7   4.36e-02           3  \n",
       "8   4.36e-02           6  \n",
       "9   6.04e-02           4  \n",
       "10  6.04e-02           2  \n",
       "11  6.15e-02           7  \n",
       "12  6.15e-02           6  \n",
       "13  6.15e-02           4  \n",
       "14  6.15e-02           3  \n",
       "15  6.76e-02           2  \n",
       "16  6.76e-02           2  \n",
       "17  6.88e-02           4  \n",
       "18  7.07e-02           3  \n",
       "19  9.33e-02           2  \n",
       "20  9.33e-02           2  \n",
       "21  1.09e-01           2  \n",
       "22  1.18e-01           5  \n",
       "23  1.26e-01           3  \n",
       "24  1.26e-01           2  \n",
       "25  1.26e-01           2  \n",
       "26  1.26e-01           7  \n",
       "27  1.40e-01           3  \n",
       "28  1.58e-01           2  \n",
       "29  1.93e-01           2  \n",
       "30  1.95e-01           3  \n",
       "31  1.96e-01           2  \n",
       "32  1.97e-01           6  \n",
       "33  1.97e-01           4  \n",
       "34  2.26e-01           4  \n",
       "35  2.26e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C2 selected: 89 with EID: 88\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.50e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.90e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.42e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.25e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                               extracellular region   \n",
       "1  anatomical structure formation involved in mor...   \n",
       "2                               cell differentiation   \n",
       "3                      Starch and sucrose metabolism   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  6.50e-05           7  \n",
       "1  (GO, biological_process)  2.90e-03           4  \n",
       "2  (GO, biological_process)  2.42e-02           4  \n",
       "3          (KEGG, Pathways)  4.25e-02           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C3 selected: 87 with EID: 87\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 39 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.58e-08</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.58e-08</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.63e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.53e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.05e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.52e-03</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.85e-03</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.48e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.28e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.90e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.90e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Glyoxylate and dicarboxylate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.82e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.22e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                           cell wall   \n",
       "1                    external encapsulating structure   \n",
       "2                      aberrant spore coat morphology   \n",
       "3                cell wall organization or biogenesis   \n",
       "4   anatomical structure formation involved in mor...   \n",
       "5                         abolished cellulose binding   \n",
       "6                                extracellular region   \n",
       "7                    anatomical structure development   \n",
       "8                                cell differentiation   \n",
       "9                         aberrant cell-cell adhesion   \n",
       "10                          decreased spore viability   \n",
       "11                               extracellular matrix   \n",
       "12                                 Metabolic pathways   \n",
       "13                             Glutathione metabolism   \n",
       "14                                 hydrolase activity   \n",
       "15              Biosynthesis of secondary metabolites   \n",
       "16                        Biosynthesis of antibiotics   \n",
       "17                          Citrate cycle (TCA cycle)   \n",
       "18                      Starch and sucrose metabolism   \n",
       "19            Glyoxylate and dicarboxylate metabolism   \n",
       "20                        cellular component assembly   \n",
       "21                       cytoskeletal protein binding   \n",
       "22                                  Carbon metabolism   \n",
       "23                                  catabolic process   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  4.58e-08           6  \n",
       "1   (GO, cellular_component)  4.58e-08           6  \n",
       "2    (Dictybase, Phenotypes)  1.63e-06           5  \n",
       "3   (GO, biological_process)  4.53e-06           5  \n",
       "4   (GO, biological_process)  1.05e-04           7  \n",
       "5    (Dictybase, Phenotypes)  1.41e-03           2  \n",
       "6   (GO, cellular_component)  1.52e-03           8  \n",
       "7   (GO, biological_process)  1.85e-03          13  \n",
       "8   (GO, biological_process)  4.48e-03           7  \n",
       "9    (Dictybase, Phenotypes)  8.28e-03           2  \n",
       "10   (Dictybase, Phenotypes)  1.90e-02           4  \n",
       "11  (GO, cellular_component)  1.90e-02           2  \n",
       "12          (KEGG, Pathways)  1.02e-01          12  \n",
       "13          (KEGG, Pathways)  1.10e-01           2  \n",
       "14  (GO, molecular_function)  1.10e-01           2  \n",
       "15          (KEGG, Pathways)  1.12e-01           6  \n",
       "16          (KEGG, Pathways)  1.12e-01           5  \n",
       "17          (KEGG, Pathways)  1.23e-01           2  \n",
       "18          (KEGG, Pathways)  1.26e-01           2  \n",
       "19          (KEGG, Pathways)  1.47e-01           2  \n",
       "20  (GO, biological_process)  1.47e-01           5  \n",
       "21  (GO, molecular_function)  1.82e-01           4  \n",
       "22          (KEGG, Pathways)  2.05e-01           3  \n",
       "23  (GO, biological_process)  2.22e-01           5  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C4 selected: 85 with EID: 85\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.24\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 7 out of 20 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.88e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                  Ontology       FDR  N in query\n",
       "0  extracellular region  (GO, cellular_component)  4.88e-04           7"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C5 selected: 73 with EID: 73\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.11\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 5 out of 8 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.91e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.03e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                  Ontology       FDR  \\\n",
       "0               cell-cell signaling  (GO, biological_process)  3.91e-03   \n",
       "1  anatomical structure development  (GO, biological_process)  5.03e-02   \n",
       "2              extracellular region  (GO, cellular_component)  1.34e-01   \n",
       "3                         wild type   (Dictybase, Phenotypes)  2.33e-01   \n",
       "4               signal transduction  (GO, biological_process)  2.33e-01   \n",
       "\n",
       "   N in query  \n",
       "0           2  \n",
       "1           4  \n",
       "2           2  \n",
       "3           2  \n",
       "4           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C6 selected: 72 with EID: 72\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.24\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 17 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.29e-09</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                  Ontology       FDR  N in query\n",
       "0  extracellular region  (GO, cellular_component)  3.29e-09          10\n",
       "1             wild type   (Dictybase, Phenotypes)  1.54e-01           5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C7 selected: 70 with EID: 70\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.17e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.17e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                  Ontology       FDR  N in query\n",
       "0  extracellular region  (GO, cellular_component)  6.17e-02           4\n",
       "1       kinase activity  (GO, molecular_function)  6.17e-02           6\n",
       "2         cell adhesion  (GO, biological_process)  1.81e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C8 selected: 65 with EID: 63\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.14\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 9 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.97e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.97e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.97e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.34e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.34e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.34e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>lipid binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.79e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.66e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>protein transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                                          cell wall   \n",
       "1                   external encapsulating structure   \n",
       "2  anatomical structure formation involved in mor...   \n",
       "3                               extracellular region   \n",
       "4                               cell differentiation   \n",
       "5                                 hydrolase activity   \n",
       "6                                      lipid binding   \n",
       "7                   anatomical structure development   \n",
       "8                                  protein transport   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  2.97e-03           2  \n",
       "1  (GO, cellular_component)  2.97e-03           2  \n",
       "2  (GO, biological_process)  2.97e-03           3  \n",
       "3  (GO, cellular_component)  1.34e-02           3  \n",
       "4  (GO, biological_process)  1.34e-02           3  \n",
       "5  (GO, molecular_function)  1.34e-02           2  \n",
       "6  (GO, molecular_function)  1.79e-02           2  \n",
       "7  (GO, biological_process)  2.66e-02           4  \n",
       "8  (GO, biological_process)  1.15e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C9 selected: 57 with EID: 56\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 11 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.25e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                  Ontology       FDR  N in query\n",
       "0              cell adhesion  (GO, biological_process)  1.25e-03           4\n",
       "1  decreased spore viability   (Dictybase, Phenotypes)  1.12e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C10 selected: 54 with EID: 54\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.57\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 26 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.18e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.39e-05</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.71e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.90e-03</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.90e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.33e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.62e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.14e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.24e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.36e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                  Ontology  \\\n",
       "0                        Oxidative phosphorylation          (KEGG, Pathways)   \n",
       "1                                    mitochondrion  (GO, cellular_component)   \n",
       "2   generation of precursor metabolites and energy  (GO, biological_process)   \n",
       "3                               Metabolic pathways          (KEGG, Pathways)   \n",
       "4                                  protein folding  (GO, biological_process)   \n",
       "5                         unfolded protein binding  (GO, molecular_function)   \n",
       "6                          oxidoreductase activity  (GO, molecular_function)   \n",
       "7                                        Phagosome          (KEGG, Pathways)   \n",
       "8                        Citrate cycle (TCA cycle)          (KEGG, Pathways)   \n",
       "9                            Propanoate metabolism          (KEGG, Pathways)   \n",
       "10              transmembrane transporter activity  (GO, molecular_function)   \n",
       "11                                        Ribosome          (KEGG, Pathways)   \n",
       "12                         transmembrane transport  (GO, biological_process)   \n",
       "13                               Carbon metabolism          (KEGG, Pathways)   \n",
       "14                     Biosynthesis of antibiotics          (KEGG, Pathways)   \n",
       "15                                         cytosol  (GO, cellular_component)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   6.18e-06           7  \n",
       "1   1.39e-05          11  \n",
       "2   4.71e-04           6  \n",
       "3   1.90e-03          14  \n",
       "4   1.90e-03           4  \n",
       "5   9.33e-03           3  \n",
       "6   3.62e-02           6  \n",
       "7   5.14e-02           3  \n",
       "8   1.24e-01           2  \n",
       "9   1.26e-01           2  \n",
       "10  1.36e-01           4  \n",
       "11  1.55e-01           3  \n",
       "12  1.55e-01           4  \n",
       "13  1.58e-01           3  \n",
       "14  1.58e-01           4  \n",
       "15  2.04e-01           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C11 selected: 54 with EID: 54\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.89\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 44 out of 48 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.62e-70</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.18e-68</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.22e-60</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.02e-53</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.23e-46</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.25e-05</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.87e-05</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                  Ontology       FDR  \\\n",
       "0                            Ribosome          (KEGG, Pathways)  7.62e-70   \n",
       "1                            ribosome  (GO, cellular_component)  2.18e-68   \n",
       "2  structural constituent of ribosome  (GO, molecular_function)  4.22e-60   \n",
       "3        structural molecule activity  (GO, molecular_function)  2.02e-53   \n",
       "4                         translation  (GO, biological_process)  1.23e-46   \n",
       "5                         RNA binding  (GO, molecular_function)  1.25e-05   \n",
       "6                        rRNA binding  (GO, molecular_function)  3.87e-05   \n",
       "\n",
       "   N in query  \n",
       "0          44  \n",
       "1          43  \n",
       "2          39  \n",
       "3          39  \n",
       "4          39  \n",
       "5          11  \n",
       "6           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C12 selected: 49 with EID: 49\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.42e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.80e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.00e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.11e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.11e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.98e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.25e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.28e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.97e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.64e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.64e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.99e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.10e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                cell differentiation   \n",
       "1   anatomical structure formation involved in mor...   \n",
       "2           DNA-binding transcription factor activity   \n",
       "3                    anatomical structure development   \n",
       "4                                        reproduction   \n",
       "5                           decreased gene expression   \n",
       "6                       Starch and sucrose metabolism   \n",
       "7                                         DNA binding   \n",
       "8                           decreased spore viability   \n",
       "9                           cytoskeleton organization   \n",
       "10     generation of precursor metabolites and energy   \n",
       "11                                      cell adhesion   \n",
       "12                     carbohydrate metabolic process   \n",
       "13                                    kinase activity   \n",
       "14                                signal transduction   \n",
       "15                                         locomotion   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  7.42e-04           6  \n",
       "1   (GO, biological_process)  5.80e-03           4  \n",
       "2   (GO, molecular_function)  1.00e-02           3  \n",
       "3   (GO, biological_process)  1.11e-02           7  \n",
       "4   (GO, biological_process)  1.11e-02           4  \n",
       "5    (Dictybase, Phenotypes)  2.98e-02           3  \n",
       "6           (KEGG, Pathways)  4.25e-02           2  \n",
       "7   (GO, molecular_function)  5.28e-02           4  \n",
       "8    (Dictybase, Phenotypes)  6.97e-02           2  \n",
       "9   (GO, biological_process)  8.64e-02           3  \n",
       "10  (GO, biological_process)  8.64e-02           2  \n",
       "11  (GO, biological_process)  8.99e-02           2  \n",
       "12  (GO, biological_process)  9.10e-02           2  \n",
       "13  (GO, molecular_function)  1.89e-01           3  \n",
       "14  (GO, biological_process)  2.17e-01           3  \n",
       "15  (GO, biological_process)  2.40e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C13 selected: 49 with EID: 49\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.94\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 35 out of 46 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.94e-52</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.64e-22</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.21e-18</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene set                  Ontology       FDR  N in query\n",
       "0          Proteasome          (KEGG, Pathways)  1.94e-52          29\n",
       "1  peptidase activity  (GO, molecular_function)  4.64e-22          21\n",
       "2   catabolic process  (GO, biological_process)  6.21e-18          25"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C14 selected: 43 with EID: 43\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.51\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.28e-09</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.28e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.28e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.28e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.44e-04</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.44e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.82e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.16e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.62e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.84e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.98e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.19e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.55e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                  Ontology       FDR  \\\n",
       "0          cytoskeletal protein binding  (GO, molecular_function)  8.28e-09   \n",
       "1           cellular component assembly  (GO, biological_process)  1.28e-04   \n",
       "2   protein-containing complex assembly  (GO, biological_process)  1.28e-04   \n",
       "3             cytoskeleton organization  (GO, biological_process)  1.28e-04   \n",
       "4                           Endocytosis          (KEGG, Pathways)  4.44e-04   \n",
       "5                          cytoskeleton  (GO, cellular_component)  4.44e-04   \n",
       "6                    response to stress  (GO, biological_process)  1.82e-02   \n",
       "7                               cytosol  (GO, cellular_component)  3.16e-02   \n",
       "8                    cell morphogenesis  (GO, biological_process)  4.62e-02   \n",
       "9               decreased cell motility   (Dictybase, Phenotypes)  4.84e-02   \n",
       "10                      plasma membrane  (GO, cellular_component)  6.98e-02   \n",
       "11             aberrant cell morphology   (Dictybase, Phenotypes)  8.19e-02   \n",
       "12                  cytoplasmic vesicle  (GO, cellular_component)  8.55e-02   \n",
       "13                   aberrant streaming   (Dictybase, Phenotypes)  1.55e-01   \n",
       "14                abolished aggregation   (Dictybase, Phenotypes)  1.71e-01   \n",
       "15     anatomical structure development  (GO, biological_process)  1.91e-01   \n",
       "16                 aberrant cytokinesis   (Dictybase, Phenotypes)  2.09e-01   \n",
       "17                   mitotic cell cycle  (GO, biological_process)  2.09e-01   \n",
       "18                            wild type   (Dictybase, Phenotypes)  2.15e-01   \n",
       "19           vesicle-mediated transport  (GO, biological_process)  2.26e-01   \n",
       "20                         reproduction  (GO, biological_process)  2.26e-01   \n",
       "21                        cell division  (GO, biological_process)  2.26e-01   \n",
       "\n",
       "    N in query  \n",
       "0           11  \n",
       "1            8  \n",
       "2            7  \n",
       "3            8  \n",
       "4            5  \n",
       "5            7  \n",
       "6            7  \n",
       "7            5  \n",
       "8            2  \n",
       "9            3  \n",
       "10           5  \n",
       "11           2  \n",
       "12           4  \n",
       "13           2  \n",
       "14           2  \n",
       "15           5  \n",
       "16           2  \n",
       "17           2  \n",
       "18           4  \n",
       "19           3  \n",
       "20           2  \n",
       "21           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C15 selected: 38 with EID: 38\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.5\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 19 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.25e-13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.81e-09</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.50e-08</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.03e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.66e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.06e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.75e-06</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.75e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.99e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.17e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.47e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Gene set                  Ontology       FDR  \\\n",
       "0                      cell cycle  (GO, biological_process)  8.25e-13   \n",
       "1          chromosome segregation  (GO, biological_process)  1.81e-09   \n",
       "2        mitotic nuclear division  (GO, biological_process)  5.50e-08   \n",
       "3         chromosome organization  (GO, biological_process)  3.03e-07   \n",
       "4              mitotic cell cycle  (GO, biological_process)  1.66e-06   \n",
       "5                 DNA replication          (KEGG, Pathways)  2.06e-06   \n",
       "6                      chromosome  (GO, cellular_component)  2.75e-06   \n",
       "7                   cell division  (GO, biological_process)  2.75e-06   \n",
       "8                    cytoskeleton  (GO, cellular_component)  5.99e-05   \n",
       "9   microtubule organizing center  (GO, cellular_component)  6.17e-03   \n",
       "10      cytoskeleton organization  (GO, biological_process)  6.47e-03   \n",
       "11   cytoskeletal protein binding  (GO, molecular_function)  1.61e-02   \n",
       "12          DNA metabolic process  (GO, biological_process)  1.74e-01   \n",
       "\n",
       "    N in query  \n",
       "0           13  \n",
       "1            6  \n",
       "2            5  \n",
       "3            7  \n",
       "4            7  \n",
       "5            5  \n",
       "6            6  \n",
       "7            7  \n",
       "8            7  \n",
       "9            3  \n",
       "10           5  \n",
       "11           4  \n",
       "12           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C16 selected: 35 with EID: 35\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.66e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell death</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.88e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.88e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.60e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.93e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.93e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.01e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.72e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.44e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0   aberrant CRAC localization to the plasma membrane   \n",
       "1           development arrests at tipped mound stage   \n",
       "2                                          cell death   \n",
       "3   anatomical structure formation involved in mor...   \n",
       "4                                     RNA degradation   \n",
       "5                                aberrant aggregation   \n",
       "6                        decreased chemotaxis to cAMP   \n",
       "7                    anatomical structure development   \n",
       "8                                cell differentiation   \n",
       "9                                     kinase activity   \n",
       "10                       decreased fruiting body size   \n",
       "11                                         locomotion   \n",
       "12                  aberrant fruiting body morphology   \n",
       "13                                          wild type   \n",
       "14                                       cytoskeleton   \n",
       "15                                        DNA binding   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0    (Dictybase, Phenotypes)  1.68e-02           2  \n",
       "1    (Dictybase, Phenotypes)  3.66e-02           2  \n",
       "2   (GO, biological_process)  4.88e-02           2  \n",
       "3   (GO, biological_process)  4.88e-02           3  \n",
       "4           (KEGG, Pathways)  5.98e-02           2  \n",
       "5    (Dictybase, Phenotypes)  5.98e-02           2  \n",
       "6    (Dictybase, Phenotypes)  5.98e-02           2  \n",
       "7   (GO, biological_process)  5.98e-02           5  \n",
       "8   (GO, biological_process)  5.98e-02           3  \n",
       "9   (GO, molecular_function)  6.60e-02           4  \n",
       "10   (Dictybase, Phenotypes)  6.93e-02           2  \n",
       "11  (GO, biological_process)  6.93e-02           3  \n",
       "12   (Dictybase, Phenotypes)  9.01e-02           2  \n",
       "13   (Dictybase, Phenotypes)  1.72e-01           3  \n",
       "14  (GO, cellular_component)  2.00e-01           2  \n",
       "15  (GO, molecular_function)  2.44e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C17 selected: 31 with EID: 31\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 8 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Gene set                  Ontology       FDR  N in query\n",
       "0  oxidoreductase activity  (GO, molecular_function)  2.31e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C18 selected: 31 with EID: 31\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.71\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.09e-15</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.23e-13</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.47e-12</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.76e-05</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.85e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.86e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.21e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.78e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.54e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.06e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Gene set                  Ontology       FDR  \\\n",
       "0    Ribosome biogenesis in eukaryotes          (KEGG, Pathways)  3.09e-15   \n",
       "1                            nucleolus  (GO, cellular_component)  2.23e-13   \n",
       "2                  ribosome biogenesis  (GO, biological_process)  1.47e-12   \n",
       "3   ribonucleoprotein complex assembly  (GO, biological_process)  1.76e-05   \n",
       "4                          RNA binding  (GO, molecular_function)  9.85e-04   \n",
       "5                         rRNA binding  (GO, molecular_function)  7.86e-03   \n",
       "6  protein-containing complex assembly  (GO, biological_process)  2.21e-02   \n",
       "7          cellular component assembly  (GO, biological_process)  6.78e-02   \n",
       "8                    helicase activity  (GO, molecular_function)  7.54e-02   \n",
       "9           methyltransferase activity  (GO, molecular_function)  8.06e-02   \n",
       "\n",
       "   N in query  \n",
       "0          12  \n",
       "1          11  \n",
       "2          10  \n",
       "3           4  \n",
       "4           6  \n",
       "5           2  \n",
       "6           4  \n",
       "7           4  \n",
       "8           2  \n",
       "9           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C19 selected: 20 with EID: 20\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.9\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.92e-26</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.10e-26</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.65e-22</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.13e-20</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.87e-16</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.21e-05</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.30e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.53e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.00e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.30e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                  Ontology       FDR  \\\n",
       "0                            ribosome  (GO, cellular_component)  1.92e-26   \n",
       "1                            Ribosome          (KEGG, Pathways)  2.10e-26   \n",
       "2  structural constituent of ribosome  (GO, molecular_function)  1.65e-22   \n",
       "3        structural molecule activity  (GO, molecular_function)  2.13e-20   \n",
       "4                         translation  (GO, biological_process)  3.87e-16   \n",
       "5                        rRNA binding  (GO, molecular_function)  8.21e-05   \n",
       "6                         RNA binding  (GO, molecular_function)  2.30e-03   \n",
       "7                             cytosol  (GO, cellular_component)  4.53e-03   \n",
       "8                 ribosome biogenesis  (GO, biological_process)  6.00e-03   \n",
       "9                 delayed development   (Dictybase, Phenotypes)  8.30e-02   \n",
       "\n",
       "   N in query  \n",
       "0          17  \n",
       "1          17  \n",
       "2          15  \n",
       "3          15  \n",
       "4          14  \n",
       "5           3  \n",
       "6           5  \n",
       "7           5  \n",
       "8           3  \n",
       "9           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C20 selected: 1 with EID: 1\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.0\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n"
     ]
    },
    {
     "ename": "UnboundLocalError",
     "evalue": "local variable 'query_in_enriched' referenced before assignment",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mUnboundLocalError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-e722b600c3e7>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mregulons\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_table\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpathRegulons\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mindex_col\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mgroup\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mregulons\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Cluster'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munique\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m     \u001b[0menriched\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgroup_diff_enrichment\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mregulons\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mgroup\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'C'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgroup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmin_overlap\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0muse_annotated_genes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0menriched\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m         \u001b[0menriched\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpathRegulons\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'enrichment/mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgroup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'.tsv'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-5-9c765ea4997f>\u001b[0m in \u001b[0;36mgroup_diff_enrichment\u001b[0;34m(data, group, padj, min_overlap, use_annotated_genes)\u001b[0m\n\u001b[1;32m     40\u001b[0m             \u001b[0mresult\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0menrichment_display\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     41\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Enrichment at FDR: '\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpadj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m' and min query - gene set overlap'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmin_overlap\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 42\u001b[0;31m     print('N query genes in displayed gene sets:',len(query_in_enriched),'out of', len(query_eids),\n\u001b[0m\u001b[1;32m     43\u001b[0m           'query genes used for enrichment calculation.')\n\u001b[1;32m     44\u001b[0m     \u001b[0mdisplay\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mUnboundLocalError\u001b[0m: local variable 'query_in_enriched' referenced before assignment"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2,use_annotated_genes=True)\n",
    "    if save_enrichment and enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
