{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Enrichment of regulons from Regulons_by_strain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from importlib import reload\n",
    "\n",
    "module_path = os.path.abspath(os.path.join('..'))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.insert(0,module_path)\n",
    "\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "pd.set_option('max_colwidth', 70)\n",
    "pd.set_option('display.max_rows', 500)\n",
    "import string\n",
    "import numpy as np\n",
    "\n",
    "from orangecontrib.bioinformatics.geneset.__init__ import list_all\n",
    "\n",
    "import jupyter_functions as jf\n",
    "reload(jf)\n",
    "import jupyter_functions as jf\n",
    "from functionsDENet import loadPickle, savePickle\n",
    "import stages_DE.enrichment_library as enr\n",
    "reload(enr)\n",
    "import stages_DE.enrichment_library as enr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Path for expression data (mergedGenes_RPKUM.tsv)\n",
    "dataPath='/home/karin/Documents/timeTrajectories/data/RPKUM/combined/'\n",
    "#Path for regulons\n",
    "pathRegulons='/home/karin/Documents/timeTrajectories/data/regulons/by_strain/kN300_mean0std1_log/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Used gene sets\n",
    "Use gene sets that do not have less than 5 or more than 500 genes. Use all ontologies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all gene sets\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=True,set_sizes=(5,500))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Min size 5 Max size 440\n",
      "N sets {('Dictybase', 'Phenotypes'): 209, ('GO', 'biological_process'): 52, ('GO', 'molecular_function'): 35, ('GO', 'cellular_component'): 24, ('KEGG', 'Pathways'): 96}\n"
     ]
    }
   ],
   "source": [
    "# Find out max/min gene set sizes and N sets\n",
    "max_n=0\n",
    "min_n=np.inf\n",
    "for k,v in GENE_SETS_ONTOLOGY.items():\n",
    "    for s in v:\n",
    "        n=len(s.genes)\n",
    "        if n<min_n:\n",
    "            min_n=n\n",
    "        if n>max_n:\n",
    "            max_n=n\n",
    "print('Min size',min_n,'Max size',max_n)  \n",
    "print('N sets',{k:len(v) for k,v in GENE_SETS_ONTOLOGY.items()})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For conversion of gene names to EID\n",
    "genes = pd.read_csv(dataPath + 'mergedGenes_RPKUM.tsv', sep='\\t', index_col=0)\n",
    "all_gene_names= genes[(genes != 0).any(axis=1)].index\n",
    "NAME_EID=enr.name_genes_entrez(gene_names=all_gene_names, key_entrez=False)\n",
    "ALL_GENE_NAMES_EID=enr.convert_EID(genes=all_gene_names, name_EID=NAME_EID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "def group_diff_enrichment(data:pd.DataFrame,group:str,padj:float=0.25,min_overlap:int=None,\n",
    "                          use_annotated_genes:bool=False):\n",
    "    #Displays only gene sets that have overlap with query greater or equal to min_overlap \n",
    "    #For p value and padj calculation uses alll that have overlap >=1 } from gene_set_enrichment\n",
    "    \"\"\"\n",
    "    :param use_annotated_genes: if True use for reference and query  only genes that have at \n",
    "    least one gene set annotation\n",
    "    \"\"\"\n",
    "    selected=list(data[data['Cluster']==group].index)\n",
    "    query_EID=enr.convert_EID(genes=selected, name_EID=NAME_EID)\n",
    "    print('***  '+group+' selected:',len(selected),'with EID:',len(query_EID))\n",
    "    \n",
    "    reference_gene_eids=ALL_GENE_NAMES_EID.copy()\n",
    "    query_eids=query_EID.copy()\n",
    "    \n",
    "    if use_annotated_genes:\n",
    "        gene_sets_genes=set()\n",
    "        for gene_set_name, gene_sets in GENE_SETS_ONTOLOGY.items():\n",
    "            for gene_set in gene_sets:\n",
    "                gene_sets_genes.update(gene_set.genes)\n",
    "        reference_gene_eids=set(reference_gene_eids) & gene_sets_genes\n",
    "        query_eids=set(query_eids) & gene_sets_genes\n",
    "        print('Ratio of genes annotated with a gene set in reference',\n",
    "              round(len(reference_gene_eids)/len(ALL_GENE_NAMES_EID),2),\n",
    "             'and query',round(len(query_eids)/len(query_EID),2))\n",
    "    \n",
    "    query_in_enriched=set()\n",
    "    result=None\n",
    "    if len(query_eids) > 0:\n",
    "        enrichment=enr.gene_set_enrichment(query_eids, reference_EID=reference_gene_eids, \n",
    "                                                padj_threshold=padj,min_overlap=min_overlap,\n",
    "                                                gene_sets_ontology=GENE_SETS_ONTOLOGY)\n",
    "        if len(enrichment)>0:\n",
    "            enrichment_display=list()\n",
    "            enrichment= sorted(enrichment, key=lambda data: data.padj)\n",
    "            for enriched in enrichment:\n",
    "                query_in_enriched.update(enriched.gene_set.genes & query_eids)\n",
    "                enrichment_display.append({'Gene set':enriched.gene_set.name,\n",
    "                                           'Ontology':enriched.ontology[0]+': '+enriched.ontology[1],\n",
    "                                           'FDR':\"{:.2e}\".format(enriched.padj),'N in query':enriched.in_query,\n",
    "                                           'Set size':len(enriched.gene_set.genes),\n",
    "                                                          'N in ref.':enriched.in_reference})\n",
    "            result=pd.DataFrame(enrichment_display)\n",
    "    print('Enrichment at FDR: '+str(padj)+' and min query - gene set overlap',str(min_overlap))\n",
    "    print('N query genes in displayed gene sets:',len(query_in_enriched),'out of', len(query_eids),\n",
    "          'query genes used for enrichment calculation.')\n",
    "    display(result)\n",
    "    print('\\n')\n",
    "    return result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enrichment reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_enrichment=False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Enrichment for expression_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 30 with EID: 30\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 30 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.61e-19</td>\n",
       "      <td>12</td>\n",
       "      <td>72</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.81e-17</td>\n",
       "      <td>11</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.97e-16</td>\n",
       "      <td>10</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.39e-07</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.74e-06</td>\n",
       "      <td>6</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.59e-04</td>\n",
       "      <td>4</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.00e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.69e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.20e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.30e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>58</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.70e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                Ontology       FDR  \\\n",
       "0     Ribosome biogenesis in eukaryotes          KEGG: Pathways  3.61e-19   \n",
       "1                             nucleolus  GO: cellular_component  4.81e-17   \n",
       "2                   ribosome biogenesis  GO: biological_process  5.97e-16   \n",
       "3    ribonucleoprotein complex assembly  GO: biological_process  3.39e-07   \n",
       "4                           RNA binding  GO: molecular_function  8.74e-06   \n",
       "5   protein-containing complex assembly  GO: biological_process  7.59e-04   \n",
       "6                          rRNA binding  GO: molecular_function  1.00e-03   \n",
       "7           cellular component assembly  GO: biological_process  2.69e-03   \n",
       "8                     helicase activity  GO: molecular_function  1.20e-02   \n",
       "9            methyltransferase activity  GO: molecular_function  1.30e-02   \n",
       "10                      ATPase activity  GO: molecular_function  7.70e-02   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           12        72         68  \n",
       "1           11        72         72  \n",
       "2           10        61         61  \n",
       "3            4        18         18  \n",
       "4            6       169        169  \n",
       "5            4       134        133  \n",
       "6            2        13         13  \n",
       "7            4       202        201  \n",
       "8            2        51         51  \n",
       "9            2        58         56  \n",
       "10           2       169        168  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C2 selected: 66 with EID: 65\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 65 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.60e-10</td>\n",
       "      <td>9</td>\n",
       "      <td>71</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.23e-04</td>\n",
       "      <td>9</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.91e-04</td>\n",
       "      <td>4</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.34e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>53</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.16e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.16e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.16e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>211</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant cellular response to stress</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.55e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.59e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.01e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.56e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.56e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.11e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>150</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.35e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.77e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>165</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.77e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.12e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.81e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>104</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.78e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>9.38e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.52e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>vacuole</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>127</td>\n",
       "      <td>127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>80</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.02e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>91</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.22e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.24e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                Ontology  \\\n",
       "0                        Oxidative phosphorylation          KEGG: Pathways   \n",
       "1                                    mitochondrion  GO: cellular_component   \n",
       "2                        Citrate cycle (TCA cycle)          KEGG: Pathways   \n",
       "3                                        Phagosome          KEGG: Pathways   \n",
       "4                                  protein folding  GO: biological_process   \n",
       "5   generation of precursor metabolites and energy  GO: biological_process   \n",
       "6                              cytoplasmic vesicle  GO: cellular_component   \n",
       "7             aberrant cellular response to stress   Dictybase: Phenotypes   \n",
       "8                                         Ribosome          KEGG: Pathways   \n",
       "9            Biosynthesis of secondary metabolites          KEGG: Pathways   \n",
       "10          Various types of N-glycan biosynthesis          KEGG: Pathways   \n",
       "11                              cell morphogenesis  GO: biological_process   \n",
       "12              transmembrane transporter activity  GO: molecular_function   \n",
       "13                        unfolded protein binding  GO: molecular_function   \n",
       "14                         transmembrane transport  GO: biological_process   \n",
       "15                           N-Glycan biosynthesis          KEGG: Pathways   \n",
       "16                           Propanoate metabolism          KEGG: Pathways   \n",
       "17                               Carbon metabolism          KEGG: Pathways   \n",
       "18                        aberrant cell morphology   Dictybase: Phenotypes   \n",
       "19                                         cytosol  GO: cellular_component   \n",
       "20                         oxidoreductase activity  GO: molecular_function   \n",
       "21                                         vacuole  GO: cellular_component   \n",
       "22                              aberrant streaming   Dictybase: Phenotypes   \n",
       "23                                 ligase activity  GO: molecular_function   \n",
       "24     Protein processing in endoplasmic reticulum          KEGG: Pathways   \n",
       "25                            aberrant cytokinesis   Dictybase: Phenotypes   \n",
       "26                              mitotic cell cycle  GO: biological_process   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   6.60e-10           9        71         54  \n",
       "1   1.23e-04           9       229        229  \n",
       "2   4.91e-04           4        30         30  \n",
       "3   3.34e-03           4        53         52  \n",
       "4   1.16e-02           3        37         37  \n",
       "5   1.16e-02           4        77         77  \n",
       "6   1.16e-02           6       211        211  \n",
       "7   1.55e-02           2        11         11  \n",
       "8   1.59e-02           4       105         97  \n",
       "9   3.01e-02           6       289        281  \n",
       "10  4.56e-02           2        22         22  \n",
       "11  4.56e-02           2        23         23  \n",
       "12  5.11e-02           4       150        149  \n",
       "13  5.35e-02           2        27         27  \n",
       "14  5.77e-02           4       165        164  \n",
       "15  5.77e-02           2        31         30  \n",
       "16  6.12e-02           2        32         32  \n",
       "17  7.81e-02           3       104        102  \n",
       "18  8.78e-02           2        42         41  \n",
       "19  9.38e-02           4       203        203  \n",
       "20  9.52e-02           4       210        207  \n",
       "21  1.13e-01           3       127        127  \n",
       "22  1.68e-01           2        72         72  \n",
       "23  1.76e-01           2        80         80  \n",
       "24  2.02e-01           2        91         88  \n",
       "25  2.22e-01           2        98         98  \n",
       "26  2.24e-01           2       100        100  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C3 selected: 68 with EID: 68\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 62 out of 68 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.41e-126</td>\n",
       "      <td>60</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.47e-124</td>\n",
       "      <td>59</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.17e-109</td>\n",
       "      <td>53</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.72e-98</td>\n",
       "      <td>53</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.79e-85</td>\n",
       "      <td>52</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.26e-15</td>\n",
       "      <td>16</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.86e-13</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.12e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.83e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                Ontology        FDR  \\\n",
       "0                            Ribosome          KEGG: Pathways  6.41e-126   \n",
       "1                            ribosome  GO: cellular_component  1.47e-124   \n",
       "2  structural constituent of ribosome  GO: molecular_function  8.17e-109   \n",
       "3        structural molecule activity  GO: molecular_function   1.72e-98   \n",
       "4                         translation  GO: biological_process   1.79e-85   \n",
       "5                         RNA binding  GO: molecular_function   3.26e-15   \n",
       "6                        rRNA binding  GO: molecular_function   8.86e-13   \n",
       "7                             cytosol  GO: cellular_component   6.12e-05   \n",
       "8                 ribosome biogenesis  GO: biological_process   1.68e-02   \n",
       "9                 delayed development   Dictybase: Phenotypes   1.83e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0          60       105         97  \n",
       "1          59        93         93  \n",
       "2          53        85         85  \n",
       "3          53       117        117  \n",
       "4          52       169        169  \n",
       "5          16       169        169  \n",
       "6           7        13         13  \n",
       "7           8       203        203  \n",
       "8           3        61         61  \n",
       "9           2        69         69  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C4 selected: 55 with EID: 55\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 55 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.19e-08</td>\n",
       "      <td>10</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.88e-04</td>\n",
       "      <td>5</td>\n",
       "      <td>77</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.98e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.98e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.98e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.98e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.56e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.63e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.47e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.60e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>211</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.85e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>233</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                Ontology       FDR  \\\n",
       "0          cytoskeletal protein binding  GO: molecular_function  6.19e-08   \n",
       "1                           Endocytosis          KEGG: Pathways  3.88e-04   \n",
       "2   protein-containing complex assembly  GO: biological_process  1.98e-03   \n",
       "3           cellular component assembly  GO: biological_process  1.98e-03   \n",
       "4             cytoskeleton organization  GO: biological_process  1.98e-03   \n",
       "5                       plasma membrane  GO: cellular_component  1.98e-03   \n",
       "6                    response to stress  GO: biological_process  5.56e-03   \n",
       "7                          cytoskeleton  GO: cellular_component  6.63e-03   \n",
       "8                               cytosol  GO: cellular_component  5.47e-02   \n",
       "9                   cytoplasmic vesicle  GO: cellular_component  5.60e-02   \n",
       "10              decreased cell motility   Dictybase: Phenotypes  1.56e-01   \n",
       "11                            wild type   Dictybase: Phenotypes  1.68e-01   \n",
       "12           vesicle-mediated transport  GO: biological_process  1.85e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           10       158        158  \n",
       "1            5        77         75  \n",
       "2            5       134        133  \n",
       "3            6       202        201  \n",
       "4            6       197        196  \n",
       "5            7       268        268  \n",
       "6            7       360        358  \n",
       "7            5       185        185  \n",
       "8            4       203        203  \n",
       "9            4       211        211  \n",
       "10           2        76         76  \n",
       "11           4       341        338  \n",
       "12           3       233        233  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C5 selected: 41 with EID: 41\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 38 out of 41 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.56e-69</td>\n",
       "      <td>29</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.88e-34</td>\n",
       "      <td>21</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.46e-31</td>\n",
       "      <td>25</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.17e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.21e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.06e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>91</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Gene set                Ontology  \\\n",
       "0                                   Proteasome          KEGG: Pathways   \n",
       "1                           peptidase activity  GO: molecular_function   \n",
       "2                            catabolic process  GO: biological_process   \n",
       "3                    enzyme regulator activity  GO: molecular_function   \n",
       "4                           response to stress  GO: biological_process   \n",
       "5                              ATPase activity  GO: molecular_function   \n",
       "6  Protein processing in endoplasmic reticulum          KEGG: Pathways   \n",
       "\n",
       "        FDR  N in query  Set size  N in ref.  \n",
       "0  2.56e-69          29        39         37  \n",
       "1  6.88e-34          21        91         91  \n",
       "2  1.46e-31          25       250        250  \n",
       "3  5.17e-02           3       105        105  \n",
       "4  5.21e-02           5       360        358  \n",
       "5  1.06e-01           3       169        168  \n",
       "6  1.64e-01           2        91         88  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C6 selected: 74 with EID: 74\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 24 out of 74 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.88e-06</td>\n",
       "      <td>15</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.03e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.53e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>9.01e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.08e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.27e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.27e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.27e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.27e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.69e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.18e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.86e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.50e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.83e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>67</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.95e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                Ontology  \\\n",
       "0           anatomical structure development  GO: biological_process   \n",
       "1                              cell adhesion  GO: biological_process   \n",
       "2   development arrests at loose mound stage   Dictybase: Phenotypes   \n",
       "3                       extracellular region  GO: cellular_component   \n",
       "4              Glycosaminoglycan degradation          KEGG: Pathways   \n",
       "5               decreased fruiting body size   Dictybase: Phenotypes   \n",
       "6                   decreased aggregate size   Dictybase: Phenotypes   \n",
       "7                   increased slug migration   Dictybase: Phenotypes   \n",
       "8                        signal transduction  GO: biological_process   \n",
       "9                         response to stress  GO: biological_process   \n",
       "10                    multiple tips in mound   Dictybase: Phenotypes   \n",
       "11                      aberrant aggregation   Dictybase: Phenotypes   \n",
       "12                                locomotion  GO: biological_process   \n",
       "13                           plasma membrane  GO: cellular_component   \n",
       "14                 abolished gene expression   Dictybase: Phenotypes   \n",
       "15                      cell differentiation  GO: biological_process   \n",
       "16                    precocious development   Dictybase: Phenotypes   \n",
       "17                 decreased gene expression   Dictybase: Phenotypes   \n",
       "18                 decreased chemotaxis rate   Dictybase: Phenotypes   \n",
       "19         aberrant fruiting body morphology   Dictybase: Phenotypes   \n",
       "20             aberrant F-actin organization   Dictybase: Phenotypes   \n",
       "21                             cell motility  GO: biological_process   \n",
       "22                   decreased cell motility   Dictybase: Phenotypes   \n",
       "23                     abolished aggregation   Dictybase: Phenotypes   \n",
       "24                              cytoskeleton  GO: cellular_component   \n",
       "25                 cytoskeleton organization  GO: biological_process   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   3.88e-06          15       440        438  \n",
       "1   7.03e-03           5        84         83  \n",
       "2   7.53e-03           3        20         20  \n",
       "3   9.01e-03           6       157        157  \n",
       "4   1.08e-02           2         6          6  \n",
       "5   1.31e-02           4        76         75  \n",
       "6   1.31e-02           3        32         32  \n",
       "7   1.31e-02           2         8          8  \n",
       "8   1.31e-02           8       352        350  \n",
       "9   1.36e-02           8       360        358  \n",
       "10  4.27e-02           2        18         18  \n",
       "11  4.27e-02           3        61         61  \n",
       "12  4.27e-02           5       191        189  \n",
       "13  4.27e-02           6       268        268  \n",
       "14  6.69e-02           2        25         25  \n",
       "15  7.18e-02           4       145        145  \n",
       "16  7.86e-02           2        30         29  \n",
       "17  8.50e-02           3        95         86  \n",
       "18  1.03e-01           2        36         36  \n",
       "19  1.03e-01           3       101         98  \n",
       "20  1.03e-01           2        37         37  \n",
       "21  1.83e-01           2        67         66  \n",
       "22  1.87e-01           2        76         76  \n",
       "23  1.95e-01           2        82         80  \n",
       "24  2.14e-01           3       185        185  \n",
       "25  2.25e-01           3       197        196  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C7 selected: 40 with EID: 40\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 40 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.28e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.39e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>translucent sorus</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.39e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.39e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.93e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell death</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.52e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.52e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.48e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.48e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.48e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.48e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.43e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.43e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.58e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0          aberrant CRAC localization to the plasma membrane   \n",
       "1                  development arrests at tipped mound stage   \n",
       "2                                          translucent sorus   \n",
       "3                         decreased intracellular cAMP level   \n",
       "4   anatomical structure formation involved in morphogenesis   \n",
       "5                                                 cell death   \n",
       "6                           anatomical structure development   \n",
       "7                                            kinase activity   \n",
       "8                                       aberrant aggregation   \n",
       "9                               decreased chemotaxis to cAMP   \n",
       "10                                      cell differentiation   \n",
       "11                                           RNA degradation   \n",
       "12                              decreased fruiting body size   \n",
       "13                                                locomotion   \n",
       "14                         aberrant fruiting body morphology   \n",
       "15                                                 wild type   \n",
       "16                                              cytoskeleton   \n",
       "17                                               DNA binding   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0    Dictybase: Phenotypes  1.28e-02           2         6          6  \n",
       "1    Dictybase: Phenotypes  1.39e-02           2        12         12  \n",
       "2    Dictybase: Phenotypes  1.39e-02           2         9          9  \n",
       "3    Dictybase: Phenotypes  1.39e-02           2        12         12  \n",
       "4   GO: biological_process  2.93e-02           3        74         74  \n",
       "5   GO: biological_process  3.52e-02           2        25         25  \n",
       "6   GO: biological_process  3.52e-02           6       440        438  \n",
       "7   GO: molecular_function  4.09e-02           5       328        327  \n",
       "8    Dictybase: Phenotypes  6.48e-02           2        61         61  \n",
       "9    Dictybase: Phenotypes  6.48e-02           2        63         63  \n",
       "10  GO: biological_process  6.48e-02           3       145        145  \n",
       "11          KEGG: Pathways  6.48e-02           2        48         48  \n",
       "12   Dictybase: Phenotypes  7.43e-02           2        76         75  \n",
       "13  GO: biological_process  7.43e-02           3       191        189  \n",
       "14   Dictybase: Phenotypes  9.58e-02           2       101         98  \n",
       "15   Dictybase: Phenotypes  1.61e-01           3       341        338  \n",
       "16  GO: cellular_component  1.88e-01           2       185        185  \n",
       "17  GO: molecular_function  2.26e-01           2       221        221  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C8 selected: 23 with EID: 23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 23 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene set                Ontology       FDR  N in query  Set size  \\\n",
       "0  kinase activity  GO: molecular_function  1.10e-01           3       328   \n",
       "\n",
       "   N in ref.  \n",
       "0        327  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C9 selected: 20 with EID: 20\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 20 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C10 selected: 36 with EID: 36\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 36 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.86e-14</td>\n",
       "      <td>13</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.19e-10</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.12e-09</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.05e-07</td>\n",
       "      <td>7</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.95e-07</td>\n",
       "      <td>7</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.95e-07</td>\n",
       "      <td>5</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.15e-07</td>\n",
       "      <td>6</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.64e-07</td>\n",
       "      <td>6</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.22e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.01e-04</td>\n",
       "      <td>5</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.22e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.05e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.73e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Gene set                Ontology       FDR  \\\n",
       "0                      cell cycle  GO: biological_process  1.86e-14   \n",
       "1          chromosome segregation  GO: biological_process  1.19e-10   \n",
       "2        mitotic nuclear division  GO: biological_process  5.12e-09   \n",
       "3              mitotic cell cycle  GO: biological_process  1.05e-07   \n",
       "4                   cell division  GO: biological_process  1.95e-07   \n",
       "5                 DNA replication          KEGG: Pathways  1.95e-07   \n",
       "6                      chromosome  GO: cellular_component  2.15e-07   \n",
       "7         chromosome organization  GO: biological_process  2.64e-07   \n",
       "8                    cytoskeleton  GO: cellular_component  3.22e-06   \n",
       "9       cytoskeleton organization  GO: biological_process  8.01e-04   \n",
       "10  microtubule organizing center  GO: cellular_component  1.22e-03   \n",
       "11   cytoskeletal protein binding  GO: molecular_function  3.05e-03   \n",
       "12          DNA metabolic process  GO: biological_process  7.73e-02   \n",
       "13                ATPase activity  GO: molecular_function  1.47e-01   \n",
       "14                plasma membrane  GO: cellular_component  2.43e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           13       167        167  \n",
       "1            6        18         18  \n",
       "2            5        15         15  \n",
       "3            7       100        100  \n",
       "4            7       115        115  \n",
       "5            5        34         33  \n",
       "6            6        70         70  \n",
       "7            6        74         74  \n",
       "8            7       185        185  \n",
       "9            5       197        196  \n",
       "10           3        50         50  \n",
       "11           4       158        158  \n",
       "12           2       105        105  \n",
       "13           2       169        168  \n",
       "14           2       268        268  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C11 selected: 77 with EID: 77\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 77 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.43e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                Ontology       FDR  \\\n",
       "0                cell differentiation  GO: biological_process  2.43e-03   \n",
       "1               decreased sporulation   Dictybase: Phenotypes  2.39e-01   \n",
       "2  development arrests at mound stage   Dictybase: Phenotypes  2.39e-01   \n",
       "3    anatomical structure development  GO: biological_process  2.39e-01   \n",
       "4                       cell adhesion  GO: biological_process  2.39e-01   \n",
       "5       Starch and sucrose metabolism          KEGG: Pathways  2.39e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           7       145        145  \n",
       "1           2        53         53  \n",
       "2           2        56         54  \n",
       "3           7       440        438  \n",
       "4           3        84         83  \n",
       "5           2        31         31  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C12 selected: 57 with EID: 57\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 57 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.15e-04</td>\n",
       "      <td>7</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.28e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                Ontology       FDR  N in query  \\\n",
       "0       extracellular region  GO: cellular_component  3.15e-04           7   \n",
       "1        cell-cell signaling  GO: biological_process  5.19e-02           2   \n",
       "2  decreased spore viability   Dictybase: Phenotypes  2.28e-01           2   \n",
       "3         peptidase activity  GO: molecular_function  2.37e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       157        157  \n",
       "1        16         16  \n",
       "2        60         59  \n",
       "3        91         91  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C13 selected: 64 with EID: 64\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 64 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.67e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Gene set               Ontology       FDR  N in query  \\\n",
       "0  aberrant cell-cell adhesion  Dictybase: Phenotypes  3.67e-03           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0         5          5  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C14 selected: 19 with EID: 19\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 19 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C15 selected: 91 with EID: 91\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 30 out of 91 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.69e-11</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.69e-11</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.01e-09</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.97e-08</td>\n",
       "      <td>9</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.10e-06</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.73e-06</td>\n",
       "      <td>10</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.59e-04</td>\n",
       "      <td>13</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.26e-04</td>\n",
       "      <td>4</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.09e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.30e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.32e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>89</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.42e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.62e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.94e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.94e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>28</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                           external encapsulating structure   \n",
       "1                                                  cell wall   \n",
       "2                             aberrant spore coat morphology   \n",
       "3   anatomical structure formation involved in morphogenesis   \n",
       "4                       cell wall organization or biogenesis   \n",
       "5                                       cell differentiation   \n",
       "6                           anatomical structure development   \n",
       "7                              Starch and sucrose metabolism   \n",
       "8                       abolished stalk cell differentiation   \n",
       "9                                       extracellular region   \n",
       "10                               cellular component assembly   \n",
       "11                            carbohydrate metabolic process   \n",
       "12                                               DNA binding   \n",
       "13                                 decreased spore viability   \n",
       "14                                     abolished sporulation   \n",
       "15                                    Glutathione metabolism   \n",
       "16                                      aberrant culmination   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  7.69e-11           7        13         13  \n",
       "1   GO: cellular_component  7.69e-11           7        13         13  \n",
       "2    Dictybase: Phenotypes  4.01e-09           6        12         12  \n",
       "3   GO: biological_process  7.97e-08           9        74         74  \n",
       "4   GO: biological_process  1.10e-06           5        15         15  \n",
       "5   GO: biological_process  1.73e-06          10       145        145  \n",
       "6   GO: biological_process  2.59e-04          13       440        438  \n",
       "7           KEGG: Pathways  9.26e-04           4        31         31  \n",
       "8    Dictybase: Phenotypes  1.09e-02           2         7          7  \n",
       "9   GO: cellular_component  1.09e-02           6       157        157  \n",
       "10  GO: biological_process  3.30e-02           6       202        201  \n",
       "11  GO: biological_process  3.32e-02           4        89         88  \n",
       "12  GO: molecular_function  4.42e-02           6       221        221  \n",
       "13   Dictybase: Phenotypes  6.62e-02           3        60         59  \n",
       "14   Dictybase: Phenotypes  8.94e-02           2        25         25  \n",
       "15          KEGG: Pathways  8.94e-02           2        28         25  \n",
       "16   Dictybase: Phenotypes  1.58e-01           3        88         88  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C16 selected: 74 with EID: 74\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 74 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.40e-04</td>\n",
       "      <td>8</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                Ontology       FDR  N in query  \\\n",
       "0  extracellular region  GO: cellular_component  1.40e-04           8   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       157        157  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C17 selected: 61 with EID: 59\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 59 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C18 selected: 46 with EID: 45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 45 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.59e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.70e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1  anatomical structure formation involved in morphogenesis   \n",
       "2                                      cell differentiation   \n",
       "3                                   oxidoreductase activity   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: cellular_component  4.59e-04           6       157        157  \n",
       "1  GO: biological_process  2.70e-02           3        74         74  \n",
       "2  GO: biological_process  1.15e-01           3       145        145  \n",
       "3  GO: molecular_function  1.46e-01           3       210        207  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C19 selected: 58 with EID: 58\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 58 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.68e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>8.68e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.57e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.96e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                             cell adhesion   \n",
       "1                                      extracellular region   \n",
       "2                                        hydrolase activity   \n",
       "3                                      cell differentiation   \n",
       "4  anatomical structure formation involved in morphogenesis   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: biological_process  8.68e-02           3        84         83  \n",
       "1  GO: cellular_component  8.68e-02           4       157        157  \n",
       "2  GO: molecular_function  1.57e-01           2        46         45  \n",
       "3  GO: biological_process  1.92e-01           3       145        145  \n",
       "4  GO: biological_process  1.96e-01           2        74         74  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C20 selected: 54 with EID: 54\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 54 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>53</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set               Ontology       FDR  N in query  \\\n",
       "0  decreased spore viability  Dictybase: Phenotypes  2.34e-01           2   \n",
       "1                  Phagosome         KEGG: Pathways  2.34e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0        60         59  \n",
       "1        53         52  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C21 selected: 45 with EID: 45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 45 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2)\n",
    "    if save_enrichment and enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Using only genes annotated with at least one gene set \n",
    "To find genes that have gene set annotations only gene sets used in enrichment are used (see above). Genes that are in at least one of these gene sets are used for enrichment calculation, both for reference and gene group of interest (query)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  C1 selected: 30 with EID: 30\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.7\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 21 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.09e-15</td>\n",
       "      <td>12</td>\n",
       "      <td>72</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>8.66e-14</td>\n",
       "      <td>11</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.22e-13</td>\n",
       "      <td>10</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.03e-06</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.86e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.46e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.10e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.00e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.27e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.64e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>58</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Gene set                Ontology       FDR  \\\n",
       "0    Ribosome biogenesis in eukaryotes          KEGG: Pathways  1.09e-15   \n",
       "1                            nucleolus  GO: cellular_component  8.66e-14   \n",
       "2                  ribosome biogenesis  GO: biological_process  6.22e-13   \n",
       "3   ribonucleoprotein complex assembly  GO: biological_process  7.03e-06   \n",
       "4                          RNA binding  GO: molecular_function  5.86e-04   \n",
       "5                         rRNA binding  GO: molecular_function  5.46e-03   \n",
       "6  protein-containing complex assembly  GO: biological_process  1.10e-02   \n",
       "7          cellular component assembly  GO: biological_process  4.00e-02   \n",
       "8                    helicase activity  GO: molecular_function  5.27e-02   \n",
       "9           methyltransferase activity  GO: molecular_function  5.64e-02   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0          12        72         68  \n",
       "1          11        72         72  \n",
       "2          10        61         61  \n",
       "3           4        18         18  \n",
       "4           6       169        169  \n",
       "5           2        13         13  \n",
       "6           4       134        133  \n",
       "7           4       202        201  \n",
       "8           2        51         51  \n",
       "9           2        58         56  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C2 selected: 66 with EID: 65\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.58\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 33 out of 38 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.92e-08</td>\n",
       "      <td>9</td>\n",
       "      <td>71</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.90e-03</td>\n",
       "      <td>9</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.90e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.07e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>53</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant cellular response to stress</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.32e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.32e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.37e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>211</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.57e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.57e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.94e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.18e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                Ontology  \\\n",
       "0                        Oxidative phosphorylation          KEGG: Pathways   \n",
       "1                                    mitochondrion  GO: cellular_component   \n",
       "2                        Citrate cycle (TCA cycle)          KEGG: Pathways   \n",
       "3                                        Phagosome          KEGG: Pathways   \n",
       "4             aberrant cellular response to stress   Dictybase: Phenotypes   \n",
       "5                                  protein folding  GO: biological_process   \n",
       "6   generation of precursor metabolites and energy  GO: biological_process   \n",
       "7                              cytoplasmic vesicle  GO: cellular_component   \n",
       "8                                         Ribosome          KEGG: Pathways   \n",
       "9                               cell morphogenesis  GO: biological_process   \n",
       "10          Various types of N-glycan biosynthesis          KEGG: Pathways   \n",
       "11                        unfolded protein binding  GO: molecular_function   \n",
       "12                           N-Glycan biosynthesis          KEGG: Pathways   \n",
       "13                           Propanoate metabolism          KEGG: Pathways   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   8.92e-08           9        71         54  \n",
       "1   6.90e-03           9       229        229  \n",
       "2   6.90e-03           4        30         30  \n",
       "3   3.07e-02           4        53         52  \n",
       "4   7.32e-02           2        11         11  \n",
       "5   7.32e-02           3        37         37  \n",
       "6   7.37e-02           4        77         77  \n",
       "7   1.37e-01           6       211        211  \n",
       "8   1.37e-01           4       105         97  \n",
       "9   1.57e-01           2        23         23  \n",
       "10  1.57e-01           2        22         22  \n",
       "11  1.94e-01           2        27         27  \n",
       "12  2.18e-01           2        31         30  \n",
       "13  2.29e-01           2        32         32  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C3 selected: 68 with EID: 68\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.97\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 62 out of 66 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.76e-98</td>\n",
       "      <td>60</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.26e-97</td>\n",
       "      <td>59</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.68e-84</td>\n",
       "      <td>53</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.91e-73</td>\n",
       "      <td>53</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.25e-61</td>\n",
       "      <td>52</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.38e-09</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.17e-08</td>\n",
       "      <td>16</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.91e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                Ontology       FDR  \\\n",
       "0                            Ribosome          KEGG: Pathways  6.76e-98   \n",
       "1                            ribosome  GO: cellular_component  6.26e-97   \n",
       "2  structural constituent of ribosome  GO: molecular_function  9.68e-84   \n",
       "3        structural molecule activity  GO: molecular_function  1.91e-73   \n",
       "4                         translation  GO: biological_process  6.25e-61   \n",
       "5                        rRNA binding  GO: molecular_function  2.38e-09   \n",
       "6                         RNA binding  GO: molecular_function  4.17e-08   \n",
       "7                             cytosol  GO: cellular_component  7.91e-02   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0          60       105         97  \n",
       "1          59        93         93  \n",
       "2          53        85         85  \n",
       "3          53       117        117  \n",
       "4          52       169        169  \n",
       "5           7        13         13  \n",
       "6          16       169        169  \n",
       "7           8       203        203  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C4 selected: 55 with EID: 55\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.44\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 24 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.91e-07</td>\n",
       "      <td>10</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.38e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>77</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.77e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.77e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.77e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.77e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.18e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.18e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.41e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>211</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                Ontology       FDR  \\\n",
       "0          cytoskeletal protein binding  GO: molecular_function  3.91e-07   \n",
       "1                           Endocytosis          KEGG: Pathways  1.38e-03   \n",
       "2   protein-containing complex assembly  GO: biological_process  6.77e-03   \n",
       "3           cellular component assembly  GO: biological_process  6.77e-03   \n",
       "4             cytoskeleton organization  GO: biological_process  6.77e-03   \n",
       "5                       plasma membrane  GO: cellular_component  6.77e-03   \n",
       "6                    response to stress  GO: biological_process  2.18e-02   \n",
       "7                          cytoskeleton  GO: cellular_component  2.18e-02   \n",
       "8                               cytosol  GO: cellular_component  1.41e-01   \n",
       "9                   cytoplasmic vesicle  GO: cellular_component  1.44e-01   \n",
       "10              decreased cell motility   Dictybase: Phenotypes  2.42e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           10       158        158  \n",
       "1            5        77         75  \n",
       "2            5       134        133  \n",
       "3            6       202        201  \n",
       "4            6       197        196  \n",
       "5            7       268        268  \n",
       "6            7       360        358  \n",
       "7            5       185        185  \n",
       "8            4       203        203  \n",
       "9            4       211        211  \n",
       "10           2        76         76  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C5 selected: 41 with EID: 41\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.98\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 35 out of 40 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.03e-55</td>\n",
       "      <td>29</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.60e-24</td>\n",
       "      <td>21</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.94e-20</td>\n",
       "      <td>25</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene set                Ontology       FDR  N in query  Set size  \\\n",
       "0          Proteasome          KEGG: Pathways  2.03e-55          29        39   \n",
       "1  peptidase activity  GO: molecular_function  7.60e-24          21        91   \n",
       "2   catabolic process  GO: biological_process  8.94e-20          25       250   \n",
       "\n",
       "   N in ref.  \n",
       "0         37  \n",
       "1         91  \n",
       "2        250  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C6 selected: 74 with EID: 74\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.42\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 24 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.30e-05</td>\n",
       "      <td>15</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.59e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.59e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.28e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.44e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.78e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.78e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.35e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.29e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.47e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.98e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.38e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.65e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.65e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.46e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>67</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                Ontology  \\\n",
       "0           anatomical structure development  GO: biological_process   \n",
       "1   development arrests at loose mound stage   Dictybase: Phenotypes   \n",
       "2                              cell adhesion  GO: biological_process   \n",
       "3              Glycosaminoglycan degradation          KEGG: Pathways   \n",
       "4                       extracellular region  GO: cellular_component   \n",
       "5                   decreased aggregate size   Dictybase: Phenotypes   \n",
       "6                   increased slug migration   Dictybase: Phenotypes   \n",
       "7               decreased fruiting body size   Dictybase: Phenotypes   \n",
       "8                        signal transduction  GO: biological_process   \n",
       "9                         response to stress  GO: biological_process   \n",
       "10                    multiple tips in mound   Dictybase: Phenotypes   \n",
       "11                      aberrant aggregation   Dictybase: Phenotypes   \n",
       "12                 abolished gene expression   Dictybase: Phenotypes   \n",
       "13                                locomotion  GO: biological_process   \n",
       "14                           plasma membrane  GO: cellular_component   \n",
       "15                    precocious development   Dictybase: Phenotypes   \n",
       "16                      cell differentiation  GO: biological_process   \n",
       "17                 decreased chemotaxis rate   Dictybase: Phenotypes   \n",
       "18                 decreased gene expression   Dictybase: Phenotypes   \n",
       "19             aberrant F-actin organization   Dictybase: Phenotypes   \n",
       "20         aberrant fruiting body morphology   Dictybase: Phenotypes   \n",
       "21                             cell motility  GO: biological_process   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   2.30e-05          15       440        438  \n",
       "1   1.59e-02           3        20         20  \n",
       "2   1.59e-02           5        84         83  \n",
       "3   2.28e-02           2         6          6  \n",
       "4   2.44e-02           6       157        157  \n",
       "5   2.78e-02           3        32         32  \n",
       "6   2.78e-02           2         8          8  \n",
       "7   3.35e-02           4        76         75  \n",
       "8   5.29e-02           8       352        350  \n",
       "9   5.47e-02           8       360        358  \n",
       "10  7.98e-02           2        18         18  \n",
       "11  1.01e-01           3        61         61  \n",
       "12  1.13e-01           2        25         25  \n",
       "13  1.13e-01           5       191        189  \n",
       "14  1.13e-01           6       268        268  \n",
       "15  1.38e-01           2        30         29  \n",
       "16  1.56e-01           4       145        145  \n",
       "17  1.65e-01           2        36         36  \n",
       "18  1.65e-01           3        95         86  \n",
       "19  1.66e-01           2        37         37  \n",
       "20  1.75e-01           3       101         98  \n",
       "21  2.46e-01           2        67         66  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C7 selected: 40 with EID: 40\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.70e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.84e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>translucent sorus</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.84e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.84e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.16e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell death</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.65e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.65e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.06e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.80e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.80e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.80e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.80e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.41e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.41e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.10e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0          aberrant CRAC localization to the plasma membrane   \n",
       "1                  development arrests at tipped mound stage   \n",
       "2                                          translucent sorus   \n",
       "3                         decreased intracellular cAMP level   \n",
       "4   anatomical structure formation involved in morphogenesis   \n",
       "5                                                 cell death   \n",
       "6                           anatomical structure development   \n",
       "7                                            kinase activity   \n",
       "8                                       aberrant aggregation   \n",
       "9                               decreased chemotaxis to cAMP   \n",
       "10                                      cell differentiation   \n",
       "11                                           RNA degradation   \n",
       "12                              decreased fruiting body size   \n",
       "13                                                locomotion   \n",
       "14                         aberrant fruiting body morphology   \n",
       "15                                                 wild type   \n",
       "16                                              cytoskeleton   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0    Dictybase: Phenotypes  1.70e-02           2         6          6  \n",
       "1    Dictybase: Phenotypes  1.84e-02           2        12         12  \n",
       "2    Dictybase: Phenotypes  1.84e-02           2         9          9  \n",
       "3    Dictybase: Phenotypes  1.84e-02           2        12         12  \n",
       "4   GO: biological_process  4.16e-02           3        74         74  \n",
       "5   GO: biological_process  4.65e-02           2        25         25  \n",
       "6   GO: biological_process  4.65e-02           6       440        438  \n",
       "7   GO: molecular_function  6.06e-02           5       328        327  \n",
       "8    Dictybase: Phenotypes  7.80e-02           2        61         61  \n",
       "9    Dictybase: Phenotypes  7.80e-02           2        63         63  \n",
       "10  GO: biological_process  7.80e-02           3       145        145  \n",
       "11          KEGG: Pathways  7.80e-02           2        48         48  \n",
       "12   Dictybase: Phenotypes  9.41e-02           2        76         75  \n",
       "13  GO: biological_process  9.41e-02           3       191        189  \n",
       "14   Dictybase: Phenotypes  1.13e-01           2       101         98  \n",
       "15   Dictybase: Phenotypes  2.10e-01           3       341        338  \n",
       "16  GO: cellular_component  2.37e-01           2       185        185  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C8 selected: 23 with EID: 23\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 6 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.72e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene set                Ontology       FDR  N in query  Set size  \\\n",
       "0  kinase activity  GO: molecular_function  4.72e-02           3       328   \n",
       "\n",
       "   N in ref.  \n",
       "0        327  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C9 selected: 20 with EID: 20\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.0\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 0 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C10 selected: 36 with EID: 36\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.53\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 19 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.67e-13</td>\n",
       "      <td>13</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.58e-09</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.78e-08</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.81e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.15e-06</td>\n",
       "      <td>5</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.74e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.74e-06</td>\n",
       "      <td>6</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.35e-06</td>\n",
       "      <td>6</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.22e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.37e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.76e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.67e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Gene set                Ontology       FDR  \\\n",
       "0                      cell cycle  GO: biological_process  6.67e-13   \n",
       "1          chromosome segregation  GO: biological_process  1.58e-09   \n",
       "2        mitotic nuclear division  GO: biological_process  4.78e-08   \n",
       "3              mitotic cell cycle  GO: biological_process  1.81e-06   \n",
       "4                 DNA replication          KEGG: Pathways  2.15e-06   \n",
       "5                   cell division  GO: biological_process  2.74e-06   \n",
       "6                      chromosome  GO: cellular_component  2.74e-06   \n",
       "7         chromosome organization  GO: biological_process  3.35e-06   \n",
       "8                    cytoskeleton  GO: cellular_component  5.22e-05   \n",
       "9   microtubule organizing center  GO: cellular_component  5.37e-03   \n",
       "10      cytoskeleton organization  GO: biological_process  5.76e-03   \n",
       "11   cytoskeletal protein binding  GO: molecular_function  1.67e-02   \n",
       "12          DNA metabolic process  GO: biological_process  1.86e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           13       167        167  \n",
       "1            6        18         18  \n",
       "2            5        15         15  \n",
       "3            7       100        100  \n",
       "4            5        34         33  \n",
       "5            7       115        115  \n",
       "6            6        70         70  \n",
       "7            6        74         74  \n",
       "8            7       185        185  \n",
       "9            3        50         50  \n",
       "10           5       197        196  \n",
       "11           4       158        158  \n",
       "12           2       105        105  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C11 selected: 77 with EID: 77\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.29\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.47e-04</td>\n",
       "      <td>7</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      cell differentiation   \n",
       "1                                     decreased sporulation   \n",
       "2                        development arrests at mound stage   \n",
       "3                          anatomical structure development   \n",
       "4                                             cell adhesion   \n",
       "5                             Starch and sucrose metabolism   \n",
       "6                                     abolished culmination   \n",
       "7                                       delayed aggregation   \n",
       "8  anatomical structure formation involved in morphogenesis   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: biological_process  6.47e-04           7       145        145  \n",
       "1   Dictybase: Phenotypes  2.07e-01           2        53         53  \n",
       "2   Dictybase: Phenotypes  2.07e-01           2        56         54  \n",
       "3  GO: biological_process  2.07e-01           7       440        438  \n",
       "4  GO: biological_process  2.07e-01           3        84         83  \n",
       "5          KEGG: Pathways  2.07e-01           2        31         31  \n",
       "6   Dictybase: Phenotypes  2.17e-01           2        62         62  \n",
       "7   Dictybase: Phenotypes  2.45e-01           2        72         72  \n",
       "8  GO: biological_process  2.45e-01           2        74         74  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C12 selected: 57 with EID: 57\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 11 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.66e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.96e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                Ontology       FDR  \\\n",
       "0              extracellular region  GO: cellular_component  1.66e-05   \n",
       "1               cell-cell signaling  GO: biological_process  2.96e-02   \n",
       "2         decreased spore viability   Dictybase: Phenotypes  1.63e-01   \n",
       "3                peptidase activity  GO: molecular_function  1.63e-01   \n",
       "4  anatomical structure development  GO: biological_process  1.88e-01   \n",
       "5                   plasma membrane  GO: cellular_component  1.88e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           7       157        157  \n",
       "1           2        16         16  \n",
       "2           2        60         59  \n",
       "3           2        91         91  \n",
       "4           4       440        438  \n",
       "5           3       268        268  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C13 selected: 64 with EID: 64\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.16\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 10 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.13e-04</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.61e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.61e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Gene set                Ontology       FDR  \\\n",
       "0        aberrant cell-cell adhesion   Dictybase: Phenotypes  8.13e-04   \n",
       "1  aberrant fruiting body morphology   Dictybase: Phenotypes  7.61e-02   \n",
       "2   anatomical structure development  GO: biological_process  7.61e-02   \n",
       "3               extracellular region  GO: cellular_component  1.16e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           2         5          5  \n",
       "1           2       101         98  \n",
       "2           4       440        438  \n",
       "3           2       157        157  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C14 selected: 19 with EID: 19\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.05\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 1 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C15 selected: 91 with EID: 91\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 28 out of 41 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.38e-10</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.38e-10</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.57e-08</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.70e-07</td>\n",
       "      <td>9</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.30e-06</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.38e-05</td>\n",
       "      <td>10</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.70e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.96e-03</td>\n",
       "      <td>13</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.16e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.14e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>89</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.38e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>28</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                           external encapsulating structure   \n",
       "1                                                  cell wall   \n",
       "2                             aberrant spore coat morphology   \n",
       "3   anatomical structure formation involved in morphogenesis   \n",
       "4                       cell wall organization or biogenesis   \n",
       "5                                       cell differentiation   \n",
       "6                              Starch and sucrose metabolism   \n",
       "7                           anatomical structure development   \n",
       "8                       abolished stalk cell differentiation   \n",
       "9                                       extracellular region   \n",
       "10                            carbohydrate metabolic process   \n",
       "11                               cellular component assembly   \n",
       "12                                     abolished sporulation   \n",
       "13                                 decreased spore viability   \n",
       "14                                               DNA binding   \n",
       "15                                    Glutathione metabolism   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  6.38e-10           7        13         13  \n",
       "1   GO: cellular_component  6.38e-10           7        13         13  \n",
       "2    Dictybase: Phenotypes  2.57e-08           6        12         12  \n",
       "3   GO: biological_process  9.70e-07           9        74         74  \n",
       "4   GO: biological_process  5.30e-06           5        15         15  \n",
       "5   GO: biological_process  2.38e-05          10       145        145  \n",
       "6           KEGG: Pathways  3.70e-03           4        31         31  \n",
       "7   GO: biological_process  3.96e-03          13       440        438  \n",
       "8    Dictybase: Phenotypes  2.37e-02           2         7          7  \n",
       "9   GO: cellular_component  5.16e-02           6       157        157  \n",
       "10  GO: biological_process  1.14e-01           4        89         88  \n",
       "11  GO: biological_process  1.38e-01           6       202        201  \n",
       "12   Dictybase: Phenotypes  1.69e-01           2        25         25  \n",
       "13   Dictybase: Phenotypes  1.69e-01           3        60         59  \n",
       "14  GO: molecular_function  1.69e-01           6       221        221  \n",
       "15          KEGG: Pathways  1.69e-01           2        28         25  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C16 selected: 74 with EID: 74\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 15 out of 17 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.21e-06</td>\n",
       "      <td>8</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.96e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.20e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                Ontology       FDR  N in query  \\\n",
       "0       extracellular region  GO: cellular_component  3.21e-06           8   \n",
       "1                  wild type   Dictybase: Phenotypes  1.96e-01           5   \n",
       "2  decreased gene expression   Dictybase: Phenotypes  2.15e-01           2   \n",
       "3  decreased spore viability   Dictybase: Phenotypes  2.15e-01           2   \n",
       "4                DNA binding  GO: molecular_function  2.20e-01           3   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       157        157  \n",
       "1       341        338  \n",
       "2        95         86  \n",
       "3        60         59  \n",
       "4       221        221  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C17 selected: 61 with EID: 59\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 12 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                Ontology       FDR  \\\n",
       "0  anatomical structure development  GO: biological_process  2.34e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           4       440        438  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C18 selected: 46 with EID: 45\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.33\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.02e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1  anatomical structure formation involved in morphogenesis   \n",
       "2                                      cell differentiation   \n",
       "3                                   oxidoreductase activity   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: cellular_component  3.02e-04           6       157        157  \n",
       "1  GO: biological_process  2.73e-02           3        74         74  \n",
       "2  GO: biological_process  1.18e-01           3       145        145  \n",
       "3  GO: molecular_function  1.53e-01           3       210        207  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C19 selected: 58 with EID: 58\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 12 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.18e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.18e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.29e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.29e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                             cell adhesion   \n",
       "1                                      extracellular region   \n",
       "2                                      cell differentiation   \n",
       "3                                        hydrolase activity   \n",
       "4  anatomical structure formation involved in morphogenesis   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: biological_process  2.18e-02           3        84         83  \n",
       "1  GO: cellular_component  2.18e-02           4       157        157  \n",
       "2  GO: biological_process  5.29e-02           3       145        145  \n",
       "3  GO: molecular_function  5.29e-02           2        46         45  \n",
       "4  GO: biological_process  1.02e-01           2        74         74  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C20 selected: 54 with EID: 54\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 11 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.51e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.51e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>53</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                Ontology       FDR  N in query  \\\n",
       "0  decreased spore viability   Dictybase: Phenotypes  1.51e-01           2   \n",
       "1                  Phagosome          KEGG: Pathways  1.51e-01           2   \n",
       "2       extracellular region  GO: cellular_component  2.37e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0        60         59  \n",
       "1        53         52  \n",
       "2       157        157  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  C21 selected: 45 with EID: 45\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.13\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 6 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene set                Ontology       FDR  N in query  Set size  \\\n",
       "0  response to stress  GO: biological_process  2.25e-01           2       360   \n",
       "1             cytosol  GO: cellular_component  2.25e-01           2       203   \n",
       "\n",
       "   N in ref.  \n",
       "0        358  \n",
       "1        203  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    enriched=group_diff_enrichment(data=regulons,group='C'+str(group),min_overlap=2,use_annotated_genes=True)\n",
    "    if  save_enrichment and enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersLouvain0.4minmaxNologPCA30kN30_annotatedOnly_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Enrichment for mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*** Cluster C1 (A)\n",
      "***  C1 selected: 31 with EID: 31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.70e-19</td>\n",
       "      <td>12</td>\n",
       "      <td>72</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>9.74e-17</td>\n",
       "      <td>11</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.15e-15</td>\n",
       "      <td>10</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.11e-07</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.41e-05</td>\n",
       "      <td>6</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.00e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.82e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>58</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.28e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                Ontology       FDR  \\\n",
       "0     Ribosome biogenesis in eukaryotes          KEGG: Pathways  7.70e-19   \n",
       "1                             nucleolus  GO: cellular_component  9.74e-17   \n",
       "2                   ribosome biogenesis  GO: biological_process  1.15e-15   \n",
       "3    ribonucleoprotein complex assembly  GO: biological_process  5.11e-07   \n",
       "4                           RNA binding  GO: molecular_function  1.41e-05   \n",
       "5   protein-containing complex assembly  GO: biological_process  1.13e-03   \n",
       "6                          rRNA binding  GO: molecular_function  1.41e-03   \n",
       "7           cellular component assembly  GO: biological_process  4.00e-03   \n",
       "8                     helicase activity  GO: molecular_function  1.68e-02   \n",
       "9            methyltransferase activity  GO: molecular_function  1.82e-02   \n",
       "10                      ATPase activity  GO: molecular_function  9.28e-02   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           12        72         68  \n",
       "1           11        72         72  \n",
       "2           10        61         61  \n",
       "3            4        18         18  \n",
       "4            6       169        169  \n",
       "5            4       134        133  \n",
       "6            2        13         13  \n",
       "7            4       202        201  \n",
       "8            2        51         51  \n",
       "9            2        58         56  \n",
       "10           2       169        168  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C2 (B)\n",
      "***  C2 selected: 54 with EID: 54\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 25 out of 54 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.89e-08</td>\n",
       "      <td>11</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.89e-08</td>\n",
       "      <td>7</td>\n",
       "      <td>71</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.55e-05</td>\n",
       "      <td>6</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.28e-04</td>\n",
       "      <td>4</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.02e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.14e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.01e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>53</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.11e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.11e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.11e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.28e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>104</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.64e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.88e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>150</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>165</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.25e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>80</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>354</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>vacuole</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>127</td>\n",
       "      <td>127</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                Ontology  \\\n",
       "0                                    mitochondrion  GO: cellular_component   \n",
       "1                        Oxidative phosphorylation          KEGG: Pathways   \n",
       "2   generation of precursor metabolites and energy  GO: biological_process   \n",
       "3                                  protein folding  GO: biological_process   \n",
       "4                         unfolded protein binding  GO: molecular_function   \n",
       "5                          oxidoreductase activity  GO: molecular_function   \n",
       "6                                        Phagosome          KEGG: Pathways   \n",
       "7                            Propanoate metabolism          KEGG: Pathways   \n",
       "8                        Citrate cycle (TCA cycle)          KEGG: Pathways   \n",
       "9                                         Ribosome          KEGG: Pathways   \n",
       "10                               Carbon metabolism          KEGG: Pathways   \n",
       "11                                         cytosol  GO: cellular_component   \n",
       "12              transmembrane transporter activity  GO: molecular_function   \n",
       "13                         transmembrane transport  GO: biological_process   \n",
       "14           Biosynthesis of secondary metabolites          KEGG: Pathways   \n",
       "15                                 ligase activity  GO: molecular_function   \n",
       "16                small molecule metabolic process  GO: biological_process   \n",
       "17                                         vacuole  GO: cellular_component   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   7.89e-08          11       229        229  \n",
       "1   7.89e-08           7        71         54  \n",
       "2   1.55e-05           6        77         77  \n",
       "3   2.28e-04           4        37         37  \n",
       "4   2.02e-03           3        27         27  \n",
       "5   2.14e-03           6       210        207  \n",
       "6   1.01e-02           3        53         52  \n",
       "7   4.11e-02           2        32         32  \n",
       "8   4.11e-02           2        30         30  \n",
       "9   4.11e-02           3       105         97  \n",
       "10  4.28e-02           3       104        102  \n",
       "11  4.64e-02           4       203        203  \n",
       "12  9.88e-02           3       150        149  \n",
       "13  1.03e-01           3       165        164  \n",
       "14  1.03e-01           4       289        281  \n",
       "15  1.25e-01           2        80         80  \n",
       "16  1.46e-01           4       354        353  \n",
       "17  2.00e-01           2       127        127  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C3 (C)\n",
      "***  C3 selected: 54 with EID: 54\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 44 out of 54 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.07e-86</td>\n",
       "      <td>44</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.46e-85</td>\n",
       "      <td>43</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.18e-76</td>\n",
       "      <td>39</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.04e-69</td>\n",
       "      <td>39</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.72e-62</td>\n",
       "      <td>39</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.40e-10</td>\n",
       "      <td>11</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.50e-07</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                Ontology       FDR  \\\n",
       "0                            Ribosome          KEGG: Pathways  1.07e-86   \n",
       "1                            ribosome  GO: cellular_component  4.46e-85   \n",
       "2  structural constituent of ribosome  GO: molecular_function  8.18e-76   \n",
       "3        structural molecule activity  GO: molecular_function  4.04e-69   \n",
       "4                         translation  GO: biological_process  5.72e-62   \n",
       "5                         RNA binding  GO: molecular_function  4.40e-10   \n",
       "6                        rRNA binding  GO: molecular_function  6.50e-07   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0          44       105         97  \n",
       "1          43        93         93  \n",
       "2          39        85         85  \n",
       "3          39       117        117  \n",
       "4          39       169        169  \n",
       "5          11       169        169  \n",
       "6           4        13         13  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C4 (D)\n",
      "***  C4 selected: 43 with EID: 43\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 43 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.38e-10</td>\n",
       "      <td>11</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.27e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.27e-06</td>\n",
       "      <td>8</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.27e-06</td>\n",
       "      <td>8</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.07e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.07e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.58e-05</td>\n",
       "      <td>5</td>\n",
       "      <td>77</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.47e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.06e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.28e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.43e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.79e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>211</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.78e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.87e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.27e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.27e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.27e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.27e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.65e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.65e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.65e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>233</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.41e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>146</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>165</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>150</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                Ontology       FDR  \\\n",
       "0          cytoskeletal protein binding  GO: molecular_function  2.38e-10   \n",
       "1   protein-containing complex assembly  GO: biological_process  5.27e-06   \n",
       "2           cellular component assembly  GO: biological_process  5.27e-06   \n",
       "3             cytoskeleton organization  GO: biological_process  5.27e-06   \n",
       "4                          cytoskeleton  GO: cellular_component  3.07e-05   \n",
       "5                       plasma membrane  GO: cellular_component  3.07e-05   \n",
       "6                           Endocytosis          KEGG: Pathways  4.58e-05   \n",
       "7                    response to stress  GO: biological_process  1.47e-03   \n",
       "8                               cytosol  GO: cellular_component  4.06e-03   \n",
       "9               decreased cell motility   Dictybase: Phenotypes  1.28e-02   \n",
       "10                   cell morphogenesis  GO: biological_process  1.43e-02   \n",
       "11                  cytoplasmic vesicle  GO: cellular_component  2.79e-02   \n",
       "12             aberrant cell morphology   Dictybase: Phenotypes  3.78e-02   \n",
       "13     anatomical structure development  GO: biological_process  6.87e-02   \n",
       "14                            wild type   Dictybase: Phenotypes  8.27e-02   \n",
       "15                abolished aggregation   Dictybase: Phenotypes  8.27e-02   \n",
       "16                   aberrant streaming   Dictybase: Phenotypes  8.27e-02   \n",
       "17                         reproduction  GO: biological_process  8.27e-02   \n",
       "18                 aberrant cytokinesis   Dictybase: Phenotypes  9.65e-02   \n",
       "19                   mitotic cell cycle  GO: biological_process  9.65e-02   \n",
       "20           vesicle-mediated transport  GO: biological_process  9.65e-02   \n",
       "21                        cell division  GO: biological_process  1.11e-01   \n",
       "22                decreased growth rate   Dictybase: Phenotypes  1.41e-01   \n",
       "23                           cell cycle  GO: biological_process  1.55e-01   \n",
       "24              transmembrane transport  GO: biological_process  1.55e-01   \n",
       "25                      ATPase activity  GO: molecular_function  1.55e-01   \n",
       "26   transmembrane transporter activity  GO: molecular_function  1.55e-01   \n",
       "27                           locomotion  GO: biological_process  1.80e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           11       158        158  \n",
       "1            7       134        133  \n",
       "2            8       202        201  \n",
       "3            8       197        196  \n",
       "4            7       185        185  \n",
       "5            8       268        268  \n",
       "6            5        77         75  \n",
       "7            7       360        358  \n",
       "8            5       203        203  \n",
       "9            3        76         76  \n",
       "10           2        23         23  \n",
       "11           4       211        211  \n",
       "12           2        42         41  \n",
       "13           5       440        438  \n",
       "14           4       341        338  \n",
       "15           2        82         80  \n",
       "16           2        72         72  \n",
       "17           2        79         79  \n",
       "18           2        98         98  \n",
       "19           2       100        100  \n",
       "20           3       233        233  \n",
       "21           2       115        115  \n",
       "22           2       146        137  \n",
       "23           2       167        167  \n",
       "24           2       165        164  \n",
       "25           2       169        168  \n",
       "26           2       150        149  \n",
       "27           2       191        189  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C5 (E)\n",
      "***  C5 selected: 20 with EID: 20\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 20 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.29e-33</td>\n",
       "      <td>17</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.70e-33</td>\n",
       "      <td>17</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.12e-28</td>\n",
       "      <td>15</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.46e-26</td>\n",
       "      <td>15</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.52e-22</td>\n",
       "      <td>14</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.78e-06</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.87e-05</td>\n",
       "      <td>5</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.00e-05</td>\n",
       "      <td>5</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.02e-04</td>\n",
       "      <td>3</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.21e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.16e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Gene set                Ontology       FDR  \\\n",
       "0                             ribosome  GO: cellular_component  4.29e-33   \n",
       "1                             Ribosome          KEGG: Pathways  4.70e-33   \n",
       "2   structural constituent of ribosome  GO: molecular_function  1.12e-28   \n",
       "3         structural molecule activity  GO: molecular_function  1.46e-26   \n",
       "4                          translation  GO: biological_process  7.52e-22   \n",
       "5                         rRNA binding  GO: molecular_function  3.78e-06   \n",
       "6                          RNA binding  GO: molecular_function  1.87e-05   \n",
       "7                              cytosol  GO: cellular_component  4.00e-05   \n",
       "8                  ribosome biogenesis  GO: biological_process  3.02e-04   \n",
       "9                  delayed development   Dictybase: Phenotypes  1.21e-02   \n",
       "10    anatomical structure development  GO: biological_process  2.16e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           17        93         93  \n",
       "1           17       105         97  \n",
       "2           15        85         85  \n",
       "3           15       117        117  \n",
       "4           14       169        169  \n",
       "5            3        13         13  \n",
       "6            5       169        169  \n",
       "7            5       203        203  \n",
       "8            3        61         61  \n",
       "9            2        69         69  \n",
       "10           2       440        438  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C6 (F)\n",
      "***  C6 selected: 49 with EID: 49\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 42 out of 49 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.09e-65</td>\n",
       "      <td>29</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.14e-31</td>\n",
       "      <td>21</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.27e-29</td>\n",
       "      <td>25</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.24e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>91</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.14e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.28e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>211</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.41e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Gene set                Ontology  \\\n",
       "0                                   Proteasome          KEGG: Pathways   \n",
       "1                           peptidase activity  GO: molecular_function   \n",
       "2                            catabolic process  GO: biological_process   \n",
       "3  Protein processing in endoplasmic reticulum          KEGG: Pathways   \n",
       "4                    enzyme regulator activity  GO: molecular_function   \n",
       "5                           response to stress  GO: biological_process   \n",
       "6                              ATPase activity  GO: molecular_function   \n",
       "7                          cytoplasmic vesicle  GO: cellular_component   \n",
       "8                        endoplasmic reticulum  GO: cellular_component   \n",
       "\n",
       "        FDR  N in query  Set size  N in ref.  \n",
       "0  1.09e-65          29        39         37  \n",
       "1  1.14e-31          21        91         91  \n",
       "2  9.27e-29          25       250        250  \n",
       "3  6.24e-02           3        91         88  \n",
       "4  8.14e-02           3       105        105  \n",
       "5  1.09e-01           5       360        358  \n",
       "6  1.77e-01           3       169        168  \n",
       "7  2.28e-01           3       211        211  \n",
       "8  2.41e-01           2       117        117  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C7 (G)\n",
      "***  C7 selected: 96 with EID: 96\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 30 out of 96 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.54e-05</td>\n",
       "      <td>16</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.20e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.28e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.28e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.90e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.15e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.15e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.20e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.31e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.96e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.29e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.29e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.29e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.75e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.59e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.59e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.59e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.59e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.59e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>phosphatase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.59e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.85e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.85e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.62e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.28e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.28e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.97e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>67</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.24e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>endosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.48e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>83</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Gene set                Ontology  \\\n",
       "0            anatomical structure development  GO: biological_process   \n",
       "1                        cell differentiation  GO: biological_process   \n",
       "2    development arrests at loose mound stage   Dictybase: Phenotypes   \n",
       "3                         signal transduction  GO: biological_process   \n",
       "4               Glycosaminoglycan degradation          KEGG: Pathways   \n",
       "5                        aberrant aggregation   Dictybase: Phenotypes   \n",
       "6                        extracellular region  GO: cellular_component   \n",
       "7                    increased slug migration   Dictybase: Phenotypes   \n",
       "8                    decreased aggregate size   Dictybase: Phenotypes   \n",
       "9                decreased fruiting body size   Dictybase: Phenotypes   \n",
       "10  development arrests at tipped mound stage   Dictybase: Phenotypes   \n",
       "11                                 locomotion  GO: biological_process   \n",
       "12                              cell adhesion  GO: biological_process   \n",
       "13                            plasma membrane  GO: cellular_component   \n",
       "14          aberrant fruiting body morphology   Dictybase: Phenotypes   \n",
       "15         development arrests at mound stage   Dictybase: Phenotypes   \n",
       "16                     multiple tips in mound   Dictybase: Phenotypes   \n",
       "17                    delayed gene expression   Dictybase: Phenotypes   \n",
       "18                        cell-cell signaling  GO: biological_process   \n",
       "19                       phosphatase activity  GO: molecular_function   \n",
       "20                      abolished sporulation   Dictybase: Phenotypes   \n",
       "21                  abolished gene expression   Dictybase: Phenotypes   \n",
       "22                  cytoskeleton organization  GO: biological_process   \n",
       "23                     precocious development   Dictybase: Phenotypes   \n",
       "24                         response to stress  GO: biological_process   \n",
       "25                  decreased gene expression   Dictybase: Phenotypes   \n",
       "26                  decreased chemotaxis rate   Dictybase: Phenotypes   \n",
       "27              aberrant F-actin organization   Dictybase: Phenotypes   \n",
       "28                         peptidase activity  GO: molecular_function   \n",
       "29                   abolished slug migration   Dictybase: Phenotypes   \n",
       "30                                  wild type   Dictybase: Phenotypes   \n",
       "31                       aberrant development   Dictybase: Phenotypes   \n",
       "32                               cytoskeleton  GO: cellular_component   \n",
       "33                      decreased sporulation   Dictybase: Phenotypes   \n",
       "34                cellular component assembly  GO: biological_process   \n",
       "35                              cell motility  GO: biological_process   \n",
       "36                        delayed aggregation   Dictybase: Phenotypes   \n",
       "37                    decreased cell motility   Dictybase: Phenotypes   \n",
       "38                      abolished aggregation   Dictybase: Phenotypes   \n",
       "39                                   endosome  GO: cellular_component   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   2.54e-05          16       440        438  \n",
       "1   7.20e-03           7       145        145  \n",
       "2   1.28e-02           3        20         20  \n",
       "3   1.28e-02          10       352        350  \n",
       "4   1.90e-02           2         6          6  \n",
       "5   2.15e-02           4        61         61  \n",
       "6   2.15e-02           6       157        157  \n",
       "7   2.20e-02           2         8          8  \n",
       "8   2.31e-02           3        32         32  \n",
       "9   2.96e-02           4        76         75  \n",
       "10  3.29e-02           2        12         12  \n",
       "11  3.29e-02           6       191        189  \n",
       "12  3.29e-02           4        84         83  \n",
       "13  3.75e-02           7       268        268  \n",
       "14  4.59e-02           4       101         98  \n",
       "15  4.59e-02           3        56         54  \n",
       "16  4.59e-02           2        18         18  \n",
       "17  4.59e-02           2        19         18  \n",
       "18  4.59e-02           2        16         16  \n",
       "19  4.59e-02           3        51         51  \n",
       "20  7.85e-02           2        25         25  \n",
       "21  7.85e-02           2        25         25  \n",
       "22  8.62e-02           5       197        196  \n",
       "23  9.28e-02           2        30         29  \n",
       "24  9.28e-02           7       360        358  \n",
       "25  1.22e-01           3        95         86  \n",
       "26  1.22e-01           2        36         36  \n",
       "27  1.22e-01           2        37         37  \n",
       "28  1.22e-01           3        91         91  \n",
       "29  1.40e-01           2        43         43  \n",
       "30  1.48e-01           6       341        338  \n",
       "31  1.64e-01           2        51         51  \n",
       "32  1.64e-01           4       185        185  \n",
       "33  1.68e-01           2        53         53  \n",
       "34  1.80e-01           4       202        201  \n",
       "35  1.97e-01           2        67         66  \n",
       "36  2.12e-01           2        72         72  \n",
       "37  2.24e-01           2        76         76  \n",
       "38  2.39e-01           2        82         80  \n",
       "39  2.48e-01           2        83         83  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C8 (H)\n",
      "***  C8 selected: 35 with EID: 35\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 35 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.78e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.13e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.31e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell death</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.40e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.05e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0          aberrant CRAC localization to the plasma membrane   \n",
       "1                  development arrests at tipped mound stage   \n",
       "2   anatomical structure formation involved in morphogenesis   \n",
       "3                                                 cell death   \n",
       "4                                       aberrant aggregation   \n",
       "5                               decreased chemotaxis to cAMP   \n",
       "6                                       cell differentiation   \n",
       "7                           anatomical structure development   \n",
       "8                                                 locomotion   \n",
       "9                                            kinase activity   \n",
       "10                                           RNA degradation   \n",
       "11                              decreased fruiting body size   \n",
       "12                         aberrant fruiting body morphology   \n",
       "13                                                 wild type   \n",
       "14                                              cytoskeleton   \n",
       "15                                               DNA binding   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0    Dictybase: Phenotypes  9.78e-03           2         6          6  \n",
       "1    Dictybase: Phenotypes  2.13e-02           2        12         12  \n",
       "2   GO: biological_process  3.31e-02           3        74         74  \n",
       "3   GO: biological_process  4.73e-02           2        25         25  \n",
       "4    Dictybase: Phenotypes  5.92e-02           2        61         61  \n",
       "5    Dictybase: Phenotypes  5.92e-02           2        63         63  \n",
       "6   GO: biological_process  5.92e-02           3       145        145  \n",
       "7   GO: biological_process  5.92e-02           5       440        438  \n",
       "8   GO: biological_process  5.92e-02           3       191        189  \n",
       "9   GO: molecular_function  5.92e-02           4       328        327  \n",
       "10          KEGG: Pathways  5.92e-02           2        48         48  \n",
       "11   Dictybase: Phenotypes  6.40e-02           2        76         75  \n",
       "12   Dictybase: Phenotypes  8.05e-02           2       101         98  \n",
       "13   Dictybase: Phenotypes  1.23e-01           3       341        338  \n",
       "14  GO: cellular_component  1.55e-01           2       185        185  \n",
       "15  GO: molecular_function  1.91e-01           2       221        221  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C9 (I)\n",
      "***  C9 selected: 38 with EID: 38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 38 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.25e-14</td>\n",
       "      <td>13</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.68e-10</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.81e-09</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.57e-07</td>\n",
       "      <td>7</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.78e-07</td>\n",
       "      <td>7</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.78e-07</td>\n",
       "      <td>5</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.02e-07</td>\n",
       "      <td>6</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.70e-07</td>\n",
       "      <td>6</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.75e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.43e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.74e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.53e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Gene set                Ontology       FDR  \\\n",
       "0                      cell cycle  GO: biological_process  4.25e-14   \n",
       "1          chromosome segregation  GO: biological_process  1.68e-10   \n",
       "2        mitotic nuclear division  GO: biological_process  6.81e-09   \n",
       "3              mitotic cell cycle  GO: biological_process  1.57e-07   \n",
       "4                   cell division  GO: biological_process  2.78e-07   \n",
       "5                 DNA replication          KEGG: Pathways  2.78e-07   \n",
       "6                      chromosome  GO: cellular_component  3.02e-07   \n",
       "7         chromosome organization  GO: biological_process  3.70e-07   \n",
       "8                    cytoskeleton  GO: cellular_component  4.75e-06   \n",
       "9       cytoskeleton organization  GO: biological_process  1.04e-03   \n",
       "10  microtubule organizing center  GO: cellular_component  1.43e-03   \n",
       "11   cytoskeletal protein binding  GO: molecular_function  3.74e-03   \n",
       "12          DNA metabolic process  GO: biological_process  8.53e-02   \n",
       "13                ATPase activity  GO: molecular_function  1.56e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           13       167        167  \n",
       "1            6        18         18  \n",
       "2            5        15         15  \n",
       "3            7       100        100  \n",
       "4            7       115        115  \n",
       "5            5        34         33  \n",
       "6            6        70         70  \n",
       "7            6        74         74  \n",
       "8            7       185        185  \n",
       "9            5       197        196  \n",
       "10           3        50         50  \n",
       "11           4       158        158  \n",
       "12           2       105        105  \n",
       "13           2       169        168  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C10 (J)\n",
      "***  C10 selected: 70 with EID: 70\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 70 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.18e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                Ontology       FDR  N in query  \\\n",
       "0       kinase activity  GO: molecular_function  1.45e-01           6   \n",
       "1  extracellular region  GO: cellular_component  1.45e-01           4   \n",
       "2         cell adhesion  GO: biological_process  2.18e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       328        327  \n",
       "1       157        157  \n",
       "2        84         83  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C11 (K)\n",
      "***  C11 selected: 85 with EID: 85\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 7 out of 85 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.09e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                Ontology       FDR  N in query  \\\n",
       "0  extracellular region  GO: cellular_component  6.09e-03           7   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       157        157  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C12 (L)\n",
      "***  C12 selected: 73 with EID: 73\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 73 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.50e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene set                Ontology       FDR  N in query  \\\n",
       "0  cell-cell signaling  GO: biological_process  5.50e-02           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0        16         16  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C13 (M)\n",
      "***  C13 selected: 49 with EID: 49\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 49 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.62e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.62e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.03e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.96e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.07e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.23e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>89</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       cell differentiation   \n",
       "1   anatomical structure formation involved in morphogenesis   \n",
       "2                  DNA-binding transcription factor activity   \n",
       "3                           anatomical structure development   \n",
       "4                                  decreased gene expression   \n",
       "5                              Starch and sucrose metabolism   \n",
       "6                                  decreased spore viability   \n",
       "7                                                DNA binding   \n",
       "8                                        signal transduction   \n",
       "9                                               reproduction   \n",
       "10                            carbohydrate metabolic process   \n",
       "11                                 cytoskeleton organization   \n",
       "12                                             cell adhesion   \n",
       "13            generation of precursor metabolites and energy   \n",
       "14                                           kinase activity   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  1.62e-03           6       145        145  \n",
       "1   GO: biological_process  7.62e-03           4        74         74  \n",
       "2   GO: molecular_function  1.03e-02           3        37         37  \n",
       "3   GO: biological_process  2.96e-02           7       440        438  \n",
       "4    Dictybase: Phenotypes  7.07e-02           3        95         86  \n",
       "5           KEGG: Pathways  8.23e-02           2        31         31  \n",
       "6    Dictybase: Phenotypes  1.09e-01           2        60         59  \n",
       "7   GO: molecular_function  1.09e-01           4       221        221  \n",
       "8   GO: biological_process  1.13e-01           4       352        350  \n",
       "9   GO: biological_process  1.13e-01           2        79         79  \n",
       "10  GO: biological_process  1.13e-01           2        89         88  \n",
       "11  GO: biological_process  1.13e-01           3       197        196  \n",
       "12  GO: biological_process  1.13e-01           2        84         83  \n",
       "13  GO: biological_process  1.13e-01           2        77         77  \n",
       "14  GO: molecular_function  2.25e-01           3       328        327  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C14 (N)\n",
      "***  C14 selected: 87 with EID: 87\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 87 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.64e-09</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.64e-09</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.07e-07</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.59e-07</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.30e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.44e-04</td>\n",
       "      <td>13</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.60e-04</td>\n",
       "      <td>8</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.96e-04</td>\n",
       "      <td>7</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.20e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.09e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>9.49e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.46e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.46e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.46e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.46e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>28</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.17e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.18e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Glyoxylate and dicarboxylate metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>104</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.27e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.43e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>lyase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.96e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>66</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.96e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>150</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                           external encapsulating structure   \n",
       "1                                                  cell wall   \n",
       "2                             aberrant spore coat morphology   \n",
       "3                       cell wall organization or biogenesis   \n",
       "4   anatomical structure formation involved in morphogenesis   \n",
       "5                           anatomical structure development   \n",
       "6                                       extracellular region   \n",
       "7                                       cell differentiation   \n",
       "8                                aberrant cell-cell adhesion   \n",
       "9                                  decreased spore viability   \n",
       "10                                      extracellular matrix   \n",
       "11                               cellular component assembly   \n",
       "12                                        hydrolase activity   \n",
       "13                     Biosynthesis of secondary metabolites   \n",
       "14                                    Glutathione metabolism   \n",
       "15                                 Citrate cycle (TCA cycle)   \n",
       "16                             Starch and sucrose metabolism   \n",
       "17                              cytoskeletal protein binding   \n",
       "18                   Glyoxylate and dicarboxylate metabolism   \n",
       "19                                         catabolic process   \n",
       "20                                         Carbon metabolism   \n",
       "21                                           plasma membrane   \n",
       "22                              structural molecule activity   \n",
       "23                                                   cytosol   \n",
       "24                                     abolished culmination   \n",
       "25                                               DNA binding   \n",
       "26                                            lyase activity   \n",
       "27                        transmembrane transporter activity   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  6.64e-09           6        13         13  \n",
       "1   GO: cellular_component  6.64e-09           6        13         13  \n",
       "2    Dictybase: Phenotypes  3.07e-07           5        12         12  \n",
       "3   GO: biological_process  8.59e-07           5        15         15  \n",
       "4   GO: biological_process  1.30e-05           7        74         74  \n",
       "5   GO: biological_process  1.44e-04          13       440        438  \n",
       "6   GO: cellular_component  1.60e-04           8       157        157  \n",
       "7   GO: biological_process  6.96e-04           7       145        145  \n",
       "8    Dictybase: Phenotypes  4.20e-03           2         5          5  \n",
       "9    Dictybase: Phenotypes  6.09e-03           4        60         59  \n",
       "10  GO: cellular_component  9.49e-03           2         8          8  \n",
       "11  GO: biological_process  7.46e-02           5       202        201  \n",
       "12  GO: molecular_function  7.46e-02           2        25         25  \n",
       "13          KEGG: Pathways  7.46e-02           6       289        281  \n",
       "14          KEGG: Pathways  7.46e-02           2        28         25  \n",
       "15          KEGG: Pathways  9.17e-02           2        30         30  \n",
       "16          KEGG: Pathways  9.18e-02           2        31         31  \n",
       "17  GO: molecular_function  1.10e-01           4       158        158  \n",
       "18          KEGG: Pathways  1.15e-01           2        37         37  \n",
       "19  GO: biological_process  1.15e-01           5       250        250  \n",
       "20          KEGG: Pathways  1.15e-01           3       104        102  \n",
       "21  GO: cellular_component  1.27e-01           5       268        268  \n",
       "22  GO: molecular_function  1.43e-01           3       117        117  \n",
       "23  GO: cellular_component  1.54e-01           4       203        203  \n",
       "24   Dictybase: Phenotypes  1.84e-01           2        62         62  \n",
       "25  GO: molecular_function  1.84e-01           4       221        221  \n",
       "26  GO: molecular_function  1.96e-01           2        66         66  \n",
       "27  GO: molecular_function  1.96e-01           3       150        149  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C15 (O)\n",
      "***  C15 selected: 31 with EID: 31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C16 (P)\n",
      "***  C16 selected: 72 with EID: 72\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 72 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.91e-07</td>\n",
       "      <td>10</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                Ontology       FDR  N in query  \\\n",
       "0  extracellular region  GO: cellular_component  5.91e-07          10   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       157        157  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C17 (Q)\n",
      "***  C17 selected: 57 with EID: 56\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 56 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.23e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                Ontology       FDR  N in query  \\\n",
       "0              cell adhesion  GO: biological_process  1.23e-02           4   \n",
       "1  decreased spore viability   Dictybase: Phenotypes  1.80e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0        84         83  \n",
       "1        60         59  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C18 (R)\n",
      "***  C18 selected: 65 with EID: 63\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 63 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.83e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.83e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.02e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.05e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>lipid binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>59</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.25e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                          external encapsulating structure   \n",
       "1                                                 cell wall   \n",
       "2  anatomical structure formation involved in morphogenesis   \n",
       "3                                        hydrolase activity   \n",
       "4                                      cell differentiation   \n",
       "5                                             lipid binding   \n",
       "6                                      extracellular region   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: cellular_component  1.83e-02           2        13         13  \n",
       "1  GO: cellular_component  1.83e-02           2        13         13  \n",
       "2  GO: biological_process  4.02e-02           3        74         74  \n",
       "3  GO: molecular_function  1.05e-01           2        46         45  \n",
       "4  GO: biological_process  1.20e-01           3       145        145  \n",
       "5  GO: molecular_function  1.20e-01           2        59         59  \n",
       "6  GO: cellular_component  1.25e-01           3       157        157  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C19 (S)\n",
      "***  C19 selected: 89 with EID: 88\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 88 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.40e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.81e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.71e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.71e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1  anatomical structure formation involved in morphogenesis   \n",
       "2                                      cell differentiation   \n",
       "3                             Starch and sucrose metabolism   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: cellular_component  2.40e-03           7       157        157  \n",
       "1  GO: biological_process  1.81e-02           4        74         74  \n",
       "2  GO: biological_process  9.71e-02           4       145        145  \n",
       "3          KEGG: Pathways  9.71e-02           2        31         31  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C20 (T)\n",
      "***  C20 selected: 1 with EID: 1\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 1 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    cluster='C'+str(group)\n",
    "    print('*** Cluster %s (%s)' %(cluster,string.ascii_uppercase[group-1]))\n",
    "    \n",
    "    enriched=group_diff_enrichment(data=regulons,group=cluster,min_overlap=2)\n",
    "    if  save_enrichment and enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log_enrichment_cluster'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Using only genes annotated with at least one gene set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*** Cluster C1 (A)\n",
      "***  C1 selected: 31 with EID: 31\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.71\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.11e-15</td>\n",
       "      <td>12</td>\n",
       "      <td>72</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.24e-13</td>\n",
       "      <td>11</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.48e-12</td>\n",
       "      <td>10</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribonucleoprotein complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.12e-05</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.02e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.87e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.71e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.17e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.54e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.06e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>58</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Gene set                Ontology       FDR  \\\n",
       "0    Ribosome biogenesis in eukaryotes          KEGG: Pathways  3.11e-15   \n",
       "1                            nucleolus  GO: cellular_component  2.24e-13   \n",
       "2                  ribosome biogenesis  GO: biological_process  1.48e-12   \n",
       "3   ribonucleoprotein complex assembly  GO: biological_process  1.12e-05   \n",
       "4                          RNA binding  GO: molecular_function  1.02e-03   \n",
       "5                         rRNA binding  GO: molecular_function  7.87e-03   \n",
       "6  protein-containing complex assembly  GO: biological_process  1.71e-02   \n",
       "7          cellular component assembly  GO: biological_process  6.17e-02   \n",
       "8                    helicase activity  GO: molecular_function  7.54e-02   \n",
       "9           methyltransferase activity  GO: molecular_function  8.06e-02   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0          12        72         68  \n",
       "1          11        72         72  \n",
       "2          10        61         61  \n",
       "3           4        18         18  \n",
       "4           6       169        169  \n",
       "5           2        13         13  \n",
       "6           4       134        133  \n",
       "7           4       202        201  \n",
       "8           2        51         51  \n",
       "9           2        58         56  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C2 (B)\n",
      "***  C2 selected: 54 with EID: 54\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.57\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 24 out of 31 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.95e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>71</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.46e-05</td>\n",
       "      <td>11</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.63e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.05e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.08e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.78e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.08e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>53</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>104</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                Ontology  \\\n",
       "0                        Oxidative phosphorylation          KEGG: Pathways   \n",
       "1                                    mitochondrion  GO: cellular_component   \n",
       "2   generation of precursor metabolites and energy  GO: biological_process   \n",
       "3                                  protein folding  GO: biological_process   \n",
       "4                         unfolded protein binding  GO: molecular_function   \n",
       "5                          oxidoreductase activity  GO: molecular_function   \n",
       "6                                        Phagosome          KEGG: Pathways   \n",
       "7                        Citrate cycle (TCA cycle)          KEGG: Pathways   \n",
       "8                            Propanoate metabolism          KEGG: Pathways   \n",
       "9                                         Ribosome          KEGG: Pathways   \n",
       "10                               Carbon metabolism          KEGG: Pathways   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   5.95e-06           7        71         54  \n",
       "1   1.46e-05          11       229        229  \n",
       "2   3.63e-04           6        77         77  \n",
       "3   2.05e-03           4        37         37  \n",
       "4   1.08e-02           3        27         27  \n",
       "5   3.78e-02           6       210        207  \n",
       "6   5.08e-02           3        53         52  \n",
       "7   1.34e-01           2        30         30  \n",
       "8   1.35e-01           2        32         32  \n",
       "9   1.87e-01           3       105         97  \n",
       "10  1.93e-01           3       104        102  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C3 (C)\n",
      "***  C3 selected: 54 with EID: 54\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.89\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 44 out of 48 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.42e-70</td>\n",
       "      <td>44</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.13e-68</td>\n",
       "      <td>43</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.10e-60</td>\n",
       "      <td>39</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.96e-53</td>\n",
       "      <td>39</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.62e-46</td>\n",
       "      <td>39</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.27e-05</td>\n",
       "      <td>11</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.69e-05</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                Ontology       FDR  \\\n",
       "0                            Ribosome          KEGG: Pathways  7.42e-70   \n",
       "1                            ribosome  GO: cellular_component  2.13e-68   \n",
       "2  structural constituent of ribosome  GO: molecular_function  4.10e-60   \n",
       "3        structural molecule activity  GO: molecular_function  1.96e-53   \n",
       "4                         translation  GO: biological_process  2.62e-46   \n",
       "5                         RNA binding  GO: molecular_function  1.27e-05   \n",
       "6                        rRNA binding  GO: molecular_function  3.69e-05   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0          44       105         97  \n",
       "1          43        93         93  \n",
       "2          39        85         85  \n",
       "3          39       117        117  \n",
       "4          39       169        169  \n",
       "5          11       169        169  \n",
       "6           4        13         13  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C4 (D)\n",
      "***  C4 selected: 43 with EID: 43\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.51\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 20 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.32e-09</td>\n",
       "      <td>11</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.64e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.64e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.64e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.15e-04</td>\n",
       "      <td>7</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.15e-04</td>\n",
       "      <td>5</td>\n",
       "      <td>77</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.57e-04</td>\n",
       "      <td>8</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.72e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.74e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.88e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.11e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.85e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>211</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.82e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.82e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.01e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>233</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                Ontology       FDR  \\\n",
       "0          cytoskeletal protein binding  GO: molecular_function  8.32e-09   \n",
       "1   protein-containing complex assembly  GO: biological_process  9.64e-05   \n",
       "2           cellular component assembly  GO: biological_process  9.64e-05   \n",
       "3             cytoskeleton organization  GO: biological_process  9.64e-05   \n",
       "4                          cytoskeleton  GO: cellular_component  4.15e-04   \n",
       "5                           Endocytosis          KEGG: Pathways  4.15e-04   \n",
       "6                       plasma membrane  GO: cellular_component  4.57e-04   \n",
       "7                    response to stress  GO: biological_process  1.72e-02   \n",
       "8                               cytosol  GO: cellular_component  2.74e-02   \n",
       "9                    cell morphogenesis  GO: biological_process  3.88e-02   \n",
       "10              decreased cell motility   Dictybase: Phenotypes  4.11e-02   \n",
       "11             aberrant cell morphology   Dictybase: Phenotypes  9.85e-02   \n",
       "12                  cytoplasmic vesicle  GO: cellular_component  1.17e-01   \n",
       "13                   aberrant streaming   Dictybase: Phenotypes  1.75e-01   \n",
       "14                abolished aggregation   Dictybase: Phenotypes  1.82e-01   \n",
       "15                         reproduction  GO: biological_process  1.82e-01   \n",
       "16     anatomical structure development  GO: biological_process  2.01e-01   \n",
       "17                 aberrant cytokinesis   Dictybase: Phenotypes  2.17e-01   \n",
       "18                   mitotic cell cycle  GO: biological_process  2.17e-01   \n",
       "19                            wild type   Dictybase: Phenotypes  2.21e-01   \n",
       "20                        cell division  GO: biological_process  2.45e-01   \n",
       "21           vesicle-mediated transport  GO: biological_process  2.45e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           11       158        158  \n",
       "1            7       134        133  \n",
       "2            8       202        201  \n",
       "3            8       197        196  \n",
       "4            7       185        185  \n",
       "5            5        77         75  \n",
       "6            8       268        268  \n",
       "7            7       360        358  \n",
       "8            5       203        203  \n",
       "9            2        23         23  \n",
       "10           3        76         76  \n",
       "11           2        42         41  \n",
       "12           4       211        211  \n",
       "13           2        72         72  \n",
       "14           2        82         80  \n",
       "15           2        79         79  \n",
       "16           5       440        438  \n",
       "17           2        98         98  \n",
       "18           2       100        100  \n",
       "19           4       341        338  \n",
       "20           2       115        115  \n",
       "21           3       233        233  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C5 (E)\n",
      "***  C5 selected: 20 with EID: 20\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.9\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.94e-26</td>\n",
       "      <td>17</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.12e-26</td>\n",
       "      <td>17</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.67e-22</td>\n",
       "      <td>15</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.14e-20</td>\n",
       "      <td>15</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.04e-16</td>\n",
       "      <td>14</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.23e-05</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.36e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.75e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.01e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.53e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                Ontology       FDR  \\\n",
       "0                            ribosome  GO: cellular_component  1.94e-26   \n",
       "1                            Ribosome          KEGG: Pathways  2.12e-26   \n",
       "2  structural constituent of ribosome  GO: molecular_function  1.67e-22   \n",
       "3        structural molecule activity  GO: molecular_function  2.14e-20   \n",
       "4                         translation  GO: biological_process  5.04e-16   \n",
       "5                        rRNA binding  GO: molecular_function  8.23e-05   \n",
       "6                         RNA binding  GO: molecular_function  2.36e-03   \n",
       "7                             cytosol  GO: cellular_component  4.75e-03   \n",
       "8                 ribosome biogenesis  GO: biological_process  6.01e-03   \n",
       "9                 delayed development   Dictybase: Phenotypes  8.53e-02   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0          17        93         93  \n",
       "1          17       105         97  \n",
       "2          15        85         85  \n",
       "3          15       117        117  \n",
       "4          14       169        169  \n",
       "5           3        13         13  \n",
       "6           5       169        169  \n",
       "7           5       203        203  \n",
       "8           3        61         61  \n",
       "9           2        69         69  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C6 (F)\n",
      "***  C6 selected: 49 with EID: 49\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.94\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 35 out of 46 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.82e-52</td>\n",
       "      <td>29</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.33e-22</td>\n",
       "      <td>21</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.31e-17</td>\n",
       "      <td>25</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene set                Ontology       FDR  N in query  Set size  \\\n",
       "0          Proteasome          KEGG: Pathways  1.82e-52          29        39   \n",
       "1  peptidase activity  GO: molecular_function  4.33e-22          21        91   \n",
       "2   catabolic process  GO: biological_process  1.31e-17          25       250   \n",
       "\n",
       "   N in ref.  \n",
       "0         37  \n",
       "1         91  \n",
       "2        250  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C7 (G)\n",
      "***  C7 selected: 96 with EID: 96\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.39\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 30 out of 37 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.54e-05</td>\n",
       "      <td>16</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.77e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Glycosaminoglycan degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.65e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.65e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.79e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.79e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.27e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.27e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.34e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.34e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.20e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.20e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.20e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>phosphatase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.20e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.20e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.27e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.40e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.57e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.73e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.94e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Gene set                Ontology  \\\n",
       "0            anatomical structure development  GO: biological_process   \n",
       "1                        cell differentiation  GO: biological_process   \n",
       "2    development arrests at loose mound stage   Dictybase: Phenotypes   \n",
       "3                         signal transduction  GO: biological_process   \n",
       "4               Glycosaminoglycan degradation          KEGG: Pathways   \n",
       "5                    increased slug migration   Dictybase: Phenotypes   \n",
       "6                        aberrant aggregation   Dictybase: Phenotypes   \n",
       "7                    decreased aggregate size   Dictybase: Phenotypes   \n",
       "8                        extracellular region  GO: cellular_component   \n",
       "9                decreased fruiting body size   Dictybase: Phenotypes   \n",
       "10  development arrests at tipped mound stage   Dictybase: Phenotypes   \n",
       "11                                 locomotion  GO: biological_process   \n",
       "12                              cell adhesion  GO: biological_process   \n",
       "13                     multiple tips in mound   Dictybase: Phenotypes   \n",
       "14                    delayed gene expression   Dictybase: Phenotypes   \n",
       "15                        cell-cell signaling  GO: biological_process   \n",
       "16                       phosphatase activity  GO: molecular_function   \n",
       "17                            plasma membrane  GO: cellular_component   \n",
       "18          aberrant fruiting body morphology   Dictybase: Phenotypes   \n",
       "19         development arrests at mound stage   Dictybase: Phenotypes   \n",
       "20                      abolished sporulation   Dictybase: Phenotypes   \n",
       "21                  abolished gene expression   Dictybase: Phenotypes   \n",
       "22                     precocious development   Dictybase: Phenotypes   \n",
       "23                  cytoskeleton organization  GO: biological_process   \n",
       "24                  decreased chemotaxis rate   Dictybase: Phenotypes   \n",
       "25                  decreased gene expression   Dictybase: Phenotypes   \n",
       "26              aberrant F-actin organization   Dictybase: Phenotypes   \n",
       "27                         response to stress  GO: biological_process   \n",
       "28                         peptidase activity  GO: molecular_function   \n",
       "29                   abolished slug migration   Dictybase: Phenotypes   \n",
       "30                       aberrant development   Dictybase: Phenotypes   \n",
       "31                      decreased sporulation   Dictybase: Phenotypes   \n",
       "32                                  wild type   Dictybase: Phenotypes   \n",
       "33                               cytoskeleton  GO: cellular_component   \n",
       "34                cellular component assembly  GO: biological_process   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   6.54e-05          16       440        438  \n",
       "1   1.77e-02           7       145        145  \n",
       "2   2.73e-02           3        20         20  \n",
       "3   2.73e-02          10       352        350  \n",
       "4   2.73e-02           2         6          6  \n",
       "5   3.65e-02           2         8          8  \n",
       "6   3.65e-02           4        61         61  \n",
       "7   3.79e-02           3        32         32  \n",
       "8   3.79e-02           6       157        157  \n",
       "9   5.27e-02           4        76         75  \n",
       "10  5.27e-02           2        12         12  \n",
       "11  6.34e-02           6       191        189  \n",
       "12  6.34e-02           4        84         83  \n",
       "13  7.20e-02           2        18         18  \n",
       "14  7.20e-02           2        19         18  \n",
       "15  7.20e-02           2        16         16  \n",
       "16  7.20e-02           3        51         51  \n",
       "17  7.20e-02           7       268        268  \n",
       "18  7.27e-02           4       101         98  \n",
       "19  7.40e-02           3        56         54  \n",
       "20  1.11e-01           2        25         25  \n",
       "21  1.11e-01           2        25         25  \n",
       "22  1.40e-01           2        30         29  \n",
       "23  1.57e-01           5       197        196  \n",
       "24  1.58e-01           2        36         36  \n",
       "25  1.58e-01           3        95         86  \n",
       "26  1.58e-01           2        37         37  \n",
       "27  1.58e-01           7       360        358  \n",
       "28  1.73e-01           3        91         91  \n",
       "29  1.94e-01           2        43         43  \n",
       "30  2.27e-01           2        51         51  \n",
       "31  2.27e-01           2        53         53  \n",
       "32  2.27e-01           6       341        338  \n",
       "33  2.27e-01           4       185        185  \n",
       "34  2.45e-01           4       202        201  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C8 (H)\n",
      "***  C8 selected: 35 with EID: 35\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 11 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant CRAC localization to the plasma membrane</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.47e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.63e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell death</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.09e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.09e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.09e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.09e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.09e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>RNA degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.09e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.77e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.77e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.77e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.05e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0          aberrant CRAC localization to the plasma membrane   \n",
       "1                  development arrests at tipped mound stage   \n",
       "2   anatomical structure formation involved in morphogenesis   \n",
       "3                                                 cell death   \n",
       "4                                       aberrant aggregation   \n",
       "5                               decreased chemotaxis to cAMP   \n",
       "6                                       cell differentiation   \n",
       "7                           anatomical structure development   \n",
       "8                                            RNA degradation   \n",
       "9                               decreased fruiting body size   \n",
       "10                                                locomotion   \n",
       "11                                           kinase activity   \n",
       "12                         aberrant fruiting body morphology   \n",
       "13                                                 wild type   \n",
       "14                                              cytoskeleton   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0    Dictybase: Phenotypes  1.47e-02           2         6          6  \n",
       "1    Dictybase: Phenotypes  3.20e-02           2        12         12  \n",
       "2   GO: biological_process  5.63e-02           3        74         74  \n",
       "3   GO: biological_process  7.09e-02           2        25         25  \n",
       "4    Dictybase: Phenotypes  8.09e-02           2        61         61  \n",
       "5    Dictybase: Phenotypes  8.09e-02           2        63         63  \n",
       "6   GO: biological_process  8.09e-02           3       145        145  \n",
       "7   GO: biological_process  8.09e-02           5       440        438  \n",
       "8           KEGG: Pathways  8.09e-02           2        48         48  \n",
       "9    Dictybase: Phenotypes  8.77e-02           2        76         75  \n",
       "10  GO: biological_process  8.77e-02           3       191        189  \n",
       "11  GO: molecular_function  8.77e-02           4       328        327  \n",
       "12   Dictybase: Phenotypes  1.05e-01           2       101         98  \n",
       "13   Dictybase: Phenotypes  1.86e-01           3       341        338  \n",
       "14  GO: cellular_component  2.14e-01           2       185        185  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C9 (I)\n",
      "***  C9 selected: 38 with EID: 38\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.5\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 18 out of 19 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.67e-13</td>\n",
       "      <td>13</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.58e-09</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.78e-08</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.81e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.15e-06</td>\n",
       "      <td>5</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.74e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.74e-06</td>\n",
       "      <td>6</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.35e-06</td>\n",
       "      <td>6</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.22e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.37e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.76e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.67e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Gene set                Ontology       FDR  \\\n",
       "0                      cell cycle  GO: biological_process  6.67e-13   \n",
       "1          chromosome segregation  GO: biological_process  1.58e-09   \n",
       "2        mitotic nuclear division  GO: biological_process  4.78e-08   \n",
       "3              mitotic cell cycle  GO: biological_process  1.81e-06   \n",
       "4                 DNA replication          KEGG: Pathways  2.15e-06   \n",
       "5                   cell division  GO: biological_process  2.74e-06   \n",
       "6                      chromosome  GO: cellular_component  2.74e-06   \n",
       "7         chromosome organization  GO: biological_process  3.35e-06   \n",
       "8                    cytoskeleton  GO: cellular_component  5.22e-05   \n",
       "9   microtubule organizing center  GO: cellular_component  5.37e-03   \n",
       "10      cytoskeleton organization  GO: biological_process  5.76e-03   \n",
       "11   cytoskeletal protein binding  GO: molecular_function  1.67e-02   \n",
       "12          DNA metabolic process  GO: biological_process  1.86e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           13       167        167  \n",
       "1            6        18         18  \n",
       "2            5        15         15  \n",
       "3            7       100        100  \n",
       "4            5        34         33  \n",
       "5            7       115        115  \n",
       "6            6        70         70  \n",
       "7            6        74         74  \n",
       "8            7       185        185  \n",
       "9            3        50         50  \n",
       "10           5       197        196  \n",
       "11           4       158        158  \n",
       "12           2       105        105  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C10 (J)\n",
      "***  C10 selected: 70 with EID: 70\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.80e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.80e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                Ontology       FDR  N in query  \\\n",
       "0       kinase activity  GO: molecular_function  5.80e-02           6   \n",
       "1  extracellular region  GO: cellular_component  5.80e-02           4   \n",
       "2         cell adhesion  GO: biological_process  1.74e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       328        327  \n",
       "1       157        157  \n",
       "2        84         83  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C11 (K)\n",
      "***  C11 selected: 85 with EID: 85\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.24\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 7 out of 20 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.26e-04</td>\n",
       "      <td>7</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                Ontology       FDR  N in query  \\\n",
       "0  extracellular region  GO: cellular_component  4.26e-04           7   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       157        157  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C12 (L)\n",
      "***  C12 selected: 73 with EID: 73\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.11\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 5 out of 8 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.01e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.17e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                Ontology       FDR  \\\n",
       "0               cell-cell signaling  GO: biological_process  6.01e-03   \n",
       "1  anatomical structure development  GO: biological_process  5.17e-02   \n",
       "2              extracellular region  GO: cellular_component  1.33e-01   \n",
       "3                         wild type   Dictybase: Phenotypes  2.33e-01   \n",
       "4               signal transduction  GO: biological_process  2.33e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           2        16         16  \n",
       "1           4       440        438  \n",
       "2           2       157        157  \n",
       "3           2       341        338  \n",
       "4           2       352        350  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C13 (M)\n",
      "***  C13 selected: 49 with EID: 49\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.40e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.00e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.98e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.09e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.63e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.32e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.59e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>89</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.98e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       cell differentiation   \n",
       "1   anatomical structure formation involved in morphogenesis   \n",
       "2                  DNA-binding transcription factor activity   \n",
       "3                           anatomical structure development   \n",
       "4                                  decreased gene expression   \n",
       "5                              Starch and sucrose metabolism   \n",
       "6                                                DNA binding   \n",
       "7                                  decreased spore viability   \n",
       "8                                        signal transduction   \n",
       "9                                               reproduction   \n",
       "10                            carbohydrate metabolic process   \n",
       "11                                 cytoskeleton organization   \n",
       "12                                             cell adhesion   \n",
       "13            generation of precursor metabolites and energy   \n",
       "14                                           kinase activity   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  6.40e-04           6       145        145  \n",
       "1   GO: biological_process  5.00e-03           4        74         74  \n",
       "2   GO: molecular_function  7.98e-03           3        37         37  \n",
       "3   GO: biological_process  1.09e-02           7       440        438  \n",
       "4    Dictybase: Phenotypes  5.63e-02           3        95         86  \n",
       "5           KEGG: Pathways  7.32e-02           2        31         31  \n",
       "6   GO: molecular_function  8.59e-02           4       221        221  \n",
       "7    Dictybase: Phenotypes  1.04e-01           2        60         59  \n",
       "8   GO: biological_process  1.04e-01           4       352        350  \n",
       "9   GO: biological_process  1.04e-01           2        79         79  \n",
       "10  GO: biological_process  1.04e-01           2        89         88  \n",
       "11  GO: biological_process  1.04e-01           3       197        196  \n",
       "12  GO: biological_process  1.04e-01           2        84         83  \n",
       "13  GO: biological_process  1.04e-01           2        77         77  \n",
       "14  GO: molecular_function  1.98e-01           3       328        327  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C14 (N)\n",
      "***  C14 selected: 87 with EID: 87\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.45\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 33 out of 39 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.07e-08</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.07e-08</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.45e-06</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.02e-06</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.33e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.51e-03</td>\n",
       "      <td>8</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.03e-03</td>\n",
       "      <td>13</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.48e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.16e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.84e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.84e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.50e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.50e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>28</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Citrate cycle (TCA cycle)</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.94e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.94e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Glyoxylate and dicarboxylate metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.36e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                           external encapsulating structure   \n",
       "1                                                  cell wall   \n",
       "2                             aberrant spore coat morphology   \n",
       "3                       cell wall organization or biogenesis   \n",
       "4   anatomical structure formation involved in morphogenesis   \n",
       "5                                       extracellular region   \n",
       "6                           anatomical structure development   \n",
       "7                                       cell differentiation   \n",
       "8                                aberrant cell-cell adhesion   \n",
       "9                                  decreased spore viability   \n",
       "10                                      extracellular matrix   \n",
       "11                                        hydrolase activity   \n",
       "12                                    Glutathione metabolism   \n",
       "13                                 Citrate cycle (TCA cycle)   \n",
       "14                             Starch and sucrose metabolism   \n",
       "15                               cellular component assembly   \n",
       "16                   Glyoxylate and dicarboxylate metabolism   \n",
       "17                     Biosynthesis of secondary metabolites   \n",
       "18                              cytoskeletal protein binding   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  4.07e-08           6        13         13  \n",
       "1   GO: cellular_component  4.07e-08           6        13         13  \n",
       "2    Dictybase: Phenotypes  1.45e-06           5        12         12  \n",
       "3   GO: biological_process  4.02e-06           5        15         15  \n",
       "4   GO: biological_process  9.33e-05           7        74         74  \n",
       "5   GO: cellular_component  1.51e-03           8       157        157  \n",
       "6   GO: biological_process  2.03e-03          13       440        438  \n",
       "7   GO: biological_process  4.48e-03           7       145        145  \n",
       "8    Dictybase: Phenotypes  8.16e-03           2         5          5  \n",
       "9    Dictybase: Phenotypes  1.84e-02           4        60         59  \n",
       "10  GO: cellular_component  1.84e-02           2         8          8  \n",
       "11  GO: molecular_function  1.50e-01           2        25         25  \n",
       "12          KEGG: Pathways  1.50e-01           2        28         25  \n",
       "13          KEGG: Pathways  1.94e-01           2        30         30  \n",
       "14          KEGG: Pathways  1.94e-01           2        31         31  \n",
       "15  GO: biological_process  2.00e-01           5       202        201  \n",
       "16          KEGG: Pathways  2.00e-01           2        37         37  \n",
       "17          KEGG: Pathways  2.08e-01           6       289        281  \n",
       "18  GO: molecular_function  2.36e-01           4       158        158  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C15 (O)\n",
      "***  C15 selected: 31 with EID: 31\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 8 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Gene set                Ontology       FDR  N in query  \\\n",
       "0  oxidoreductase activity  GO: molecular_function  2.39e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       210        207  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C16 (P)\n",
      "***  C16 selected: 72 with EID: 72\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 12 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.74e-09</td>\n",
       "      <td>10</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                Ontology       FDR  N in query  \\\n",
       "0  extracellular region  GO: cellular_component  6.74e-09          10   \n",
       "1             wild type   Dictybase: Phenotypes  1.99e-01           5   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       157        157  \n",
       "1       341        338  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C17 (Q)\n",
      "***  C17 selected: 57 with EID: 56\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 11 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.25e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                Ontology       FDR  N in query  \\\n",
       "0              cell adhesion  GO: biological_process  1.25e-03           4   \n",
       "1  decreased spore viability   Dictybase: Phenotypes  1.12e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0        84         83  \n",
       "1        60         59  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C18 (R)\n",
      "***  C18 selected: 65 with EID: 63\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.14\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 8 out of 9 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.14e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.14e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.14e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.39e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.39e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.39e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>lipid binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.02e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>59</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.88e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>protein transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>197</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0  anatomical structure formation involved in morphogenesis   \n",
       "1                          external encapsulating structure   \n",
       "2                                                 cell wall   \n",
       "3                                      cell differentiation   \n",
       "4                                        hydrolase activity   \n",
       "5                                      extracellular region   \n",
       "6                                             lipid binding   \n",
       "7                          anatomical structure development   \n",
       "8                                         protein transport   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: biological_process  3.14e-03           3        74         74  \n",
       "1  GO: cellular_component  3.14e-03           2        13         13  \n",
       "2  GO: cellular_component  3.14e-03           2        13         13  \n",
       "3  GO: biological_process  1.39e-02           3       145        145  \n",
       "4  GO: molecular_function  1.39e-02           2        46         45  \n",
       "5  GO: cellular_component  1.39e-02           3       157        157  \n",
       "6  GO: molecular_function  2.02e-02           2        59         59  \n",
       "7  GO: biological_process  2.88e-02           4       440        438  \n",
       "8  GO: biological_process  1.23e-01           2       197        197  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C19 (S)\n",
      "***  C19 selected: 89 with EID: 88\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.2\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 10 out of 18 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.93e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.76e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.30e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.05e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1  anatomical structure formation involved in morphogenesis   \n",
       "2                                      cell differentiation   \n",
       "3                             Starch and sucrose metabolism   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: cellular_component  5.93e-05           7       157        157  \n",
       "1  GO: biological_process  2.76e-03           4        74         74  \n",
       "2  GO: biological_process  2.30e-02           4       145        145  \n",
       "3          KEGG: Pathways  4.05e-02           2        31         31  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "*** Cluster C20 (T)\n",
      "***  C20 selected: 1 with EID: 1\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.0\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 0 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "regulons=pd.read_table(pathRegulons+'mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab',index_col=0)\n",
    "for group in range(1,len(regulons['Cluster'].unique())+1):\n",
    "    cluster='C'+str(group)\n",
    "    print('*** Cluster %s (%s)' %(cluster,string.ascii_uppercase[group-1]))\n",
    "\n",
    "    enriched=group_diff_enrichment(data=regulons,group=cluster,min_overlap=2,use_annotated_genes=True)\n",
    "    if save_enrichment and enriched is not None:\n",
    "        enriched.to_csv(pathRegulons+'enrichment/mergedGenes_minExpressed0.990.1Strains1Min1Max18_clustersAX4Louvain0.4m0s1log.tab'+str(group)+'.tsv', sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
