{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from importlib import reload\n",
    "\n",
    "module_path = os.path.abspath(os.path.join('..'))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.insert(0,module_path)\n",
    "\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "pd.set_option('max_colwidth', 70)\n",
    "pd.set_option('display.max_rows', 500)\n",
    "import string\n",
    "import numpy as np\n",
    "\n",
    "from orangecontrib.bioinformatics.geneset.__init__ import list_all\n",
    "\n",
    "import jupyter_functions as jf\n",
    "reload(jf)\n",
    "import jupyter_functions as jf\n",
    "from networks.functionsDENet import loadPickle, savePickle\n",
    "import enrichment_library as enr\n",
    "reload(enr)\n",
    "import enrichment_library as enr\n",
    "from stages_DE.stages_library import PHENOTYPES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Path for expression data (mergedGenes_RPKUM.tsv)\n",
    "dataPath='/home/karin/Documents/timeTrajectories/data/RPKUM/combined/'\n",
    "path_abberant='/home/karin/Documents/timeTrajectories/data/regulons/selected_genes/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all gene sets\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=True,set_sizes=(5,500))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For conversion of gene names to EID\n",
    "genes = pd.read_csv(dataPath + 'mergedGenes_RPKUM.tsv', sep='\\t', index_col=0)\n",
    "all_gene_names= genes[(genes != 0).any(axis=1)].index\n",
    "NAME_EID=enr.name_genes_entrez(gene_names=all_gene_names, key_entrez=False)\n",
    "ALL_GENE_NAMES_EID=enr.convert_EID(genes=all_gene_names, name_EID=NAME_EID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "def group_diff_enrichment(query_names,group:str,padj:float=0.25,min_overlap:int=None,\n",
    "                          use_annotated_genes:bool=False):\n",
    "    #Displays only gene sets that have overlap with query greater or equal to min_overlap \n",
    "    #For p value and padj calculation uses alll that have overlap >=1 } from gene_set_enrichment\n",
    "    \"\"\"\n",
    "    :param use_annotated_genes: if True use for reference and query  only genes that have at \n",
    "    least one gene set annotation\n",
    "    \"\"\"\n",
    "    query_EID=enr.convert_EID(genes=query_names, name_EID=NAME_EID)\n",
    "    print('***  '+group+' selected:',len(query_names),'with EID:',len(query_EID))\n",
    "    \n",
    "    reference_gene_eids=ALL_GENE_NAMES_EID.copy()\n",
    "    query_eids=query_EID.copy()\n",
    "    \n",
    "    if use_annotated_genes:\n",
    "        gene_sets_genes=set()\n",
    "        for gene_set_name, gene_sets in GENE_SETS_ONTOLOGY.items():\n",
    "            for gene_set in gene_sets:\n",
    "                gene_sets_genes.update(gene_set.genes)\n",
    "        reference_gene_eids=set(reference_gene_eids) & gene_sets_genes\n",
    "        query_eids=set(query_eids) & gene_sets_genes\n",
    "        \n",
    "        query_annotated_reatio='NA'\n",
    "        if len(query_EID)>0:\n",
    "            query_annotated_reatio=round(len(query_eids)/len(query_EID),2)\n",
    "        print('Ratio of genes annotated with a gene set in reference',\n",
    "              round(len(reference_gene_eids)/len(ALL_GENE_NAMES_EID),2),\n",
    "             'and query',query_annotated_reatio)\n",
    "    \n",
    "    query_in_enriched=set()\n",
    "    result=None\n",
    "    if len(query_eids) > 0:\n",
    "        enrichment=enr.gene_set_enrichment(query_eids, reference_EID=reference_gene_eids, \n",
    "                                                padj_threshold=padj,min_overlap=min_overlap,\n",
    "                                                gene_sets_ontology=GENE_SETS_ONTOLOGY)\n",
    "        if len(enrichment)>0:\n",
    "            enrichment_display=list()\n",
    "            enrichment= sorted(enrichment, key=lambda data: data.padj)\n",
    "            for enriched in enrichment:\n",
    "                query_in_enriched.update(enriched.gene_set.genes & query_eids)\n",
    "                enrichment_display.append({'Gene set':enriched.gene_set.name,'Ontology':enriched.ontology,\n",
    "                                           'FDR':\"{:.2e}\".format(enriched.padj),'N in query':enriched.in_query})\n",
    "            result=pd.DataFrame(enrichment_display)\n",
    "    print('Enrichment at FDR: '+str(padj)+' and min query - gene set overlap',str(min_overlap))\n",
    "    print('N query genes in displayed gene sets:',len(query_in_enriched),'out of', len(query_eids),\n",
    "          'query genes used for enrichment calculation.')\n",
    "    display(result)\n",
    "    print('\\n')\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "comparison_order=('agg-','dis','tag','cud')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  agg- selected: 162 with EID: 160\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 30 out of 45 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.47e-05</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.89e-03</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.44e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.37e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.37e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.37e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.12e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>aberrant stalk morphogenesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.18e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.84e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>mislocalized gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.70e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.85e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.20e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>aberrant protein localization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       cell differentiation   \n",
       "1                           anatomical structure development   \n",
       "2   anatomical structure formation involved in morphogenesis   \n",
       "3                             aberrant spore coat morphology   \n",
       "4                                                  cell wall   \n",
       "5                           external encapsulating structure   \n",
       "6                       cell wall organization or biogenesis   \n",
       "7                                 multiple tips in culminant   \n",
       "8                               aberrant stalk morphogenesis   \n",
       "9                                  decreased spore viability   \n",
       "10                              mislocalized gene expression   \n",
       "11                                 cytoskeleton organization   \n",
       "12                                     decreased sporulation   \n",
       "13                                      extracellular region   \n",
       "14                                       signal transduction   \n",
       "15                             aberrant protein localization   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  5.47e-05          11  \n",
       "1   (GO, biological_process)  3.89e-03          15  \n",
       "2   (GO, biological_process)  7.44e-03           6  \n",
       "3    (Dictybase, Phenotypes)  8.37e-03           3  \n",
       "4   (GO, cellular_component)  8.37e-03           3  \n",
       "5   (GO, cellular_component)  8.37e-03           3  \n",
       "6   (GO, biological_process)  1.12e-02           3  \n",
       "7    (Dictybase, Phenotypes)  2.13e-02           2  \n",
       "8    (Dictybase, Phenotypes)  5.18e-02           2  \n",
       "9    (Dictybase, Phenotypes)  5.84e-02           4  \n",
       "10   (Dictybase, Phenotypes)  9.70e-02           2  \n",
       "11  (GO, biological_process)  1.63e-01           6  \n",
       "12   (Dictybase, Phenotypes)  1.63e-01           3  \n",
       "13  (GO, cellular_component)  1.85e-01           5  \n",
       "14  (GO, biological_process)  2.20e-01           8  \n",
       "15   (Dictybase, Phenotypes)  2.30e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  dis selected: 550 with EID: 549\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 52 out of 142 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.29e-07</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.68e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.68e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.42e-02</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.49e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.49e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1                                                 cell wall   \n",
       "2                          external encapsulating structure   \n",
       "3                          anatomical structure development   \n",
       "4                             Starch and sucrose metabolism   \n",
       "5  anatomical structure formation involved in morphogenesis   \n",
       "6                            aberrant spore coat morphology   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  8.29e-07          23  \n",
       "1  (GO, cellular_component)  5.68e-02           4  \n",
       "2  (GO, cellular_component)  5.68e-02           4  \n",
       "3  (GO, biological_process)  7.42e-02          28  \n",
       "4          (KEGG, Pathways)  1.49e-01           5  \n",
       "5  (GO, biological_process)  1.49e-01           8  \n",
       "6   (Dictybase, Phenotypes)  2.15e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 39 with EID: 39\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 8 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.72e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                  Ontology       FDR  N in query\n",
       "0  extracellular region  (GO, cellular_component)  5.72e-02           3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cud selected: 517 with EID: 510\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 39 out of 109 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.67e-05</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.18e-04</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.76e-03</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>decreased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1  anatomical structure formation involved in morphogenesis   \n",
       "2                                      cell differentiation   \n",
       "3                                        hydrolase activity   \n",
       "4                               secondary metabolic process   \n",
       "5                                             cell adhesion   \n",
       "6                                 decreased spore viability   \n",
       "7                   decreased prespore cell differentiation   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  2.67e-05          18  \n",
       "1  (GO, biological_process)  3.18e-04          11  \n",
       "2  (GO, biological_process)  1.76e-03          14  \n",
       "3  (GO, molecular_function)  1.40e-01           5  \n",
       "4  (GO, biological_process)  1.40e-01           2  \n",
       "5  (GO, biological_process)  1.40e-01           7  \n",
       "6   (Dictybase, Phenotypes)  1.40e-01           6  \n",
       "7   (Dictybase, Phenotypes)  1.40e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "abberant=pd.read_table(\n",
    "    path_abberant+'comparisonsAvgSimsSingle2STDAny0.2_lessComparisons2_AX4basedNeigh_u-less_newGenes_noAll-removeZeroRep_simsDict_scalemean0std1_logTrue_kN11_splitStrain.tsv',\n",
    "                       index_col=0,sep='\\t')\n",
    "abberant=abberant[(abberant['Difference mean']>=0.3) & (abberant['FDR']<=0.05)]\n",
    "for comparison in comparison_order:\n",
    "    query_names=list(abberant.query('Comparison == \"'+comparison+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=comparison,min_overlap=2,use_annotated_genes=True)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "orange3venv",
   "language": "python",
   "name": "orange3venv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
