{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Enrichment for milestone genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from importlib import reload\n",
    "\n",
    "module_path = os.path.abspath(os.path.join('..'))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.insert(0,module_path)\n",
    "\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "pd.set_option('max_colwidth', 70)\n",
    "pd.set_option('display.max_rows', 500)\n",
    "import string\n",
    "import numpy as np\n",
    "\n",
    "from orangecontrib.bioinformatics.geneset.__init__ import list_all\n",
    "\n",
    "import jupyter_functions as jf\n",
    "reload(jf)\n",
    "import jupyter_functions as jf\n",
    "from networks.functionsDENet import loadPickle, savePickle\n",
    "import enrichment_library as enr\n",
    "reload(enr)\n",
    "import enrichment_library as enr\n",
    "from stages_DE.stages_library import PHENOTYPES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Path for expression data (mergedGenes_RPKUM.tsv)\n",
    "dataPath='/home/karin/Documents/timeTrajectories/data/RPKUM/combined/'\n",
    "path_de='/home/karin/Documents/timeTrajectories/data/deTime/stage_vs_other/'\n",
    "path_impulse='/home/karin/Documents/timeTrajectories/data/stages/DE_across_stages/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene sets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Uses gene sets that do not have less than 5 or more than 500 genes. Use all ontologies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all gene sets\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=True,set_sizes=(5,500))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Min size 5 Max size 440\n",
      "N sets {('GO', 'biological_process'): 52, ('GO', 'cellular_component'): 24, ('Dictybase', 'Phenotypes'): 209, ('KEGG', 'Pathways'): 96, ('GO', 'molecular_function'): 35}\n"
     ]
    }
   ],
   "source": [
    "# Find out max/min gene set sizes and N sets\n",
    "max_n=0\n",
    "min_n=np.inf\n",
    "for k,v in GENE_SETS_ONTOLOGY.items():\n",
    "    for s in v:\n",
    "        n=len(s.genes)\n",
    "        if n<min_n:\n",
    "            min_n=n\n",
    "        if n>max_n:\n",
    "            max_n=n\n",
    "print('Min size',min_n,'Max size',max_n)  \n",
    "print('N sets',{k:len(v) for k,v in GENE_SETS_ONTOLOGY.items()})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For conversion of gene names to EID\n",
    "genes = pd.read_csv(dataPath + 'mergedGenes_RPKUM.tsv', sep='\\t', index_col=0)\n",
    "all_gene_names= genes[(genes != 0).any(axis=1)].index\n",
    "NAME_EID=enr.name_genes_entrez(gene_names=all_gene_names, key_entrez=False)\n",
    "ALL_GENE_NAMES_EID=enr.convert_EID(genes=all_gene_names, name_EID=NAME_EID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def group_diff_enrichment(query_names,group:str,padj:float=0.25,min_overlap:int=None,\n",
    "                          use_annotated_genes:bool=False):\n",
    "    #Displays only gene sets that have overlap with query greater or equal to min_overlap \n",
    "    #For p value and padj calculation uses alll that have overlap >=1 } from gene_set_enrichment\n",
    "    \"\"\"\n",
    "    :param use_annotated_genes: if True use for reference and query  only genes that have at \n",
    "    least one gene set annotation\n",
    "    \"\"\"\n",
    "    query_EID=enr.convert_EID(genes=query_names, name_EID=NAME_EID)\n",
    "    print('***  '+group+' selected:',len(query_names),'with EID:',len(query_EID))\n",
    "    \n",
    "    reference_gene_eids=ALL_GENE_NAMES_EID.copy()\n",
    "    query_eids=query_EID.copy()\n",
    "    \n",
    "    if use_annotated_genes:\n",
    "        gene_sets_genes=set()\n",
    "        for gene_set_name, gene_sets in GENE_SETS_ONTOLOGY.items():\n",
    "            for gene_set in gene_sets:\n",
    "                gene_sets_genes.update(gene_set.genes)\n",
    "        reference_gene_eids=set(reference_gene_eids) & gene_sets_genes\n",
    "        query_eids=set(query_eids) & gene_sets_genes\n",
    "        \n",
    "        query_annotated_reatio='NA'\n",
    "        if len(query_EID)>0:\n",
    "            query_annotated_reatio=round(len(query_eids)/len(query_EID),2)\n",
    "        print('Ratio of genes annotated with a gene set in reference',\n",
    "              round(len(reference_gene_eids)/len(ALL_GENE_NAMES_EID),2),\n",
    "             'and query',query_annotated_reatio)\n",
    "    \n",
    "    query_in_enriched=set()\n",
    "    result=None\n",
    "    if len(query_eids) > 0:\n",
    "        enrichment=enr.gene_set_enrichment(query_eids, reference_EID=reference_gene_eids, \n",
    "                                                padj_threshold=padj,min_overlap=min_overlap,\n",
    "                                                gene_sets_ontology=GENE_SETS_ONTOLOGY)\n",
    "        if len(enrichment)>0:\n",
    "            enrichment_display=list()\n",
    "            enrichment= sorted(enrichment, key=lambda data: data.padj)\n",
    "            for enriched in enrichment:\n",
    "                query_in_enriched.update(enriched.gene_set.genes & query_eids)\n",
    "                enrichment_display.append({'Gene set':enriched.gene_set.name,\n",
    "                                           'Ontology':enriched.ontology[0]+': '+enriched.ontology[1],\n",
    "                                           'FDR':\"{:.2e}\".format(enriched.padj),'N in query':enriched.in_query,\n",
    "                                           'Set size':len(enriched.gene_set.genes),\n",
    "                                                          'N in ref.':enriched.in_reference})\n",
    "            result=pd.DataFrame(enrichment_display)\n",
    "    print('Enrichment at FDR: '+str(padj)+' and min query - gene set overlap',str(min_overlap))\n",
    "    print('N query genes in displayed gene sets:',len(query_in_enriched),'out of', len(query_eids),\n",
    "          'query genes used for enrichment calculation.')\n",
    "    display(result)\n",
    "    print('\\n')\n",
    "    return result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enrichment report"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DESeq2\n",
    "Only positive log2 fold changes (lFC) are used to select only overexpressed genes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DESeq2 all strains"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### DESeq2 all strains lFC >= 2 and padj <= 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 216 with EID: 215\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 58 out of 82 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.94e-04</td>\n",
       "      <td>14</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.11e-03</td>\n",
       "      <td>15</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.24e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>homeostatic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.77e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>lysosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.44e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Lysine biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.78e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>increased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant cellular metabolic process</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant cellular response to stress</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Gene set                Ontology       FDR  \\\n",
       "0                   extracellular region  GO: cellular_component  3.94e-04   \n",
       "1                oxidoreductase activity  GO: molecular_function  1.11e-03   \n",
       "2                     hydrolase activity  GO: molecular_function  1.24e-02   \n",
       "3                    homeostatic process  GO: biological_process  1.77e-02   \n",
       "4                     response to stress  GO: biological_process  6.15e-02   \n",
       "5                               lysosome  GO: cellular_component  6.15e-02   \n",
       "6           cytoskeletal protein binding  GO: molecular_function  9.44e-02   \n",
       "7                    Lysine biosynthesis          KEGG: Pathways  9.78e-02   \n",
       "8                     peptidase activity  GO: molecular_function  1.59e-01   \n",
       "9                      catabolic process  GO: biological_process  1.76e-01   \n",
       "10                       plasma membrane  GO: cellular_component  2.27e-01   \n",
       "11              increased aggregate size   Dictybase: Phenotypes  2.27e-01   \n",
       "12   aberrant cellular metabolic process   Dictybase: Phenotypes  2.27e-01   \n",
       "13  aberrant cellular response to stress   Dictybase: Phenotypes  2.27e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           14       157        157  \n",
       "1           15       210        207  \n",
       "2            6        46         45  \n",
       "3            7        70         70  \n",
       "4           16       360        358  \n",
       "5            6        69         69  \n",
       "6            9       158        158  \n",
       "7            2         6          6  \n",
       "8            6        91         91  \n",
       "9           11       250        250  \n",
       "10          11       268        268  \n",
       "11           3        31         30  \n",
       "12           2        12         12  \n",
       "13           2        11         11  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 15 with EID: 15\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.13\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 2 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 17 with EID: 17\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.18\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 3 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 37 with EID: 37\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.11\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 4 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene set                Ontology       FDR  N in query  Set size  \\\n",
       "0  kinase activity  GO: molecular_function  1.11e-01           2       328   \n",
       "\n",
       "   N in ref.  \n",
       "0        327  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 48 with EID: 48\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.08\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 4 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.10e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.97e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>354</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                Ontology       FDR  \\\n",
       "0              extracellular region  GO: cellular_component  7.10e-02   \n",
       "1  small molecule metabolic process  GO: biological_process  9.97e-02   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           2       157        157  \n",
       "1           2       354        353  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 411 with EID: 410\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.18\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 43 out of 73 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.06e-14</td>\n",
       "      <td>24</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.30e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.30e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.79e-03</td>\n",
       "      <td>19</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.77e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.47e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.27e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.49e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.49e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.49e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                Ontology       FDR  \\\n",
       "0                  extracellular region  GO: cellular_component  1.06e-14   \n",
       "1                             cell wall  GO: cellular_component  2.30e-03   \n",
       "2      external encapsulating structure  GO: cellular_component  2.30e-03   \n",
       "3      anatomical structure development  GO: biological_process  6.79e-03   \n",
       "4  cell wall organization or biogenesis  GO: biological_process  4.77e-02   \n",
       "5           aberrant cell-cell adhesion   Dictybase: Phenotypes  5.47e-02   \n",
       "6                  extracellular matrix  GO: cellular_component  1.27e-01   \n",
       "7        aberrant spore coat morphology   Dictybase: Phenotypes  2.49e-01   \n",
       "8                       plasma membrane  GO: cellular_component  2.49e-01   \n",
       "9             decreased spore viability   Dictybase: Phenotypes  2.49e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0          24       157        157  \n",
       "1           4        13         13  \n",
       "2           4        13         13  \n",
       "3          19       440        438  \n",
       "4           3        15         15  \n",
       "5           2         5          5  \n",
       "6           2         8          8  \n",
       "7           2        12         12  \n",
       "8          10       268        268  \n",
       "9           4        60         59  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 527 with EID: 526\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.27\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 89 out of 144 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.31e-10</td>\n",
       "      <td>28</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-07</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.60e-07</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.84e-06</td>\n",
       "      <td>15</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.40e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.40e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.86e-06</td>\n",
       "      <td>20</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.43e-05</td>\n",
       "      <td>36</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.99e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.01e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.72e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.65e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>89</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.65e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.65e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.57e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.24e-01</td>\n",
       "      <td>20</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1                             aberrant spore coat morphology   \n",
       "2                       cell wall organization or biogenesis   \n",
       "3   anatomical structure formation involved in morphogenesis   \n",
       "4                                                  cell wall   \n",
       "5                           external encapsulating structure   \n",
       "6                                       cell differentiation   \n",
       "7                           anatomical structure development   \n",
       "8                              Starch and sucrose metabolism   \n",
       "9                                       extracellular matrix   \n",
       "10                          abolished cellulose biosynthesis   \n",
       "11                            carbohydrate metabolic process   \n",
       "12                                 decreased spore viability   \n",
       "13                                        hydrolase activity   \n",
       "14                                      aberrant culmination   \n",
       "15                                       cell-cell signaling   \n",
       "16                                                 wild type   \n",
       "17                                     abolished culmination   \n",
       "18                      abolished stalk cell differentiation   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  1.31e-10          28       157        157  \n",
       "1    Dictybase: Phenotypes  1.09e-07           8        12         12  \n",
       "2   GO: biological_process  8.60e-07           8        15         15  \n",
       "3   GO: biological_process  1.84e-06          15        74         74  \n",
       "4   GO: cellular_component  3.40e-06           7        13         13  \n",
       "5   GO: cellular_component  3.40e-06           7        13         13  \n",
       "6   GO: biological_process  4.86e-06          20       145        145  \n",
       "7   GO: biological_process  3.43e-05          36       440        438  \n",
       "8           KEGG: Pathways  1.99e-03           7        31         31  \n",
       "9   GO: cellular_component  2.01e-03           4         8          8  \n",
       "10   Dictybase: Phenotypes  7.72e-03           3         5          5  \n",
       "11  GO: biological_process  1.65e-02          10        89         88  \n",
       "12   Dictybase: Phenotypes  1.65e-02           8        60         59  \n",
       "13  GO: molecular_function  1.65e-02           7        46         45  \n",
       "14   Dictybase: Phenotypes  5.57e-02           9        88         88  \n",
       "15  GO: biological_process  2.21e-01           3        16         16  \n",
       "16   Dictybase: Phenotypes  2.24e-01          20       341        338  \n",
       "17   Dictybase: Phenotypes  2.43e-01           6        62         62  \n",
       "18   Dictybase: Phenotypes  2.43e-01           2         7          7  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 933 with EID: 926\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 96 out of 192 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>8.76e-13</td>\n",
       "      <td>35</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.31e-06</td>\n",
       "      <td>24</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.85e-05</td>\n",
       "      <td>16</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.85e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.06e-04</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.95e-03</td>\n",
       "      <td>38</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.95e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.95e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.10e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.81e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.94e-02</td>\n",
       "      <td>19</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.94e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.19e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>development arrests at slug stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant stalk formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1                                       cell differentiation   \n",
       "2   anatomical structure formation involved in morphogenesis   \n",
       "3                             aberrant spore coat morphology   \n",
       "4                       cell wall organization or biogenesis   \n",
       "5                           anatomical structure development   \n",
       "6                                                  cell wall   \n",
       "7                           external encapsulating structure   \n",
       "8                                       extracellular matrix   \n",
       "9                                  decreased spore viability   \n",
       "10                                   oxidoreductase activity   \n",
       "11                                        hydrolase activity   \n",
       "12                                 decreased gene expression   \n",
       "13                             Starch and sucrose metabolism   \n",
       "14                                             cell adhesion   \n",
       "15                         development arrests at slug stage   \n",
       "16                               secondary metabolic process   \n",
       "17                                  aberrant stalk formation   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  8.76e-13          35       157        157  \n",
       "1   GO: biological_process  7.31e-06          24       145        145  \n",
       "2   GO: biological_process  1.85e-05          16        74         74  \n",
       "3    Dictybase: Phenotypes  1.85e-05           7        12         12  \n",
       "4   GO: biological_process  1.06e-04           7        15         15  \n",
       "5   GO: biological_process  5.95e-03          38       440        438  \n",
       "6   GO: cellular_component  5.95e-03           5        13         13  \n",
       "7   GO: cellular_component  5.95e-03           5        13         13  \n",
       "8   GO: cellular_component  7.10e-03           4         8          8  \n",
       "9    Dictybase: Phenotypes  3.81e-02           9        60         59  \n",
       "10  GO: molecular_function  8.94e-02          19       210        207  \n",
       "11  GO: molecular_function  8.94e-02           7        46         45  \n",
       "12   Dictybase: Phenotypes  1.26e-01          10        95         86  \n",
       "13          KEGG: Pathways  2.19e-01           5        31         31  \n",
       "14  GO: biological_process  2.27e-01           9        84         83  \n",
       "15   Dictybase: Phenotypes  2.27e-01           3        12         12  \n",
       "16  GO: biological_process  2.31e-01           2         5          5  \n",
       "17   Dictybase: Phenotypes  2.31e-01           2         5          5  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 709 with EID: 705\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.22\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 65 out of 155 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.28e-09</td>\n",
       "      <td>27</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.77e-03</td>\n",
       "      <td>17</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.51e-03</td>\n",
       "      <td>20</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.04e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.04e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1                                      cell differentiation   \n",
       "2                                   oxidoreductase activity   \n",
       "3  anatomical structure formation involved in morphogenesis   \n",
       "4                               secondary metabolic process   \n",
       "5                                             cell adhesion   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: cellular_component  5.28e-09          27       157        157  \n",
       "1  GO: biological_process  3.77e-03          17       145        145  \n",
       "2  GO: molecular_function  8.51e-03          20       210        207  \n",
       "3  GO: biological_process  2.04e-02          10        74         74  \n",
       "4  GO: biological_process  2.04e-02           3         5          5  \n",
       "5  GO: biological_process  1.48e-01           9        84         83  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  yem selected: 1001 with EID: 995\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.32\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 97 out of 320 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.67e-04</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.38e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.76e-03</td>\n",
       "      <td>17</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.31e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.99e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Cysteine and methionine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.99e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Fatty acid biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.06e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.06e-02</td>\n",
       "      <td>29</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>50</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Phenylalanine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.16e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.46e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                secondary metabolic process   \n",
       "1                                       cell differentiation   \n",
       "2   anatomical structure formation involved in morphogenesis   \n",
       "3        Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "4                                  decreased gene expression   \n",
       "5                         Cysteine and methionine metabolism   \n",
       "6                                    Fatty acid biosynthesis   \n",
       "7                                    oxidoreductase activity   \n",
       "8                           anatomical structure development   \n",
       "9                                   Phenylalanine metabolism   \n",
       "10                 SNARE interactions in vesicular transport   \n",
       "11                                  aberrant gene expression   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  9.67e-04           5         5          5  \n",
       "1   GO: biological_process  1.38e-03          28       145        145  \n",
       "2   GO: biological_process  5.76e-03          17        74         74  \n",
       "3           KEGG: Pathways  4.31e-02           5        10         10  \n",
       "4    Dictybase: Phenotypes  5.99e-02          16        95         86  \n",
       "5           KEGG: Pathways  5.99e-02           8        28         28  \n",
       "6           KEGG: Pathways  8.06e-02           4         8          8  \n",
       "7   GO: molecular_function  8.06e-02          29       210        207  \n",
       "8   GO: biological_process  1.88e-01          50       440        438  \n",
       "9           KEGG: Pathways  1.88e-01           4        12         10  \n",
       "10          KEGG: Pathways  2.16e-01           6        23         23  \n",
       "11   Dictybase: Phenotypes  2.46e-01           6        24         24  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_de+'nobatchrep_combined.tsv',index_col=0,sep='\\t')\n",
    "for stage in [phenotype for phenotype in PHENOTYPES if phenotype in list(milestones['Stage'].unique())]:\n",
    "    query_names=list(milestones.query('log2FoldChange>=2 & padj <= 0.01 & Stage==\"'+stage+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)\n",
    "  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### DESeq2 all strains lFC >= 1 and padj <= 0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 1652 with EID: 1643\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.36\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 359 out of 598 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.27e-65</td>\n",
       "      <td>90</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.79e-60</td>\n",
       "      <td>85</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.38e-57</td>\n",
       "      <td>79</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.89e-44</td>\n",
       "      <td>85</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.91e-39</td>\n",
       "      <td>99</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.15e-09</td>\n",
       "      <td>73</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.11e-07</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.90e-06</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.67e-04</td>\n",
       "      <td>48</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.76e-04</td>\n",
       "      <td>68</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.56e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>71</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.56e-02</td>\n",
       "      <td>49</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.66e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>64</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>mitochondrion organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.01e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.01e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>58</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.37e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.85e-02</td>\n",
       "      <td>38</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Porphyrin and chlorophyll metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.66e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Biosynthesis of unsaturated fatty acids</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.90e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.90e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.20e-01</td>\n",
       "      <td>16</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Gene set                Ontology       FDR  \\\n",
       "0                                  Ribosome          KEGG: Pathways  8.27e-65   \n",
       "1                                  ribosome  GO: cellular_component  6.79e-60   \n",
       "2        structural constituent of ribosome  GO: molecular_function  4.38e-57   \n",
       "3              structural molecule activity  GO: molecular_function  5.89e-44   \n",
       "4                               translation  GO: biological_process  8.91e-39   \n",
       "5                             mitochondrion  GO: cellular_component  2.15e-09   \n",
       "6                      Steroid biosynthesis          KEGG: Pathways  2.11e-07   \n",
       "7                              rRNA binding  GO: molecular_function  1.90e-06   \n",
       "8                               RNA binding  GO: molecular_function  1.67e-04   \n",
       "9     Biosynthesis of secondary metabolites          KEGG: Pathways  7.76e-04   \n",
       "10                Oxidative phosphorylation          KEGG: Pathways  1.56e-02   \n",
       "11                  oxidoreductase activity  GO: molecular_function  1.56e-02   \n",
       "12              Aminoacyl-tRNA biosynthesis          KEGG: Pathways  1.66e-02   \n",
       "13               mitochondrion organization  GO: biological_process  2.01e-02   \n",
       "14               methyltransferase activity  GO: molecular_function  2.01e-02   \n",
       "15                    Fatty acid elongation          KEGG: Pathways  2.37e-02   \n",
       "16                     extracellular region  GO: cellular_component  2.85e-02   \n",
       "17     Porphyrin and chlorophyll metabolism          KEGG: Pathways  3.66e-02   \n",
       "18  Biosynthesis of unsaturated fatty acids          KEGG: Pathways  1.80e-01   \n",
       "19                       hydrolase activity  GO: molecular_function  1.90e-01   \n",
       "20                       isomerase activity  GO: molecular_function  1.90e-01   \n",
       "21                      ribosome biogenesis  GO: biological_process  2.20e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           90       105         97  \n",
       "1           85        93         93  \n",
       "2           79        85         85  \n",
       "3           85       117        117  \n",
       "4           99       169        169  \n",
       "5           73       229        229  \n",
       "6           14        18         18  \n",
       "7           11        13         13  \n",
       "8           48       169        169  \n",
       "9           68       289        281  \n",
       "10          18        71         54  \n",
       "11          49       210        207  \n",
       "12          16        64         46  \n",
       "13          10        23         23  \n",
       "14          18        58         56  \n",
       "15           7        13         13  \n",
       "16          38       157        157  \n",
       "17           7        14         14  \n",
       "18           6        15         14  \n",
       "19          13        46         45  \n",
       "20          14        50         50  \n",
       "21          16        61         61  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 170 with EID: 167\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 28 out of 41 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.40e-05</td>\n",
       "      <td>11</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.28e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.32e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.01e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.27e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.27e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.90e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.19e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                Ontology  \\\n",
       "0                       extracellular region  GO: cellular_component   \n",
       "1   development arrests at loose mound stage   Dictybase: Phenotypes   \n",
       "2                              cell adhesion  GO: biological_process   \n",
       "3                       aberrant aggregation   Dictybase: Phenotypes   \n",
       "4                            plasma membrane  GO: cellular_component   \n",
       "5                         aberrant streaming   Dictybase: Phenotypes   \n",
       "6                        signal transduction  GO: biological_process   \n",
       "7                   abolished slug migration   Dictybase: Phenotypes   \n",
       "8                       aberrant culmination   Dictybase: Phenotypes   \n",
       "9                     precocious aggregation   Dictybase: Phenotypes   \n",
       "10              aberrant cAMP wave formation   Dictybase: Phenotypes   \n",
       "11                   aberrant slug migration   Dictybase: Phenotypes   \n",
       "12                 abolished gene expression   Dictybase: Phenotypes   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   3.40e-05          11       157        157  \n",
       "1   2.28e-03           4        20         20  \n",
       "2   5.32e-02           5        84         83  \n",
       "3   9.01e-02           4        61         61  \n",
       "4   1.09e-01           8       268        268  \n",
       "5   1.09e-01           4        72         72  \n",
       "6   1.27e-01           9       352        350  \n",
       "7   1.27e-01           3        43         43  \n",
       "8   1.46e-01           4        88         88  \n",
       "9   1.69e-01           2        20         20  \n",
       "10  1.69e-01           4        96         96  \n",
       "11  1.90e-01           3        58         58  \n",
       "12  2.19e-01           2        25         25  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 193 with EID: 190\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.3\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 43 out of 57 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.38e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.24e-03</td>\n",
       "      <td>13</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.81e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.56e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.56e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.61e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.62e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.93e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.76e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.06e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>abolished cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>delayed tip formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.41e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.19e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.19e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.32e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>AGE-RAGE signaling pathway in diabetic complications</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Gene set  \\\n",
       "0                                   extracellular region   \n",
       "1                                        plasma membrane   \n",
       "2               development arrests at loose mound stage   \n",
       "3                                    signal transduction   \n",
       "4                               abolished slug migration   \n",
       "5                                   aberrant aggregation   \n",
       "6                             multiple tips in culminant   \n",
       "7                                     response to stress   \n",
       "8                                  decreased culmination   \n",
       "9                       anatomical structure development   \n",
       "10                         abolished cAMP wave formation   \n",
       "11                                   abolished streaming   \n",
       "12                                 delayed tip formation   \n",
       "13                                    aberrant streaming   \n",
       "14                                 abolished aggregation   \n",
       "15                                multiple tips in mound   \n",
       "16                                         cell adhesion   \n",
       "17                                precocious aggregation   \n",
       "18                                  aberrant development   \n",
       "19                                  aberrant culmination   \n",
       "20                    development arrests at mound stage   \n",
       "21                                 abolished sporulation   \n",
       "22                             abolished gene expression   \n",
       "23                          aberrant cAMP wave formation   \n",
       "24                               aberrant slug migration   \n",
       "25  AGE-RAGE signaling pathway in diabetic complications   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  1.38e-03          11       157        157  \n",
       "1   GO: cellular_component  5.24e-03          13       268        268  \n",
       "2    Dictybase: Phenotypes  6.81e-03           4        20         20  \n",
       "3   GO: biological_process  8.56e-03          14       352        350  \n",
       "4    Dictybase: Phenotypes  8.56e-03           5        43         43  \n",
       "5    Dictybase: Phenotypes  3.61e-02           5        61         61  \n",
       "6    Dictybase: Phenotypes  3.62e-02           2         5          5  \n",
       "7   GO: biological_process  6.93e-02          12       360        358  \n",
       "8    Dictybase: Phenotypes  9.76e-02           2         9          9  \n",
       "9   GO: biological_process  1.06e-01          13       440        438  \n",
       "10   Dictybase: Phenotypes  1.20e-01           2        11         11  \n",
       "11   Dictybase: Phenotypes  1.22e-01           3        33         33  \n",
       "12   Dictybase: Phenotypes  1.41e-01           2        13         13  \n",
       "13   Dictybase: Phenotypes  1.77e-01           4        72         72  \n",
       "14   Dictybase: Phenotypes  2.19e-01           4        82         80  \n",
       "15   Dictybase: Phenotypes  2.19e-01           2        18         18  \n",
       "16  GO: biological_process  2.32e-01           4        84         83  \n",
       "17   Dictybase: Phenotypes  2.33e-01           2        20         20  \n",
       "18   Dictybase: Phenotypes  2.37e-01           3        51         51  \n",
       "19   Dictybase: Phenotypes  2.37e-01           4        88         88  \n",
       "20   Dictybase: Phenotypes  2.47e-01           3        56         54  \n",
       "21   Dictybase: Phenotypes  2.47e-01           2        25         25  \n",
       "22   Dictybase: Phenotypes  2.47e-01           2        25         25  \n",
       "23   Dictybase: Phenotypes  2.47e-01           4        96         96  \n",
       "24   Dictybase: Phenotypes  2.47e-01           3        58         58  \n",
       "25          KEGG: Pathways  2.47e-01           2        24         24  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 384 with EID: 382\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.18\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 44 out of 70 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.34e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.75e-03</td>\n",
       "      <td>12</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.38e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.08e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.08e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>increased cell-substrate adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.37e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.53e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>delayed tip formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                Ontology       FDR  \\\n",
       "0              chromosome segregation  GO: biological_process  1.34e-03   \n",
       "1                          cell cycle  GO: biological_process  1.75e-03   \n",
       "2            mitotic nuclear division  GO: biological_process  4.38e-03   \n",
       "3             chromosome organization  GO: biological_process  7.08e-03   \n",
       "4                     DNA replication          KEGG: Pathways  7.08e-03   \n",
       "5                  mitotic cell cycle  GO: biological_process  2.73e-02   \n",
       "6                          chromosome  GO: cellular_component  2.73e-02   \n",
       "7   increased cell-substrate adhesion   Dictybase: Phenotypes  2.73e-02   \n",
       "8                       cell division  GO: biological_process  5.37e-02   \n",
       "9                extracellular region  GO: cellular_component  7.53e-02   \n",
       "10                          cell wall  GO: cellular_component  1.86e-01   \n",
       "11   external encapsulating structure  GO: cellular_component  1.86e-01   \n",
       "12              delayed tip formation   Dictybase: Phenotypes  1.86e-01   \n",
       "13          decreased spore viability   Dictybase: Phenotypes  1.86e-01   \n",
       "14                    kinase activity  GO: molecular_function  2.27e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0            5        18         18  \n",
       "1           12       167        167  \n",
       "2            4        15         15  \n",
       "3            7        74         74  \n",
       "4            5        34         33  \n",
       "5            7       100        100  \n",
       "6            6        70         70  \n",
       "7            4        30         30  \n",
       "8            7       115        115  \n",
       "9            8       157        157  \n",
       "10           2        13         13  \n",
       "11           2        13         13  \n",
       "12           2        13         13  \n",
       "13           4        60         59  \n",
       "14          11       328        327  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 616 with EID: 613\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 92 out of 157 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.17e-12</td>\n",
       "      <td>22</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.17e-12</td>\n",
       "      <td>16</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.08e-08</td>\n",
       "      <td>27</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.38e-07</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.51e-07</td>\n",
       "      <td>16</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.13e-05</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.46e-05</td>\n",
       "      <td>17</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.71e-04</td>\n",
       "      <td>14</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.71e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.19e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.82e-03</td>\n",
       "      <td>20</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.11e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.43e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.03e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.03e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.42e-02</td>\n",
       "      <td>15</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>delayed tip formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.42e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.39e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.73e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>62</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant myosin II organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>increased macropinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>increased chemotaxis to folate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>28</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased cell-cell adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased prestalk cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.51e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Gene set                Ontology       FDR  \\\n",
       "0                                chromosome  GO: cellular_component  1.17e-12   \n",
       "1                           DNA replication          KEGG: Pathways  1.17e-12   \n",
       "2                                cell cycle  GO: biological_process  1.08e-08   \n",
       "3                    chromosome segregation  GO: biological_process  3.38e-07   \n",
       "4                   chromosome organization  GO: biological_process  6.51e-07   \n",
       "5                  mitotic nuclear division  GO: biological_process  2.13e-05   \n",
       "6                             cell division  GO: biological_process  5.46e-05   \n",
       "7                        mitotic cell cycle  GO: biological_process  6.71e-04   \n",
       "8                        nuclear chromosome  GO: cellular_component  6.71e-04   \n",
       "9                           Mismatch repair          KEGG: Pathways  1.19e-03   \n",
       "10                              DNA binding  GO: molecular_function  6.82e-03   \n",
       "11               Nucleotide excision repair          KEGG: Pathways  8.11e-03   \n",
       "12                     Base excision repair          KEGG: Pathways  9.43e-03   \n",
       "13                    DNA metabolic process  GO: biological_process  1.03e-02   \n",
       "14                    Pyrimidine metabolism          KEGG: Pathways  1.03e-02   \n",
       "15                     extracellular region  GO: cellular_component  1.42e-02   \n",
       "16                    delayed tip formation   Dictybase: Phenotypes  1.42e-02   \n",
       "17            microtubule organizing center  GO: cellular_component  3.39e-02   \n",
       "18           Ubiquitin mediated proteolysis          KEGG: Pathways  9.73e-02   \n",
       "19          aberrant myosin II organization   Dictybase: Phenotypes  1.17e-01   \n",
       "20               increased macropinocytosis   Dictybase: Phenotypes  1.17e-01   \n",
       "21                        cell cycle arrest   Dictybase: Phenotypes  1.17e-01   \n",
       "22           increased chemotaxis to folate   Dictybase: Phenotypes  1.17e-01   \n",
       "23                   Glutathione metabolism          KEGG: Pathways  1.23e-01   \n",
       "24             decreased cell-cell adhesion   Dictybase: Phenotypes  1.46e-01   \n",
       "25  decreased prestalk cell differentiation   Dictybase: Phenotypes  1.51e-01   \n",
       "26                             cytoskeleton  GO: cellular_component  2.15e-01   \n",
       "27               precocious gene expression   Dictybase: Phenotypes  2.15e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           22        70         70  \n",
       "1           16        34         33  \n",
       "2           27       167        167  \n",
       "3            9        18         18  \n",
       "4           16        74         74  \n",
       "5            7        15         15  \n",
       "6           17       115        115  \n",
       "7           14       100        100  \n",
       "8            6        17         17  \n",
       "9            6        19         19  \n",
       "10          20       221        221  \n",
       "11           7        37         37  \n",
       "12           6        28         28  \n",
       "13          12       105        105  \n",
       "14           6        31         29  \n",
       "15          15       157        157  \n",
       "16           4        13         13  \n",
       "17           7        50         50  \n",
       "18           7        62         61  \n",
       "19           2         5          5  \n",
       "20           2         5          5  \n",
       "21           2         5          5  \n",
       "22           2         5          5  \n",
       "23           4        28         25  \n",
       "24           3        15         15  \n",
       "25           2         6          6  \n",
       "26          13       185        185  \n",
       "27           3        18         18  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 1373 with EID: 1367\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 166 out of 292 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.35e-09</td>\n",
       "      <td>38</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.80e-05</td>\n",
       "      <td>62</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.63e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.84e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.84e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.13e-04</td>\n",
       "      <td>27</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.44e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.80e-04</td>\n",
       "      <td>17</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.21e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.01e-03</td>\n",
       "      <td>44</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.16e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.23e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.23e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant thermotaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.23e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.23e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.23e-02</td>\n",
       "      <td>28</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant phototaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.84e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.84e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>28</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>development arrests at slug stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.07e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.58e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>increased stalk length</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.41e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1                           anatomical structure development   \n",
       "2                             aberrant spore coat morphology   \n",
       "3                                                  cell wall   \n",
       "4                           external encapsulating structure   \n",
       "5                                       cell differentiation   \n",
       "6                              Starch and sucrose metabolism   \n",
       "7   anatomical structure formation involved in morphogenesis   \n",
       "8                       cell wall organization or biogenesis   \n",
       "9                                        signal transduction   \n",
       "10                                      aberrant culmination   \n",
       "11                                 decreased spore viability   \n",
       "12                 DNA-binding transcription factor activity   \n",
       "13                          abolished cellulose biosynthesis   \n",
       "14                                      aberrant thermotaxis   \n",
       "15                                           spores inviable   \n",
       "16                                               DNA binding   \n",
       "17                                       aberrant phototaxis   \n",
       "18                   increased prespore cell differentiation   \n",
       "19                                    Glutathione metabolism   \n",
       "20                         development arrests at slug stage   \n",
       "21                         aberrant fruiting body morphology   \n",
       "22                                     abolished culmination   \n",
       "23                                      extracellular matrix   \n",
       "24                                    increased stalk length   \n",
       "25                              aberrant cAMP wave formation   \n",
       "26                                       cell-cell signaling   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  4.35e-09          38       157        157  \n",
       "1   GO: biological_process  1.80e-05          62       440        438  \n",
       "2    Dictybase: Phenotypes  2.63e-05           8        12         12  \n",
       "3   GO: cellular_component  3.84e-05           8        13         13  \n",
       "4   GO: cellular_component  3.84e-05           8        13         13  \n",
       "5   GO: biological_process  2.13e-04          27       145        145  \n",
       "6           KEGG: Pathways  2.44e-04          11        31         31  \n",
       "7   GO: biological_process  5.80e-04          17        74         74  \n",
       "8   GO: biological_process  1.21e-03           7        15         15  \n",
       "9   GO: biological_process  6.01e-03          44       352        350  \n",
       "10   Dictybase: Phenotypes  1.36e-02          16        88         88  \n",
       "11   Dictybase: Phenotypes  2.16e-02          12        60         59  \n",
       "12  GO: molecular_function  2.23e-02           9        37         37  \n",
       "13   Dictybase: Phenotypes  5.23e-02           3         5          5  \n",
       "14   Dictybase: Phenotypes  5.23e-02           3         5          5  \n",
       "15   Dictybase: Phenotypes  5.23e-02           3         5          5  \n",
       "16  GO: molecular_function  5.23e-02          28       221        221  \n",
       "17   Dictybase: Phenotypes  8.84e-02           3        14          6  \n",
       "18   Dictybase: Phenotypes  8.84e-02           3         6          6  \n",
       "19          KEGG: Pathways  1.04e-01           6        28         25  \n",
       "20   Dictybase: Phenotypes  1.07e-01           4        12         12  \n",
       "21   Dictybase: Phenotypes  1.40e-01          14       101         98  \n",
       "22   Dictybase: Phenotypes  1.58e-01          10        62         62  \n",
       "23  GO: cellular_component  1.68e-01           3         8          8  \n",
       "24   Dictybase: Phenotypes  1.68e-01           3         8          8  \n",
       "25   Dictybase: Phenotypes  2.25e-01          13        96         96  \n",
       "26  GO: biological_process  2.41e-01           4        16         16  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 1871 with EID: 1864\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.3\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 340 out of 567 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.80e-07</td>\n",
       "      <td>107</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.87e-05</td>\n",
       "      <td>29</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.08e-05</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.65e-05</td>\n",
       "      <td>44</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.28e-04</td>\n",
       "      <td>45</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.17e-04</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.17e-04</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.52e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.67e-03</td>\n",
       "      <td>20</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.14e-03</td>\n",
       "      <td>74</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.14e-03</td>\n",
       "      <td>70</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.25e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.34e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.02e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.02e-02</td>\n",
       "      <td>25</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.10e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.60e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.60e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant spore morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.03e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.03e-02</td>\n",
       "      <td>25</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.03e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant phototaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.83e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.88e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.53e-02</td>\n",
       "      <td>65</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>45</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>increased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.36e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.42e-01</td>\n",
       "      <td>21</td>\n",
       "      <td>89</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>12</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>increased stalk length</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>20</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>aberrant autophagic cell death</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>abolished protein localization to nucleus</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>increased macropinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>aberrant thermotaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>precocious sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>translucent sorus</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>Histidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>Amino sugar and nucleotide sugar metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                           anatomical structure development   \n",
       "1   anatomical structure formation involved in morphogenesis   \n",
       "2                             aberrant spore coat morphology   \n",
       "3                                       cell differentiation   \n",
       "4                                       extracellular region   \n",
       "5                                                  cell wall   \n",
       "6                           external encapsulating structure   \n",
       "7                              Starch and sucrose metabolism   \n",
       "8                                  decreased spore viability   \n",
       "9                                        signal transduction   \n",
       "10                                           kinase activity   \n",
       "11                   increased prespore cell differentiation   \n",
       "12                 DNA-binding transcription factor activity   \n",
       "13                      cell wall organization or biogenesis   \n",
       "14                                      aberrant culmination   \n",
       "15                 development arrests at tipped mound stage   \n",
       "16                          abolished cellulose biosynthesis   \n",
       "17                                           spores inviable   \n",
       "18                                 aberrant spore morphology   \n",
       "19                         aberrant fruiting body morphology   \n",
       "20                                   aberrant slug migration   \n",
       "21                                       aberrant phototaxis   \n",
       "22                                     abolished culmination   \n",
       "23                                                 wild type   \n",
       "24                                               DNA binding   \n",
       "25                                      Galactose metabolism   \n",
       "26                                 increased gene expression   \n",
       "27                            carbohydrate metabolic process   \n",
       "28                                          ABC transporters   \n",
       "29                                  abolished slug migration   \n",
       "30                                     decreased sporulation   \n",
       "31                                      extracellular matrix   \n",
       "32                                    increased stalk length   \n",
       "33                                 decreased gene expression   \n",
       "34                                       cell-cell signaling   \n",
       "35                        decreased intracellular cAMP level   \n",
       "36                                 abolished gene expression   \n",
       "37                            aberrant autophagic cell death   \n",
       "38                 abolished protein localization to nucleus   \n",
       "39                                increased macropinocytosis   \n",
       "40                                      aberrant thermotaxis   \n",
       "41                                    precocious sporulation   \n",
       "42                                         translucent sorus   \n",
       "43                                      Histidine metabolism   \n",
       "44               Amino sugar and nucleotide sugar metabolism   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  7.80e-07         107       440        438  \n",
       "1   GO: biological_process  1.87e-05          29        74         74  \n",
       "2    Dictybase: Phenotypes  2.08e-05          10        12         12  \n",
       "3   GO: biological_process  3.65e-05          44       145        145  \n",
       "4   GO: cellular_component  1.28e-04          45       157        157  \n",
       "5   GO: cellular_component  5.17e-04           9        13         13  \n",
       "6   GO: cellular_component  5.17e-04           9        13         13  \n",
       "7           KEGG: Pathways  1.52e-03          14        31         31  \n",
       "8    Dictybase: Phenotypes  4.67e-03          20        60         59  \n",
       "9   GO: biological_process  7.14e-03          74       352        350  \n",
       "10  GO: molecular_function  7.14e-03          70       328        327  \n",
       "11   Dictybase: Phenotypes  9.25e-03           5         6          6  \n",
       "12  GO: molecular_function  9.34e-03          14        37         37  \n",
       "13  GO: biological_process  1.02e-02           8        15         15  \n",
       "14   Dictybase: Phenotypes  1.02e-02          25        88         88  \n",
       "15   Dictybase: Phenotypes  1.10e-02           7        12         12  \n",
       "16   Dictybase: Phenotypes  3.60e-02           4         5          5  \n",
       "17   Dictybase: Phenotypes  3.60e-02           4         5          5  \n",
       "18   Dictybase: Phenotypes  4.03e-02           5         8          8  \n",
       "19   Dictybase: Phenotypes  4.03e-02          25       101         98  \n",
       "20   Dictybase: Phenotypes  4.03e-02          17        58         58  \n",
       "21   Dictybase: Phenotypes  7.83e-02           4        14          6  \n",
       "22   Dictybase: Phenotypes  7.88e-02          17        62         62  \n",
       "23   Dictybase: Phenotypes  9.53e-02          65       341        338  \n",
       "24  GO: molecular_function  1.13e-01          45       221        221  \n",
       "25          KEGG: Pathways  1.23e-01           6        15         14  \n",
       "26   Dictybase: Phenotypes  1.36e-01          13        46         46  \n",
       "27  GO: biological_process  1.42e-01          21        89         88  \n",
       "28          KEGG: Pathways  1.62e-01           6        15         15  \n",
       "29   Dictybase: Phenotypes  1.71e-01          12        43         43  \n",
       "30   Dictybase: Phenotypes  1.71e-01          14        53         53  \n",
       "31  GO: cellular_component  1.86e-01           4         8          8  \n",
       "32   Dictybase: Phenotypes  1.86e-01           4         8          8  \n",
       "33   Dictybase: Phenotypes  1.86e-01          20        95         86  \n",
       "34  GO: biological_process  1.87e-01           6        16         16  \n",
       "35   Dictybase: Phenotypes  1.87e-01           5        12         12  \n",
       "36   Dictybase: Phenotypes  1.87e-01           8        25         25  \n",
       "37   Dictybase: Phenotypes  1.93e-01           3         5          5  \n",
       "38   Dictybase: Phenotypes  1.93e-01           3         5          5  \n",
       "39   Dictybase: Phenotypes  1.93e-01           3         5          5  \n",
       "40   Dictybase: Phenotypes  1.93e-01           3         5          5  \n",
       "41   Dictybase: Phenotypes  1.93e-01           3         5          5  \n",
       "42   Dictybase: Phenotypes  2.30e-01           4         9          9  \n",
       "43          KEGG: Pathways  2.30e-01           5        16         13  \n",
       "44          KEGG: Pathways  2.42e-01           8        27         27  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 2344 with EID: 2328\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.27\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 344 out of 629 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.03e-11</td>\n",
       "      <td>40</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.49e-11</td>\n",
       "      <td>59</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.85e-09</td>\n",
       "      <td>58</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.95e-05</td>\n",
       "      <td>109</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.45e-05</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.09e-03</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.09e-03</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.09e-03</td>\n",
       "      <td>15</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.61e-03</td>\n",
       "      <td>21</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.29e-02</td>\n",
       "      <td>27</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.60e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.60e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.48e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.48e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.48e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.87e-02</td>\n",
       "      <td>13</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant spore morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.23e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>18</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>69</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>abolished autophagic vacuole formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>abolished autophagic cell death</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>increased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>22</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>22</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>aberrant prestalk cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>15</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.02e-01</td>\n",
       "      <td>24</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.19e-01</td>\n",
       "      <td>47</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>70</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0   anatomical structure formation involved in morphogenesis   \n",
       "1                                       cell differentiation   \n",
       "2                                       extracellular region   \n",
       "3                           anatomical structure development   \n",
       "4                             aberrant spore coat morphology   \n",
       "5                                                  cell wall   \n",
       "6                           external encapsulating structure   \n",
       "7                              Starch and sucrose metabolism   \n",
       "8                                  decreased spore viability   \n",
       "9                                       aberrant culmination   \n",
       "10                   increased prespore cell differentiation   \n",
       "11                      cell wall organization or biogenesis   \n",
       "12                 development arrests at tipped mound stage   \n",
       "13                               secondary metabolic process   \n",
       "14                                   aberrant slug migration   \n",
       "15                                           spores inviable   \n",
       "16                 DNA-binding transcription factor activity   \n",
       "17                                 aberrant spore morphology   \n",
       "18                                     abolished culmination   \n",
       "19                 SNARE interactions in vesicular transport   \n",
       "20                                           kinase activity   \n",
       "21                    abolished autophagic vacuole formation   \n",
       "22                           abolished autophagic cell death   \n",
       "23                                 increased gene expression   \n",
       "24                                           Golgi apparatus   \n",
       "25       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "26                                 decreased gene expression   \n",
       "27                    aberrant prestalk cell differentiation   \n",
       "28                                     decreased sporulation   \n",
       "29                                      Galactose metabolism   \n",
       "30                         aberrant fruiting body morphology   \n",
       "31                                               DNA binding   \n",
       "32                                       signal transduction   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  1.03e-11          40        74         74  \n",
       "1   GO: biological_process  5.49e-11          59       145        145  \n",
       "2   GO: cellular_component  6.85e-09          58       157        157  \n",
       "3   GO: biological_process  1.95e-05         109       440        438  \n",
       "4    Dictybase: Phenotypes  3.45e-05          10        12         12  \n",
       "5   GO: cellular_component  1.09e-03           9        13         13  \n",
       "6   GO: cellular_component  1.09e-03           9        13         13  \n",
       "7           KEGG: Pathways  1.09e-03          15        31         31  \n",
       "8    Dictybase: Phenotypes  6.61e-03          21        60         59  \n",
       "9    Dictybase: Phenotypes  1.29e-02          27        88         88  \n",
       "10   Dictybase: Phenotypes  1.68e-02           5         6          6  \n",
       "11  GO: biological_process  2.60e-02           8        15         15  \n",
       "12   Dictybase: Phenotypes  2.60e-02           7        12         12  \n",
       "13  GO: biological_process  6.48e-02           4         5          5  \n",
       "14   Dictybase: Phenotypes  6.48e-02          18        58         58  \n",
       "15   Dictybase: Phenotypes  6.48e-02           4         5          5  \n",
       "16  GO: molecular_function  6.87e-02          13        37         37  \n",
       "17   Dictybase: Phenotypes  7.23e-02           5         8          8  \n",
       "18   Dictybase: Phenotypes  1.09e-01          18        62         62  \n",
       "19          KEGG: Pathways  1.09e-01           9        23         23  \n",
       "20  GO: molecular_function  1.09e-01          69       328        327  \n",
       "21   Dictybase: Phenotypes  1.11e-01           4         6          6  \n",
       "22   Dictybase: Phenotypes  1.11e-01           4         6          6  \n",
       "23   Dictybase: Phenotypes  1.46e-01          14        46         46  \n",
       "24  GO: cellular_component  1.70e-01          22        85         85  \n",
       "25          KEGG: Pathways  1.70e-01           5        10         10  \n",
       "26   Dictybase: Phenotypes  1.80e-01          22        95         86  \n",
       "27   Dictybase: Phenotypes  1.84e-01           4         7          7  \n",
       "28   Dictybase: Phenotypes  1.84e-01          15        53         53  \n",
       "29          KEGG: Pathways  1.84e-01           6        15         14  \n",
       "30   Dictybase: Phenotypes  2.02e-01          24       101         98  \n",
       "31  GO: molecular_function  2.19e-01          47       221        221  \n",
       "32  GO: biological_process  2.26e-01          70       352        350  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 1970 with EID: 1958\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 277 out of 494 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.61e-06</td>\n",
       "      <td>44</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.61e-06</td>\n",
       "      <td>46</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.33e-06</td>\n",
       "      <td>28</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.24e-04</td>\n",
       "      <td>86</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.64e-04</td>\n",
       "      <td>14</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.01e-02</td>\n",
       "      <td>23</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.82e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.82e-02</td>\n",
       "      <td>62</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.82e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.82e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.72e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.72e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.72e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased prestalk cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.57e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.57e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.67e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.70e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>19</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>56</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>19</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.78e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.90e-01</td>\n",
       "      <td>58</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.90e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>aberrant autophagic cell death</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>delayed culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.20e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.20e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       cell differentiation   \n",
       "1                                       extracellular region   \n",
       "2   anatomical structure formation involved in morphogenesis   \n",
       "3                           anatomical structure development   \n",
       "4                              Starch and sucrose metabolism   \n",
       "5                                       aberrant culmination   \n",
       "6                                secondary metabolic process   \n",
       "7                                                  wild type   \n",
       "8                                            spores inviable   \n",
       "9                  SNARE interactions in vesicular transport   \n",
       "10                 development arrests at tipped mound stage   \n",
       "11                                 decreased spore viability   \n",
       "12                            aberrant spore coat morphology   \n",
       "13                   decreased prestalk cell differentiation   \n",
       "14                   increased prespore cell differentiation   \n",
       "15                                     abolished culmination   \n",
       "16       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "17                                           Golgi apparatus   \n",
       "18                                           kinase activity   \n",
       "19                                 decreased gene expression   \n",
       "20                                      extracellular matrix   \n",
       "21                                       signal transduction   \n",
       "22                 DNA-binding transcription factor activity   \n",
       "23                        development arrests at mound stage   \n",
       "24                            aberrant autophagic cell death   \n",
       "25                          abolished cellulose biosynthesis   \n",
       "26                                       delayed culmination   \n",
       "27                                   delayed gene expression   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  1.61e-06          44       145        145  \n",
       "1   GO: cellular_component  1.61e-06          46       157        157  \n",
       "2   GO: biological_process  2.33e-06          28        74         74  \n",
       "3   GO: biological_process  4.24e-04          86       440        438  \n",
       "4           KEGG: Pathways  4.64e-04          14        31         31  \n",
       "5    Dictybase: Phenotypes  2.01e-02          23        88         88  \n",
       "6   GO: biological_process  3.82e-02           4         5          5  \n",
       "7    Dictybase: Phenotypes  3.82e-02          62       341        338  \n",
       "8    Dictybase: Phenotypes  3.82e-02           4         5          5  \n",
       "9           KEGG: Pathways  3.82e-02           9        23         23  \n",
       "10   Dictybase: Phenotypes  4.72e-02           6        12         12  \n",
       "11   Dictybase: Phenotypes  4.72e-02          16        60         59  \n",
       "12   Dictybase: Phenotypes  4.72e-02           6        12         12  \n",
       "13   Dictybase: Phenotypes  6.57e-02           4         6          6  \n",
       "14   Dictybase: Phenotypes  6.57e-02           4         6          6  \n",
       "15   Dictybase: Phenotypes  6.67e-02          16        62         62  \n",
       "16          KEGG: Pathways  8.70e-02           5        10         10  \n",
       "17  GO: cellular_component  1.35e-01          19        85         85  \n",
       "18  GO: molecular_function  1.35e-01          56       328        327  \n",
       "19   Dictybase: Phenotypes  1.46e-01          19        95         86  \n",
       "20  GO: cellular_component  1.78e-01           4         8          8  \n",
       "21  GO: biological_process  1.90e-01          58       352        350  \n",
       "22  GO: molecular_function  1.90e-01          10        37         37  \n",
       "23   Dictybase: Phenotypes  1.92e-01          13        56         54  \n",
       "24   Dictybase: Phenotypes  2.05e-01           3         5          5  \n",
       "25   Dictybase: Phenotypes  2.05e-01           3         5          5  \n",
       "26   Dictybase: Phenotypes  2.20e-01           8        28         28  \n",
       "27   Dictybase: Phenotypes  2.20e-01           6        19         18  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  yem selected: 2231 with EID: 2218\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.34\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 69 out of 760 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.98e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>26</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Glycerolipid metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>16</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>42</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                               secondary metabolic process   \n",
       "1  anatomical structure formation involved in morphogenesis   \n",
       "2                                   Glycerolipid metabolism   \n",
       "3                 DNA-binding transcription factor activity   \n",
       "4                                      cell differentiation   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: biological_process  9.98e-02           5         5          5  \n",
       "1  GO: biological_process  1.09e-01          26        74         74  \n",
       "2          KEGG: Pathways  1.09e-01          10        20         19  \n",
       "3  GO: molecular_function  1.09e-01          16        37         37  \n",
       "4  GO: biological_process  2.13e-01          42       145        145  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_de+'nobatchrep_combined.tsv',index_col=0,sep='\\t')\n",
    "for stage in [phenotype for phenotype in PHENOTYPES if phenotype in list(milestones['Stage'].unique())]:\n",
    "    query_names=list(milestones.query('log2FoldChange>=1 & padj <= 0.05 & Stage==\"'+stage+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DESeq2 WT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### DESeq2 WT lFC >= 2 and padj <= 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 878 with EID: 873\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.39\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 248 out of 337 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.46e-19</td>\n",
       "      <td>43</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.98e-15</td>\n",
       "      <td>36</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.23e-15</td>\n",
       "      <td>51</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.23e-15</td>\n",
       "      <td>37</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.32e-11</td>\n",
       "      <td>36</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.66e-06</td>\n",
       "      <td>45</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.75e-04</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.06e-03</td>\n",
       "      <td>17</td>\n",
       "      <td>72</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.29e-03</td>\n",
       "      <td>35</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.89e-03</td>\n",
       "      <td>43</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>lipid metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.00e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.32e-03</td>\n",
       "      <td>16</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Porphyrin and chlorophyll metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.09e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.77e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.01e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>64</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.18e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>58</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.04e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Glycine</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.04e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cellular amino acid metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.57e-02</td>\n",
       "      <td>20</td>\n",
       "      <td>125</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.87e-02</td>\n",
       "      <td>25</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.09e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>tRNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.09e-02</td>\n",
       "      <td>13</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>Vitamin B6 metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.34e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Lysine biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.59e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>22</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>63</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.43e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>80</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>Butanoate metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>protein-macromolecule adaptor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>lysosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Fatty acid metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.48e-01</td>\n",
       "      <td>40</td>\n",
       "      <td>354</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Gene set                Ontology       FDR  \\\n",
       "0                                 Ribosome          KEGG: Pathways  2.46e-19   \n",
       "1       structural constituent of ribosome  GO: molecular_function  1.98e-15   \n",
       "2                              translation  GO: biological_process  4.23e-15   \n",
       "3                                 ribosome  GO: cellular_component  4.23e-15   \n",
       "4             structural molecule activity  GO: molecular_function  9.32e-11   \n",
       "5                            mitochondrion  GO: cellular_component  1.66e-06   \n",
       "6                     Steroid biosynthesis          KEGG: Pathways  1.75e-04   \n",
       "7        Ribosome biogenesis in eukaryotes          KEGG: Pathways  1.06e-03   \n",
       "8                  oxidoreductase activity  GO: molecular_function  1.29e-03   \n",
       "9    Biosynthesis of secondary metabolites          KEGG: Pathways  1.89e-03   \n",
       "10                 lipid metabolic process  GO: biological_process  3.00e-03   \n",
       "11                               nucleolus  GO: cellular_component  5.32e-03   \n",
       "12    Porphyrin and chlorophyll metabolism          KEGG: Pathways  1.09e-02   \n",
       "13                      hydrolase activity  GO: molecular_function  1.77e-02   \n",
       "14             Aminoacyl-tRNA biosynthesis          KEGG: Pathways  2.01e-02   \n",
       "15              methyltransferase activity  GO: molecular_function  3.18e-02   \n",
       "16                   Fatty acid elongation          KEGG: Pathways  4.04e-02   \n",
       "17                                 Glycine          KEGG: Pathways  4.04e-02   \n",
       "18   cellular amino acid metabolic process  GO: biological_process  4.57e-02   \n",
       "19                             RNA binding  GO: molecular_function  4.87e-02   \n",
       "20                     ribosome biogenesis  GO: biological_process  5.09e-02   \n",
       "21                  tRNA metabolic process  GO: biological_process  5.09e-02   \n",
       "22                   Vitamin B6 metabolism          KEGG: Pathways  5.34e-02   \n",
       "23                     Lysine biosynthesis          KEGG: Pathways  9.59e-02   \n",
       "24                    extracellular region  GO: cellular_component  1.22e-01   \n",
       "25             Biosynthesis of amino acids          KEGG: Pathways  1.28e-01   \n",
       "26                         ligase activity  GO: molecular_function  1.43e-01   \n",
       "27                    Butanoate metabolism          KEGG: Pathways  1.63e-01   \n",
       "28                      isomerase activity  GO: molecular_function  1.87e-01   \n",
       "29  protein-macromolecule adaptor activity  GO: molecular_function  1.89e-01   \n",
       "30                                lysosome  GO: cellular_component  2.25e-01   \n",
       "31                   Fatty acid metabolism          KEGG: Pathways  2.33e-01   \n",
       "32        small molecule metabolic process  GO: biological_process  2.48e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           43       105         97  \n",
       "1           36        85         85  \n",
       "2           51       169        169  \n",
       "3           37        93         93  \n",
       "4           36       117        117  \n",
       "5           45       229        229  \n",
       "6            9        18         18  \n",
       "7           17        72         68  \n",
       "8           35       210        207  \n",
       "9           43       289        281  \n",
       "10          28       159        159  \n",
       "11          16        72         72  \n",
       "12           6        14         14  \n",
       "13          11        46         45  \n",
       "14          11        64         46  \n",
       "15          12        58         56  \n",
       "16           5        13         13  \n",
       "17           7        27         24  \n",
       "18          20       125        124  \n",
       "19          25       169        169  \n",
       "20          12        61         61  \n",
       "21          13        69         69  \n",
       "22           3         5          5  \n",
       "23           3         6          6  \n",
       "24          22       157        157  \n",
       "25          11        63         62  \n",
       "26          13        80         80  \n",
       "27           4        13         13  \n",
       "28           9        50         50  \n",
       "29           3         8          8  \n",
       "30          11        69         69  \n",
       "31           6        30         29  \n",
       "32          40       354        353  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 87 with EID: 86\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.17\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 6 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.88e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.88e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.88e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Gene set                Ontology       FDR  \\\n",
       "0                             cell adhesion  GO: biological_process  6.88e-02   \n",
       "1                    precocious aggregation   Dictybase: Phenotypes  6.88e-02   \n",
       "2  development arrests at loose mound stage   Dictybase: Phenotypes  6.88e-02   \n",
       "3                           plasma membrane  GO: cellular_component  1.56e-01   \n",
       "4                     decreased sporulation   Dictybase: Phenotypes  1.56e-01   \n",
       "5              decreased fruiting body size   Dictybase: Phenotypes  1.56e-01   \n",
       "6                   decreased cell motility   Dictybase: Phenotypes  1.56e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           3        84         83  \n",
       "1           2        20         20  \n",
       "2           2        20         20  \n",
       "3           4       268        268  \n",
       "4           2        53         53  \n",
       "5           2        76         75  \n",
       "6           2        76         76  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 70 with EID: 69\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.22\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.69e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.31e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.59e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>GTPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Gene set                Ontology       FDR  N in query  \\\n",
       "0      abolished slug migration   Dictybase: Phenotypes  2.69e-02           3   \n",
       "1               plasma membrane  GO: cellular_component  3.20e-02           5   \n",
       "2            aberrant streaming   Dictybase: Phenotypes  3.20e-02           3   \n",
       "3          aberrant aggregation   Dictybase: Phenotypes  3.20e-02           3   \n",
       "4          aberrant culmination   Dictybase: Phenotypes  4.31e-02           3   \n",
       "5  aberrant cAMP wave formation   Dictybase: Phenotypes  4.59e-02           3   \n",
       "6       aberrant slug migration   Dictybase: Phenotypes  1.17e-01           2   \n",
       "7               GTPase activity  GO: molecular_function  1.17e-01           3   \n",
       "8                  reproduction  GO: biological_process  1.34e-01           2   \n",
       "9                 cell adhesion  GO: biological_process  1.37e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0        43         43  \n",
       "1       268        268  \n",
       "2        72         72  \n",
       "3        61         61  \n",
       "4        88         88  \n",
       "5        96         96  \n",
       "6        58         58  \n",
       "7       145        145  \n",
       "8        79         79  \n",
       "9        84         83  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 24 with EID: 24\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.17\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 4 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.15e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>354</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.15e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.15e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Gene set                Ontology       FDR  \\\n",
       "0       small molecule metabolic process  GO: biological_process  9.15e-02   \n",
       "1  Biosynthesis of secondary metabolites          KEGG: Pathways  9.15e-02   \n",
       "2                        kinase activity  GO: molecular_function  9.15e-02   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           2       354        353  \n",
       "1           2       289        281  \n",
       "2           2       328        327  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 60 with EID: 60\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.37\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.36e-08</td>\n",
       "      <td>11</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.33e-08</td>\n",
       "      <td>8</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.55e-08</td>\n",
       "      <td>8</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.20e-07</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.58e-07</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.58e-07</td>\n",
       "      <td>8</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.07e-06</td>\n",
       "      <td>5</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.81e-03</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.89e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.48e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.72e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.03e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.82e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.83e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.83e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.04e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.54e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.54e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>354</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Gene set                Ontology       FDR  \\\n",
       "0                         cell cycle  GO: biological_process  1.36e-08   \n",
       "1                         chromosome  GO: cellular_component  4.33e-08   \n",
       "2            chromosome organization  GO: biological_process  4.55e-08   \n",
       "3           mitotic nuclear division  GO: biological_process  1.20e-07   \n",
       "4             chromosome segregation  GO: biological_process  2.58e-07   \n",
       "5                 mitotic cell cycle  GO: biological_process  2.58e-07   \n",
       "6                    DNA replication          KEGG: Pathways  5.07e-06   \n",
       "7                  cell cycle arrest   Dictybase: Phenotypes  1.81e-03   \n",
       "8                      cell division  GO: biological_process  1.89e-03   \n",
       "9              Pyrimidine metabolism          KEGG: Pathways  2.48e-03   \n",
       "10                   ATPase activity  GO: molecular_function  8.72e-03   \n",
       "11     microtubule organizing center  GO: cellular_component  1.03e-02   \n",
       "12                   Mismatch repair          KEGG: Pathways  1.82e-02   \n",
       "13                      cytoskeleton  GO: cellular_component  5.83e-02   \n",
       "14        Nucleotide excision repair          KEGG: Pathways  5.83e-02   \n",
       "15             DNA metabolic process  GO: biological_process  6.04e-02   \n",
       "16                       DNA binding  GO: molecular_function  7.54e-02   \n",
       "17                 helicase activity  GO: molecular_function  7.54e-02   \n",
       "18  small molecule metabolic process  GO: biological_process  2.47e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           11       167        167  \n",
       "1            8        70         70  \n",
       "2            8        74         74  \n",
       "3            5        15         15  \n",
       "4            5        18         18  \n",
       "5            8       100        100  \n",
       "6            5        34         33  \n",
       "7            2         5          5  \n",
       "8            5       115        115  \n",
       "9            3        31         29  \n",
       "10           5       169        168  \n",
       "11           3        50         50  \n",
       "12           2        19         19  \n",
       "13           4       185        185  \n",
       "14           2        37         37  \n",
       "15           3       105        105  \n",
       "16           4       221        221  \n",
       "17           2        51         51  \n",
       "18           4       354        353  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 148 with EID: 148\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.14\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 11 out of 21 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.83e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.69e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.69e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>28</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                Ontology       FDR  \\\n",
       "0              extracellular region  GO: cellular_component  3.83e-05   \n",
       "1                         cell wall  GO: cellular_component  3.69e-02   \n",
       "2  external encapsulating structure  GO: cellular_component  3.69e-02   \n",
       "3            Glutathione metabolism          KEGG: Pathways  1.02e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           8       157        157  \n",
       "1           2        13         13  \n",
       "2           2        13         13  \n",
       "3           2        28         25  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 545 with EID: 544\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.24\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 63 out of 132 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.61e-07</td>\n",
       "      <td>23</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.25e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.21e-05</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.21e-05</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.59e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.03e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.40e-03</td>\n",
       "      <td>10</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.52e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.21e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.59e-02</td>\n",
       "      <td>13</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.82e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>20</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1                             aberrant spore coat morphology   \n",
       "2                                                  cell wall   \n",
       "3                           external encapsulating structure   \n",
       "4                       cell wall organization or biogenesis   \n",
       "5                                       extracellular matrix   \n",
       "6   anatomical structure formation involved in morphogenesis   \n",
       "7                           anatomical structure development   \n",
       "8                                  decreased spore viability   \n",
       "9                                       cell differentiation   \n",
       "10                             Starch and sucrose metabolism   \n",
       "11                                                 wild type   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  1.61e-07          23       157        157  \n",
       "1    Dictybase: Phenotypes  2.25e-06           7        12         12  \n",
       "2   GO: cellular_component  7.21e-05           6        13         13  \n",
       "3   GO: cellular_component  7.21e-05           6        13         13  \n",
       "4   GO: biological_process  1.59e-04           6        15         15  \n",
       "5   GO: cellular_component  2.03e-03           4         8          8  \n",
       "6   GO: biological_process  3.40e-03          10        74         74  \n",
       "7   GO: biological_process  9.52e-03          28       440        438  \n",
       "8    Dictybase: Phenotypes  1.21e-02           8        60         59  \n",
       "9   GO: biological_process  1.59e-02          13       145        145  \n",
       "10          KEGG: Pathways  4.82e-02           5        31         31  \n",
       "11   Dictybase: Phenotypes  1.09e-01          20       341        338  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 944 with EID: 936\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 94 out of 211 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.59e-10</td>\n",
       "      <td>33</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.25e-04</td>\n",
       "      <td>16</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.30e-04</td>\n",
       "      <td>23</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.64e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.27e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.75e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.48e-02</td>\n",
       "      <td>39</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.79e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.44e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.99e-02</td>\n",
       "      <td>31</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.65e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.49e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1   anatomical structure formation involved in morphogenesis   \n",
       "2                                       cell differentiation   \n",
       "3                             aberrant spore coat morphology   \n",
       "4                       cell wall organization or biogenesis   \n",
       "5                                       extracellular matrix   \n",
       "6                           anatomical structure development   \n",
       "7                              Starch and sucrose metabolism   \n",
       "8                                secondary metabolic process   \n",
       "9                                                  wild type   \n",
       "10                                        hydrolase activity   \n",
       "11                                 decreased spore viability   \n",
       "12       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "13                                 decreased gene expression   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: cellular_component  6.59e-10          33       157        157  \n",
       "1   GO: biological_process  1.25e-04          16        74         74  \n",
       "2   GO: biological_process  1.30e-04          23       145        145  \n",
       "3    Dictybase: Phenotypes  8.64e-04           6        12         12  \n",
       "4   GO: biological_process  3.27e-03           6        15         15  \n",
       "5   GO: cellular_component  1.75e-02           4         8          8  \n",
       "6   GO: biological_process  2.48e-02          39       440        438  \n",
       "7           KEGG: Pathways  2.79e-02           7        31         31  \n",
       "8   GO: biological_process  3.44e-02           3         5          5  \n",
       "9    Dictybase: Phenotypes  3.99e-02          31       341        338  \n",
       "10  GO: molecular_function  4.65e-02           8        46         45  \n",
       "11   Dictybase: Phenotypes  2.30e-01           8        60         59  \n",
       "12          KEGG: Pathways  2.34e-01           3        10         10  \n",
       "13   Dictybase: Phenotypes  2.49e-01          10        95         86  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 307 with EID: 303\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 69 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>8.87e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.87e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Fructose and mannose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Pentose and glucuronate interconversions</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Gene set                Ontology       FDR  \\\n",
       "0                      extracellular region  GO: cellular_component  8.87e-02   \n",
       "1                   oxidoreductase activity  GO: molecular_function  8.87e-02   \n",
       "2           Fructose and mannose metabolism          KEGG: Pathways  1.30e-01   \n",
       "3  Pentose and glucuronate interconversions          KEGG: Pathways  1.30e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0           9       157        157  \n",
       "1          11       210        207  \n",
       "2           3        19         19  \n",
       "3           2         6          6  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_de+'WT_batchrep_combined.tsv',index_col=0,sep='\\t')\n",
    "for stage in [phenotype for phenotype in PHENOTYPES if phenotype in list(milestones['Stage'].unique())]:\n",
    "    query_names=list(milestones.query('log2FoldChange>=2 & padj <= 0.01 & Stage==\"'+stage+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### DESeq2 WT lFC >= 1 and padj <= 0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 2578 with EID: 2560\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 596 out of 966 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.19e-52</td>\n",
       "      <td>94</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.38e-46</td>\n",
       "      <td>88</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.46e-45</td>\n",
       "      <td>82</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.01e-37</td>\n",
       "      <td>120</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.99e-31</td>\n",
       "      <td>89</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.62e-22</td>\n",
       "      <td>125</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.32e-19</td>\n",
       "      <td>50</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.26e-12</td>\n",
       "      <td>48</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.10e-11</td>\n",
       "      <td>45</td>\n",
       "      <td>72</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.89e-09</td>\n",
       "      <td>79</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>tRNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.57e-08</td>\n",
       "      <td>41</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.26e-08</td>\n",
       "      <td>31</td>\n",
       "      <td>64</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.49e-07</td>\n",
       "      <td>33</td>\n",
       "      <td>71</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.04e-05</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.40e-05</td>\n",
       "      <td>30</td>\n",
       "      <td>58</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.88e-05</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.08e-04</td>\n",
       "      <td>38</td>\n",
       "      <td>80</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.97e-04</td>\n",
       "      <td>121</td>\n",
       "      <td>354</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cellular amino acid metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.97e-04</td>\n",
       "      <td>52</td>\n",
       "      <td>125</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.41e-04</td>\n",
       "      <td>99</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-02</td>\n",
       "      <td>32</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.15e-02</td>\n",
       "      <td>71</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>cofactor metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>30</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>mitochondrion organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.32e-02</td>\n",
       "      <td>13</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Porphyrin and chlorophyll metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.33e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.40e-02</td>\n",
       "      <td>14</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.39e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.39e-02</td>\n",
       "      <td>14</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.39e-02</td>\n",
       "      <td>21</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>Sulfur relay system</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.50e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.69e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Purine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>20</td>\n",
       "      <td>52</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.85e-01</td>\n",
       "      <td>48</td>\n",
       "      <td>150</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                Ontology  \\\n",
       "0                                         Ribosome          KEGG: Pathways   \n",
       "1                                         ribosome  GO: cellular_component   \n",
       "2               structural constituent of ribosome  GO: molecular_function   \n",
       "3                                      translation  GO: biological_process   \n",
       "4                     structural molecule activity  GO: molecular_function   \n",
       "5                                    mitochondrion  GO: cellular_component   \n",
       "6                              ribosome biogenesis  GO: biological_process   \n",
       "7                                        nucleolus  GO: cellular_component   \n",
       "8                Ribosome biogenesis in eukaryotes          KEGG: Pathways   \n",
       "9                                      RNA binding  GO: molecular_function   \n",
       "10                          tRNA metabolic process  GO: biological_process   \n",
       "11                     Aminoacyl-tRNA biosynthesis          KEGG: Pathways   \n",
       "12                       Oxidative phosphorylation          KEGG: Pathways   \n",
       "13                                    rRNA binding  GO: molecular_function   \n",
       "14                      methyltransferase activity  GO: molecular_function   \n",
       "15                            Steroid biosynthesis          KEGG: Pathways   \n",
       "16                                 ligase activity  GO: molecular_function   \n",
       "17                small molecule metabolic process  GO: biological_process   \n",
       "18           cellular amino acid metabolic process  GO: biological_process   \n",
       "19           Biosynthesis of secondary metabolites          KEGG: Pathways   \n",
       "20  generation of precursor metabolites and energy  GO: biological_process   \n",
       "21                         oxidoreductase activity  GO: molecular_function   \n",
       "22                      cofactor metabolic process  GO: biological_process   \n",
       "23                      mitochondrion organization  GO: biological_process   \n",
       "24            Porphyrin and chlorophyll metabolism          KEGG: Pathways   \n",
       "25                        unfolded protein binding  GO: molecular_function   \n",
       "26                           Fatty acid elongation          KEGG: Pathways   \n",
       "27                           Pyrimidine metabolism          KEGG: Pathways   \n",
       "28                              isomerase activity  GO: molecular_function   \n",
       "29                             Sulfur relay system          KEGG: Pathways   \n",
       "30                                 protein folding  GO: biological_process   \n",
       "31                               Purine metabolism          KEGG: Pathways   \n",
       "32              transmembrane transporter activity  GO: molecular_function   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   3.19e-52          94       105         97  \n",
       "1   2.38e-46          88        93         93  \n",
       "2   3.46e-45          82        85         85  \n",
       "3   8.01e-37         120       169        169  \n",
       "4   8.99e-31          89       117        117  \n",
       "5   4.62e-22         125       229        229  \n",
       "6   1.32e-19          50        61         61  \n",
       "7   1.26e-12          48        72         72  \n",
       "8   1.10e-11          45        72         68  \n",
       "9   3.89e-09          79       169        169  \n",
       "10  1.57e-08          41        69         69  \n",
       "11  2.26e-08          31        64         46  \n",
       "12  2.49e-07          33        71         54  \n",
       "13  1.04e-05          12        13         13  \n",
       "14  5.40e-05          30        58         56  \n",
       "15  5.88e-05          14        18         18  \n",
       "16  1.08e-04          38        80         80  \n",
       "17  1.97e-04         121       354        353  \n",
       "18  1.97e-04          52       125        124  \n",
       "19  3.41e-04          99       289        281  \n",
       "20  1.04e-02          32        77         77  \n",
       "21  1.15e-02          71       210        207  \n",
       "22  1.31e-02          30        72         72  \n",
       "23  1.32e-02          13        23         23  \n",
       "24  2.33e-02           9        14         14  \n",
       "25  2.40e-02          14        27         27  \n",
       "26  5.39e-02           8        13         13  \n",
       "27  5.39e-02          14        31         29  \n",
       "28  5.39e-02          21        50         50  \n",
       "29  9.50e-02           6         9          9  \n",
       "30  9.69e-02          16        37         37  \n",
       "31  1.09e-01          20        52         50  \n",
       "32  1.85e-01          48       150        149  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 557 with EID: 553\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.35\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 86 out of 193 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.36e-40</td>\n",
       "      <td>34</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.55e-09</td>\n",
       "      <td>23</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.49e-07</td>\n",
       "      <td>35</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.57e-06</td>\n",
       "      <td>18</td>\n",
       "      <td>91</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.65e-05</td>\n",
       "      <td>20</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.73e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Protein export</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.07e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.55e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.55e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.93e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>31</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Gene set                Ontology  \\\n",
       "0                                   Proteasome          KEGG: Pathways   \n",
       "1                           peptidase activity  GO: molecular_function   \n",
       "2                            catabolic process  GO: biological_process   \n",
       "3  Protein processing in endoplasmic reticulum          KEGG: Pathways   \n",
       "4                        endoplasmic reticulum  GO: cellular_component   \n",
       "5       Various types of N-glycan biosynthesis          KEGG: Pathways   \n",
       "6                               Protein export          KEGG: Pathways   \n",
       "7                           aberrant streaming   Dictybase: Phenotypes   \n",
       "8     development arrests at loose mound stage   Dictybase: Phenotypes   \n",
       "9                        N-Glycan biosynthesis          KEGG: Pathways   \n",
       "\n",
       "        FDR  N in query  Set size  N in ref.  \n",
       "0  2.36e-40          34        39         37  \n",
       "1  2.55e-09          23        91         91  \n",
       "2  4.49e-07          35       250        250  \n",
       "3  7.57e-06          18        91         88  \n",
       "4  2.65e-05          20       117        117  \n",
       "5  1.73e-02           6        22         22  \n",
       "6  4.07e-02           5        19         18  \n",
       "7  5.55e-02          10        72         72  \n",
       "8  5.55e-02           5        20         20  \n",
       "9  5.93e-02           6        31         30  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 509 with EID: 505\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.36\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 136 out of 182 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.73e-22</td>\n",
       "      <td>24</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.64e-03</td>\n",
       "      <td>27</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.64e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.17e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.25e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.71e-02</td>\n",
       "      <td>23</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>19</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>28</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>13</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.09e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.72e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant contractile vacuole morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.72e-02</td>\n",
       "      <td>32</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.76e-02</td>\n",
       "      <td>15</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.12e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.12e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.12e-02</td>\n",
       "      <td>26</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.16e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>67</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased pinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.46e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>Other glycan degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.97e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>aberrant positive regulation of chemorepellant activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>91</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>12</td>\n",
       "      <td>146</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>decreased phagocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.49e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>increased protein level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>31</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                                Proteasome   \n",
       "1                                           plasma membrane   \n",
       "2                                        peptidase activity   \n",
       "3                                        aberrant streaming   \n",
       "4                                  decreased aggregate size   \n",
       "5                                         catabolic process   \n",
       "6                                                locomotion   \n",
       "7                                 cytoskeleton organization   \n",
       "8                                       signal transduction   \n",
       "9                                     endoplasmic reticulum   \n",
       "10                 development arrests at loose mound stage   \n",
       "11                                     aberrant cytokinesis   \n",
       "12                                 aberrant cell morphology   \n",
       "13                   Various types of N-glycan biosynthesis   \n",
       "14                             cytoskeletal protein binding   \n",
       "15                                       cell morphogenesis   \n",
       "16                                             cytoskeleton   \n",
       "17                  aberrant contractile vacuole morphology   \n",
       "18                             aberrant cAMP wave formation   \n",
       "19                         anatomical structure development   \n",
       "20                                     extracellular region   \n",
       "21                                    decreased culmination   \n",
       "22                            increased basal F-actin level   \n",
       "23                                                wild type   \n",
       "24                                            cell motility   \n",
       "25                                    decreased pinocytosis   \n",
       "26                                 Other glycan degradation   \n",
       "27                                            cell adhesion   \n",
       "28                                      delayed development   \n",
       "29                               multiple tips in culminant   \n",
       "30  aberrant positive regulation of chemorepellant activity   \n",
       "31                                     aberrant culmination   \n",
       "32              Protein processing in endoplasmic reticulum   \n",
       "33                             decreased fruiting body size   \n",
       "34                                     aberrant aggregation   \n",
       "35                                    decreased growth rate   \n",
       "36                                   decreased phagocytosis   \n",
       "37                            aberrant F-actin organization   \n",
       "38                                         ABC transporters   \n",
       "39                       development arrests at mound stage   \n",
       "40                                       mitotic cell cycle   \n",
       "41                                 abolished slug migration   \n",
       "42                                  increased protein level   \n",
       "43                                 increased slug migration   \n",
       "44                abolished activation of adenylate cyclase   \n",
       "45                                    N-Glycan biosynthesis   \n",
       "46                                    increased growth rate   \n",
       "47                                enzyme regulator activity   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0           KEGG: Pathways  9.73e-22          24        39         37  \n",
       "1   GO: cellular_component  5.64e-03          27       268        268  \n",
       "2   GO: molecular_function  5.64e-03          14        91         91  \n",
       "3    Dictybase: Phenotypes  2.17e-02          11        72         72  \n",
       "4    Dictybase: Phenotypes  2.25e-02           7        32         32  \n",
       "5   GO: biological_process  3.71e-02          23       250        250  \n",
       "6   GO: biological_process  4.09e-02          18       191        189  \n",
       "7   GO: biological_process  4.09e-02          19       197        196  \n",
       "8   GO: biological_process  4.09e-02          28       352        350  \n",
       "9   GO: cellular_component  4.09e-02          13       117        117  \n",
       "10   Dictybase: Phenotypes  4.09e-02           5        20         20  \n",
       "11   Dictybase: Phenotypes  4.09e-02          12        98         98  \n",
       "12   Dictybase: Phenotypes  4.09e-02           7        42         41  \n",
       "13          KEGG: Pathways  4.09e-02           5        22         22  \n",
       "14  GO: molecular_function  4.09e-02          16       158        158  \n",
       "15  GO: biological_process  4.72e-02           5        23         23  \n",
       "16  GO: cellular_component  5.49e-02          17       185        185  \n",
       "17   Dictybase: Phenotypes  5.49e-02           3         8          8  \n",
       "18   Dictybase: Phenotypes  5.49e-02          11        96         96  \n",
       "19  GO: biological_process  5.72e-02          32       440        438  \n",
       "20  GO: cellular_component  5.76e-02          15       157        157  \n",
       "21   Dictybase: Phenotypes  6.12e-02           3         9          9  \n",
       "22   Dictybase: Phenotypes  6.12e-02           4        18         17  \n",
       "23   Dictybase: Phenotypes  6.12e-02          26       341        338  \n",
       "24  GO: biological_process  9.16e-02           8        67         66  \n",
       "25   Dictybase: Phenotypes  9.46e-02           5        30         30  \n",
       "26          KEGG: Pathways  9.97e-02           3        11         11  \n",
       "27  GO: biological_process  1.04e-01           9        84         83  \n",
       "28   Dictybase: Phenotypes  1.04e-01           8        69         69  \n",
       "29   Dictybase: Phenotypes  1.30e-01           2         5          5  \n",
       "30   Dictybase: Phenotypes  1.30e-01           2         5          5  \n",
       "31   Dictybase: Phenotypes  1.30e-01           9        88         88  \n",
       "32          KEGG: Pathways  1.30e-01           9        91         88  \n",
       "33   Dictybase: Phenotypes  1.35e-01           8        76         75  \n",
       "34   Dictybase: Phenotypes  1.35e-01           7        61         61  \n",
       "35   Dictybase: Phenotypes  1.44e-01          12       146        137  \n",
       "36   Dictybase: Phenotypes  1.49e-01           7        63         63  \n",
       "37   Dictybase: Phenotypes  1.53e-01           5        37         37  \n",
       "38          KEGG: Pathways  1.66e-01           3        15         15  \n",
       "39   Dictybase: Phenotypes  2.04e-01           6        56         54  \n",
       "40  GO: biological_process  2.17e-01           9       100        100  \n",
       "41   Dictybase: Phenotypes  2.38e-01           5        43         43  \n",
       "42   Dictybase: Phenotypes  2.38e-01           2         9          8  \n",
       "43   Dictybase: Phenotypes  2.38e-01           2         8          8  \n",
       "44   Dictybase: Phenotypes  2.38e-01           2         8          8  \n",
       "45          KEGG: Pathways  2.38e-01           4        31         30  \n",
       "46   Dictybase: Phenotypes  2.42e-01           4        32         31  \n",
       "47  GO: molecular_function  2.42e-01           9       105        105  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 305 with EID: 305\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.36\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 93 out of 109 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.17e-29</td>\n",
       "      <td>24</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.41e-26</td>\n",
       "      <td>29</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.62e-19</td>\n",
       "      <td>33</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.22e-16</td>\n",
       "      <td>22</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.88e-16</td>\n",
       "      <td>25</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.35e-15</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.19e-14</td>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.77e-11</td>\n",
       "      <td>13</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.31e-10</td>\n",
       "      <td>20</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.74e-10</td>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.28e-10</td>\n",
       "      <td>11</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.50e-09</td>\n",
       "      <td>25</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.66e-07</td>\n",
       "      <td>15</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.42e-06</td>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.87e-04</td>\n",
       "      <td>15</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.24e-04</td>\n",
       "      <td>8</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.70e-03</td>\n",
       "      <td>8</td>\n",
       "      <td>62</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.49e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.21e-03</td>\n",
       "      <td>21</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>9.81e-03</td>\n",
       "      <td>13</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.23e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.36e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.82e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.19e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.19e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>increased cell-cell adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Gene set                Ontology       FDR  \\\n",
       "0                  DNA replication          KEGG: Pathways  1.17e-29   \n",
       "1                       chromosome  GO: cellular_component  1.41e-26   \n",
       "2                       cell cycle  GO: biological_process  5.62e-19   \n",
       "3          chromosome organization  GO: biological_process  2.22e-16   \n",
       "4            DNA metabolic process  GO: biological_process  2.88e-16   \n",
       "5                  Mismatch repair          KEGG: Pathways  1.35e-15   \n",
       "6           chromosome segregation  GO: biological_process  3.19e-14   \n",
       "7       Nucleotide excision repair          KEGG: Pathways  8.77e-11   \n",
       "8                    cell division  GO: biological_process  2.31e-10   \n",
       "9         mitotic nuclear division  GO: biological_process  3.74e-10   \n",
       "10            Base excision repair          KEGG: Pathways  7.28e-10   \n",
       "11                     DNA binding  GO: molecular_function  6.50e-09   \n",
       "12              mitotic cell cycle  GO: biological_process  5.66e-07   \n",
       "13              nuclear chromosome  GO: cellular_component  1.42e-06   \n",
       "14                 ATPase activity  GO: molecular_function  3.87e-04   \n",
       "15   microtubule organizing center  GO: cellular_component  4.24e-04   \n",
       "16  Ubiquitin mediated proteolysis          KEGG: Pathways  1.70e-03   \n",
       "17        Homologous recombination          KEGG: Pathways  3.49e-03   \n",
       "18              response to stress  GO: biological_process  4.21e-03   \n",
       "19                    cytoskeleton  GO: cellular_component  9.81e-03   \n",
       "20               cell cycle arrest   Dictybase: Phenotypes  4.23e-02   \n",
       "21           Pyrimidine metabolism          KEGG: Pathways  4.36e-02   \n",
       "22               helicase activity  GO: molecular_function  6.82e-02   \n",
       "23      Non-homologous end-joining          KEGG: Pathways  7.19e-02   \n",
       "24                 histone binding  GO: molecular_function  7.19e-02   \n",
       "25    increased cell-cell adhesion   Dictybase: Phenotypes  2.30e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           24        34         33  \n",
       "1           29        70         70  \n",
       "2           33       167        167  \n",
       "3           22        74         74  \n",
       "4           25       105        105  \n",
       "5           13        19         19  \n",
       "6           12        18         18  \n",
       "7           13        37         37  \n",
       "8           20       115        115  \n",
       "9            9        15         15  \n",
       "10          11        28         28  \n",
       "11          25       221        221  \n",
       "12          15       100        100  \n",
       "13           7        17         17  \n",
       "14          15       169        168  \n",
       "15           8        50         50  \n",
       "16           8        62         61  \n",
       "17           5        26         25  \n",
       "18          21       360        358  \n",
       "19          13       185        185  \n",
       "20           2         5          5  \n",
       "21           4        31         29  \n",
       "22           5        51         51  \n",
       "23           2         7          7  \n",
       "24           2         7          7  \n",
       "25           2        13         13  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 309 with EID: 308\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.39\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 101 out of 119 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.23e-33</td>\n",
       "      <td>34</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.00e-29</td>\n",
       "      <td>24</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.42e-23</td>\n",
       "      <td>28</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.97e-21</td>\n",
       "      <td>36</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.78e-19</td>\n",
       "      <td>28</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.18e-17</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.16e-15</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.94e-13</td>\n",
       "      <td>11</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.25e-12</td>\n",
       "      <td>31</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.30e-12</td>\n",
       "      <td>23</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.95e-12</td>\n",
       "      <td>14</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.85e-11</td>\n",
       "      <td>12</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.07e-08</td>\n",
       "      <td>17</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.88e-08</td>\n",
       "      <td>8</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.11e-05</td>\n",
       "      <td>10</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.64e-04</td>\n",
       "      <td>9</td>\n",
       "      <td>62</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.93e-04</td>\n",
       "      <td>15</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.03e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.07e-03</td>\n",
       "      <td>15</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.90e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.21e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.69e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>aberrant nucleus morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.64e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.64e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.05e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.10e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.85e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>nuclear envelope</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.03e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Gene set                Ontology       FDR  \\\n",
       "0                       chromosome  GO: cellular_component  7.23e-33   \n",
       "1                  DNA replication          KEGG: Pathways  5.00e-29   \n",
       "2          chromosome organization  GO: biological_process  2.42e-23   \n",
       "3                       cell cycle  GO: biological_process  4.97e-21   \n",
       "4            DNA metabolic process  GO: biological_process  8.78e-19   \n",
       "5           chromosome segregation  GO: biological_process  1.18e-17   \n",
       "6                  Mismatch repair          KEGG: Pathways  3.16e-15   \n",
       "7         mitotic nuclear division  GO: biological_process  1.94e-13   \n",
       "8                      DNA binding  GO: molecular_function  1.25e-12   \n",
       "9                    cell division  GO: biological_process  1.30e-12   \n",
       "10      Nucleotide excision repair          KEGG: Pathways  7.95e-12   \n",
       "11            Base excision repair          KEGG: Pathways  5.85e-11   \n",
       "12              mitotic cell cycle  GO: biological_process  3.07e-08   \n",
       "13              nuclear chromosome  GO: cellular_component  7.88e-08   \n",
       "14   microtubule organizing center  GO: cellular_component  1.11e-05   \n",
       "15  Ubiquitin mediated proteolysis          KEGG: Pathways  4.64e-04   \n",
       "16                 ATPase activity  GO: molecular_function  7.93e-04   \n",
       "17           Pyrimidine metabolism          KEGG: Pathways  1.03e-03   \n",
       "18                    cytoskeleton  GO: cellular_component  2.07e-03   \n",
       "19        Homologous recombination          KEGG: Pathways  3.90e-03   \n",
       "20                 histone binding  GO: molecular_function  4.21e-03   \n",
       "21                     nucleoplasm  GO: cellular_component  2.69e-02   \n",
       "22     aberrant nucleus morphology   Dictybase: Phenotypes  3.64e-02   \n",
       "23               cell cycle arrest   Dictybase: Phenotypes  3.64e-02   \n",
       "24      Non-homologous end-joining          KEGG: Pathways  7.05e-02   \n",
       "25               helicase activity  GO: molecular_function  7.10e-02   \n",
       "26              response to stress  GO: biological_process  7.85e-02   \n",
       "27                nuclear envelope  GO: cellular_component  2.03e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           34        70         70  \n",
       "1           24        34         33  \n",
       "2           28        74         74  \n",
       "3           36       167        167  \n",
       "4           28       105        105  \n",
       "5           14        18         18  \n",
       "6           13        19         19  \n",
       "7           11        15         15  \n",
       "8           31       221        221  \n",
       "9           23       115        115  \n",
       "10          14        37         37  \n",
       "11          12        28         28  \n",
       "12          17       100        100  \n",
       "13           8        17         17  \n",
       "14          10        50         50  \n",
       "15           9        62         61  \n",
       "16          15       169        168  \n",
       "17           6        31         29  \n",
       "18          15       185        185  \n",
       "19           5        26         25  \n",
       "20           3         7          7  \n",
       "21           8        91         91  \n",
       "22           2         5          5  \n",
       "23           2         5          5  \n",
       "24           2         7          7  \n",
       "25           5        51         51  \n",
       "26          18       360        358  \n",
       "27           3        29         29  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 757 with EID: 753\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.24\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 105 out of 182 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-04</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.77e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.77e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.77e-04</td>\n",
       "      <td>13</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.77e-04</td>\n",
       "      <td>9</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.01e-04</td>\n",
       "      <td>13</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.00e-03</td>\n",
       "      <td>20</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.09e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.67e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.67e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>28</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.60e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.89e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.49e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.49e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.92e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.92e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.92e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.92e-02</td>\n",
       "      <td>15</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.92e-02</td>\n",
       "      <td>19</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.92e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.02e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.02e-02</td>\n",
       "      <td>14</td>\n",
       "      <td>150</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.87e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>165</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>aberrant myosin II organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>increased macropinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>increased prestalk 0 cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>increased sorting to prestalk region</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.83e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>extracellular space</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                     chromosome segregation   \n",
       "1                                                  cell wall   \n",
       "2                           external encapsulating structure   \n",
       "3                                                 chromosome   \n",
       "4                                            DNA replication   \n",
       "5                                    chromosome organization   \n",
       "6                                                 cell cycle   \n",
       "7                                   mitotic nuclear division   \n",
       "8                                         nuclear chromosome   \n",
       "9                                     Glutathione metabolism   \n",
       "10                                 decreased spore viability   \n",
       "11                            aberrant spore coat morphology   \n",
       "12                                          ABC transporters   \n",
       "13                                           ATPase activity   \n",
       "14  anatomical structure formation involved in morphogenesis   \n",
       "15                                             cell division   \n",
       "16                                        mitotic cell cycle   \n",
       "17                                      extracellular region   \n",
       "18                                               DNA binding   \n",
       "19                 DNA-binding transcription factor activity   \n",
       "20                                      Base excision repair   \n",
       "21                        transmembrane transporter activity   \n",
       "22                                     Pyrimidine metabolism   \n",
       "23                                   transmembrane transport   \n",
       "24                                      cell differentiation   \n",
       "25                           aberrant myosin II organization   \n",
       "26                                increased macropinocytosis   \n",
       "27                                         cell cycle arrest   \n",
       "28                 increased prestalk 0 cell differentiation   \n",
       "29                      increased sorting to prestalk region   \n",
       "30                             microtubule organizing center   \n",
       "31                                Nucleotide excision repair   \n",
       "32                   increased prespore cell differentiation   \n",
       "33                      cell wall organization or biogenesis   \n",
       "34                                       extracellular space   \n",
       "35                                           histone binding   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  1.13e-04           8        18         18  \n",
       "1   GO: cellular_component  5.77e-04           6        13         13  \n",
       "2   GO: cellular_component  5.77e-04           6        13         13  \n",
       "3   GO: cellular_component  5.77e-04          13        70         70  \n",
       "4           KEGG: Pathways  5.77e-04           9        34         33  \n",
       "5   GO: biological_process  9.01e-04          13        74         74  \n",
       "6   GO: biological_process  2.00e-03          20       167        167  \n",
       "7   GO: biological_process  1.09e-02           5        15         15  \n",
       "8   GO: cellular_component  1.67e-02           5        17         17  \n",
       "9           KEGG: Pathways  1.67e-02           6        28         25  \n",
       "10   Dictybase: Phenotypes  2.60e-02           9        60         59  \n",
       "11   Dictybase: Phenotypes  2.89e-02           4        12         12  \n",
       "12          KEGG: Pathways  6.49e-02           4        15         15  \n",
       "13  GO: molecular_function  6.49e-02          16       169        168  \n",
       "14  GO: biological_process  6.92e-02           9        74         74  \n",
       "15  GO: biological_process  6.92e-02          12       115        115  \n",
       "16  GO: biological_process  6.92e-02          11       100        100  \n",
       "17  GO: cellular_component  6.92e-02          15       157        157  \n",
       "18  GO: molecular_function  6.92e-02          19       221        221  \n",
       "19  GO: molecular_function  6.92e-02           6        37         37  \n",
       "20          KEGG: Pathways  8.02e-02           5        28         28  \n",
       "21  GO: molecular_function  8.02e-02          14       150        149  \n",
       "22          KEGG: Pathways  8.87e-02           5        31         29  \n",
       "23  GO: biological_process  1.37e-01          14       165        164  \n",
       "24  GO: biological_process  1.37e-01          13       145        145  \n",
       "25   Dictybase: Phenotypes  1.37e-01           2         5          5  \n",
       "26   Dictybase: Phenotypes  1.37e-01           2         5          5  \n",
       "27   Dictybase: Phenotypes  1.37e-01           2         5          5  \n",
       "28   Dictybase: Phenotypes  1.37e-01           2         5          5  \n",
       "29   Dictybase: Phenotypes  1.37e-01           2         5          5  \n",
       "30  GO: cellular_component  1.75e-01           6        50         50  \n",
       "31          KEGG: Pathways  1.75e-01           5        37         37  \n",
       "32   Dictybase: Phenotypes  1.81e-01           2         6          6  \n",
       "33  GO: biological_process  1.83e-01           3        15         15  \n",
       "34  GO: cellular_component  2.25e-01           2         7          7  \n",
       "35  GO: molecular_function  2.25e-01           2         7          7  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 1615 with EID: 1609\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.29\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 224 out of 468 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.36e-04</td>\n",
       "      <td>9</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.99e-03</td>\n",
       "      <td>22</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.99e-03</td>\n",
       "      <td>35</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.57e-03</td>\n",
       "      <td>36</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.23e-03</td>\n",
       "      <td>78</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.58e-03</td>\n",
       "      <td>18</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.62e-03</td>\n",
       "      <td>12</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.10e-02</td>\n",
       "      <td>22</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.00e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.00e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.27e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.27e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.27e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Phenylalanine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.63e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.63e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Alanine</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.63e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>19</td>\n",
       "      <td>89</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant spore morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>56</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>delayed culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant thermotaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>54</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant stalk formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>18</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>Tyrosine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>Taurine and hypotaurine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.90e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>abolished catalytic activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.49e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>aberrant phototaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.49e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>Arginine and proline metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.49e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>Pentose and glucuronate interconversions</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.49e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                             aberrant spore coat morphology   \n",
       "1   anatomical structure formation involved in morphogenesis   \n",
       "2                                       cell differentiation   \n",
       "3                                       extracellular region   \n",
       "4                           anatomical structure development   \n",
       "5                                  decreased spore viability   \n",
       "6                              Starch and sucrose metabolism   \n",
       "7                                            Golgi apparatus   \n",
       "8                       cell wall organization or biogenesis   \n",
       "9                           abolished cellulose biosynthesis   \n",
       "10                                                 cell wall   \n",
       "11                          external encapsulating structure   \n",
       "12                   increased prespore cell differentiation   \n",
       "13                                  Phenylalanine metabolism   \n",
       "14                                      Galactose metabolism   \n",
       "15                                                   Alanine   \n",
       "16                            carbohydrate metabolic process   \n",
       "17                                      extracellular matrix   \n",
       "18                                 aberrant spore morphology   \n",
       "19                                       signal transduction   \n",
       "20                                       delayed culmination   \n",
       "21                                      aberrant thermotaxis   \n",
       "22                                                 wild type   \n",
       "23                                  aberrant stalk formation   \n",
       "24                                      aberrant culmination   \n",
       "25                                           spores inviable   \n",
       "26                                       Tyrosine metabolism   \n",
       "27                        Taurine and hypotaurine metabolism   \n",
       "28                                   aberrant slug migration   \n",
       "29                              abolished catalytic activity   \n",
       "30       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "31                                       aberrant phototaxis   \n",
       "32                           Arginine and proline metabolism   \n",
       "33                  Pentose and glucuronate interconversions   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0    Dictybase: Phenotypes  2.36e-04           9        12         12  \n",
       "1   GO: biological_process  2.99e-03          22        74         74  \n",
       "2   GO: biological_process  2.99e-03          35       145        145  \n",
       "3   GO: cellular_component  4.57e-03          36       157        157  \n",
       "4   GO: biological_process  5.23e-03          78       440        438  \n",
       "5    Dictybase: Phenotypes  5.58e-03          18        60         59  \n",
       "6           KEGG: Pathways  5.62e-03          12        31         31  \n",
       "7   GO: cellular_component  1.10e-02          22        85         85  \n",
       "8   GO: biological_process  3.00e-02           7        15         15  \n",
       "9    Dictybase: Phenotypes  3.00e-02           4         5          5  \n",
       "10  GO: cellular_component  6.27e-02           6        13         13  \n",
       "11  GO: cellular_component  6.27e-02           6        13         13  \n",
       "12   Dictybase: Phenotypes  6.27e-02           4         6          6  \n",
       "13          KEGG: Pathways  7.63e-02           5        12         10  \n",
       "14          KEGG: Pathways  7.63e-02           6        15         14  \n",
       "15          KEGG: Pathways  7.63e-02           8        23         23  \n",
       "16  GO: biological_process  1.26e-01          19        89         88  \n",
       "17  GO: cellular_component  1.64e-01           4         8          8  \n",
       "18   Dictybase: Phenotypes  1.64e-01           4         8          8  \n",
       "19  GO: biological_process  1.66e-01          56       352        350  \n",
       "20   Dictybase: Phenotypes  1.66e-01           8        28         28  \n",
       "21   Dictybase: Phenotypes  1.66e-01           3         5          5  \n",
       "22   Dictybase: Phenotypes  1.66e-01          54       341        338  \n",
       "23   Dictybase: Phenotypes  1.66e-01           3         5          5  \n",
       "24   Dictybase: Phenotypes  1.66e-01          18        88         88  \n",
       "25   Dictybase: Phenotypes  1.66e-01           3         5          5  \n",
       "26          KEGG: Pathways  1.66e-01           5        15         13  \n",
       "27          KEGG: Pathways  1.66e-01           3         6          5  \n",
       "28   Dictybase: Phenotypes  1.90e-01          13        58         58  \n",
       "29   Dictybase: Phenotypes  2.08e-01           5        14         14  \n",
       "30          KEGG: Pathways  2.49e-01           4        10         10  \n",
       "31   Dictybase: Phenotypes  2.49e-01           3        14          6  \n",
       "32          KEGG: Pathways  2.49e-01           6        23         20  \n",
       "33          KEGG: Pathways  2.49e-01           3         6          6  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 2289 with EID: 2274\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 373 out of 631 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.19e-08</td>\n",
       "      <td>35</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.19e-08</td>\n",
       "      <td>54</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.28e-05</td>\n",
       "      <td>110</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.28e-05</td>\n",
       "      <td>18</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.80e-05</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>9.08e-04</td>\n",
       "      <td>46</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.58e-02</td>\n",
       "      <td>15</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.81e-02</td>\n",
       "      <td>27</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.23e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.38e-02</td>\n",
       "      <td>26</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.16e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.68e-02</td>\n",
       "      <td>73</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.65e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.69e-02</td>\n",
       "      <td>76</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.25e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>6.25e-02</td>\n",
       "      <td>24</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.25e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.25e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.25e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.69e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant spore morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.69e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>increased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.69e-02</td>\n",
       "      <td>15</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>abolished autophagic vacuole formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>abolished autophagic cell death</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>aberrant phototaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>Pentose and glucuronate interconversions</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>delayed culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>70</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>Phenylalanine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>autophagy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.50e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>17</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>aberrant prestalk cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.83e-01</td>\n",
       "      <td>24</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>aberrant protein localization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.24e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>aberrant autophagic cell death</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>aberrant gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>aberrant thermotaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>precocious sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>aberrant stalk formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>Arginine and proline metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>Nitrogen metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>46</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>12</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0   anatomical structure formation involved in morphogenesis   \n",
       "1                                       cell differentiation   \n",
       "2                           anatomical structure development   \n",
       "3                              Starch and sucrose metabolism   \n",
       "4                             aberrant spore coat morphology   \n",
       "5                                       extracellular region   \n",
       "6                  DNA-binding transcription factor activity   \n",
       "7                                       aberrant culmination   \n",
       "8                    increased prespore cell differentiation   \n",
       "9                                  decreased gene expression   \n",
       "10                      cell wall organization or biogenesis   \n",
       "11                                           kinase activity   \n",
       "12                 SNARE interactions in vesicular transport   \n",
       "13                                       signal transduction   \n",
       "14                               secondary metabolic process   \n",
       "15                                           Golgi apparatus   \n",
       "16                          abolished cellulose biosynthesis   \n",
       "17                                   aberrant slug migration   \n",
       "18                                      Galactose metabolism   \n",
       "19                                 decreased spore viability   \n",
       "20                                 aberrant spore morphology   \n",
       "21                                 increased gene expression   \n",
       "22                 development arrests at tipped mound stage   \n",
       "23                    abolished autophagic vacuole formation   \n",
       "24                           abolished autophagic cell death   \n",
       "25                                       aberrant phototaxis   \n",
       "26                  Pentose and glucuronate interconversions   \n",
       "27                                       delayed culmination   \n",
       "28                                                 cell wall   \n",
       "29                          external encapsulating structure   \n",
       "30                                                 wild type   \n",
       "31       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "32                                  Phenylalanine metabolism   \n",
       "33                                                 autophagy   \n",
       "34                                     abolished culmination   \n",
       "35                    aberrant prestalk cell differentiation   \n",
       "36                                   delayed gene expression   \n",
       "37                         aberrant fruiting body morphology   \n",
       "38                             aberrant protein localization   \n",
       "39                                       cell-cell signaling   \n",
       "40                                      extracellular matrix   \n",
       "41                            aberrant autophagic cell death   \n",
       "42                                  aberrant gene expression   \n",
       "43                                      aberrant thermotaxis   \n",
       "44                                    precocious sporulation   \n",
       "45                                  aberrant stalk formation   \n",
       "46                                           spores inviable   \n",
       "47                           Arginine and proline metabolism   \n",
       "48                                       Nitrogen metabolism   \n",
       "49                                               DNA binding   \n",
       "50                                  abolished slug migration   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  4.19e-08          35        74         74  \n",
       "1   GO: biological_process  4.19e-08          54       145        145  \n",
       "2   GO: biological_process  1.28e-05         110       440        438  \n",
       "3           KEGG: Pathways  1.28e-05          18        31         31  \n",
       "4    Dictybase: Phenotypes  3.80e-05          10        12         12  \n",
       "5   GO: cellular_component  9.08e-04          46       157        157  \n",
       "6   GO: molecular_function  1.58e-02          15        37         37  \n",
       "7    Dictybase: Phenotypes  1.81e-02          27        88         88  \n",
       "8    Dictybase: Phenotypes  2.23e-02           5         6          6  \n",
       "9    Dictybase: Phenotypes  2.38e-02          26        95         86  \n",
       "10  GO: biological_process  3.16e-02           8        15         15  \n",
       "11  GO: molecular_function  3.68e-02          73       328        327  \n",
       "12          KEGG: Pathways  4.65e-02          10        23         23  \n",
       "13  GO: biological_process  5.69e-02          76       352        350  \n",
       "14  GO: biological_process  6.25e-02           4         5          5  \n",
       "15  GO: cellular_component  6.25e-02          24        85         85  \n",
       "16   Dictybase: Phenotypes  6.25e-02           4         5          5  \n",
       "17   Dictybase: Phenotypes  6.25e-02          18        58         58  \n",
       "18          KEGG: Pathways  6.25e-02           7        15         14  \n",
       "19   Dictybase: Phenotypes  6.69e-02          18        60         59  \n",
       "20   Dictybase: Phenotypes  6.69e-02           5         8          8  \n",
       "21   Dictybase: Phenotypes  6.69e-02          15        46         46  \n",
       "22   Dictybase: Phenotypes  1.02e-01           6        12         12  \n",
       "23   Dictybase: Phenotypes  1.02e-01           4         6          6  \n",
       "24   Dictybase: Phenotypes  1.02e-01           4         6          6  \n",
       "25   Dictybase: Phenotypes  1.02e-01           4        14          6  \n",
       "26          KEGG: Pathways  1.02e-01           4         6          6  \n",
       "27   Dictybase: Phenotypes  1.16e-01          10        28         28  \n",
       "28  GO: cellular_component  1.26e-01           6        13         13  \n",
       "29  GO: cellular_component  1.26e-01           6        13         13  \n",
       "30   Dictybase: Phenotypes  1.26e-01          70       341        338  \n",
       "31          KEGG: Pathways  1.45e-01           5        10         10  \n",
       "32          KEGG: Pathways  1.45e-01           5        12         10  \n",
       "33  GO: biological_process  1.50e-01          11        34         34  \n",
       "34   Dictybase: Phenotypes  1.56e-01          17        62         62  \n",
       "35   Dictybase: Phenotypes  1.56e-01           4         7          7  \n",
       "36   Dictybase: Phenotypes  1.66e-01           7        19         18  \n",
       "37   Dictybase: Phenotypes  1.83e-01          24       101         98  \n",
       "38   Dictybase: Phenotypes  2.24e-01          10        32         32  \n",
       "39  GO: biological_process  2.37e-01           6        16         16  \n",
       "40  GO: cellular_component  2.37e-01           4         8          8  \n",
       "41   Dictybase: Phenotypes  2.37e-01           3         5          5  \n",
       "42   Dictybase: Phenotypes  2.37e-01           8        24         24  \n",
       "43   Dictybase: Phenotypes  2.37e-01           3         5          5  \n",
       "44   Dictybase: Phenotypes  2.37e-01           3         5          5  \n",
       "45   Dictybase: Phenotypes  2.37e-01           3         5          5  \n",
       "46   Dictybase: Phenotypes  2.37e-01           3         5          5  \n",
       "47          KEGG: Pathways  2.37e-01           7        23         20  \n",
       "48          KEGG: Pathways  2.37e-01           4         8          8  \n",
       "49  GO: molecular_function  2.37e-01          46       221        221  \n",
       "50   Dictybase: Phenotypes  2.42e-01          12        43         43  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 1096 with EID: 1085\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 129 out of 304 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>25</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.35e-02</td>\n",
       "      <td>54</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.48e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.07e-02</td>\n",
       "      <td>14</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.07e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.07e-02</td>\n",
       "      <td>30</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.23e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.27e-01</td>\n",
       "      <td>39</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant prestalk cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>autophagy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       cell differentiation   \n",
       "1                           anatomical structure development   \n",
       "2                  DNA-binding transcription factor activity   \n",
       "3   anatomical structure formation involved in morphogenesis   \n",
       "4                  SNARE interactions in vesicular transport   \n",
       "5                                                DNA binding   \n",
       "6                                   aberrant gene expression   \n",
       "7                                            kinase activity   \n",
       "8                                secondary metabolic process   \n",
       "9                                  decreased gene expression   \n",
       "10                                  decreased slug migration   \n",
       "11                                      aberrant culmination   \n",
       "12                   increased prespore cell differentiation   \n",
       "13                 development arrests at tipped mound stage   \n",
       "14                    aberrant prestalk cell differentiation   \n",
       "15                                   aberrant slug migration   \n",
       "16                                                 autophagy   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  2.73e-02          25       145        145  \n",
       "1   GO: biological_process  3.35e-02          54       440        438  \n",
       "2   GO: molecular_function  3.48e-02          10        37         37  \n",
       "3   GO: biological_process  7.07e-02          14        74         74  \n",
       "4           KEGG: Pathways  7.07e-02           7        23         23  \n",
       "5   GO: molecular_function  7.07e-02          30       221        221  \n",
       "6    Dictybase: Phenotypes  7.23e-02           7        24         24  \n",
       "7   GO: molecular_function  1.27e-01          39       328        327  \n",
       "8   GO: biological_process  1.35e-01           3         5          5  \n",
       "9    Dictybase: Phenotypes  1.68e-01          14        95         86  \n",
       "10   Dictybase: Phenotypes  1.77e-01           4        11         11  \n",
       "11   Dictybase: Phenotypes  1.77e-01          14        88         88  \n",
       "12   Dictybase: Phenotypes  1.77e-01           3         6          6  \n",
       "13   Dictybase: Phenotypes  2.25e-01           4        12         12  \n",
       "14   Dictybase: Phenotypes  2.37e-01           3         7          7  \n",
       "15   Dictybase: Phenotypes  2.37e-01          10        58         58  \n",
       "16  GO: biological_process  2.37e-01           7        34         34  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_de+'WT_batchrep_combined.tsv',index_col=0,sep='\\t')\n",
    "for stage in [phenotype for phenotype in PHENOTYPES if phenotype in list(milestones['Stage'].unique())]:\n",
    "    query_names=list(milestones.query('log2FoldChange>=1 & padj <= 0.05 & Stage==\"'+stage+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ImpulseDE2\n",
    "Milestone genes are selected as those that are at up-level in a stage and at down level at non neighbouring stages (to account for imperfect annotation of up/down). no_agg is not considered a neighbour of stream and vice versa."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### ImpulseDE2 all strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 1199 with EID: 1193\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 328 out of 479 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.00e-38</td>\n",
       "      <td>67</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.31e-30</td>\n",
       "      <td>59</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.68e-30</td>\n",
       "      <td>56</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.62e-23</td>\n",
       "      <td>59</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.67e-20</td>\n",
       "      <td>69</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.13e-08</td>\n",
       "      <td>60</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.22e-05</td>\n",
       "      <td>49</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.62e-05</td>\n",
       "      <td>20</td>\n",
       "      <td>71</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.32e-03</td>\n",
       "      <td>19</td>\n",
       "      <td>63</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.75e-03</td>\n",
       "      <td>55</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.75e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.10e-03</td>\n",
       "      <td>64</td>\n",
       "      <td>354</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cellular amino acid metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.37e-02</td>\n",
       "      <td>28</td>\n",
       "      <td>125</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.75e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>lysosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.88e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Glycine</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.08e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.76e-02</td>\n",
       "      <td>30</td>\n",
       "      <td>150</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.08e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.55e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cofactor metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.01e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Pentose phosphate pathway</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.06e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>21</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.06e-02</td>\n",
       "      <td>20</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.06e-02</td>\n",
       "      <td>13</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Vitamin B6 metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.50e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.50e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Sulfur relay system</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>beta-Alanine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>vacuole</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>24</td>\n",
       "      <td>127</td>\n",
       "      <td>127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>20</td>\n",
       "      <td>104</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>12</td>\n",
       "      <td>53</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.83e-01</td>\n",
       "      <td>28</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Lysine biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                Ontology  \\\n",
       "0                                         Ribosome          KEGG: Pathways   \n",
       "1                                         ribosome  GO: cellular_component   \n",
       "2               structural constituent of ribosome  GO: molecular_function   \n",
       "3                     structural molecule activity  GO: molecular_function   \n",
       "4                                      translation  GO: biological_process   \n",
       "5                                    mitochondrion  GO: cellular_component   \n",
       "6                          oxidoreductase activity  GO: molecular_function   \n",
       "7                        Oxidative phosphorylation          KEGG: Pathways   \n",
       "8                      Biosynthesis of amino acids          KEGG: Pathways   \n",
       "9            Biosynthesis of secondary metabolites          KEGG: Pathways   \n",
       "10                        unfolded protein binding  GO: molecular_function   \n",
       "11                small molecule metabolic process  GO: biological_process   \n",
       "12           cellular amino acid metabolic process  GO: biological_process   \n",
       "13                                 protein folding  GO: biological_process   \n",
       "14                                        lysosome  GO: cellular_component   \n",
       "15                                         Glycine          KEGG: Pathways   \n",
       "16              transmembrane transporter activity  GO: molecular_function   \n",
       "17                            Steroid biosynthesis          KEGG: Pathways   \n",
       "18  generation of precursor metabolites and energy  GO: biological_process   \n",
       "19                      cofactor metabolic process  GO: biological_process   \n",
       "20                       Pentose phosphate pathway          KEGG: Pathways   \n",
       "21                              peptidase activity  GO: molecular_function   \n",
       "22                              isomerase activity  GO: molecular_function   \n",
       "23                           Vitamin B6 metabolism          KEGG: Pathways   \n",
       "24                           Fatty acid elongation          KEGG: Pathways   \n",
       "25                             Sulfur relay system          KEGG: Pathways   \n",
       "26                         beta-Alanine metabolism          KEGG: Pathways   \n",
       "27                                         vacuole  GO: cellular_component   \n",
       "28                               Carbon metabolism          KEGG: Pathways   \n",
       "29                                       Phagosome          KEGG: Pathways   \n",
       "30                            extracellular region  GO: cellular_component   \n",
       "31                             Lysine biosynthesis          KEGG: Pathways   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   7.00e-38          67       105         97  \n",
       "1   3.31e-30          59        93         93  \n",
       "2   4.68e-30          56        85         85  \n",
       "3   4.62e-23          59       117        117  \n",
       "4   1.67e-20          69       169        169  \n",
       "5   7.13e-08          60       229        229  \n",
       "6   6.22e-05          49       210        207  \n",
       "7   6.62e-05          20        71         54  \n",
       "8   2.32e-03          19        63         62  \n",
       "9   3.75e-03          55       289        281  \n",
       "10  3.75e-03          11        27         27  \n",
       "11  9.10e-03          64       354        353  \n",
       "12  1.37e-02          28       125        124  \n",
       "13  1.75e-02          12        37         37  \n",
       "14  1.88e-02          18        69         69  \n",
       "15  2.08e-02           9        27         24  \n",
       "16  4.76e-02          30       150        149  \n",
       "17  5.08e-02           7        18         18  \n",
       "18  5.55e-02          18        77         77  \n",
       "19  6.01e-02          17        72         72  \n",
       "20  6.06e-02           7        21         19  \n",
       "21  6.06e-02          20        91         91  \n",
       "22  6.06e-02          13        50         50  \n",
       "23  1.50e-01           3         5          5  \n",
       "24  1.50e-01           5        13         13  \n",
       "25  1.54e-01           4         9          9  \n",
       "26  1.54e-01           7        23         23  \n",
       "27  1.56e-01          24       127        127  \n",
       "28  1.66e-01          20       104        102  \n",
       "29  1.66e-01          12        53         52  \n",
       "30  1.83e-01          28       157        157  \n",
       "31  2.13e-01           3         6          6  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 956 with EID: 941\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 276 out of 377 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.13e-08</td>\n",
       "      <td>27</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.13e-08</td>\n",
       "      <td>32</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.06e-06</td>\n",
       "      <td>43</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.79e-06</td>\n",
       "      <td>31</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.33e-06</td>\n",
       "      <td>64</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.33e-06</td>\n",
       "      <td>28</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.33e-06</td>\n",
       "      <td>16</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.03e-05</td>\n",
       "      <td>42</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.68e-05</td>\n",
       "      <td>50</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.10e-04</td>\n",
       "      <td>14</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.68e-04</td>\n",
       "      <td>34</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased activation of adenylate cyclase</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-03</td>\n",
       "      <td>20</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.55e-03</td>\n",
       "      <td>22</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.55e-03</td>\n",
       "      <td>13</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.25e-03</td>\n",
       "      <td>36</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.25e-03</td>\n",
       "      <td>34</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Protein export</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.25e-03</td>\n",
       "      <td>8</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>abolished chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.80e-03</td>\n",
       "      <td>20</td>\n",
       "      <td>91</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.32e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.81e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.46e-03</td>\n",
       "      <td>15</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.28e-03</td>\n",
       "      <td>23</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.28e-03</td>\n",
       "      <td>10</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.87e-03</td>\n",
       "      <td>62</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.87e-03</td>\n",
       "      <td>15</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.87e-03</td>\n",
       "      <td>17</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>8.56e-03</td>\n",
       "      <td>21</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.61e-03</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.04e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>aberrant chemotaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.04e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>decreased protein kinase activation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.22e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.40e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.40e-02</td>\n",
       "      <td>49</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.52e-02</td>\n",
       "      <td>14</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.54e-02</td>\n",
       "      <td>15</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.54e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>abolished protein phosphorylation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.81e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>aberrant protein localization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.86e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.18e-02</td>\n",
       "      <td>32</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.55e-02</td>\n",
       "      <td>23</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>decreased growth rate in suspension</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.99e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>increased intracellular cAMP level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.99e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>aberrant phagocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.99e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>GTPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.43e-02</td>\n",
       "      <td>24</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>decreased cAMP-induced cAMP pulse</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.56e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>decreased macropinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.42e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>aberrant chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.42e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>aberrant pseudopod formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.42e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>decreased lateral pseudopod suppression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.42e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.42e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.45e-02</td>\n",
       "      <td>34</td>\n",
       "      <td>233</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.51e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.85e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>13</td>\n",
       "      <td>67</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.61e-02</td>\n",
       "      <td>22</td>\n",
       "      <td>146</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.78e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>decreased cell polarity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.78e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>aberrant slug formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.78e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.97e-02</td>\n",
       "      <td>47</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>protein targeting</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.89e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>abolished cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.89e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>enzyme binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.89e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.91e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>decreased myosin II assembly</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.91e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.57e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>68</td>\n",
       "      <td>decreased pinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.57e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>69</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.57e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>31</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>70</td>\n",
       "      <td>decreased exocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.89e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>71</td>\n",
       "      <td>decreased chemotaxis to folate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.89e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>abolished activation of GTPase activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>42</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>76</td>\n",
       "      <td>Glycosylphosphatidylinositol (GPI)-anchor biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.24e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>77</td>\n",
       "      <td>homeostatic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>12</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>79</td>\n",
       "      <td>aberrant stalk morphogenesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80</td>\n",
       "      <td>decreased phagocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.39e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>81</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>82</td>\n",
       "      <td>abolished positive regulation of chemorepellant activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>aberrant cell separation during cytokinesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>84</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>21</td>\n",
       "      <td>150</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>85</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>86</td>\n",
       "      <td>decreased stalk width</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.72e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>87</td>\n",
       "      <td>decreased cAMP-induced cGMP pulse</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.72e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>88</td>\n",
       "      <td>abolished cell polarity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.72e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>89</td>\n",
       "      <td>multiple tips</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>90</td>\n",
       "      <td>increased lateral pseudopod formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>91</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>92</td>\n",
       "      <td>aberrant cytoskeleton organisation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>93</td>\n",
       "      <td>decreased basal F-actin level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>94</td>\n",
       "      <td>aberrant cellular homeostasis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>95</td>\n",
       "      <td>Nicotinate and nicotinamide metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>96</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>22</td>\n",
       "      <td>165</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>97</td>\n",
       "      <td>cell population proliferation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>98</td>\n",
       "      <td>cytoplasmic vesicle</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>27</td>\n",
       "      <td>211</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>99</td>\n",
       "      <td>aberrant macropinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>100</td>\n",
       "      <td>aberrant myosin II organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>101</td>\n",
       "      <td>abolished protein localization to nucleus</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>102</td>\n",
       "      <td>decreased infection rate by a gram-negative pathogenic bacterium</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>103</td>\n",
       "      <td>aberrant gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>104</td>\n",
       "      <td>aberrant filopodium formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>105</td>\n",
       "      <td>aberrant cortical actin cytoskeleton organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>106</td>\n",
       "      <td>abolished protein kinase activation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>107</td>\n",
       "      <td>aberrant signal transduction</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>108</td>\n",
       "      <td>abolished protein localization to plasma membrane</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>109</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>21</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>110</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>101</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>111</td>\n",
       "      <td>decreased protein level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.49e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             Gene set  \\\n",
       "0                                                  aberrant streaming   \n",
       "1                                                aberrant cytokinesis   \n",
       "2                                                          locomotion   \n",
       "3                                                       cell division   \n",
       "4                                                 signal transduction   \n",
       "5                                                  mitotic cell cycle   \n",
       "6                                                          Proteasome   \n",
       "7                                           cytoskeleton organization   \n",
       "8                                                     plasma membrane   \n",
       "9                                       aberrant F-actin organization   \n",
       "10                                                         cell cycle   \n",
       "11                          decreased activation of adenylate cyclase   \n",
       "12                                              abolished aggregation   \n",
       "13                                       aberrant cAMP wave formation   \n",
       "14                                           aberrant cell morphology   \n",
       "15                                        cellular component assembly   \n",
       "16                                                       cytoskeleton   \n",
       "17                                                     Protein export   \n",
       "18                                       abolished chemotaxis to cAMP   \n",
       "19                        Protein processing in endoplasmic reticulum   \n",
       "20                          abolished activation of adenylate cyclase   \n",
       "21                                 development arrests at mound stage   \n",
       "22                                               aberrant aggregation   \n",
       "23                                              endoplasmic reticulum   \n",
       "24                                           decreased aggregate size   \n",
       "25                                   anatomical structure development   \n",
       "26                                       decreased chemotaxis to cAMP   \n",
       "27                                            decreased cell motility   \n",
       "28                                          enzyme regulator activity   \n",
       "29                                                 cell morphogenesis   \n",
       "30                                                       reproduction   \n",
       "31                                                delayed aggregation   \n",
       "32                                                aberrant chemotaxis   \n",
       "33                                decreased protein kinase activation   \n",
       "34                                          decreased chemotaxis rate   \n",
       "35                                                          wild type   \n",
       "36                                              abolished culmination   \n",
       "37                                                delayed development   \n",
       "38                                           abolished slug migration   \n",
       "39                                  abolished protein phosphorylation   \n",
       "40                                      aberrant protein localization   \n",
       "41                                                            cytosol   \n",
       "42                                protein-containing complex assembly   \n",
       "43                                decreased growth rate in suspension   \n",
       "44                                 increased intracellular cAMP level   \n",
       "45                                              aberrant phagocytosis   \n",
       "46                                                    GTPase activity   \n",
       "47                                  decreased cAMP-induced cAMP pulse   \n",
       "48                                         decreased macropinocytosis   \n",
       "49                                        aberrant chemotaxis to cAMP   \n",
       "50                                       aberrant pseudopod formation   \n",
       "51                            decreased lateral pseudopod suppression   \n",
       "52                                              increased growth rate   \n",
       "53                                         vesicle-mediated transport   \n",
       "54                                            aberrant slug migration   \n",
       "55                                               aberrant culmination   \n",
       "56                                                      cell motility   \n",
       "57                                              decreased growth rate   \n",
       "58                                                abolished streaming   \n",
       "59                                            decreased cell polarity   \n",
       "60                                            aberrant slug formation   \n",
       "61                                                 response to stress   \n",
       "62                                                  protein targeting   \n",
       "63                                      abolished cAMP wave formation   \n",
       "64                                                     enzyme binding   \n",
       "65                                      increased basal F-actin level   \n",
       "66                                       decreased myosin II assembly   \n",
       "67                                               aberrant development   \n",
       "68                                              decreased pinocytosis   \n",
       "69                                              N-Glycan biosynthesis   \n",
       "70                                               decreased exocytosis   \n",
       "71                                     decreased chemotaxis to folate   \n",
       "72                            abolished activation of GTPase activity   \n",
       "73                                         precocious gene expression   \n",
       "74                                       decreased fruiting body size   \n",
       "75                                                    kinase activity   \n",
       "76             Glycosylphosphatidylinositol (GPI)-anchor biosynthesis   \n",
       "77                                                homeostatic process   \n",
       "78                                          decreased gene expression   \n",
       "79                                       aberrant stalk morphogenesis   \n",
       "80                                             decreased phagocytosis   \n",
       "81                                             precocious aggregation   \n",
       "82           abolished positive regulation of chemorepellant activity   \n",
       "83                        aberrant cell separation during cytokinesis   \n",
       "84                                 transmembrane transporter activity   \n",
       "85                                      microtubule organizing center   \n",
       "86                                              decreased stalk width   \n",
       "87                                  decreased cAMP-induced cGMP pulse   \n",
       "88                                            abolished cell polarity   \n",
       "89                                                      multiple tips   \n",
       "90                              increased lateral pseudopod formation   \n",
       "91                                             precocious development   \n",
       "92                                 aberrant cytoskeleton organisation   \n",
       "93                                      decreased basal F-actin level   \n",
       "94                                      aberrant cellular homeostasis   \n",
       "95                             Nicotinate and nicotinamide metabolism   \n",
       "96                                            transmembrane transport   \n",
       "97                                      cell population proliferation   \n",
       "98                                                cytoplasmic vesicle   \n",
       "99                                          aberrant macropinocytosis   \n",
       "100                                   aberrant myosin II organization   \n",
       "101                         abolished protein localization to nucleus   \n",
       "102  decreased infection rate by a gram-negative pathogenic bacterium   \n",
       "103                                          aberrant gene expression   \n",
       "104                                     aberrant filopodium formation   \n",
       "105                 aberrant cortical actin cytoskeleton organization   \n",
       "106                               abolished protein kinase activation   \n",
       "107                                      aberrant signal transduction   \n",
       "108                 abolished protein localization to plasma membrane   \n",
       "109                                      cytoskeletal protein binding   \n",
       "110                                 aberrant fruiting body morphology   \n",
       "111                                           decreased protein level   \n",
       "\n",
       "                   Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0     Dictybase: Phenotypes  2.13e-08          27        72         72  \n",
       "1     Dictybase: Phenotypes  2.13e-08          32        98         98  \n",
       "2    GO: biological_process  3.06e-06          43       191        189  \n",
       "3    GO: biological_process  3.79e-06          31       115        115  \n",
       "4    GO: biological_process  4.33e-06          64       352        350  \n",
       "5    GO: biological_process  4.33e-06          28       100        100  \n",
       "6            KEGG: Pathways  4.33e-06          16        39         37  \n",
       "7    GO: biological_process  1.03e-05          42       197        196  \n",
       "8    GO: cellular_component  5.68e-05          50       268        268  \n",
       "9     Dictybase: Phenotypes  1.10e-04          14        37         37  \n",
       "10   GO: biological_process  3.68e-04          34       167        167  \n",
       "11    Dictybase: Phenotypes  1.09e-03           5         6          6  \n",
       "12    Dictybase: Phenotypes  1.09e-03          20        82         80  \n",
       "13    Dictybase: Phenotypes  1.55e-03          22        96         96  \n",
       "14    Dictybase: Phenotypes  1.55e-03          13        42         41  \n",
       "15   GO: biological_process  2.25e-03          36       202        201  \n",
       "16   GO: cellular_component  2.25e-03          34       185        185  \n",
       "17           KEGG: Pathways  2.25e-03           8        19         18  \n",
       "18    Dictybase: Phenotypes  2.27e-03           7        14         14  \n",
       "19           KEGG: Pathways  2.80e-03          20        91         88  \n",
       "20    Dictybase: Phenotypes  5.32e-03           5         8          8  \n",
       "21    Dictybase: Phenotypes  5.81e-03          14        56         54  \n",
       "22    Dictybase: Phenotypes  6.46e-03          15        61         61  \n",
       "23   GO: cellular_component  7.28e-03          23       117        117  \n",
       "24    Dictybase: Phenotypes  7.28e-03          10        32         32  \n",
       "25   GO: biological_process  7.87e-03          62       440        438  \n",
       "26    Dictybase: Phenotypes  7.87e-03          15        63         63  \n",
       "27    Dictybase: Phenotypes  7.87e-03          17        76         76  \n",
       "28   GO: molecular_function  8.56e-03          21       105        105  \n",
       "29   GO: biological_process  9.61e-03           8        23         23  \n",
       "30   GO: biological_process  1.04e-02          17        79         79  \n",
       "31    Dictybase: Phenotypes  1.04e-02          16        72         72  \n",
       "32    Dictybase: Phenotypes  1.04e-02           4         6          6  \n",
       "33    Dictybase: Phenotypes  1.22e-02           7        19         19  \n",
       "34    Dictybase: Phenotypes  1.40e-02          10        36         36  \n",
       "35    Dictybase: Phenotypes  1.40e-02          49       341        338  \n",
       "36    Dictybase: Phenotypes  1.52e-02          14        62         62  \n",
       "37    Dictybase: Phenotypes  1.54e-02          15        69         69  \n",
       "38    Dictybase: Phenotypes  1.54e-02          11        43         43  \n",
       "39    Dictybase: Phenotypes  1.81e-02           5        11         11  \n",
       "40    Dictybase: Phenotypes  1.86e-02           9        32         32  \n",
       "41   GO: cellular_component  2.18e-02          32       203        203  \n",
       "42   GO: biological_process  2.55e-02          23       134        133  \n",
       "43    Dictybase: Phenotypes  2.99e-02           4         8          8  \n",
       "44    Dictybase: Phenotypes  2.99e-02           4         8          8  \n",
       "45    Dictybase: Phenotypes  2.99e-02           4         8          8  \n",
       "46   GO: molecular_function  3.43e-02          24       145        145  \n",
       "47    Dictybase: Phenotypes  3.56e-02           5        14         13  \n",
       "48    Dictybase: Phenotypes  4.42e-02           6        19         19  \n",
       "49    Dictybase: Phenotypes  4.42e-02           8        31         31  \n",
       "50    Dictybase: Phenotypes  4.42e-02           4         9          9  \n",
       "51    Dictybase: Phenotypes  4.42e-02           4         9          9  \n",
       "52    Dictybase: Phenotypes  4.42e-02           8        32         31  \n",
       "53   GO: biological_process  4.45e-02          34       233        233  \n",
       "54    Dictybase: Phenotypes  4.51e-02          12        58         58  \n",
       "55    Dictybase: Phenotypes  4.85e-02          16        88         88  \n",
       "56   GO: biological_process  4.89e-02          13        67         66  \n",
       "57    Dictybase: Phenotypes  5.61e-02          22       146        137  \n",
       "58    Dictybase: Phenotypes  5.78e-02           8        33         33  \n",
       "59    Dictybase: Phenotypes  5.78e-02           8        34         33  \n",
       "60    Dictybase: Phenotypes  5.78e-02           4        10         10  \n",
       "61   GO: biological_process  6.97e-02          47       360        358  \n",
       "62   GO: biological_process  7.89e-02           8        35         35  \n",
       "63    Dictybase: Phenotypes  7.89e-02           4        11         11  \n",
       "64   GO: molecular_function  7.89e-02          18       110        110  \n",
       "65    Dictybase: Phenotypes  8.91e-02           5        18         17  \n",
       "66    Dictybase: Phenotypes  8.91e-02           5        17         17  \n",
       "67    Dictybase: Phenotypes  9.57e-02          10        51         51  \n",
       "68    Dictybase: Phenotypes  9.57e-02           7        30         30  \n",
       "69           KEGG: Pathways  9.57e-02           7        31         30  \n",
       "70    Dictybase: Phenotypes  9.89e-02           4        12         12  \n",
       "71    Dictybase: Phenotypes  9.89e-02           4        12         12  \n",
       "72    Dictybase: Phenotypes  1.01e-01           3         7          7  \n",
       "73    Dictybase: Phenotypes  1.03e-01           5        18         18  \n",
       "74    Dictybase: Phenotypes  1.04e-01          13        76         75  \n",
       "75   GO: molecular_function  1.13e-01          42       328        327  \n",
       "76           KEGG: Pathways  1.24e-01           5        20         19  \n",
       "77   GO: biological_process  1.33e-01          12        70         70  \n",
       "78    Dictybase: Phenotypes  1.33e-01          14        95         86  \n",
       "79    Dictybase: Phenotypes  1.37e-01           3         8          8  \n",
       "80    Dictybase: Phenotypes  1.39e-01          11        63         63  \n",
       "81    Dictybase: Phenotypes  1.44e-01           5        20         20  \n",
       "82    Dictybase: Phenotypes  1.47e-01           4        14         14  \n",
       "83    Dictybase: Phenotypes  1.47e-01           4        14         14  \n",
       "84   GO: molecular_function  1.69e-01          21       150        149  \n",
       "85   GO: cellular_component  1.70e-01           9        50         50  \n",
       "86    Dictybase: Phenotypes  1.72e-01           3         9          9  \n",
       "87    Dictybase: Phenotypes  1.72e-01           3        10          9  \n",
       "88    Dictybase: Phenotypes  1.72e-01           3         9          9  \n",
       "89    Dictybase: Phenotypes  1.74e-01           4        15         15  \n",
       "90    Dictybase: Phenotypes  1.86e-01           5        22         22  \n",
       "91    Dictybase: Phenotypes  1.86e-01           6        30         29  \n",
       "92    Dictybase: Phenotypes  2.12e-01           3        10         10  \n",
       "93    Dictybase: Phenotypes  2.12e-01           3        10         10  \n",
       "94    Dictybase: Phenotypes  2.12e-01           3        10         10  \n",
       "95           KEGG: Pathways  2.12e-01           3        10         10  \n",
       "96   GO: biological_process  2.12e-01          22       165        164  \n",
       "97   GO: biological_process  2.21e-01           2         5          5  \n",
       "98   GO: cellular_component  2.21e-01          27       211        211  \n",
       "99    Dictybase: Phenotypes  2.21e-01           2         5          5  \n",
       "100   Dictybase: Phenotypes  2.21e-01           2         5          5  \n",
       "101   Dictybase: Phenotypes  2.21e-01           2         5          5  \n",
       "102   Dictybase: Phenotypes  2.21e-01           2         5          5  \n",
       "103   Dictybase: Phenotypes  2.21e-01           5        24         24  \n",
       "104   Dictybase: Phenotypes  2.21e-01           2         5          5  \n",
       "105   Dictybase: Phenotypes  2.21e-01           2         5          5  \n",
       "106   Dictybase: Phenotypes  2.21e-01           2         5          5  \n",
       "107   Dictybase: Phenotypes  2.21e-01           2         5          5  \n",
       "108   Dictybase: Phenotypes  2.21e-01           2         5          5  \n",
       "109  GO: molecular_function  2.21e-01          21       158        158  \n",
       "110   Dictybase: Phenotypes  2.39e-01          14       101         98  \n",
       "111   Dictybase: Phenotypes  2.49e-01           4        18         18  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 1190 with EID: 1175\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.39\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 335 out of 461 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.91e-14</td>\n",
       "      <td>95</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.65e-11</td>\n",
       "      <td>58</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.58e-09</td>\n",
       "      <td>36</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.03e-07</td>\n",
       "      <td>27</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.15e-06</td>\n",
       "      <td>50</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.79e-06</td>\n",
       "      <td>27</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.79e-06</td>\n",
       "      <td>70</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.16e-06</td>\n",
       "      <td>17</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>enzyme binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.82e-05</td>\n",
       "      <td>32</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.96e-05</td>\n",
       "      <td>17</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.51e-05</td>\n",
       "      <td>24</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.21e-04</td>\n",
       "      <td>26</td>\n",
       "      <td>91</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.23e-04</td>\n",
       "      <td>15</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.42e-04</td>\n",
       "      <td>56</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>decreased lateral pseudopod suppression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.94e-04</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.97e-04</td>\n",
       "      <td>66</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.38e-04</td>\n",
       "      <td>30</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.38e-04</td>\n",
       "      <td>20</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.38e-04</td>\n",
       "      <td>13</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>decreased cAMP-induced cAMP pulse</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.34e-04</td>\n",
       "      <td>8</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.90e-04</td>\n",
       "      <td>43</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.91e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.11e-03</td>\n",
       "      <td>20</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.45e-03</td>\n",
       "      <td>12</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased protein kinase activation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.60e-03</td>\n",
       "      <td>9</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased macropinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.60e-03</td>\n",
       "      <td>9</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>aberrant pseudopod formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.96e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.57e-03</td>\n",
       "      <td>38</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>decreased myosin II assembly</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.68e-03</td>\n",
       "      <td>8</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>abolished positive regulation of chemorepellant activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.44e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Protein export</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.57e-03</td>\n",
       "      <td>8</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased cell-substrate adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.30e-03</td>\n",
       "      <td>10</td>\n",
       "      <td>28</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.81e-03</td>\n",
       "      <td>73</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.81e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>abolished protein phosphorylation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.81e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.74e-03</td>\n",
       "      <td>24</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.07e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>67</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>increased filopodium formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>increased F-actin polymerization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.09e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.42e-02</td>\n",
       "      <td>20</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.56e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>increased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.56e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.64e-02</td>\n",
       "      <td>22</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>decreased activation of adenylate cyclase</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.64e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>aberrant chemotaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.64e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>decreased taxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.64e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>increased lateral pseudopod formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.72e-02</td>\n",
       "      <td>27</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>aberrant protein localization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.72e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.72e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.90e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.92e-02</td>\n",
       "      <td>21</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.03e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>abolished chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.03e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.69e-02</td>\n",
       "      <td>30</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>aberrant cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.88e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>decreased positive regulation of GTPase activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.17e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>increased cell-substrate adhesion</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.28e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.51e-02</td>\n",
       "      <td>36</td>\n",
       "      <td>203</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.54e-02</td>\n",
       "      <td>23</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.22e-02</td>\n",
       "      <td>57</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.52e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>aberrant stalk morphogenesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.52e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>decreased growth rate in suspension</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.52e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>increased intracellular cAMP level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.52e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.03e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>68</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.39e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>69</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.51e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>70</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>71</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.16e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>aberrant signal transduction</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.16e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73</td>\n",
       "      <td>decreased activation of GTPase activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.16e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.43e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75</td>\n",
       "      <td>decreased stalk width</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.43e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>76</td>\n",
       "      <td>decreased cAMP-induced cGMP pulse</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.43e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>77</td>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.55e-02</td>\n",
       "      <td>25</td>\n",
       "      <td>146</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78</td>\n",
       "      <td>GTPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.10e-02</td>\n",
       "      <td>26</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>79</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.11e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>31</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.96e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>81</td>\n",
       "      <td>aberrant slug formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.15e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>82</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.52e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>decreased protein phosphorylation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.69e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>84</td>\n",
       "      <td>increased activation of GTPase activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.69e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>85</td>\n",
       "      <td>abolished protein localization to cell cortex</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.69e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>86</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.82e-02</td>\n",
       "      <td>13</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>87</td>\n",
       "      <td>decreased phagocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>88</td>\n",
       "      <td>aberrant activation of GTPase activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>89</td>\n",
       "      <td>abolished cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>90</td>\n",
       "      <td>decreased F-actin polymerization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>91</td>\n",
       "      <td>Other glycan degradation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>92</td>\n",
       "      <td>phosphatase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>93</td>\n",
       "      <td>increased phagocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>94</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.21e-01</td>\n",
       "      <td>12</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>95</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>96</td>\n",
       "      <td>abolished activation of GTPase activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>97</td>\n",
       "      <td>decreased exocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>98</td>\n",
       "      <td>decreased chemotaxis to folate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>99</td>\n",
       "      <td>increased myosin II assembly</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>100</td>\n",
       "      <td>protein targeting</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>101</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>102</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.57e-01</td>\n",
       "      <td>16</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>103</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>27</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>104</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>105</td>\n",
       "      <td>aberrant vegetative growth</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>106</td>\n",
       "      <td>growth</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>107</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>15</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>108</td>\n",
       "      <td>increased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>31</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>109</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>110</td>\n",
       "      <td>aberrant response to hyperosmotic stress</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>111</td>\n",
       "      <td>increased infection rate by a gram-negative pathogenic bacterium</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>112</td>\n",
       "      <td>aberrant phagocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>113</td>\n",
       "      <td>increased protein kinase activation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.71e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>114</td>\n",
       "      <td>cell death</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.90e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>115</td>\n",
       "      <td>increased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>116</td>\n",
       "      <td>abolished positive chemotaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>117</td>\n",
       "      <td>abolished catalytic activity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>118</td>\n",
       "      <td>decreased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>119</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.16e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>120</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>23</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>121</td>\n",
       "      <td>mRNA surveillance pathway</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>122</td>\n",
       "      <td>decreased cell polarity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.32e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>123</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             Gene set  \\\n",
       "0                                                 signal transduction   \n",
       "1                                                          locomotion   \n",
       "2                                                aberrant cytokinesis   \n",
       "3                                                  aberrant streaming   \n",
       "4                                           cytoskeleton organization   \n",
       "5                                               abolished aggregation   \n",
       "6                                                     kinase activity   \n",
       "7                                       aberrant F-actin organization   \n",
       "8                                                      enzyme binding   \n",
       "9                                            aberrant cell morphology   \n",
       "10                                            decreased cell motility   \n",
       "11                        Protein processing in endoplasmic reticulum   \n",
       "12                                          decreased chemotaxis rate   \n",
       "13                                                    plasma membrane   \n",
       "14                            decreased lateral pseudopod suppression   \n",
       "15                                                          wild type   \n",
       "16                                              endoplasmic reticulum   \n",
       "17                                       decreased chemotaxis to cAMP   \n",
       "18                                              increased growth rate   \n",
       "19                                  decreased cAMP-induced cAMP pulse   \n",
       "20                                        cellular component assembly   \n",
       "21                          abolished activation of adenylate cyclase   \n",
       "22                                                delayed development   \n",
       "23                                        aberrant chemotaxis to cAMP   \n",
       "24                                decreased protein kinase activation   \n",
       "25                                         decreased macropinocytosis   \n",
       "26                                       aberrant pseudopod formation   \n",
       "27                                                       cytoskeleton   \n",
       "28                                       decreased myosin II assembly   \n",
       "29           abolished positive regulation of chemorepellant activity   \n",
       "30                                                     Protein export   \n",
       "31                                  decreased cell-substrate adhesion   \n",
       "32                                   anatomical structure development   \n",
       "33                                           decreased aggregate size   \n",
       "34                                  abolished protein phosphorylation   \n",
       "35                                          enzyme regulator activity   \n",
       "36                                                      cell motility   \n",
       "37                                                delayed aggregation   \n",
       "38                                     increased filopodium formation   \n",
       "39                                               aberrant aggregation   \n",
       "40                                   increased F-actin polymerization   \n",
       "41                                          decreased gene expression   \n",
       "42                                       decreased fruiting body size   \n",
       "43                                            increased cell motility   \n",
       "44                                                 mitotic cell cycle   \n",
       "45                          decreased activation of adenylate cyclase   \n",
       "46                                                aberrant chemotaxis   \n",
       "47                                                    decreased taxis   \n",
       "48                              increased lateral pseudopod formation   \n",
       "49                                protein-containing complex assembly   \n",
       "50                                      aberrant protein localization   \n",
       "51                                                         Proteasome   \n",
       "52                                           abolished slug migration   \n",
       "53                                       aberrant cAMP wave formation   \n",
       "54                                                 cell morphogenesis   \n",
       "55                                       abolished chemotaxis to cAMP   \n",
       "56                                       cytoskeletal protein binding   \n",
       "57                                             aberrant cell motility   \n",
       "58                   decreased positive regulation of GTPase activity   \n",
       "59                                  increased cell-substrate adhesion   \n",
       "60                                                            cytosol   \n",
       "61                                                      cell division   \n",
       "62                                                 response to stress   \n",
       "63                                                       reproduction   \n",
       "64                                       aberrant stalk morphogenesis   \n",
       "65                                decreased growth rate in suspension   \n",
       "66                                 increased intracellular cAMP level   \n",
       "67                                      increased basal F-actin level   \n",
       "68                                      microtubule organizing center   \n",
       "69                                                abolished streaming   \n",
       "70                                               aberrant development   \n",
       "71                                         multiple tips in culminant   \n",
       "72                                       aberrant signal transduction   \n",
       "73                            decreased activation of GTPase activity   \n",
       "74                                              decreased culmination   \n",
       "75                                              decreased stalk width   \n",
       "76                                  decreased cAMP-induced cGMP pulse   \n",
       "77                                              decreased growth rate   \n",
       "78                                                    GTPase activity   \n",
       "79                                              N-Glycan biosynthesis   \n",
       "80                                          abolished gene expression   \n",
       "81                                            aberrant slug formation   \n",
       "82                           development arrests at loose mound stage   \n",
       "83                                  decreased protein phosphorylation   \n",
       "84                            increased activation of GTPase activity   \n",
       "85                      abolished protein localization to cell cortex   \n",
       "86                                              abolished culmination   \n",
       "87                                             decreased phagocytosis   \n",
       "88                             aberrant activation of GTPase activity   \n",
       "89                                      abolished cAMP wave formation   \n",
       "90                                   decreased F-actin polymerization   \n",
       "91                                           Other glycan degradation   \n",
       "92                                               phosphatase activity   \n",
       "93                                             increased phagocytosis   \n",
       "94                                            aberrant slug migration   \n",
       "95                             Various types of N-glycan biosynthesis   \n",
       "96                            abolished activation of GTPase activity   \n",
       "97                                               decreased exocytosis   \n",
       "98                                     decreased chemotaxis to folate   \n",
       "99                                       increased myosin II assembly   \n",
       "100                                                 protein targeting   \n",
       "101                                            precocious development   \n",
       "102                                              aberrant culmination   \n",
       "103                                                        cell cycle   \n",
       "104                                        precocious gene expression   \n",
       "105                                        aberrant vegetative growth   \n",
       "106                                                            growth   \n",
       "107                                                     cell adhesion   \n",
       "108                                          increased aggregate size   \n",
       "109                                          increased slug migration   \n",
       "110                          aberrant response to hyperosmotic stress   \n",
       "111  increased infection rate by a gram-negative pathogenic bacterium   \n",
       "112                                             aberrant phagocytosis   \n",
       "113                               increased protein kinase activation   \n",
       "114                                                        cell death   \n",
       "115                                      increased fruiting body size   \n",
       "116                                     abolished positive chemotaxis   \n",
       "117                                      abolished catalytic activity   \n",
       "118                           decreased prespore cell differentiation   \n",
       "119                                            precocious aggregation   \n",
       "120                                              cell differentiation   \n",
       "121                                         mRNA surveillance pathway   \n",
       "122                                           decreased cell polarity   \n",
       "123                                             decreased sporulation   \n",
       "\n",
       "                   Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0    GO: biological_process  1.91e-14          95       352        350  \n",
       "1    GO: biological_process  9.65e-11          58       191        189  \n",
       "2     Dictybase: Phenotypes  7.58e-09          36        98         98  \n",
       "3     Dictybase: Phenotypes  9.03e-07          27        72         72  \n",
       "4    GO: biological_process  2.15e-06          50       197        196  \n",
       "5     Dictybase: Phenotypes  6.79e-06          27        82         80  \n",
       "6    GO: molecular_function  6.79e-06          70       328        327  \n",
       "7     Dictybase: Phenotypes  8.16e-06          17        37         37  \n",
       "8    GO: molecular_function  1.82e-05          32       110        110  \n",
       "9     Dictybase: Phenotypes  3.96e-05          17        42         41  \n",
       "10    Dictybase: Phenotypes  8.51e-05          24        76         76  \n",
       "11           KEGG: Pathways  1.21e-04          26        91         88  \n",
       "12    Dictybase: Phenotypes  1.23e-04          15        36         36  \n",
       "13   GO: cellular_component  1.42e-04          56       268        268  \n",
       "14    Dictybase: Phenotypes  1.94e-04           7         9          9  \n",
       "15    Dictybase: Phenotypes  1.97e-04          66       341        338  \n",
       "16   GO: cellular_component  3.38e-04          30       117        117  \n",
       "17    Dictybase: Phenotypes  3.38e-04          20        63         63  \n",
       "18    Dictybase: Phenotypes  3.38e-04          13        32         31  \n",
       "19    Dictybase: Phenotypes  4.34e-04           8        14         13  \n",
       "20   GO: biological_process  6.90e-04          43       202        201  \n",
       "21    Dictybase: Phenotypes  8.91e-04           6         8          8  \n",
       "22    Dictybase: Phenotypes  1.11e-03          20        69         69  \n",
       "23    Dictybase: Phenotypes  1.45e-03          12        31         31  \n",
       "24    Dictybase: Phenotypes  1.60e-03           9        19         19  \n",
       "25    Dictybase: Phenotypes  1.60e-03           9        19         19  \n",
       "26    Dictybase: Phenotypes  1.96e-03           6         9          9  \n",
       "27   GO: cellular_component  3.57e-03          38       185        185  \n",
       "28    Dictybase: Phenotypes  3.68e-03           8        17         17  \n",
       "29    Dictybase: Phenotypes  5.44e-03           7        14         14  \n",
       "30           KEGG: Pathways  5.57e-03           8        19         18  \n",
       "31    Dictybase: Phenotypes  6.30e-03          10        28         27  \n",
       "32   GO: biological_process  6.81e-03          73       440        438  \n",
       "33    Dictybase: Phenotypes  6.81e-03          11        32         32  \n",
       "34    Dictybase: Phenotypes  6.81e-03           6        11         11  \n",
       "35   GO: molecular_function  7.74e-03          24       105        105  \n",
       "36   GO: biological_process  1.07e-02          17        67         66  \n",
       "37    Dictybase: Phenotypes  1.09e-02          18        72         72  \n",
       "38    Dictybase: Phenotypes  1.09e-02           7        16         16  \n",
       "39    Dictybase: Phenotypes  1.09e-02          16        61         61  \n",
       "40    Dictybase: Phenotypes  1.09e-02           6        12         12  \n",
       "41    Dictybase: Phenotypes  1.42e-02          20        95         86  \n",
       "42    Dictybase: Phenotypes  1.56e-02          18        76         75  \n",
       "43    Dictybase: Phenotypes  1.56e-02           7        17         17  \n",
       "44   GO: biological_process  1.64e-02          22       100        100  \n",
       "45    Dictybase: Phenotypes  1.64e-02           4         6          6  \n",
       "46    Dictybase: Phenotypes  1.64e-02           4         6          6  \n",
       "47    Dictybase: Phenotypes  1.64e-02           4         6          6  \n",
       "48    Dictybase: Phenotypes  1.68e-02           8        22         22  \n",
       "49   GO: biological_process  1.72e-02          27       134        133  \n",
       "50    Dictybase: Phenotypes  1.72e-02          10        32         32  \n",
       "51           KEGG: Pathways  1.72e-02          11        39         37  \n",
       "52    Dictybase: Phenotypes  1.90e-02          12        43         43  \n",
       "53    Dictybase: Phenotypes  1.92e-02          21        96         96  \n",
       "54   GO: biological_process  2.03e-02           8        23         23  \n",
       "55    Dictybase: Phenotypes  2.03e-02           6        14         14  \n",
       "56   GO: molecular_function  2.69e-02          30       158        158  \n",
       "57    Dictybase: Phenotypes  2.88e-02           4         7          7  \n",
       "58    Dictybase: Phenotypes  3.17e-02           5        11         11  \n",
       "59    Dictybase: Phenotypes  3.28e-02           9        30         30  \n",
       "60   GO: cellular_component  3.51e-02          36       203        203  \n",
       "61   GO: biological_process  3.54e-02          23       115        115  \n",
       "62   GO: biological_process  4.22e-02          57       360        358  \n",
       "63   GO: biological_process  4.52e-02          17        79         79  \n",
       "64    Dictybase: Phenotypes  4.52e-02           4         8          8  \n",
       "65    Dictybase: Phenotypes  4.52e-02           4         8          8  \n",
       "66    Dictybase: Phenotypes  4.52e-02           4         8          8  \n",
       "67    Dictybase: Phenotypes  5.03e-02           6        18         17  \n",
       "68   GO: cellular_component  5.39e-02          12        50         50  \n",
       "69    Dictybase: Phenotypes  5.51e-02           9        33         33  \n",
       "70    Dictybase: Phenotypes  6.15e-02          12        51         51  \n",
       "71    Dictybase: Phenotypes  6.16e-02           3         5          5  \n",
       "72    Dictybase: Phenotypes  6.16e-02           3         5          5  \n",
       "73    Dictybase: Phenotypes  6.16e-02           3         5          5  \n",
       "74    Dictybase: Phenotypes  6.43e-02           4         9          9  \n",
       "75    Dictybase: Phenotypes  6.43e-02           4         9          9  \n",
       "76    Dictybase: Phenotypes  6.43e-02           4        10          9  \n",
       "77    Dictybase: Phenotypes  6.55e-02          25       146        137  \n",
       "78   GO: molecular_function  7.10e-02          26       145        145  \n",
       "79           KEGG: Pathways  8.11e-02           8        31         30  \n",
       "80    Dictybase: Phenotypes  8.96e-02           7        25         25  \n",
       "81    Dictybase: Phenotypes  9.15e-02           4        10         10  \n",
       "82    Dictybase: Phenotypes  9.52e-02           6        20         20  \n",
       "83    Dictybase: Phenotypes  9.69e-02           3         6          6  \n",
       "84    Dictybase: Phenotypes  9.69e-02           3         6          6  \n",
       "85    Dictybase: Phenotypes  9.69e-02           3         6          6  \n",
       "86    Dictybase: Phenotypes  9.82e-02          13        62         62  \n",
       "87    Dictybase: Phenotypes  1.10e-01          13        63         63  \n",
       "88    Dictybase: Phenotypes  1.16e-01           4        11         11  \n",
       "89    Dictybase: Phenotypes  1.16e-01           4        11         11  \n",
       "90    Dictybase: Phenotypes  1.16e-01           5        16         16  \n",
       "91           KEGG: Pathways  1.16e-01           4        11         11  \n",
       "92   GO: molecular_function  1.16e-01          11        51         51  \n",
       "93    Dictybase: Phenotypes  1.16e-01           7        27         27  \n",
       "94    Dictybase: Phenotypes  1.21e-01          12        58         58  \n",
       "95           KEGG: Pathways  1.30e-01           6        22         22  \n",
       "96    Dictybase: Phenotypes  1.37e-01           3         7          7  \n",
       "97    Dictybase: Phenotypes  1.46e-01           4        12         12  \n",
       "98    Dictybase: Phenotypes  1.46e-01           4        12         12  \n",
       "99    Dictybase: Phenotypes  1.46e-01           4        12         12  \n",
       "100  GO: biological_process  1.53e-01           8        35         35  \n",
       "101   Dictybase: Phenotypes  1.53e-01           7        30         29  \n",
       "102   Dictybase: Phenotypes  1.57e-01          16        88         88  \n",
       "103  GO: biological_process  1.61e-01          27       167        167  \n",
       "104   Dictybase: Phenotypes  1.64e-01           5        18         18  \n",
       "105   Dictybase: Phenotypes  1.64e-01           9        42         42  \n",
       "106  GO: biological_process  1.71e-01           3         8          8  \n",
       "107  GO: biological_process  1.71e-01          15        84         83  \n",
       "108   Dictybase: Phenotypes  1.71e-01           7        31         30  \n",
       "109   Dictybase: Phenotypes  1.71e-01           3         8          8  \n",
       "110   Dictybase: Phenotypes  1.71e-01           4        13         13  \n",
       "111   Dictybase: Phenotypes  1.71e-01           4        21         13  \n",
       "112   Dictybase: Phenotypes  1.71e-01           3         8          8  \n",
       "113   Dictybase: Phenotypes  1.71e-01           4        13         13  \n",
       "114  GO: biological_process  1.90e-01           6        25         25  \n",
       "115   Dictybase: Phenotypes  2.04e-01           2         5          4  \n",
       "116   Dictybase: Phenotypes  2.04e-01           2         5          4  \n",
       "117   Dictybase: Phenotypes  2.06e-01           4        14         14  \n",
       "118   Dictybase: Phenotypes  2.06e-01           4        14         14  \n",
       "119   Dictybase: Phenotypes  2.16e-01           5        20         20  \n",
       "120  GO: biological_process  2.21e-01          23       145        145  \n",
       "121          KEGG: Pathways  2.31e-01           9        46         46  \n",
       "122   Dictybase: Phenotypes  2.32e-01           7        34         33  \n",
       "123   Dictybase: Phenotypes  2.34e-01          10        53         53  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 191 with EID: 191\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 61 out of 76 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.78e-25</td>\n",
       "      <td>20</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.36e-23</td>\n",
       "      <td>24</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.88e-14</td>\n",
       "      <td>18</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.25e-12</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.03e-10</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.61e-10</td>\n",
       "      <td>11</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.16e-09</td>\n",
       "      <td>19</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.73e-09</td>\n",
       "      <td>15</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.98e-08</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.16e-07</td>\n",
       "      <td>8</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.20e-06</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.07e-05</td>\n",
       "      <td>11</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.07e-04</td>\n",
       "      <td>15</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.07e-04</td>\n",
       "      <td>13</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.01e-04</td>\n",
       "      <td>10</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.03e-03</td>\n",
       "      <td>8</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.95e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.95e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.06e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.19e-02</td>\n",
       "      <td>13</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>nuclear envelope</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>8.19e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.19e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>nuclease activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Gene set                Ontology       FDR  N in query  \\\n",
       "0              DNA replication          KEGG: Pathways  1.78e-25          20   \n",
       "1                   chromosome  GO: cellular_component  1.36e-23          24   \n",
       "2      chromosome organization  GO: biological_process  1.88e-14          18   \n",
       "3              Mismatch repair          KEGG: Pathways  8.25e-12          10   \n",
       "4       chromosome segregation  GO: biological_process  2.03e-10           9   \n",
       "5   Nucleotide excision repair          KEGG: Pathways  6.61e-10          11   \n",
       "6                   cell cycle  GO: biological_process  2.16e-09          19   \n",
       "7        DNA metabolic process  GO: biological_process  8.73e-09          15   \n",
       "8     mitotic nuclear division  GO: biological_process  4.98e-08           7   \n",
       "9         Base excision repair          KEGG: Pathways  3.16e-07           8   \n",
       "10          nuclear chromosome  GO: cellular_component  4.20e-06           6   \n",
       "11          mitotic cell cycle  GO: biological_process  2.07e-05          11   \n",
       "12                 DNA binding  GO: molecular_function  1.07e-04          15   \n",
       "13             ATPase activity  GO: molecular_function  1.07e-04          13   \n",
       "14               cell division  GO: biological_process  4.01e-04          10   \n",
       "15                 nucleoplasm  GO: cellular_component  2.03e-03           8   \n",
       "16           helicase activity  GO: molecular_function  1.68e-02           5   \n",
       "17  Non-homologous end-joining          KEGG: Pathways  3.95e-02           2   \n",
       "18             histone binding  GO: molecular_function  3.95e-02           2   \n",
       "19    Homologous recombination          KEGG: Pathways  6.06e-02           3   \n",
       "20          response to stress  GO: biological_process  8.19e-02          13   \n",
       "21            nuclear envelope  GO: cellular_component  8.19e-02           3   \n",
       "22       Pyrimidine metabolism          KEGG: Pathways  8.19e-02           3   \n",
       "23           nuclease activity  GO: molecular_function  1.88e-01           3   \n",
       "24                   nucleolus  GO: cellular_component  2.04e-01           4   \n",
       "\n",
       "    Set size  N in ref.  \n",
       "0         34         33  \n",
       "1         70         70  \n",
       "2         74         74  \n",
       "3         19         19  \n",
       "4         18         18  \n",
       "5         37         37  \n",
       "6        167        167  \n",
       "7        105        105  \n",
       "8         15         15  \n",
       "9         28         28  \n",
       "10        17         17  \n",
       "11       100        100  \n",
       "12       221        221  \n",
       "13       169        168  \n",
       "14       115        115  \n",
       "15        91         91  \n",
       "16        51         51  \n",
       "17         7          7  \n",
       "18         7          7  \n",
       "19        26         25  \n",
       "20       360        358  \n",
       "21        29         29  \n",
       "22        31         29  \n",
       "23        41         41  \n",
       "24        72         72  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 261 with EID: 260\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 73 out of 99 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.17e-27</td>\n",
       "      <td>29</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.75e-23</td>\n",
       "      <td>20</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.99e-13</td>\n",
       "      <td>19</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.45e-13</td>\n",
       "      <td>26</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.25e-12</td>\n",
       "      <td>11</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.17e-11</td>\n",
       "      <td>13</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.94e-09</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.61e-09</td>\n",
       "      <td>17</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>3.26e-08</td>\n",
       "      <td>8</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.79e-07</td>\n",
       "      <td>16</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.07e-07</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.48e-06</td>\n",
       "      <td>8</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.16e-06</td>\n",
       "      <td>20</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.35e-06</td>\n",
       "      <td>13</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.98e-03</td>\n",
       "      <td>13</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.01e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>9.96e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.22e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.56e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.88e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.29e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.29e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>nucleobase-containing compound catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set                Ontology  \\\n",
       "0                                         chromosome  GO: cellular_component   \n",
       "1                                    DNA replication          KEGG: Pathways   \n",
       "2                            chromosome organization  GO: biological_process   \n",
       "3                                         cell cycle  GO: biological_process   \n",
       "4                                    Mismatch repair          KEGG: Pathways   \n",
       "5                         Nucleotide excision repair          KEGG: Pathways   \n",
       "6                             chromosome segregation  GO: biological_process   \n",
       "7                              DNA metabolic process  GO: biological_process   \n",
       "8                                 nuclear chromosome  GO: cellular_component   \n",
       "9                                      cell division  GO: biological_process   \n",
       "10                          mitotic nuclear division  GO: biological_process   \n",
       "11                              Base excision repair          KEGG: Pathways   \n",
       "12                                       DNA binding  GO: molecular_function   \n",
       "13                                mitotic cell cycle  GO: biological_process   \n",
       "14                                   ATPase activity  GO: molecular_function   \n",
       "15                             Pyrimidine metabolism          KEGG: Pathways   \n",
       "16                     microtubule organizing center  GO: cellular_component   \n",
       "17                                       nucleoplasm  GO: cellular_component   \n",
       "18                                      cytoskeleton  GO: cellular_component   \n",
       "19                                 cell cycle arrest   Dictybase: Phenotypes   \n",
       "20                                 helicase activity  GO: molecular_function   \n",
       "21                        Non-homologous end-joining          KEGG: Pathways   \n",
       "22                                   histone binding  GO: molecular_function   \n",
       "23                          Homologous recombination          KEGG: Pathways   \n",
       "24  nucleobase-containing compound catabolic process  GO: biological_process   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   1.17e-27          29        70         70  \n",
       "1   3.75e-23          20        34         33  \n",
       "2   1.99e-13          19        74         74  \n",
       "3   3.45e-13          26       167        167  \n",
       "4   2.25e-12          11        19         19  \n",
       "5   3.17e-11          13        37         37  \n",
       "6   1.94e-09           9        18         18  \n",
       "7   6.61e-09          17       105        105  \n",
       "8   3.26e-08           8        17         17  \n",
       "9   1.79e-07          16       115        115  \n",
       "10  3.07e-07           7        15         15  \n",
       "11  2.48e-06           8        28         28  \n",
       "12  3.16e-06          20       221        221  \n",
       "13  7.35e-06          13       100        100  \n",
       "14  1.98e-03          13       169        168  \n",
       "15  5.01e-03           5        31         29  \n",
       "16  9.96e-03           6        50         50  \n",
       "17  1.22e-02           8        91         91  \n",
       "18  1.31e-02          12       185        185  \n",
       "19  3.56e-02           2         5          5  \n",
       "20  4.88e-02           5        51         51  \n",
       "21  6.29e-02           2         7          7  \n",
       "22  6.29e-02           2         7          7  \n",
       "23  1.18e-01           3        26         25  \n",
       "24  1.26e-01           3        26         26  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 76 with EID: 76\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 21 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.91e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Gene set               Ontology       FDR  N in query  Set size  \\\n",
       "0  cell cycle arrest  Dictybase: Phenotypes  1.91e-02           2         5   \n",
       "\n",
       "   N in ref.  \n",
       "0          5  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 8 with EID: 8\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.0\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 0 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 565 with EID: 559\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 32 out of 156 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>8.59e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.10e-01</td>\n",
       "      <td>18</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                Ontology  \\\n",
       "0                            Golgi apparatus  GO: cellular_component   \n",
       "1  SNARE interactions in vesicular transport          KEGG: Pathways   \n",
       "2                    oxidoreductase activity  GO: molecular_function   \n",
       "\n",
       "        FDR  N in query  Set size  N in ref.  \n",
       "0  8.59e-02          11        85         85  \n",
       "1  1.10e-01           5        23         23  \n",
       "2  1.10e-01          18       210        207  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 555 with EID: 546\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 36 out of 123 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.29e-05</td>\n",
       "      <td>13</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.98e-03</td>\n",
       "      <td>16</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.04e-03</td>\n",
       "      <td>15</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.36e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.95e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>8.86e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0  anatomical structure formation involved in morphogenesis   \n",
       "1                                      extracellular region   \n",
       "2                                      cell differentiation   \n",
       "3                 SNARE interactions in vesicular transport   \n",
       "4                               secondary metabolic process   \n",
       "5       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "6                                 decreased spore viability   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: biological_process  5.29e-05          13        74         74  \n",
       "1  GO: cellular_component  1.98e-03          16       157        157  \n",
       "2  GO: biological_process  2.04e-03          15       145        145  \n",
       "3          KEGG: Pathways  2.36e-03           6        23         23  \n",
       "4  GO: biological_process  9.95e-03           3         5          5  \n",
       "5          KEGG: Pathways  8.86e-02           3        10         10  \n",
       "6   Dictybase: Phenotypes  2.40e-01           6        60         59  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_impulse+'DEacrossStages_summary_noyem.tsv',index_col=0,sep='\\t')\n",
    "stages=np.array([phenotype for phenotype in PHENOTYPES if phenotype in list(milestones.columns)])\n",
    "for stage_idx,stage in enumerate(stages):\n",
    "    \n",
    "    #Nonneighbouring stages\n",
    "    neigh_previous=1\n",
    "    neigh_next=1\n",
    "    #Assumes phenotypes order\n",
    "    if stage_idx==1:\n",
    "        neigh_previous=0\n",
    "    if stage_idx==0:\n",
    "        neigh_next=0\n",
    "    non_neighbours = stages[[(idx <stage_idx-neigh_previous or idx>stage_idx+neigh_next) \n",
    "                 for idx in range(len(stages)) ]]\n",
    "    #print(stage,':',non_neighbours)\n",
    "    \n",
    "    #Find genes up at stage and down at nonenighbouring\n",
    "    query_condition=stage+'==\"up\"'\n",
    "    for non_neigh in non_neighbours:\n",
    "        query_condition=query_condition+' & '+non_neigh+' ==\"down\"'\n",
    "    #print(query_condition)\n",
    "    \n",
    "    query_names=list(milestones.query(query_condition).index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### ImpulseDE2 WT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 2330 with EID: 2310\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 539 out of 923 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.53e-28</td>\n",
       "      <td>76</td>\n",
       "      <td>105</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>translation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.76e-25</td>\n",
       "      <td>104</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.37e-23</td>\n",
       "      <td>65</td>\n",
       "      <td>85</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.22e-22</td>\n",
       "      <td>68</td>\n",
       "      <td>93</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.17e-19</td>\n",
       "      <td>118</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.03e-17</td>\n",
       "      <td>47</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.36e-16</td>\n",
       "      <td>51</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.82e-15</td>\n",
       "      <td>69</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.89e-14</td>\n",
       "      <td>47</td>\n",
       "      <td>72</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>tRNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.96e-12</td>\n",
       "      <td>45</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.61e-09</td>\n",
       "      <td>31</td>\n",
       "      <td>64</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.49e-08</td>\n",
       "      <td>34</td>\n",
       "      <td>71</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.75e-07</td>\n",
       "      <td>72</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.93e-05</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.94e-05</td>\n",
       "      <td>38</td>\n",
       "      <td>80</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>7.22e-05</td>\n",
       "      <td>29</td>\n",
       "      <td>58</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.03e-04</td>\n",
       "      <td>36</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>cellular amino acid metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.03e-04</td>\n",
       "      <td>49</td>\n",
       "      <td>125</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.76e-04</td>\n",
       "      <td>73</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cofactor metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.77e-03</td>\n",
       "      <td>31</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.11e-03</td>\n",
       "      <td>91</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.91e-03</td>\n",
       "      <td>110</td>\n",
       "      <td>354</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.78e-02</td>\n",
       "      <td>14</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Porphyrin and chlorophyll metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.81e-02</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.84e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>mitochondrion organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.29e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.50e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>RNA polymerase</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.34e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>19</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>sulfur compound metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Butanoate metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                Ontology  \\\n",
       "0                                         Ribosome          KEGG: Pathways   \n",
       "1                                      translation  GO: biological_process   \n",
       "2               structural constituent of ribosome  GO: molecular_function   \n",
       "3                                         ribosome  GO: cellular_component   \n",
       "4                                    mitochondrion  GO: cellular_component   \n",
       "5                              ribosome biogenesis  GO: biological_process   \n",
       "6                                        nucleolus  GO: cellular_component   \n",
       "7                     structural molecule activity  GO: molecular_function   \n",
       "8                Ribosome biogenesis in eukaryotes          KEGG: Pathways   \n",
       "9                           tRNA metabolic process  GO: biological_process   \n",
       "10                     Aminoacyl-tRNA biosynthesis          KEGG: Pathways   \n",
       "11                       Oxidative phosphorylation          KEGG: Pathways   \n",
       "12                                     RNA binding  GO: molecular_function   \n",
       "13                            Steroid biosynthesis          KEGG: Pathways   \n",
       "14                                 ligase activity  GO: molecular_function   \n",
       "15                      methyltransferase activity  GO: molecular_function   \n",
       "16  generation of precursor metabolites and energy  GO: biological_process   \n",
       "17           cellular amino acid metabolic process  GO: biological_process   \n",
       "18                         oxidoreductase activity  GO: molecular_function   \n",
       "19                      cofactor metabolic process  GO: biological_process   \n",
       "20           Biosynthesis of secondary metabolites          KEGG: Pathways   \n",
       "21                small molecule metabolic process  GO: biological_process   \n",
       "22                        unfolded protein binding  GO: molecular_function   \n",
       "23            Porphyrin and chlorophyll metabolism          KEGG: Pathways   \n",
       "24                                 protein folding  GO: biological_process   \n",
       "25                      mitochondrion organization  GO: biological_process   \n",
       "26                                    rRNA binding  GO: molecular_function   \n",
       "27                                  RNA polymerase          KEGG: Pathways   \n",
       "28                              isomerase activity  GO: molecular_function   \n",
       "29               sulfur compound metabolic process  GO: biological_process   \n",
       "30                            Butanoate metabolism          KEGG: Pathways   \n",
       "31                           Fatty acid elongation          KEGG: Pathways   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   2.53e-28          76       105         97  \n",
       "1   4.76e-25         104       169        169  \n",
       "2   2.37e-23          65        85         85  \n",
       "3   1.22e-22          68        93         93  \n",
       "4   1.17e-19         118       229        229  \n",
       "5   4.03e-17          47        61         61  \n",
       "6   5.36e-16          51        72         72  \n",
       "7   2.82e-15          69       117        117  \n",
       "8   3.89e-14          47        72         68  \n",
       "9   3.96e-12          45        69         69  \n",
       "10  7.61e-09          31        64         46  \n",
       "11  1.49e-08          34        71         54  \n",
       "12  3.75e-07          72       169        169  \n",
       "13  3.93e-05          14        18         18  \n",
       "14  3.94e-05          38        80         80  \n",
       "15  7.22e-05          29        58         56  \n",
       "16  1.03e-04          36        77         77  \n",
       "17  7.03e-04          49       125        124  \n",
       "18  9.76e-04          73       210        207  \n",
       "19  2.77e-03          31        72         72  \n",
       "20  4.11e-03          91       289        281  \n",
       "21  4.91e-03         110       354        353  \n",
       "22  1.78e-02          14        27         27  \n",
       "23  1.81e-02           9        14         14  \n",
       "24  2.84e-02          17        37         37  \n",
       "25  3.29e-02          12        23         23  \n",
       "26  4.50e-02           8        13         13  \n",
       "27  7.34e-02          12        26         25  \n",
       "28  1.68e-01          19        50         50  \n",
       "29  1.77e-01          14        34         34  \n",
       "30  1.77e-01           7        13         13  \n",
       "31  1.77e-01           7        13         13  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 409 with EID: 403\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.41\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 127 out of 166 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.00e-06</td>\n",
       "      <td>12</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.47e-04</td>\n",
       "      <td>15</td>\n",
       "      <td>91</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>8.30e-04</td>\n",
       "      <td>22</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.86e-04</td>\n",
       "      <td>32</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.82e-03</td>\n",
       "      <td>12</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.11e-03</td>\n",
       "      <td>24</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.98e-03</td>\n",
       "      <td>10</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.98e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.24e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.24e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.46e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.43e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.43e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased exocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.94e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.94e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.97e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>aberrant chemotaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.97e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.20e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.77e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.77e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>increased filopodium formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.24e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.24e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.50e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.50e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.54e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.54e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.75e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.21e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>67</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.21e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.60e-02</td>\n",
       "      <td>25</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>decreased macropinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.60e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased pinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.60e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>aberrant cytoskeleton organisation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.60e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>aberrant slug formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.60e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.79e-02</td>\n",
       "      <td>29</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.79e-02</td>\n",
       "      <td>20</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.81e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>abolished protein phosphorylation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.07e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.07e-02</td>\n",
       "      <td>23</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.05e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>decreased cAMP-induced cAMP pulse</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.18e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>aberrant response to hyperosmotic stress</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.18e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.44e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>enzyme binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.51e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>abolished chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.06e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>nucleobase-containing compound catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>aberrant chemotaxis to folate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>decreased activation of adenylate cyclase</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>decreased growth rate in suspension</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>increased protein level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>increased intracellular cAMP level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>decreased protein kinase activation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>10</td>\n",
       "      <td>134</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>13</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>abolished cell polarity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>aberrant pseudopod formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>abolished growth</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>decreased lateral pseudopod suppression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>68</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>12</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>69</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>70</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>71</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.19e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>95</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.37e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set                Ontology  \\\n",
       "0                                         Proteasome          KEGG: Pathways   \n",
       "1        Protein processing in endoplasmic reticulum          KEGG: Pathways   \n",
       "2                                         locomotion  GO: biological_process   \n",
       "3                                signal transduction  GO: biological_process   \n",
       "4                                 aberrant streaming   Dictybase: Phenotypes   \n",
       "5                                  catabolic process  GO: biological_process   \n",
       "6                               aberrant aggregation   Dictybase: Phenotypes   \n",
       "7          abolished activation of adenylate cyclase   Dictybase: Phenotypes   \n",
       "8                           decreased aggregate size   Dictybase: Phenotypes   \n",
       "9                            decreased cell motility   Dictybase: Phenotypes   \n",
       "10                                         wild type   Dictybase: Phenotypes   \n",
       "11                     aberrant F-actin organization   Dictybase: Phenotypes   \n",
       "12                              aberrant cytokinesis   Dictybase: Phenotypes   \n",
       "13                              decreased exocytosis   Dictybase: Phenotypes   \n",
       "14          development arrests at loose mound stage   Dictybase: Phenotypes   \n",
       "15                         cytoskeleton organization  GO: biological_process   \n",
       "16                               aberrant chemotaxis   Dictybase: Phenotypes   \n",
       "17                                      reproduction  GO: biological_process   \n",
       "18                               abolished streaming   Dictybase: Phenotypes   \n",
       "19                      aberrant cAMP wave formation   Dictybase: Phenotypes   \n",
       "20                    increased filopodium formation   Dictybase: Phenotypes   \n",
       "21                              aberrant culmination   Dictybase: Phenotypes   \n",
       "22                       cellular component assembly  GO: biological_process   \n",
       "23                         enzyme regulator activity  GO: molecular_function   \n",
       "24                      decreased chemotaxis to cAMP   Dictybase: Phenotypes   \n",
       "25                     increased basal F-actin level   Dictybase: Phenotypes   \n",
       "26                             decreased culmination   Dictybase: Phenotypes   \n",
       "27                                     cell motility  GO: biological_process   \n",
       "28                        precocious gene expression   Dictybase: Phenotypes   \n",
       "29                                response to stress  GO: biological_process   \n",
       "30                        decreased macropinocytosis   Dictybase: Phenotypes   \n",
       "31                             decreased pinocytosis   Dictybase: Phenotypes   \n",
       "32                aberrant cytoskeleton organisation   Dictybase: Phenotypes   \n",
       "33                           aberrant slug formation   Dictybase: Phenotypes   \n",
       "34                  anatomical structure development  GO: biological_process   \n",
       "35                                   plasma membrane  GO: cellular_component   \n",
       "36                          abolished slug migration   Dictybase: Phenotypes   \n",
       "37                 abolished protein phosphorylation   Dictybase: Phenotypes   \n",
       "38                                   kinase activity  GO: molecular_function   \n",
       "39                      decreased fruiting body size   Dictybase: Phenotypes   \n",
       "40                 decreased cAMP-induced cAMP pulse   Dictybase: Phenotypes   \n",
       "41          aberrant response to hyperosmotic stress   Dictybase: Phenotypes   \n",
       "42                        multiple tips in culminant   Dictybase: Phenotypes   \n",
       "43                                    enzyme binding  GO: molecular_function   \n",
       "44                      abolished chemotaxis to cAMP   Dictybase: Phenotypes   \n",
       "45  nucleobase-containing compound catabolic process  GO: biological_process   \n",
       "46                             endoplasmic reticulum  GO: cellular_component   \n",
       "47                     aberrant chemotaxis to folate   Dictybase: Phenotypes   \n",
       "48         decreased activation of adenylate cyclase   Dictybase: Phenotypes   \n",
       "49                               delayed development   Dictybase: Phenotypes   \n",
       "50                development arrests at mound stage   Dictybase: Phenotypes   \n",
       "51                          aberrant cell morphology   Dictybase: Phenotypes   \n",
       "52                           aberrant slug migration   Dictybase: Phenotypes   \n",
       "53                                peptidase activity  GO: molecular_function   \n",
       "54                             increased growth rate   Dictybase: Phenotypes   \n",
       "55               decreased growth rate in suspension   Dictybase: Phenotypes   \n",
       "56                           increased protein level   Dictybase: Phenotypes   \n",
       "57                increased intracellular cAMP level   Dictybase: Phenotypes   \n",
       "58               decreased protein kinase activation   Dictybase: Phenotypes   \n",
       "59               protein-containing complex assembly  GO: biological_process   \n",
       "60                                     cell division  GO: biological_process   \n",
       "61                                      cytoskeleton  GO: cellular_component   \n",
       "62                             abolished aggregation   Dictybase: Phenotypes   \n",
       "63                            precocious aggregation   Dictybase: Phenotypes   \n",
       "64                           abolished cell polarity   Dictybase: Phenotypes   \n",
       "65                      aberrant pseudopod formation   Dictybase: Phenotypes   \n",
       "66                                  abolished growth   Dictybase: Phenotypes   \n",
       "67           decreased lateral pseudopod suppression   Dictybase: Phenotypes   \n",
       "68                                   ATPase activity  GO: molecular_function   \n",
       "69                                     cell adhesion  GO: biological_process   \n",
       "70                                mitotic cell cycle  GO: biological_process   \n",
       "71                         decreased chemotaxis rate   Dictybase: Phenotypes   \n",
       "72                         decreased gene expression   Dictybase: Phenotypes   \n",
       "73                                cell morphogenesis  GO: biological_process   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   4.00e-06          12        39         37  \n",
       "1   3.47e-04          15        91         88  \n",
       "2   8.30e-04          22       191        189  \n",
       "3   9.86e-04          32       352        350  \n",
       "4   1.82e-03          12        72         72  \n",
       "5   4.11e-03          24       250        250  \n",
       "6   5.98e-03          10        61         61  \n",
       "7   5.98e-03           4         8          8  \n",
       "8   7.24e-03           7        32         32  \n",
       "9   7.24e-03          11        76         76  \n",
       "10  7.46e-03          28       341        338  \n",
       "11  1.43e-02           7        37         37  \n",
       "12  1.43e-02          12        98         98  \n",
       "13  1.94e-02           4        12         12  \n",
       "14  1.94e-02           5        20         20  \n",
       "15  1.97e-02          18       197        196  \n",
       "16  1.97e-02           3         6          6  \n",
       "17  2.20e-02          10        79         79  \n",
       "18  2.77e-02           6        33         33  \n",
       "19  2.77e-02          11        96         96  \n",
       "20  4.24e-02           4        16         16  \n",
       "21  4.24e-02          10        88         88  \n",
       "22  4.50e-02          17       202        201  \n",
       "23  4.50e-02          11       105        105  \n",
       "24  4.54e-02           8        63         63  \n",
       "25  4.54e-02           4        18         17  \n",
       "26  4.75e-02           3         9          9  \n",
       "27  5.21e-02           8        67         66  \n",
       "28  5.21e-02           4        18         18  \n",
       "29  5.60e-02          25       360        358  \n",
       "30  5.60e-02           4        19         19  \n",
       "31  5.60e-02           5        30         30  \n",
       "32  5.60e-02           3        10         10  \n",
       "33  5.60e-02           3        10         10  \n",
       "34  5.79e-02          29       440        438  \n",
       "35  5.79e-02          20       268        268  \n",
       "36  5.81e-02           6        43         43  \n",
       "37  6.07e-02           3        11         11  \n",
       "38  6.07e-02          23       328        327  \n",
       "39  8.05e-02           8        76         75  \n",
       "40  9.18e-02           3        14         13  \n",
       "41  9.18e-02           3        13         13  \n",
       "42  9.44e-02           2         5          5  \n",
       "43  9.51e-02          10       110        110  \n",
       "44  1.06e-01           3        14         14  \n",
       "45  1.20e-01           4        26         26  \n",
       "46  1.23e-01          10       117        117  \n",
       "47  1.23e-01           2         6          6  \n",
       "48  1.23e-01           2         6          6  \n",
       "49  1.23e-01           7        69         69  \n",
       "50  1.23e-01           6        56         54  \n",
       "51  1.33e-01           5        42         41  \n",
       "52  1.60e-01           6        58         58  \n",
       "53  1.70e-01           8        91         91  \n",
       "54  1.80e-01           4        32         31  \n",
       "55  1.80e-01           2         8          8  \n",
       "56  1.80e-01           2         9          8  \n",
       "57  1.80e-01           2         8          8  \n",
       "58  1.84e-01           3        19         19  \n",
       "59  1.91e-01          10       134        133  \n",
       "60  1.91e-01           9       115        115  \n",
       "61  1.91e-01          13       185        185  \n",
       "62  1.91e-01           7        82         80  \n",
       "63  1.91e-01           3        20         20  \n",
       "64  1.91e-01           2         9          9  \n",
       "65  1.91e-01           2         9          9  \n",
       "66  1.91e-01           4        34         33  \n",
       "67  1.91e-01           2         9          9  \n",
       "68  1.91e-01          12       169        168  \n",
       "69  2.09e-01           7        84         83  \n",
       "70  2.09e-01           8       100        100  \n",
       "71  2.19e-01           4        36         36  \n",
       "72  2.37e-01           7        95         86  \n",
       "73  2.37e-01           3        23         23  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 429 with EID: 423\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.42\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 125 out of 176 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.99e-06</td>\n",
       "      <td>12</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.50e-05</td>\n",
       "      <td>37</td>\n",
       "      <td>352</td>\n",
       "      <td>350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.71e-04</td>\n",
       "      <td>23</td>\n",
       "      <td>191</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.30e-04</td>\n",
       "      <td>12</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.40e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>91</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.39e-03</td>\n",
       "      <td>8</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.39e-03</td>\n",
       "      <td>12</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.56e-03</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.21e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.21e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.21e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.51e-02</td>\n",
       "      <td>24</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.08e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.27e-02</td>\n",
       "      <td>22</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant chemotaxis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>decreased exocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.27e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>98</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.45e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.32e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.74e-02</td>\n",
       "      <td>26</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.99e-02</td>\n",
       "      <td>27</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>increased filopodium formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.16e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.81e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.98e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>6.14e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.85e-02</td>\n",
       "      <td>17</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>aberrant cytoskeleton organisation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.16e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>decreased macropinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.33e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased pinocytosis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.56e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.56e-02</td>\n",
       "      <td>7</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.07e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>abolished protein phosphorylation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.17e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.05e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>117</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>29</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>15</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>decreased cAMP-induced cAMP pulse</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>76</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>aberrant response to hyperosmotic stress</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>abolished chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.39e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>67</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>aberrant chemotaxis to folate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>decreased activation of adenylate cyclase</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.65e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>aberrant chemotaxis to cAMP</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>decreased growth rate in suspension</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>increased protein level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>increased intracellular cAMP level</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>decreased protein kinase activation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.15e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.23e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>abolished cell polarity</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.23e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>aberrant pseudopod formation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.23e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>abolished growth</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.23e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>decreased lateral pseudopod suppression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.23e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.23e-01</td>\n",
       "      <td>21</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>enzyme binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.23e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.46e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>82</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Gene set                Ontology  \\\n",
       "0                                    Proteasome          KEGG: Pathways   \n",
       "1                           signal transduction  GO: biological_process   \n",
       "2                                    locomotion  GO: biological_process   \n",
       "3                          aberrant aggregation   Dictybase: Phenotypes   \n",
       "4   Protein processing in endoplasmic reticulum          KEGG: Pathways   \n",
       "5                      decreased aggregate size   Dictybase: Phenotypes   \n",
       "6                            aberrant streaming   Dictybase: Phenotypes   \n",
       "7     abolished activation of adenylate cyclase   Dictybase: Phenotypes   \n",
       "8                           abolished streaming   Dictybase: Phenotypes   \n",
       "9                      abolished slug migration   Dictybase: Phenotypes   \n",
       "10                      decreased cell motility   Dictybase: Phenotypes   \n",
       "11                              plasma membrane  GO: cellular_component   \n",
       "12                aberrant F-actin organization   Dictybase: Phenotypes   \n",
       "13                            catabolic process  GO: biological_process   \n",
       "14                          aberrant chemotaxis   Dictybase: Phenotypes   \n",
       "15                         decreased exocytosis   Dictybase: Phenotypes   \n",
       "16     development arrests at loose mound stage   Dictybase: Phenotypes   \n",
       "17                         aberrant cytokinesis   Dictybase: Phenotypes   \n",
       "18                    cytoskeleton organization  GO: biological_process   \n",
       "19                 aberrant cAMP wave formation   Dictybase: Phenotypes   \n",
       "20                                    wild type   Dictybase: Phenotypes   \n",
       "21                           response to stress  GO: biological_process   \n",
       "22               increased filopodium formation   Dictybase: Phenotypes   \n",
       "23                         aberrant culmination   Dictybase: Phenotypes   \n",
       "24                        decreased culmination   Dictybase: Phenotypes   \n",
       "25                increased basal F-actin level   Dictybase: Phenotypes   \n",
       "26                 decreased chemotaxis to cAMP   Dictybase: Phenotypes   \n",
       "27                    enzyme regulator activity  GO: molecular_function   \n",
       "28                  cellular component assembly  GO: biological_process   \n",
       "29           aberrant cytoskeleton organisation   Dictybase: Phenotypes   \n",
       "30                   decreased macropinocytosis   Dictybase: Phenotypes   \n",
       "31                        decreased pinocytosis   Dictybase: Phenotypes   \n",
       "32           development arrests at mound stage   Dictybase: Phenotypes   \n",
       "33                       precocious aggregation   Dictybase: Phenotypes   \n",
       "34            abolished protein phosphorylation   Dictybase: Phenotypes   \n",
       "35                        endoplasmic reticulum  GO: cellular_component   \n",
       "36             anatomical structure development  GO: biological_process   \n",
       "37                                 cytoskeleton  GO: cellular_component   \n",
       "38            decreased cAMP-induced cAMP pulse   Dictybase: Phenotypes   \n",
       "39                   multiple tips in culminant   Dictybase: Phenotypes   \n",
       "40                 decreased fruiting body size   Dictybase: Phenotypes   \n",
       "41     aberrant response to hyperosmotic stress   Dictybase: Phenotypes   \n",
       "42                 abolished chemotaxis to cAMP   Dictybase: Phenotypes   \n",
       "43                                 reproduction  GO: biological_process   \n",
       "44                                cell motility  GO: biological_process   \n",
       "45                aberrant chemotaxis to folate   Dictybase: Phenotypes   \n",
       "46    decreased activation of adenylate cyclase   Dictybase: Phenotypes   \n",
       "47                                cell adhesion  GO: biological_process   \n",
       "48                          delayed development   Dictybase: Phenotypes   \n",
       "49                     aberrant cell morphology   Dictybase: Phenotypes   \n",
       "50                  aberrant chemotaxis to cAMP   Dictybase: Phenotypes   \n",
       "51                      delayed gene expression   Dictybase: Phenotypes   \n",
       "52          decreased growth rate in suspension   Dictybase: Phenotypes   \n",
       "53                      increased protein level   Dictybase: Phenotypes   \n",
       "54                     increased slug migration   Dictybase: Phenotypes   \n",
       "55                   precocious gene expression   Dictybase: Phenotypes   \n",
       "56           increased intracellular cAMP level   Dictybase: Phenotypes   \n",
       "57                      aberrant slug migration   Dictybase: Phenotypes   \n",
       "58                        increased growth rate   Dictybase: Phenotypes   \n",
       "59          decreased protein kinase activation   Dictybase: Phenotypes   \n",
       "60                        abolished culmination   Dictybase: Phenotypes   \n",
       "61                      abolished cell polarity   Dictybase: Phenotypes   \n",
       "62                 aberrant pseudopod formation   Dictybase: Phenotypes   \n",
       "63                             abolished growth   Dictybase: Phenotypes   \n",
       "64      decreased lateral pseudopod suppression   Dictybase: Phenotypes   \n",
       "65                              kinase activity  GO: molecular_function   \n",
       "66                               enzyme binding  GO: molecular_function   \n",
       "67                        abolished aggregation   Dictybase: Phenotypes   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   7.99e-06          12        39         37  \n",
       "1   4.50e-05          37       352        350  \n",
       "2   6.71e-04          23       191        189  \n",
       "3   7.30e-04          12        61         61  \n",
       "4   1.40e-03          14        91         88  \n",
       "5   2.39e-03           8        32         32  \n",
       "6   2.39e-03          12        72         72  \n",
       "7   7.56e-03           4         8          8  \n",
       "8   1.21e-02           7        33         33  \n",
       "9   1.21e-02           8        43         43  \n",
       "10  1.21e-02          11        76         76  \n",
       "11  1.51e-02          24       268        268  \n",
       "12  2.08e-02           7        37         37  \n",
       "13  2.27e-02          22       250        250  \n",
       "14  2.27e-02           3         6          6  \n",
       "15  2.27e-02           4        12         12  \n",
       "16  2.27e-02           5        20         20  \n",
       "17  2.27e-02          12        98         98  \n",
       "18  3.45e-02          18       197        196  \n",
       "19  4.32e-02          11        96         96  \n",
       "20  4.74e-02          26       341        338  \n",
       "21  4.99e-02          27       360        358  \n",
       "22  5.16e-02           4        16         16  \n",
       "23  5.81e-02          10        88         88  \n",
       "24  5.98e-02           3         9          9  \n",
       "25  5.98e-02           4        18         17  \n",
       "26  6.14e-02           8        63         63  \n",
       "27  6.15e-02          11       105        105  \n",
       "28  6.85e-02          17       202        201  \n",
       "29  7.16e-02           3        10         10  \n",
       "30  7.33e-02           4        19         19  \n",
       "31  7.56e-02           5        30         30  \n",
       "32  7.56e-02           7        56         54  \n",
       "33  8.07e-02           4        20         20  \n",
       "34  8.17e-02           3        11         11  \n",
       "35  1.05e-01          11       117        117  \n",
       "36  1.12e-01          29       440        438  \n",
       "37  1.12e-01          15       185        185  \n",
       "38  1.12e-01           3        14         13  \n",
       "39  1.12e-01           2         5          5  \n",
       "40  1.12e-01           8        76         75  \n",
       "41  1.12e-01           3        13         13  \n",
       "42  1.33e-01           3        14         14  \n",
       "43  1.39e-01           8        79         79  \n",
       "44  1.46e-01           7        67         66  \n",
       "45  1.46e-01           2         6          6  \n",
       "46  1.46e-01           2         6          6  \n",
       "47  1.65e-01           8        84         83  \n",
       "48  1.74e-01           7        69         69  \n",
       "49  1.77e-01           5        42         41  \n",
       "50  2.07e-01           4        31         31  \n",
       "51  2.07e-01           3        19         18  \n",
       "52  2.07e-01           2         8          8  \n",
       "53  2.07e-01           2         9          8  \n",
       "54  2.07e-01           2         8          8  \n",
       "55  2.07e-01           3        18         18  \n",
       "56  2.07e-01           2         8          8  \n",
       "57  2.07e-01           6        58         58  \n",
       "58  2.07e-01           4        32         31  \n",
       "59  2.15e-01           3        19         19  \n",
       "60  2.23e-01           6        62         62  \n",
       "61  2.23e-01           2         9          9  \n",
       "62  2.23e-01           2         9          9  \n",
       "63  2.23e-01           4        34         33  \n",
       "64  2.23e-01           2         9          9  \n",
       "65  2.23e-01          21       328        327  \n",
       "66  2.23e-01           9       110        110  \n",
       "67  2.46e-01           7        82         80  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 273 with EID: 272\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.35\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 65 out of 95 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.44e-16</td>\n",
       "      <td>21</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.44e-16</td>\n",
       "      <td>16</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.43e-14</td>\n",
       "      <td>22</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.58e-11</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.50e-10</td>\n",
       "      <td>16</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.58e-09</td>\n",
       "      <td>24</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.72e-08</td>\n",
       "      <td>20</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.62e-08</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.40e-08</td>\n",
       "      <td>9</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.27e-06</td>\n",
       "      <td>9</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.05e-06</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.81e-04</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.81e-04</td>\n",
       "      <td>14</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.45e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.70e-03</td>\n",
       "      <td>9</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.11e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.50e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.06e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>62</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.41e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Gene set                Ontology       FDR  \\\n",
       "0                       chromosome  GO: cellular_component  1.44e-16   \n",
       "1                  DNA replication          KEGG: Pathways  1.44e-16   \n",
       "2            DNA metabolic process  GO: biological_process  3.43e-14   \n",
       "3                  Mismatch repair          KEGG: Pathways  7.58e-11   \n",
       "4          chromosome organization  GO: biological_process  1.50e-10   \n",
       "5                      DNA binding  GO: molecular_function  2.58e-09   \n",
       "6                       cell cycle  GO: biological_process  1.72e-08   \n",
       "7           chromosome segregation  GO: biological_process  3.62e-08   \n",
       "8             Base excision repair          KEGG: Pathways  9.40e-08   \n",
       "9       Nucleotide excision repair          KEGG: Pathways  1.27e-06   \n",
       "10        mitotic nuclear division  GO: biological_process  6.05e-06   \n",
       "11              nuclear chromosome  GO: cellular_component  2.81e-04   \n",
       "12                 ATPase activity  GO: molecular_function  2.81e-04   \n",
       "13                   cell division  GO: biological_process  5.45e-04   \n",
       "14              mitotic cell cycle  GO: biological_process  3.70e-03   \n",
       "15              response to stress  GO: biological_process  1.11e-02   \n",
       "16        Homologous recombination          KEGG: Pathways  1.50e-02   \n",
       "17      Non-homologous end-joining          KEGG: Pathways  5.49e-02   \n",
       "18                 histone binding  GO: molecular_function  5.49e-02   \n",
       "19  Ubiquitin mediated proteolysis          KEGG: Pathways  7.06e-02   \n",
       "20           Pyrimidine metabolism          KEGG: Pathways  1.41e-01   \n",
       "\n",
       "    N in query  Set size  N in ref.  \n",
       "0           21        70         70  \n",
       "1           16        34         33  \n",
       "2           22       105        105  \n",
       "3           10        19         19  \n",
       "4           16        74         74  \n",
       "5           24       221        221  \n",
       "6           20       167        167  \n",
       "7            8        18         18  \n",
       "8            9        28         28  \n",
       "9            9        37         37  \n",
       "10           6        15         15  \n",
       "11           5        17         17  \n",
       "12          14       169        168  \n",
       "13          11       115        115  \n",
       "14           9       100        100  \n",
       "15          18       360        358  \n",
       "16           4        26         25  \n",
       "17           2         7          7  \n",
       "18           2         7          7  \n",
       "19           5        62         61  \n",
       "20           3        31         29  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 486 with EID: 485\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.34\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 119 out of 166 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.00e-31</td>\n",
       "      <td>37</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.03e-27</td>\n",
       "      <td>25</td>\n",
       "      <td>34</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.89e-21</td>\n",
       "      <td>30</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.75e-20</td>\n",
       "      <td>41</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.13e-16</td>\n",
       "      <td>30</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.13e-16</td>\n",
       "      <td>15</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.72e-15</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.10e-11</td>\n",
       "      <td>11</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>3.39e-11</td>\n",
       "      <td>35</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.51e-10</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.22e-10</td>\n",
       "      <td>24</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>9.92e-10</td>\n",
       "      <td>14</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.17e-08</td>\n",
       "      <td>20</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.46e-06</td>\n",
       "      <td>8</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.43e-05</td>\n",
       "      <td>11</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.45e-05</td>\n",
       "      <td>21</td>\n",
       "      <td>169</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.76e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.04e-03</td>\n",
       "      <td>18</td>\n",
       "      <td>185</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>6.44e-03</td>\n",
       "      <td>9</td>\n",
       "      <td>62</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.55e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.97e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.08e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>4.43e-02</td>\n",
       "      <td>25</td>\n",
       "      <td>360</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant centrosome organization</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.21e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>aberrant nucleus morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.21e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.21e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>aberrant cell cycle</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.21e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>9.39e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>nuclear envelope</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>increased sensitivity to DNA-damaging agents</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Gene set                Ontology  \\\n",
       "0                                     chromosome  GO: cellular_component   \n",
       "1                                DNA replication          KEGG: Pathways   \n",
       "2                        chromosome organization  GO: biological_process   \n",
       "3                                     cell cycle  GO: biological_process   \n",
       "4                          DNA metabolic process  GO: biological_process   \n",
       "5                                Mismatch repair          KEGG: Pathways   \n",
       "6                         chromosome segregation  GO: biological_process   \n",
       "7                       mitotic nuclear division  GO: biological_process   \n",
       "8                                    DNA binding  GO: molecular_function   \n",
       "9                           Base excision repair          KEGG: Pathways   \n",
       "10                                 cell division  GO: biological_process   \n",
       "11                    Nucleotide excision repair          KEGG: Pathways   \n",
       "12                            mitotic cell cycle  GO: biological_process   \n",
       "13                            nuclear chromosome  GO: cellular_component   \n",
       "14                 microtubule organizing center  GO: cellular_component   \n",
       "15                               ATPase activity  GO: molecular_function   \n",
       "16                      Homologous recombination          KEGG: Pathways   \n",
       "17                                  cytoskeleton  GO: cellular_component   \n",
       "18                Ubiquitin mediated proteolysis          KEGG: Pathways   \n",
       "19                               histone binding  GO: molecular_function   \n",
       "20                                   nucleoplasm  GO: cellular_component   \n",
       "21                         Pyrimidine metabolism          KEGG: Pathways   \n",
       "22                            response to stress  GO: biological_process   \n",
       "23              aberrant centrosome organization   Dictybase: Phenotypes   \n",
       "24                   aberrant nucleus morphology   Dictybase: Phenotypes   \n",
       "25                             cell cycle arrest   Dictybase: Phenotypes   \n",
       "26                           aberrant cell cycle   Dictybase: Phenotypes   \n",
       "27                             helicase activity  GO: molecular_function   \n",
       "28                              nuclear envelope  GO: cellular_component   \n",
       "29                    Non-homologous end-joining          KEGG: Pathways   \n",
       "30  increased sensitivity to DNA-damaging agents   Dictybase: Phenotypes   \n",
       "\n",
       "         FDR  N in query  Set size  N in ref.  \n",
       "0   1.00e-31          37        70         70  \n",
       "1   5.03e-27          25        34         33  \n",
       "2   1.89e-21          30        74         74  \n",
       "3   2.75e-20          41       167        167  \n",
       "4   1.13e-16          30       105        105  \n",
       "5   1.13e-16          15        19         19  \n",
       "6   1.72e-15          14        18         18  \n",
       "7   1.10e-11          11        15         15  \n",
       "8   3.39e-11          35       221        221  \n",
       "9   2.51e-10          13        28         28  \n",
       "10  3.22e-10          24       115        115  \n",
       "11  9.92e-10          14        37         37  \n",
       "12  3.17e-08          20       100        100  \n",
       "13  1.46e-06           8        17         17  \n",
       "14  4.43e-05          11        50         50  \n",
       "15  4.45e-05          21       169        168  \n",
       "16  3.76e-03           6        26         25  \n",
       "17  5.04e-03          18       185        185  \n",
       "18  6.44e-03           9        62         61  \n",
       "19  1.55e-02           3         7          7  \n",
       "20  2.97e-02          10        91         91  \n",
       "21  4.08e-02           5        31         29  \n",
       "22  4.43e-02          25       360        358  \n",
       "23  8.21e-02           2         5          5  \n",
       "24  8.21e-02           2         5          5  \n",
       "25  8.21e-02           2         5          5  \n",
       "26  8.21e-02           2         5          5  \n",
       "27  9.39e-02           6        51         51  \n",
       "28  1.47e-01           4        29         29  \n",
       "29  1.47e-01           2         7          7  \n",
       "30  1.84e-01           2         8          8  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 117 with EID: 116\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.22\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 25 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.87e-03</td>\n",
       "      <td>5</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>9.85e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>91</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell death</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>RNA polymerase</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>105</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell division</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.85e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>transferase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.02e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                Ontology       FDR  N in query  \\\n",
       "0    chromosome organization  GO: biological_process  6.87e-03           5   \n",
       "1                nucleoplasm  GO: cellular_component  9.85e-02           4   \n",
       "2                 cell cycle  GO: biological_process  1.76e-01           4   \n",
       "3                 cell death  GO: biological_process  1.76e-01           2   \n",
       "4      DNA metabolic process  GO: biological_process  1.76e-01           3   \n",
       "5                 chromosome  GO: cellular_component  1.76e-01           3   \n",
       "6             RNA polymerase          KEGG: Pathways  1.76e-01           2   \n",
       "7  enzyme regulator activity  GO: molecular_function  1.76e-01           3   \n",
       "8              cell division  GO: biological_process  1.85e-01           3   \n",
       "9       transferase activity  GO: molecular_function  2.02e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0        74         74  \n",
       "1        91         91  \n",
       "2       167        167  \n",
       "3        25         25  \n",
       "4       105        105  \n",
       "5        70         70  \n",
       "6        26         25  \n",
       "7       105        105  \n",
       "8       115        115  \n",
       "9        51         51  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 28 with EID: 28\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.5\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 7 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.36e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.36e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>89</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.36e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Amino sugar and nucleotide sugar metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.36e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>250</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>354</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.49e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>210</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.88e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>104</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>2.19e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>289</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.44e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>202</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         Gene set                Ontology  \\\n",
       "0  generation of precursor metabolites and energy  GO: biological_process   \n",
       "1                  carbohydrate metabolic process  GO: biological_process   \n",
       "2                   Starch and sucrose metabolism          KEGG: Pathways   \n",
       "3     Amino sugar and nucleotide sugar metabolism          KEGG: Pathways   \n",
       "4                               catabolic process  GO: biological_process   \n",
       "5                small molecule metabolic process  GO: biological_process   \n",
       "6                         oxidoreductase activity  GO: molecular_function   \n",
       "7                               Carbon metabolism          KEGG: Pathways   \n",
       "8           Biosynthesis of secondary metabolites          KEGG: Pathways   \n",
       "9                     cellular component assembly  GO: biological_process   \n",
       "\n",
       "        FDR  N in query  Set size  N in ref.  \n",
       "0  7.36e-02           3        77         77  \n",
       "1  7.36e-02           3        89         88  \n",
       "2  7.36e-02           2        31         31  \n",
       "3  7.36e-02           2        27         27  \n",
       "4  1.01e-01           4       250        250  \n",
       "5  1.47e-01           4       354        353  \n",
       "6  1.49e-01           3       210        207  \n",
       "7  1.88e-01           2       104        102  \n",
       "8  2.19e-01           3       289        281  \n",
       "9  2.44e-01           2       202        201  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 1412 with EID: 1403\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 112 out of 431 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>3.24e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.41e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>autophagy</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.41e-02</td>\n",
       "      <td>11</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>7.73e-02</td>\n",
       "      <td>18</td>\n",
       "      <td>77</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>28</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Autophagy - other</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>25</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.79e-01</td>\n",
       "      <td>14</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>wild type</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>52</td>\n",
       "      <td>341</td>\n",
       "      <td>338</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                Ontology  \\\n",
       "0  SNARE interactions in vesicular transport          KEGG: Pathways   \n",
       "1                secondary metabolic process  GO: biological_process   \n",
       "2                                  autophagy  GO: biological_process   \n",
       "3                                Endocytosis          KEGG: Pathways   \n",
       "4                       cell differentiation  GO: biological_process   \n",
       "5                          Autophagy - other          KEGG: Pathways   \n",
       "6                  decreased spore viability   Dictybase: Phenotypes   \n",
       "7                                  wild type   Dictybase: Phenotypes   \n",
       "\n",
       "        FDR  N in query  Set size  N in ref.  \n",
       "0  3.24e-03          11        23         23  \n",
       "1  7.41e-02           4         5          5  \n",
       "2  7.41e-02          11        34         34  \n",
       "3  7.73e-02          18        77         75  \n",
       "4  1.17e-01          28       145        145  \n",
       "5  1.64e-01           8        25         24  \n",
       "6  1.79e-01          14        60         59  \n",
       "7  2.39e-01          52       341        338  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 564 with EID: 560\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 33 out of 141 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.45e-02</td>\n",
       "      <td>16</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.56e-02</td>\n",
       "      <td>10</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>4.44e-02</td>\n",
       "      <td>6</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>5.47e-02</td>\n",
       "      <td>5</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>12</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1  anatomical structure formation involved in morphogenesis   \n",
       "2                             Starch and sucrose metabolism   \n",
       "3                 SNARE interactions in vesicular transport   \n",
       "4                                      cell differentiation   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: cellular_component  2.45e-02          16       157        157  \n",
       "1  GO: biological_process  2.56e-02          10        74         74  \n",
       "2          KEGG: Pathways  4.44e-02           6        31         31  \n",
       "3          KEGG: Pathways  5.47e-02           5        23         23  \n",
       "4  GO: biological_process  2.11e-01          12       145        145  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_impulse+'DEacrossStages_summary_WT_noyem_batchrep.tsv',index_col=0,sep='\\t')\n",
    "stages=np.array([phenotype for phenotype in PHENOTYPES if phenotype in list(milestones.columns)])\n",
    "for stage_idx,stage in enumerate(stages):\n",
    "    \n",
    "    #Nonneighbouring stages\n",
    "    neigh_previous=1\n",
    "    neigh_next=1\n",
    "    #Assumes phenotypes order\n",
    "    if stage_idx==1:\n",
    "        neigh_previous=0\n",
    "    if stage_idx==0:\n",
    "        neigh_next=0\n",
    "    non_neighbours = stages[[(idx <stage_idx-neigh_previous or idx>stage_idx+neigh_next) \n",
    "                 for idx in range(len(stages)) ]]\n",
    "    #print(stage,':',non_neighbours)\n",
    "    \n",
    "    #Find genes up at stage and down at nonenighbouring\n",
    "    query_condition=stage+'==\"up\"'\n",
    "    for non_neigh in non_neighbours:\n",
    "        query_condition=query_condition+' & '+non_neigh+' ==\"down\"'\n",
    "    #print(query_condition)\n",
    "    \n",
    "    query_names=list(milestones.query(query_condition).index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
