{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Comparison of selected/DE genes across groups\n",
    "For each strain genes that have closely co-expressed neighbours or that are DE compared to Wt were selected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<module 'deR.enrichment_library' from '/home/karin/Documents/git/baylor-dicty/deR/enrichment_library.py'>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import sys\n",
    "from importlib import reload\n",
    "\n",
    "module_path = os.path.abspath(os.path.join('..'))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.insert(0,module_path)\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "\n",
    "from orangecontrib.bioinformatics.geneset.__init__ import (list_all, load_gene_sets)\n",
    "\n",
    "import deR.enrichment_library as enr\n",
    "reload(enr)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathSelected = '/home/karin/Documents/timeTrajectories/data/regulons/selected_genes/'\n",
    "pathDE = '/home/karin/Documents/timeTrajectories/data/deTime/de_time_impulse/'\n",
    "dataPath='/home/karin/Documents/timeTrajectories/data/RPKUM/combined/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all gene sets\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For conversion of gene names to EID\n",
    "genes = pd.read_csv(dataPath + 'mergedGenes_RPKUM.tsv', sep='\\t', index_col=0)\n",
    "all_gene_names= genes[(genes != 0).any(axis=1)].index\n",
    "NAME_EID=enr.name_genes_entrez(gene_names=all_gene_names, key_entrez=False)\n",
    "ALL_GENE_NAMES_EID=enr.convert_EID(genes=all_gene_names, name_EID=NAME_EID)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Loads different sets of selected genes:\n",
    "- DE of mutants compared to WT through time\n",
    "- Selected genes based on co-expression within strains (at different thresholds - e.g. top 1000 genes or based on strain specific similarity threshold)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename strain groups to work with pd query\n",
    "selected_genes=pd.read_table(pathSelected+'newGenes_noAll-removeSelf_selectedGenesByGroup_scalemean0std1_logTrue_kN6_splitStrain.tsv',\n",
    "                             index_col=0).T.rename(\n",
    "    {'1Ag-':'Ag','2LAg':'LAg','3TA':'TA','4CD':'CD','5WT':'WT','6SFB':'SFB','7PD':'PD'},axis=1)\n",
    "\n",
    "# Genes that were never expressed were never selected - thus they are not used in enrichment\n",
    "# selected_genes.loc[genes[(genes == 0).all(axis=1)].index,:].max().max()\n",
    "selected_genes=selected_genes.loc[all_gene_names,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename strain groups to work with pd query\n",
    "selected_genes1000=pd.read_table(pathSelected+'newGenes_noAll-removeSelf_selectedGenes1000ByGroup_scalemean0std1_logTrue_kN6_splitStrain.tsv',\n",
    "                             index_col=0).T.rename(\n",
    "    {'1Ag-':'Ag','2LAg':'LAg','3TA':'TA','4CD':'CD','5WT':'WT','6SFB':'SFB','7PD':'PD'},axis=1)\n",
    "\n",
    "\n",
    "selected_genes1000=selected_genes1000.loc[all_gene_names,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename strain groups to work with pd query\n",
    "selected_genes_DE=pd.read_table(pathDE+'DE_FDRe-11group.tsv',\n",
    "                             index_col=0).T.rename(\n",
    "    {'1Ag-_DE':'Ag','2LAg_DE':'LAg','3TA_DE':'TA','4CD_DE':'CD','5WT_DE':'WT','6SFB_DE':'SFB','7PD_DE':'PD'},axis=1)\n",
    "\n",
    "selected_genes_DE=selected_genes_DE.loc[all_gene_names,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def group_diff_enrichment(selected_genes:pd.DataFrame, condition:str,padj:float=0.25):\n",
    "    selected=selected_genes.query(condition).index\n",
    "    query_EID=enr.convert_EID(genes=selected, name_EID=NAME_EID)\n",
    "    print('*** Condition:',condition, ' selected:',len(selected),'with EID:',len(query_EID))\n",
    "    result=None\n",
    "    if len(query_EID) > 0:\n",
    "        enrichment=enr.gene_set_enrichment(query_EID, reference_EID=ALL_GENE_NAMES_EID, \n",
    "                                                padj_threshold=padj,\n",
    "                                                gene_sets_ontology=GENE_SETS_ONTOLOGY)\n",
    "        if len(enrichment)>0:\n",
    "            enrichment_display=list()\n",
    "            enrichment= sorted(enrichment, key=lambda data: data.padj)\n",
    "            for enriched in enrichment:\n",
    "                enrichment_display.append({'Gene set':enriched.gene_set.name,'Ontology':enriched.ontology,\n",
    "                                           'FDR':\"{:.2e}\".format(enriched.padj),'N in query':enriched.in_query})\n",
    "            result=pd.DataFrame(enrichment_display)\n",
    "    print('Enrichment:')\n",
    "    display(result)\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enrichment of selected genes\n",
    "Comparison of selected genes (co-expressed/DE) across grouops. For each group the comparison denotes the number of strains in which the gene must have been selected (based on co-expression or DE) to be now included in the enrichment analysis. This differs across groups, as they have different number of strains. For co-expression all strains are used, for DE only mutants are used (WT AX4 was used for reference)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Co-expressed genes\n",
    "Enrichment of groups of selected co-expressed genes present in some strain groups but noth the others (conditions shown below). Groups are ordered by developmental arrest in the comparisons. Group 6SFB is not used in comparisons as it seems to be an outlier from the developmental progression. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selected genes based on co-expression similarity threshold:\n",
      "*** Condition: Ag<=1 & LAg>=3 & TA>=1 & CD>=4 & WT>=1 & PD >=1  selected: 739 with EID: 735\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.76e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.80e-04</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.80e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.33e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.33e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Gene set                  Ontology       FDR  \\\n",
       "0  cell wall organization or biogenesis  (GO, biological_process)  1.76e-04   \n",
       "1                  extracellular region  (GO, cellular_component)  1.80e-04   \n",
       "2        aberrant spore coat morphology   (Dictybase, Phenotypes)  1.80e-04   \n",
       "3      external encapsulating structure  (GO, cellular_component)  3.33e-02   \n",
       "4                             cell wall  (GO, cellular_component)  3.33e-02   \n",
       "5      anatomical structure development  (GO, biological_process)  1.02e-01   \n",
       "\n",
       "   N in query  \n",
       "0           8  \n",
       "1          26  \n",
       "2           7  \n",
       "3           5  \n",
       "4           5  \n",
       "5          41  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 1000 co-expressed genes:\n",
      "*** Condition: Ag<=1 & LAg>=3 & TA>=1 & CD>=4 & WT>=1 & PD >=1  selected: 18 with EID: 18\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.89e-10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.89e-10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.89e-10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.68e-09</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.58e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>protein-containing complex</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.35e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>biosynthetic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.35e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular nitrogen compound metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.81e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased cell fate determination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.30e-03</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant upper cup formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.30e-03</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased cell fate commitment</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.30e-03</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cytoplasm</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.70e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>organelle</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.62e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased spore size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.66e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>increased stalk width</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.82e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>increased prestalk 0 cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.82e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>increased number of aggregates</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.94e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.94e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.14e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>intracellular</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.18e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant culminant morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.49e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.16e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.27e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>decreased cell-cell adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.61e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.91e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.01e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>cellular_component</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.51e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.32e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                  structural constituent of ribosome   \n",
       "1                                            ribosome   \n",
       "2                                            Ribosome   \n",
       "3                        structural molecule activity   \n",
       "4                                         translation   \n",
       "5                          protein-containing complex   \n",
       "6                                biosynthetic process   \n",
       "7        cellular nitrogen compound metabolic process   \n",
       "8                   decreased cell fate determination   \n",
       "9                        aberrant upper cup formation   \n",
       "10                     decreased cell fate commitment   \n",
       "11                                          cytoplasm   \n",
       "12                                          organelle   \n",
       "13                               decreased spore size   \n",
       "14                              increased stalk width   \n",
       "15          increased prestalk 0 cell differentiation   \n",
       "16                     increased number of aggregates   \n",
       "17            increased prespore cell differentiation   \n",
       "18                             aberrant cell motility   \n",
       "19                                      intracellular   \n",
       "20                      aberrant culminant morphology   \n",
       "21                                               cell   \n",
       "22                                       rRNA binding   \n",
       "23                       decreased cell-cell adhesion   \n",
       "24                                        RNA binding   \n",
       "25                                 biological_process   \n",
       "26                                 cellular_component   \n",
       "27                          decreased spore viability   \n",
       "28                                 aberrant streaming   \n",
       "29  anatomical structure formation involved in mor...   \n",
       "30                                      cell adhesion   \n",
       "31                                 molecular_function   \n",
       "32                               cell differentiation   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, molecular_function)  5.89e-10           7  \n",
       "1   (GO, cellular_component)  5.89e-10           7  \n",
       "2           (KEGG, Pathways)  5.89e-10           7  \n",
       "3   (GO, molecular_function)  1.68e-09           7  \n",
       "4   (GO, biological_process)  1.58e-08           7  \n",
       "5   (GO, cellular_component)  1.35e-04           7  \n",
       "6   (GO, biological_process)  1.35e-04           7  \n",
       "7   (GO, biological_process)  2.81e-04           7  \n",
       "8    (Dictybase, Phenotypes)  5.30e-03           1  \n",
       "9    (Dictybase, Phenotypes)  5.30e-03           1  \n",
       "10   (Dictybase, Phenotypes)  5.30e-03           1  \n",
       "11  (GO, cellular_component)  6.70e-03           7  \n",
       "12  (GO, cellular_component)  8.62e-03           7  \n",
       "13   (Dictybase, Phenotypes)  1.66e-02           1  \n",
       "14   (Dictybase, Phenotypes)  1.82e-02           1  \n",
       "15   (Dictybase, Phenotypes)  1.82e-02           1  \n",
       "16   (Dictybase, Phenotypes)  1.94e-02           1  \n",
       "17   (Dictybase, Phenotypes)  1.94e-02           1  \n",
       "18   (Dictybase, Phenotypes)  2.14e-02           1  \n",
       "19  (GO, cellular_component)  2.18e-02           7  \n",
       "20   (Dictybase, Phenotypes)  2.49e-02           1  \n",
       "21  (GO, cellular_component)  3.16e-02           7  \n",
       "22  (GO, molecular_function)  3.27e-02           1  \n",
       "23   (Dictybase, Phenotypes)  3.61e-02           1  \n",
       "24  (GO, molecular_function)  3.91e-02           2  \n",
       "25  (GO, biological_process)  7.01e-02           9  \n",
       "26  (GO, cellular_component)  7.51e-02           9  \n",
       "27   (Dictybase, Phenotypes)  1.18e-01           1  \n",
       "28   (Dictybase, Phenotypes)  1.37e-01           1  \n",
       "29  (GO, biological_process)  1.37e-01           1  \n",
       "30  (GO, biological_process)  1.48e-01           1  \n",
       "31  (GO, molecular_function)  1.55e-01           8  \n",
       "32  (GO, biological_process)  2.32e-01           1  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selected genes based on co-expression similarity threshold:\n",
      "*** Condition: Ag<=1 & LAg<=1 & TA>=1 & CD>=4 & WT>=1 & PD >=1  selected: 157 with EID: 154\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 1000 co-expressed genes:\n",
      "*** Condition: Ag<=1 & LAg<=1 & TA>=1 & CD>=4 & WT>=1 & PD >=1  selected: 73 with EID: 73\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.25e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.15e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.40e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.40e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Glyoxylate and dicarboxylate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.57e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.57e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.26e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.26e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.87e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.15e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>increased spore sensitivity to hydrogen peroxide</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.43e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>abolished stalk formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.98e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>increased cell-substrate adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.98e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>increased sensitivity to UV light</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.38e-02</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>extracellular matrix organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant stalk formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Peroxisome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.67e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>delayed tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>aberrant cellular metabolic process</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>aberrant cAMP signaling during aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>aberrant response to hyperosmotic stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>mislocalized gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>Histidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>cellular_component</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.33e-01</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.35e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                      aberrant spore coat morphology   \n",
       "1                    anatomical structure development   \n",
       "2   anatomical structure formation involved in mor...   \n",
       "3                       Starch and sucrose metabolism   \n",
       "4             Glyoxylate and dicarboxylate metabolism   \n",
       "5                                  Metabolic pathways   \n",
       "6                    external encapsulating structure   \n",
       "7                                           cell wall   \n",
       "8                cell wall organization or biogenesis   \n",
       "9                           decreased spore viability   \n",
       "10   increased spore sensitivity to hydrogen peroxide   \n",
       "11                          abolished stalk formation   \n",
       "12                        abolished cellulose binding   \n",
       "13                            decreased cell motility   \n",
       "14                               cell differentiation   \n",
       "15                             Glutathione metabolism   \n",
       "16                               extracellular region   \n",
       "17                  increased cell-substrate adhesion   \n",
       "18                  increased sensitivity to UV light   \n",
       "19                  extracellular matrix organization   \n",
       "20                                  Carbon metabolism   \n",
       "21                           aberrant stalk formation   \n",
       "22                        aberrant cell-cell adhesion   \n",
       "23                   abolished cellulose biosynthesis   \n",
       "24                                 biological_process   \n",
       "25                                         Peroxisome   \n",
       "26               abolished stalk cell differentiation   \n",
       "27                               extracellular matrix   \n",
       "28                                  catabolic process   \n",
       "29                 transmembrane transporter activity   \n",
       "30                                    plasma membrane   \n",
       "31                              delayed tip formation   \n",
       "32                aberrant cellular metabolic process   \n",
       "33         aberrant cAMP signaling during aggregation   \n",
       "34                                delayed aggregation   \n",
       "35           aberrant response to hyperosmotic stress   \n",
       "36                       mislocalized gene expression   \n",
       "37                                cell-cell signaling   \n",
       "38                               Histidine metabolism   \n",
       "39                                 cellular_component   \n",
       "40                               aberrant culmination   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0    (Dictybase, Phenotypes)  4.25e-03           3  \n",
       "1   (GO, biological_process)  1.15e-02          10  \n",
       "2   (GO, biological_process)  2.40e-02           4  \n",
       "3           (KEGG, Pathways)  2.40e-02           3  \n",
       "4           (KEGG, Pathways)  2.57e-02           3  \n",
       "5           (KEGG, Pathways)  2.57e-02          11  \n",
       "6   (GO, cellular_component)  3.26e-02           2  \n",
       "7   (GO, cellular_component)  3.26e-02           2  \n",
       "8   (GO, biological_process)  3.87e-02           2  \n",
       "9    (Dictybase, Phenotypes)  5.15e-02           3  \n",
       "10   (Dictybase, Phenotypes)  5.43e-02           1  \n",
       "11   (Dictybase, Phenotypes)  7.45e-02           1  \n",
       "12   (Dictybase, Phenotypes)  7.45e-02           1  \n",
       "13   (Dictybase, Phenotypes)  7.45e-02           3  \n",
       "14  (GO, biological_process)  7.45e-02           4  \n",
       "15          (KEGG, Pathways)  7.45e-02           2  \n",
       "16  (GO, cellular_component)  7.98e-02           4  \n",
       "17   (Dictybase, Phenotypes)  7.98e-02           2  \n",
       "18   (Dictybase, Phenotypes)  9.38e-02           1  \n",
       "19  (GO, biological_process)  1.13e-01           1  \n",
       "20          (KEGG, Pathways)  1.13e-01           3  \n",
       "21   (Dictybase, Phenotypes)  1.18e-01           1  \n",
       "22   (Dictybase, Phenotypes)  1.18e-01           1  \n",
       "23   (Dictybase, Phenotypes)  1.18e-01           1  \n",
       "24  (GO, biological_process)  1.18e-01          29  \n",
       "25          (KEGG, Pathways)  1.37e-01           2  \n",
       "26   (Dictybase, Phenotypes)  1.52e-01           1  \n",
       "27  (GO, cellular_component)  1.67e-01           1  \n",
       "28  (GO, biological_process)  1.92e-01           4  \n",
       "29  (GO, molecular_function)  1.92e-01           3  \n",
       "30  (GO, cellular_component)  1.92e-01           4  \n",
       "31   (Dictybase, Phenotypes)  1.92e-01           1  \n",
       "32   (Dictybase, Phenotypes)  1.92e-01           1  \n",
       "33   (Dictybase, Phenotypes)  1.92e-01           1  \n",
       "34   (Dictybase, Phenotypes)  1.92e-01           2  \n",
       "35   (Dictybase, Phenotypes)  1.92e-01           1  \n",
       "36   (Dictybase, Phenotypes)  1.92e-01           1  \n",
       "37  (GO, biological_process)  1.92e-01           1  \n",
       "38          (KEGG, Pathways)  1.92e-01           1  \n",
       "39  (GO, cellular_component)  2.33e-01          27  \n",
       "40   (Dictybase, Phenotypes)  2.35e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selected genes based on co-expression similarity threshold:\n",
      "*** Condition: Ag<=1 & LAg<=1 & TA<=1 & CD>=4 & WT>=1 & PD >=1  selected: 87 with EID: 86\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>decreased actin nucleation on plasma membranes</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>decreased lateral pseudopod traction</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>increased endocytic recycling</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         Gene set                 Ontology  \\\n",
       "0                          precocious aggregation  (Dictybase, Phenotypes)   \n",
       "1  decreased actin nucleation on plasma membranes  (Dictybase, Phenotypes)   \n",
       "2            decreased lateral pseudopod traction  (Dictybase, Phenotypes)   \n",
       "3                   increased endocytic recycling  (Dictybase, Phenotypes)   \n",
       "\n",
       "        FDR  N in query  \n",
       "0  1.60e-01           2  \n",
       "1  1.60e-01           1  \n",
       "2  1.60e-01           1  \n",
       "3  1.60e-01           1  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 1000 co-expressed genes:\n",
      "*** Condition: Ag<=1 & LAg<=1 & TA<=1 & CD>=4 & WT>=1 & PD >=1  selected: 14 with EID: 14\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selected genes based on co-expression similarity threshold:\n",
      "*** Condition: Ag<=1 & LAg<=1 & TA<=1 & CD<=1 & WT>=1 & PD >=1  selected: 256 with EID: 250\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 1000 co-expressed genes:\n",
      "*** Condition: Ag<=1 & LAg<=1 & TA<=1 & CD<=1 & WT>=1 & PD >=1  selected: 217 with EID: 214\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant response to osmotic stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Gene set                  Ontology       FDR  \\\n",
       "0                    extracellular region  (GO, cellular_component)  2.12e-01   \n",
       "1                            cytoskeleton  (GO, cellular_component)  2.12e-01   \n",
       "2     aberrant response to osmotic stress   (Dictybase, Phenotypes)  2.12e-01   \n",
       "3  Various types of N-glycan biosynthesis          (KEGG, Pathways)  2.12e-01   \n",
       "\n",
       "   N in query  \n",
       "0           8  \n",
       "1           9  \n",
       "2           2  \n",
       "3           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for condition in ['Ag<=1 & LAg>=3 & TA>=1 & CD>=4 & WT>=1 & PD >=1',\n",
    "                 'Ag<=1 & LAg<=1 & TA>=1 & CD>=4 & WT>=1 & PD >=1',\n",
    "                 'Ag<=1 & LAg<=1 & TA<=1 & CD>=4 & WT>=1 & PD >=1',\n",
    "                 'Ag<=1 & LAg<=1 & TA<=1 & CD<=1 & WT>=1 & PD >=1']:\n",
    "    print('Selected genes based on co-expression similarity threshold:')\n",
    "    group_diff_enrichment(selected_genes,condition)\n",
    "    print('Top 1000 co-expressed genes:')\n",
    "    group_diff_enrichment(selected_genes1000,condition)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DE genes\n",
    "Comparison of DE genes (mutant vs WT through time) within the strain groups. Retained are genes DE in all (or almost all) strains of individual groups. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*** Condition: Ag==4  selected: 834 with EID: 827\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.46e-06</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.46e-06</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.80e-06</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.25e-03</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>delayed tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.99e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.99e-03</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.15e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.62e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.79e-02</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.48e-02</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.48e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>increased prestalk 0 cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.16e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>decreased catalytic activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>aberrant defense response to bacterium</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.31e-01</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>abolished SDF production</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>thin standing slug</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>increased cell-substrate adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>abolished DIF production</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>aberrant chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.98e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.98e-01</td>\n",
       "      <td>265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>AGE-RAGE signaling pathway in diabetic complic...</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>cellular_component</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                extracellular region   \n",
       "1                    anatomical structure development   \n",
       "2                                cell differentiation   \n",
       "3   anatomical structure formation involved in mor...   \n",
       "4                               delayed tip formation   \n",
       "5                           decreased spore viability   \n",
       "6                abolished stalk cell differentiation   \n",
       "7                      aberrant spore coat morphology   \n",
       "8                       Starch and sucrose metabolism   \n",
       "9                               abolished sporulation   \n",
       "10                                    plasma membrane   \n",
       "11                  aberrant fruiting body morphology   \n",
       "12               cell wall organization or biogenesis   \n",
       "13          increased prestalk 0 cell differentiation   \n",
       "14                       decreased catalytic activity   \n",
       "15                        abolished cellulose binding   \n",
       "16             aberrant defense response to bacterium   \n",
       "17                               aberrant development   \n",
       "18                                          wild type   \n",
       "19                   external encapsulating structure   \n",
       "20                                          cell wall   \n",
       "21                           abolished SDF production   \n",
       "22           development arrests at loose mound stage   \n",
       "23                                cell-cell signaling   \n",
       "24                                 thin standing slug   \n",
       "25                  increased cell-substrate adhesion   \n",
       "26                           abolished DIF production   \n",
       "27                        aberrant chemotaxis to cAMP   \n",
       "28                                 biological_process   \n",
       "29                                       reproduction   \n",
       "30                               aberrant aggregation   \n",
       "31  AGE-RAGE signaling pathway in diabetic complic...   \n",
       "32                                 cellular_component   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  2.46e-06          32  \n",
       "1   (GO, biological_process)  2.46e-06          62  \n",
       "2   (GO, biological_process)  2.80e-06          30  \n",
       "3   (GO, biological_process)  2.25e-03          16  \n",
       "4    (Dictybase, Phenotypes)  6.99e-03           6  \n",
       "5    (Dictybase, Phenotypes)  6.99e-03          13  \n",
       "6    (Dictybase, Phenotypes)  3.15e-02           4  \n",
       "7    (Dictybase, Phenotypes)  3.20e-02           5  \n",
       "8           (KEGG, Pathways)  3.20e-02           8  \n",
       "9    (Dictybase, Phenotypes)  3.62e-02           7  \n",
       "10  (GO, cellular_component)  4.79e-02          31  \n",
       "11   (Dictybase, Phenotypes)  6.48e-02          15  \n",
       "12  (GO, biological_process)  6.48e-02           5  \n",
       "13   (Dictybase, Phenotypes)  7.16e-02           3  \n",
       "14   (Dictybase, Phenotypes)  1.03e-01           5  \n",
       "15   (Dictybase, Phenotypes)  1.04e-01           2  \n",
       "16   (Dictybase, Phenotypes)  1.12e-01           3  \n",
       "17   (Dictybase, Phenotypes)  1.28e-01           9  \n",
       "18   (Dictybase, Phenotypes)  1.31e-01          35  \n",
       "19  (GO, cellular_component)  1.37e-01           4  \n",
       "20  (GO, cellular_component)  1.37e-01           4  \n",
       "21   (Dictybase, Phenotypes)  1.37e-01           4  \n",
       "22   (Dictybase, Phenotypes)  1.37e-01           5  \n",
       "23  (GO, biological_process)  1.37e-01           4  \n",
       "24   (Dictybase, Phenotypes)  1.81e-01           2  \n",
       "25   (Dictybase, Phenotypes)  1.81e-01           6  \n",
       "26   (Dictybase, Phenotypes)  1.81e-01           2  \n",
       "27   (Dictybase, Phenotypes)  1.98e-01           6  \n",
       "28  (GO, biological_process)  1.98e-01         265  \n",
       "29  (GO, biological_process)  1.99e-01          14  \n",
       "30   (Dictybase, Phenotypes)  2.27e-01           9  \n",
       "31          (KEGG, Pathways)  2.27e-01           5  \n",
       "32  (GO, cellular_component)  2.29e-01         268  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*** Condition: LAg==4  selected: 1888 with EID: 1884\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.24e-04</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.58e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.58e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.58e-03</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.58e-03</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.58e-03</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.83e-03</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>extracellular matrix organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.00e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.00e-02</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.71e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.81e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.23e-02</td>\n",
       "      <td>594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.52e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.73e-01</td>\n",
       "      <td>598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                extracellular region   \n",
       "1                    external encapsulating structure   \n",
       "2                                           cell wall   \n",
       "3                           decreased spore viability   \n",
       "4   anatomical structure formation involved in mor...   \n",
       "5                                cell differentiation   \n",
       "6                    anatomical structure development   \n",
       "7                   extracellular matrix organization   \n",
       "8                                 ribosome biogenesis   \n",
       "9                      aberrant spore coat morphology   \n",
       "10                      Starch and sucrose metabolism   \n",
       "11                                 biological_process   \n",
       "12               cell wall organization or biogenesis   \n",
       "13                                 molecular_function   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  5.24e-04          48  \n",
       "1   (GO, cellular_component)  1.58e-03           9  \n",
       "2   (GO, cellular_component)  1.58e-03           9  \n",
       "3    (Dictybase, Phenotypes)  1.58e-03          23  \n",
       "4   (GO, biological_process)  1.58e-03          26  \n",
       "5   (GO, biological_process)  1.58e-03          42  \n",
       "6   (GO, biological_process)  5.83e-03          96  \n",
       "7   (GO, biological_process)  3.00e-02           4  \n",
       "8   (GO, biological_process)  3.00e-02          20  \n",
       "9    (Dictybase, Phenotypes)  3.71e-02           7  \n",
       "10          (KEGG, Pathways)  5.81e-02          12  \n",
       "11  (GO, biological_process)  7.23e-02         594  \n",
       "12  (GO, biological_process)  1.52e-01           7  \n",
       "13  (GO, molecular_function)  1.73e-01         598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*** Condition: TA==2  selected: 444 with EID: 442\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.99e-04</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.41e-03</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Amino sugar and nucleotide sugar metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.32e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.40e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.68e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.39e-02</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.04e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.22e-02</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.65e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.65e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>aberrant response to abiotic stimulus</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Alanine</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Pantothenate and CoA biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>aberrant positive regulation of chemorepellant...</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Glyoxylate and dicarboxylate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Cyanoamino acid metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.28e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>decreased rate of endocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>decreased sorting to prestalk region</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>Pyruvate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Pentose and glucuronate interconversions</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>aberrant plasma membrane organization and biog...</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>abolished small molecule binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>increased prestalk B cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>increased protein kinase B activation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>increased spore sensitivity to hydrogen peroxide</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased lipid binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>decreased protein localization to cytoplasm</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>abolished response to SDF-1</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>decreased phosphorylation of actin</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>decreased catalase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>Nitrogen metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>Sphingolipid metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>Biosynthesis of antibiotics</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                  Metabolic pathways   \n",
       "1                       Starch and sucrose metabolism   \n",
       "2         Amino sugar and nucleotide sugar metabolism   \n",
       "3                                Galactose metabolism   \n",
       "4   anatomical structure formation involved in mor...   \n",
       "5                                cell differentiation   \n",
       "6                      carbohydrate metabolic process   \n",
       "7                    anatomical structure development   \n",
       "8               Biosynthesis of secondary metabolites   \n",
       "9                                           Phagosome   \n",
       "10              aberrant response to abiotic stimulus   \n",
       "11                                            Alanine   \n",
       "12                  Pantothenate and CoA biosynthesis   \n",
       "13                                 hydrolase activity   \n",
       "14                               extracellular region   \n",
       "15  aberrant positive regulation of chemorepellant...   \n",
       "16            Glyoxylate and dicarboxylate metabolism   \n",
       "17                         Cyanoamino acid metabolism   \n",
       "18                           aberrant cell morphology   \n",
       "19                      decreased rate of endocytosis   \n",
       "20               decreased sorting to prestalk region   \n",
       "21                          decreased spore viability   \n",
       "22                                Pyruvate metabolism   \n",
       "23           Pentose and glucuronate interconversions   \n",
       "24                                 molecular_function   \n",
       "25  aberrant plasma membrane organization and biog...   \n",
       "26                   abolished small molecule binding   \n",
       "27          increased prestalk B cell differentiation   \n",
       "28              increased protein kinase B activation   \n",
       "29                            delayed gene expression   \n",
       "30   increased spore sensitivity to hydrogen peroxide   \n",
       "31                            decreased lipid binding   \n",
       "32        decreased protein localization to cytoplasm   \n",
       "33                        abolished response to SDF-1   \n",
       "34                 decreased phosphorylation of actin   \n",
       "35                        decreased catalase activity   \n",
       "36                                          transport   \n",
       "37                                 biological_process   \n",
       "38                   small molecule metabolic process   \n",
       "39                                Nitrogen metabolism   \n",
       "40                            Sphingolipid metabolism   \n",
       "41                                  Carbon metabolism   \n",
       "42                        Biosynthesis of antibiotics   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0           (KEGG, Pathways)  4.99e-04          48  \n",
       "1           (KEGG, Pathways)  1.41e-03           8  \n",
       "2           (KEGG, Pathways)  3.32e-03           7  \n",
       "3           (KEGG, Pathways)  6.40e-03           5  \n",
       "4   (GO, biological_process)  1.68e-02          10  \n",
       "5   (GO, biological_process)  3.39e-02          14  \n",
       "6   (GO, biological_process)  4.04e-02          10  \n",
       "7   (GO, biological_process)  7.22e-02          28  \n",
       "8           (KEGG, Pathways)  8.65e-02          18  \n",
       "9           (KEGG, Pathways)  8.65e-02           7  \n",
       "10   (Dictybase, Phenotypes)  1.87e-01           2  \n",
       "11          (KEGG, Pathways)  1.87e-01           4  \n",
       "12          (KEGG, Pathways)  1.87e-01           3  \n",
       "13  (GO, molecular_function)  1.92e-01           4  \n",
       "14  (GO, cellular_component)  1.92e-01          12  \n",
       "15   (Dictybase, Phenotypes)  1.92e-01           2  \n",
       "16          (KEGG, Pathways)  1.92e-01           5  \n",
       "17          (KEGG, Pathways)  1.92e-01           2  \n",
       "18   (Dictybase, Phenotypes)  2.28e-01           5  \n",
       "19   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "20   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "21   (Dictybase, Phenotypes)  2.29e-01           6  \n",
       "22          (KEGG, Pathways)  2.29e-01           4  \n",
       "23          (KEGG, Pathways)  2.29e-01           2  \n",
       "24  (GO, molecular_function)  2.42e-01         148  \n",
       "25   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "26   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "27   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "28   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "29   (Dictybase, Phenotypes)  2.42e-01           3  \n",
       "30   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "31   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "32   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "33   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "34   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "35   (Dictybase, Phenotypes)  2.42e-01           1  \n",
       "36  (GO, biological_process)  2.42e-01          28  \n",
       "37  (GO, biological_process)  2.42e-01         146  \n",
       "38  (GO, biological_process)  2.42e-01          20  \n",
       "39          (KEGG, Pathways)  2.42e-01           2  \n",
       "40          (KEGG, Pathways)  2.42e-01           3  \n",
       "41          (KEGG, Pathways)  2.42e-01           8  \n",
       "42          (KEGG, Pathways)  2.42e-01          12  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*** Condition: CD>=4  selected: 27 with EID: 27\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*** Condition: PD==2  selected: 2055 with EID: 2045\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.77e-10</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.68e-03</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.68e-03</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.34e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.34e-02</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.34e-02</td>\n",
       "      <td>143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>One carbon pool by folate</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.41e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.70e-02</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Amino sugar and nucleotide sugar metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.36e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.36e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Valine</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.46e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.29e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Nitrogen metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Gene set  \\\n",
       "0                                extracellular region   \n",
       "1   anatomical structure formation involved in mor...   \n",
       "2                       Starch and sucrose metabolism   \n",
       "3                                  hydrolase activity   \n",
       "4                                cell differentiation   \n",
       "5                                  Metabolic pathways   \n",
       "6                           One carbon pool by folate   \n",
       "7                    anatomical structure development   \n",
       "8         Amino sugar and nucleotide sugar metabolism   \n",
       "9                                Galactose metabolism   \n",
       "10                                             Valine   \n",
       "11                   small molecule metabolic process   \n",
       "12                        secondary metabolic process   \n",
       "13                                 hydrolase activity   \n",
       "14                                Nitrogen metabolism   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  9.77e-10          63  \n",
       "1   (GO, biological_process)  6.68e-03          27  \n",
       "2           (KEGG, Pathways)  6.68e-03          15  \n",
       "3   (GO, molecular_function)  2.34e-02          17  \n",
       "4   (GO, biological_process)  2.34e-02          41  \n",
       "5           (KEGG, Pathways)  2.34e-02         143  \n",
       "6           (KEGG, Pathways)  2.41e-02           8  \n",
       "7   (GO, biological_process)  2.70e-02          99  \n",
       "8           (KEGG, Pathways)  3.36e-02          12  \n",
       "9           (KEGG, Pathways)  3.36e-02           8  \n",
       "10          (KEGG, Pathways)  9.46e-02          13  \n",
       "11  (GO, biological_process)  1.28e-01          78  \n",
       "12  (GO, biological_process)  1.29e-01           4  \n",
       "13  (GO, molecular_function)  1.54e-01          10  \n",
       "14          (KEGG, Pathways)  1.54e-01           5  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*** Condition: SFB==2  selected: 1665 with EID: 1663\n",
      "Enrichment:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure formation involved in mor...</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.52e-03</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.52e-03</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.92e-02</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>tRNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.92e-02</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.49e-02</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.49e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.63e-02</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.75e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>abolished DIF production</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.25e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0  anatomical structure formation involved in mor...   \n",
       "1                               cell differentiation   \n",
       "2                                    ligase activity   \n",
       "3                             tRNA metabolic process   \n",
       "4                               extracellular region   \n",
       "5                      Starch and sucrose metabolism   \n",
       "6                   anatomical structure development   \n",
       "7                          decreased spore viability   \n",
       "8                           abolished DIF production   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, biological_process)  1.52e-03          25  \n",
       "1  (GO, biological_process)  1.52e-03          40  \n",
       "2  (GO, molecular_function)  2.92e-02          23  \n",
       "3  (GO, biological_process)  2.92e-02          21  \n",
       "4  (GO, cellular_component)  3.49e-02          37  \n",
       "5          (KEGG, Pathways)  3.49e-02          12  \n",
       "6  (GO, biological_process)  5.63e-02          82  \n",
       "7   (Dictybase, Phenotypes)  8.75e-02          17  \n",
       "8   (Dictybase, Phenotypes)  1.25e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for condition in ['Ag==4',\n",
    "                 'LAg==4',\n",
    "                 'TA==2',\n",
    "                  # Use 4 instead of 5 (all strains) for CD as gtaG has almost no DE genes\n",
    "                 'CD>=4',\n",
    "                 'PD==2',\n",
    "                 'SFB==2']:\n",
    "    group_diff_enrichment(selected_genes_DE,condition)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The number of DE genes present across 4CD samples is very small. This happens even after reducing the presence threshold due to gtaG having unusually low N of DE genes. In DEStages.R.ipynb it is shown that DE genes between 4CD strains seem to be more diverse than in other groups, possibly explaining the low overlap. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
