{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from importlib import reload\n",
    "\n",
    "module_path = os.path.abspath(os.path.join('..'))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.insert(0,module_path)\n",
    "\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "pd.set_option('max_colwidth', 70)\n",
    "pd.set_option('display.max_rows', 500)\n",
    "import string\n",
    "import numpy as np\n",
    "\n",
    "from orangecontrib.bioinformatics.geneset.__init__ import list_all\n",
    "\n",
    "import jupyter_functions as jf\n",
    "reload(jf)\n",
    "import jupyter_functions as jf\n",
    "from networks.functionsDENet import loadPickle, savePickle\n",
    "import enrichment_library as enr\n",
    "reload(enr)\n",
    "import enrichment_library as enr\n",
    "from stages_DE.stages_library import PHENOTYPES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Path for expression data (mergedGenes_RPKUM.tsv)\n",
    "dataPath='/home/karin/Documents/timeTrajectories/data/RPKUM/combined/'\n",
    "path_abberant='/home/karin/Documents/timeTrajectories/data/regulons/selected_genes/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For conversion of gene names to EID\n",
    "genes = pd.read_csv(dataPath + 'mergedGenes_RPKUM.tsv', sep='\\t', index_col=0)\n",
    "all_gene_names= genes[(genes != 0).any(axis=1)].index\n",
    "NAME_EID=enr.name_genes_entrez(gene_names=all_gene_names, key_entrez=False)\n",
    "ALL_GENE_NAMES_EID=enr.convert_EID(genes=all_gene_names, name_EID=NAME_EID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "def group_diff_enrichment(query_names,group:str,padj:float=0.25,min_overlap:int=None,\n",
    "                          use_annotated_genes:bool=False):\n",
    "    #Displays only gene sets that have overlap with query greater or equal to min_overlap \n",
    "    #For p value and padj calculation uses alll that have overlap >=1 } from gene_set_enrichment\n",
    "    \"\"\"\n",
    "    :param use_annotated_genes: if True use for reference and query  only genes that have at \n",
    "    least one gene set annotation\n",
    "    \"\"\"\n",
    "    query_EID=enr.convert_EID(genes=query_names, name_EID=NAME_EID)\n",
    "    print('***  '+group+' selected:',len(query_names),'with EID:',len(query_EID))\n",
    "    \n",
    "    reference_gene_eids=ALL_GENE_NAMES_EID.copy()\n",
    "    query_eids=query_EID.copy()\n",
    "    \n",
    "    if use_annotated_genes:\n",
    "        gene_sets_genes=set()\n",
    "        for gene_set_name, gene_sets in GENE_SETS_ONTOLOGY.items():\n",
    "            for gene_set in gene_sets:\n",
    "                gene_sets_genes.update(gene_set.genes)\n",
    "        reference_gene_eids=set(reference_gene_eids) & gene_sets_genes\n",
    "        query_eids=set(query_eids) & gene_sets_genes\n",
    "        \n",
    "        query_annotated_reatio='NA'\n",
    "        if len(query_EID)>0:\n",
    "            query_annotated_reatio=round(len(query_eids)/len(query_EID),2)\n",
    "        print('Ratio of genes annotated with a gene set in reference',\n",
    "              round(len(reference_gene_eids)/len(ALL_GENE_NAMES_EID),2),\n",
    "             'and query',query_annotated_reatio)\n",
    "    \n",
    "    query_in_enriched=set()\n",
    "    result=None\n",
    "    if len(query_eids) > 0:\n",
    "        enrichment=enr.gene_set_enrichment(query_eids, reference_EID=reference_gene_eids, \n",
    "                                                padj_threshold=padj,min_overlap=min_overlap,\n",
    "                                                gene_sets_ontology=GENE_SETS_ONTOLOGY)\n",
    "        if len(enrichment)>0:\n",
    "            enrichment_display=list()\n",
    "            enrichment= sorted(enrichment, key=lambda data: data.padj)\n",
    "            for enriched in enrichment:\n",
    "                query_in_enriched.update(enriched.gene_set.genes & query_eids)\n",
    "                enrichment_display.append({'Gene set':enriched.gene_set.name,\n",
    "                                           'Ontology':enriched.ontology[0]+': '+enriched.ontology[1],\n",
    "                                           'FDR':\"{:.2e}\".format(enriched.padj),'N in query':enriched.in_query,\n",
    "                                          'Set size':len(enriched.gene_set.genes),\n",
    "                                                          'N in ref.':enriched.in_reference})\n",
    "            result=pd.DataFrame(enrichment_display)\n",
    "    print('Enrichment at FDR: '+str(padj)+' and min query - gene set overlap',str(min_overlap))\n",
    "    print('N query genes in displayed gene sets:',len(query_in_enriched),'out of', len(query_eids),\n",
    "          'query genes used for enrichment calculation.')\n",
    "    display(result)\n",
    "    print('\\n')\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "comparison_order=('agg-','dis','tag','cud')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enrichment report"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Filter results based on FDR (<=0.05) and Difference between group means (>=0.3)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "abberant=pd.read_table(\n",
    "    path_abberant+'comparisonsAvgSimsSingle2STDAny0.2_lessComparisons2_AX4basedNeigh_u-less_newGenes_noAll-removeZeroRep_simsDict_scalemean0std1_logTrue_kN11_splitStrain.tsv',\n",
    "                       index_col=0,sep='\\t')\n",
    "abberant=abberant[(abberant['Difference mean']>=0.3) & (abberant['FDR']<=0.05)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### KEGG, DictyPhenotypes, GO slims"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all gene sets\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=True,set_sizes=(5,500))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Min size 5 Max size 440\n",
      "N sets {('Dictybase', 'Phenotypes'): 209, ('GO', 'molecular_function'): 35, ('GO', 'biological_process'): 52, ('KEGG', 'Pathways'): 96, ('GO', 'cellular_component'): 24}\n"
     ]
    }
   ],
   "source": [
    "# Find out max/min gene set sizes and N sets\n",
    "max_n=0\n",
    "min_n=np.inf\n",
    "for k,v in GENE_SETS_ONTOLOGY.items():\n",
    "    for s in v:\n",
    "        n=len(s.genes)\n",
    "        if n<min_n:\n",
    "            min_n=n\n",
    "        if n>max_n:\n",
    "            max_n=n\n",
    "print('Min size',min_n,'Max size',max_n)  \n",
    "print('N sets',{k:len(v) for k,v in GENE_SETS_ONTOLOGY.items()})   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  agg- selected: 162 with EID: 160\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 22 out of 45 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.05e-05</td>\n",
       "      <td>11</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.60e-03</td>\n",
       "      <td>15</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.83e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.66e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.66e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.66e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.03e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.94e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>aberrant stalk morphogenesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>5.35e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>mislocalized gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>8.87e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.18e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.18e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>197</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       cell differentiation   \n",
       "1                           anatomical structure development   \n",
       "2   anatomical structure formation involved in morphogenesis   \n",
       "3                             aberrant spore coat morphology   \n",
       "4                                                  cell wall   \n",
       "5                           external encapsulating structure   \n",
       "6                       cell wall organization or biogenesis   \n",
       "7                                 multiple tips in culminant   \n",
       "8                               aberrant stalk morphogenesis   \n",
       "9                                  decreased spore viability   \n",
       "10                              mislocalized gene expression   \n",
       "11                                     decreased sporulation   \n",
       "12                                 cytoskeleton organization   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  5.05e-05          11       145        145  \n",
       "1   GO: biological_process  3.60e-03          15       440        438  \n",
       "2   GO: biological_process  6.83e-03           6        74         74  \n",
       "3    Dictybase: Phenotypes  7.66e-03           3        12         12  \n",
       "4   GO: cellular_component  7.66e-03           3        13         13  \n",
       "5   GO: cellular_component  7.66e-03           3        13         13  \n",
       "6   GO: biological_process  1.03e-02           3        15         15  \n",
       "7    Dictybase: Phenotypes  1.94e-02           2         5          5  \n",
       "8    Dictybase: Phenotypes  4.73e-02           2         8          8  \n",
       "9    Dictybase: Phenotypes  5.35e-02           4        60         59  \n",
       "10   Dictybase: Phenotypes  8.87e-02           2        12         12  \n",
       "11   Dictybase: Phenotypes  2.18e-01           3        53         53  \n",
       "12  GO: biological_process  2.18e-01           6       197        196  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  dis selected: 550 with EID: 549\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 52 out of 142 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>7.51e-07</td>\n",
       "      <td>23</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.04e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.04e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>6.68e-02</td>\n",
       "      <td>28</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>KEGG: Pathways</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.91e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1                                                 cell wall   \n",
       "2                          external encapsulating structure   \n",
       "3                          anatomical structure development   \n",
       "4  anatomical structure formation involved in morphogenesis   \n",
       "5                             Starch and sucrose metabolism   \n",
       "6                            aberrant spore coat morphology   \n",
       "7                                     abolished sporulation   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: cellular_component  7.51e-07          23       157        157  \n",
       "1  GO: cellular_component  5.04e-02           4        13         13  \n",
       "2  GO: cellular_component  5.04e-02           4        13         13  \n",
       "3  GO: biological_process  6.68e-02          28       440        438  \n",
       "4  GO: biological_process  1.33e-01           8        74         74  \n",
       "5          KEGG: Pathways  1.33e-01           5        31         31  \n",
       "6   Dictybase: Phenotypes  1.91e-01           3        12         12  \n",
       "7   Dictybase: Phenotypes  2.39e-01           4        25         25  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 39 with EID: 39\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 8 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.44e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene set                Ontology       FDR  N in query  \\\n",
       "0  extracellular region  GO: cellular_component  5.44e-02           3   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0       157        157  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cud selected: 517 with EID: 510\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 39 out of 109 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.50e-05</td>\n",
       "      <td>18</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.96e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.64e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>decreased prespore cell differentiation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>46</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1  anatomical structure formation involved in morphogenesis   \n",
       "2                                      cell differentiation   \n",
       "3                                 decreased spore viability   \n",
       "4                   decreased prespore cell differentiation   \n",
       "5                                        hydrolase activity   \n",
       "6                               secondary metabolic process   \n",
       "7                                             cell adhesion   \n",
       "\n",
       "                 Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0  GO: cellular_component  2.50e-05          18       157        157  \n",
       "1  GO: biological_process  2.96e-04          11        74         74  \n",
       "2  GO: biological_process  1.64e-03          14       145        145  \n",
       "3   Dictybase: Phenotypes  1.30e-01           6        60         59  \n",
       "4   Dictybase: Phenotypes  1.30e-01           3        14         14  \n",
       "5  GO: molecular_function  1.30e-01           5        46         45  \n",
       "6  GO: biological_process  1.30e-01           2         5          5  \n",
       "7  GO: biological_process  1.30e-01           7        84         83  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for comparison in comparison_order:\n",
    "    query_names=list(abberant.query('Comparison == \"'+comparison+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=comparison,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### KEGG, DictyPhenotypes, GO "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all gene sets\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=False,set_sizes=(5,500))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Min size 5 Max size 481\n",
      "N sets {('Dictybase', 'Phenotypes'): 209, ('GO', 'molecular_function'): 424, ('GO', 'biological_process'): 1379, ('KEGG', 'Pathways'): 96, ('GO', 'cellular_component'): 305}\n"
     ]
    }
   ],
   "source": [
    "# Find out max/min gene set sizes and N sets\n",
    "max_n=0\n",
    "min_n=np.inf\n",
    "for k,v in GENE_SETS_ONTOLOGY.items():\n",
    "    for s in v:\n",
    "        n=len(s.genes)\n",
    "        if n<min_n:\n",
    "            min_n=n\n",
    "        if n>max_n:\n",
    "            max_n=n\n",
    "print('Min size',min_n,'Max size',max_n)  \n",
    "print('N sets',{k:len(v) for k,v in GENE_SETS_ONTOLOGY.items()})  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  agg- selected: 162 with EID: 160\n",
      "Ratio of genes annotated with a gene set in reference 0.33 and query 0.31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 40 out of 50 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cellular developmental process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.61e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.61e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell wall biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>external encapsulating structure organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>spore wall assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell wall assembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cellular component morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>developmental process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-03</td>\n",
       "      <td>16</td>\n",
       "      <td>449</td>\n",
       "      <td>447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cellular component assembly involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>spore wall biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.19e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.13e-03</td>\n",
       "      <td>15</td>\n",
       "      <td>440</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>sporulation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.65e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>75</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>slug development involved in sorocarp development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.65e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>sporulation resulting in formation of a cellular spore</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.65e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.65e-03</td>\n",
       "      <td>6</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>spore wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>9.65e-03</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.20e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>anatomical structure morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.21e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>149</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.33e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.33e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.99e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>intraspecies interaction between organisms</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.51e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>365</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>sorocarp development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.51e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>365</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>non-reproductive fruiting body development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.51e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>365</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>socially cooperative development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.51e-02</td>\n",
       "      <td>12</td>\n",
       "      <td>365</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>3.21e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>protein depolymerization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.21e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>cell surface</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>4.18e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>intrinsic component of plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.30e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>external side of plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.93e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>aberrant stalk morphogenesis</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>7.61e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>cellular protein complex disassembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>7.68e-02</td>\n",
       "      <td>3</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>9.97e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>60</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>cellular component disassembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>regulation of protein depolymerization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>protein-containing complex disassembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>guanyl ribonucleotide binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>guanyl nucleotide binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.14e-01</td>\n",
       "      <td>7</td>\n",
       "      <td>188</td>\n",
       "      <td>188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>mislocalized gene expression</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>1.29e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>regulation of protein-containing complex disassembly</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.29e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>sterol transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.29e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>organic hydroxy compound transport</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.29e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>sorocarp spore cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>side of membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>multicellular organism development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>regulation of cytoskeleton organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>75</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>multicellular organismal process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>sorocarp stalk development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>sorocarp sorus development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>protein dephosphorylation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>regulation of cellular component organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>120</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>signaling receptor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>transmembrane signaling receptor activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>43</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>signal transduction by protein phosphorylation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>44</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>cellulose binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>molecular transducer activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.85e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>actin filament depolymerization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.90e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>regulation of organelle organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>87</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>supramolecular fiber organization</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>131</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>cell periphery</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>340</td>\n",
       "      <td>340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>GTP binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>182</td>\n",
       "      <td>182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>nucleoside binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>182</td>\n",
       "      <td>182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>purine ribonucleoside binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>182</td>\n",
       "      <td>182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>ribonucleoside binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>182</td>\n",
       "      <td>182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>purine nucleoside binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>182</td>\n",
       "      <td>182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>culmination involved in sorocarp development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>5</td>\n",
       "      <td>132</td>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>Dictybase: Phenotypes</td>\n",
       "      <td>2.36e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>68</td>\n",
       "      <td>catalytic activity</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>2.36e-01</td>\n",
       "      <td>11</td>\n",
       "      <td>481</td>\n",
       "      <td>479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>69</td>\n",
       "      <td>integral component of plasma membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>2.36e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                             cellular developmental process   \n",
       "1                                       cell differentiation   \n",
       "2                                       cell wall biogenesis   \n",
       "3              external encapsulating structure organization   \n",
       "4                                     cell wall organization   \n",
       "5                                        spore wall assembly   \n",
       "6                                         cell wall assembly   \n",
       "7                           cellular component morphogenesis   \n",
       "8                                      developmental process   \n",
       "9      cellular component assembly involved in morphogenesis   \n",
       "10                                     spore wall biogenesis   \n",
       "11                          anatomical structure development   \n",
       "12                                               sporulation   \n",
       "13         slug development involved in sorocarp development   \n",
       "14    sporulation resulting in formation of a cellular spore   \n",
       "15  anatomical structure formation involved in morphogenesis   \n",
       "16                                                spore wall   \n",
       "17                            aberrant spore coat morphology   \n",
       "18                        anatomical structure morphogenesis   \n",
       "19                                                 cell wall   \n",
       "20                          external encapsulating structure   \n",
       "21                      cell wall organization or biogenesis   \n",
       "22                intraspecies interaction between organisms   \n",
       "23                                      sorocarp development   \n",
       "24                non-reproductive fruiting body development   \n",
       "25                          socially cooperative development   \n",
       "26                                multiple tips in culminant   \n",
       "27                                  protein depolymerization   \n",
       "28                                              cell surface   \n",
       "29                    intrinsic component of plasma membrane   \n",
       "30                          external side of plasma membrane   \n",
       "31                              aberrant stalk morphogenesis   \n",
       "32                      cellular protein complex disassembly   \n",
       "33                                 decreased spore viability   \n",
       "34                            cellular component disassembly   \n",
       "35                    regulation of protein depolymerization   \n",
       "36                    protein-containing complex disassembly   \n",
       "37                             guanyl ribonucleotide binding   \n",
       "38                                 guanyl nucleotide binding   \n",
       "39                              mislocalized gene expression   \n",
       "40      regulation of protein-containing complex disassembly   \n",
       "41                                          sterol transport   \n",
       "42                        organic hydroxy compound transport   \n",
       "43                       sorocarp spore cell differentiation   \n",
       "44                                          side of membrane   \n",
       "45                        multicellular organism development   \n",
       "46                   regulation of cytoskeleton organization   \n",
       "47                          multicellular organismal process   \n",
       "48                                sorocarp stalk development   \n",
       "49                                sorocarp sorus development   \n",
       "50                                 protein dephosphorylation   \n",
       "51             regulation of cellular component organization   \n",
       "52                               signaling receptor activity   \n",
       "53                 transmembrane signaling receptor activity   \n",
       "54            signal transduction by protein phosphorylation   \n",
       "55                                         cellulose binding   \n",
       "56                             molecular transducer activity   \n",
       "57                           actin filament depolymerization   \n",
       "58                      regulation of organelle organization   \n",
       "59                         supramolecular fiber organization   \n",
       "60                                            cell periphery   \n",
       "61                                               GTP binding   \n",
       "62                                        nucleoside binding   \n",
       "63                             purine ribonucleoside binding   \n",
       "64                                    ribonucleoside binding   \n",
       "65                                 purine nucleoside binding   \n",
       "66              culmination involved in sorocarp development   \n",
       "67                                     decreased sporulation   \n",
       "68                                        catalytic activity   \n",
       "69                     integral component of plasma membrane   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: biological_process  2.61e-04          11       145        145  \n",
       "1   GO: biological_process  2.61e-04          11       145        145  \n",
       "2   GO: biological_process  5.19e-03           3         8          8  \n",
       "3   GO: biological_process  5.19e-03           3         8          8  \n",
       "4   GO: biological_process  5.19e-03           3         8          8  \n",
       "5   GO: biological_process  5.19e-03           3         8          8  \n",
       "6   GO: biological_process  5.19e-03           3         8          8  \n",
       "7   GO: biological_process  5.19e-03           3         8          8  \n",
       "8   GO: biological_process  5.19e-03          16       449        447  \n",
       "9   GO: biological_process  5.19e-03           3         8          8  \n",
       "10  GO: biological_process  5.19e-03           3         8          8  \n",
       "11  GO: biological_process  7.13e-03          15       440        438  \n",
       "12  GO: biological_process  9.65e-03           6        75         75  \n",
       "13  GO: biological_process  9.65e-03           3        11         11  \n",
       "14  GO: biological_process  9.65e-03           6        74         74  \n",
       "15  GO: biological_process  9.65e-03           6        74         74  \n",
       "16  GO: cellular_component  9.65e-03           3        11         11  \n",
       "17   Dictybase: Phenotypes  1.20e-02           3        12         12  \n",
       "18  GO: biological_process  1.21e-02           8       149        149  \n",
       "19  GO: cellular_component  1.33e-02           3        13         13  \n",
       "20  GO: cellular_component  1.33e-02           3        13         13  \n",
       "21  GO: biological_process  1.99e-02           3        15         15  \n",
       "22  GO: biological_process  2.51e-02          12       365        363  \n",
       "23  GO: biological_process  2.51e-02          12       365        363  \n",
       "24  GO: biological_process  2.51e-02          12       365        363  \n",
       "25  GO: biological_process  2.51e-02          12       365        363  \n",
       "26   Dictybase: Phenotypes  3.21e-02           2         5          5  \n",
       "27  GO: biological_process  3.21e-02           3        19         19  \n",
       "28  GO: cellular_component  4.18e-02           3        21         21  \n",
       "29  GO: cellular_component  5.30e-02           3        23         23  \n",
       "30  GO: cellular_component  5.93e-02           2         7          7  \n",
       "31   Dictybase: Phenotypes  7.61e-02           2         8          8  \n",
       "32  GO: biological_process  7.68e-02           3        27         27  \n",
       "33   Dictybase: Phenotypes  9.97e-02           4        60         59  \n",
       "34  GO: biological_process  1.04e-01           3        31         31  \n",
       "35  GO: biological_process  1.04e-01           2        10         10  \n",
       "36  GO: biological_process  1.04e-01           3        31         31  \n",
       "37  GO: molecular_function  1.13e-01           7       187        187  \n",
       "38  GO: molecular_function  1.14e-01           7       188        188  \n",
       "39   Dictybase: Phenotypes  1.29e-01           2        12         12  \n",
       "40  GO: biological_process  1.29e-01           2        12         12  \n",
       "41  GO: biological_process  1.29e-01           2        12         12  \n",
       "42  GO: biological_process  1.29e-01           2        12         12  \n",
       "43  GO: biological_process  1.30e-01           3        36         36  \n",
       "44  GO: cellular_component  1.45e-01           2        13         13  \n",
       "45  GO: biological_process  1.48e-01           4        72         72  \n",
       "46  GO: biological_process  1.60e-01           4        75         74  \n",
       "47  GO: biological_process  1.68e-01           4        76         76  \n",
       "48  GO: biological_process  1.68e-01           3        41         41  \n",
       "49  GO: biological_process  1.69e-01           3        42         42  \n",
       "50  GO: biological_process  1.69e-01           3        42         42  \n",
       "51  GO: biological_process  1.69e-01           5       120        119  \n",
       "52  GO: molecular_function  1.70e-01           3        43         43  \n",
       "53  GO: molecular_function  1.70e-01           3        43         43  \n",
       "54  GO: biological_process  1.70e-01           3        44         43  \n",
       "55  GO: molecular_function  1.75e-01           2        16         16  \n",
       "56  GO: molecular_function  1.85e-01           3        45         45  \n",
       "57  GO: biological_process  1.90e-01           2        17         17  \n",
       "58  GO: biological_process  2.05e-01           4        87         86  \n",
       "59  GO: biological_process  2.05e-01           5       131        130  \n",
       "60  GO: cellular_component  2.05e-01           9       340        340  \n",
       "61  GO: molecular_function  2.05e-01           6       182        182  \n",
       "62  GO: molecular_function  2.05e-01           6       182        182  \n",
       "63  GO: molecular_function  2.05e-01           6       182        182  \n",
       "64  GO: molecular_function  2.05e-01           6       182        182  \n",
       "65  GO: molecular_function  2.05e-01           6       182        182  \n",
       "66  GO: biological_process  2.05e-01           5       132        131  \n",
       "67   Dictybase: Phenotypes  2.36e-01           3        53         53  \n",
       "68  GO: molecular_function  2.36e-01          11       481        479  \n",
       "69  GO: cellular_component  2.36e-01           2        21         21  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  dis selected: 550 with EID: 549\n",
      "Ratio of genes annotated with a gene set in reference 0.33 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 38 out of 151 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.67e-06</td>\n",
       "      <td>23</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell surface receptor signaling pathway</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>8</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>multicellular organism development</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>multicellular organismal process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.36e-01</td>\n",
       "      <td>9</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Gene set                Ontology       FDR  \\\n",
       "0                     extracellular region  GO: cellular_component  5.67e-06   \n",
       "1  cell surface receptor signaling pathway  GO: biological_process  2.73e-02   \n",
       "2       multicellular organism development  GO: biological_process  1.92e-01   \n",
       "3                                cell wall  GO: cellular_component  1.92e-01   \n",
       "4         external encapsulating structure  GO: cellular_component  1.92e-01   \n",
       "5         multicellular organismal process  GO: biological_process  2.36e-01   \n",
       "\n",
       "   N in query  Set size  N in ref.  \n",
       "0          23       157        157  \n",
       "1           8        39         39  \n",
       "2           9        72         72  \n",
       "3           4        13         13  \n",
       "4           4        13         13  \n",
       "5           9        76         76  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 39 with EID: 39\n",
      "Ratio of genes annotated with a gene set in reference 0.33 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 3 out of 8 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>polysaccharide binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>6.91e-02</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.03e-01</td>\n",
       "      <td>3</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>carbohydrate binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>2</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Gene set                Ontology       FDR  N in query  \\\n",
       "0  polysaccharide binding  GO: molecular_function  6.91e-02           2   \n",
       "1    extracellular region  GO: cellular_component  1.03e-01           3   \n",
       "2    carbohydrate binding  GO: molecular_function  1.33e-01           2   \n",
       "\n",
       "   Set size  N in ref.  \n",
       "0        24         24  \n",
       "1       157        157  \n",
       "2        58         58  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cud selected: 517 with EID: 510\n",
      "Ratio of genes annotated with a gene set in reference 0.33 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 39 out of 116 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "      <th>Set size</th>\n",
       "      <th>N in ref.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>carbohydrate binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>1.63e-08</td>\n",
       "      <td>15</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>polysaccharide binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>5.22e-05</td>\n",
       "      <td>8</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>5.54e-05</td>\n",
       "      <td>18</td>\n",
       "      <td>157</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cellulose binding</td>\n",
       "      <td>GO: molecular_function</td>\n",
       "      <td>4.62e-04</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sporulation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.63e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>75</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>sporulation resulting in formation of a cellular spore</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.63e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>5.63e-04</td>\n",
       "      <td>11</td>\n",
       "      <td>74</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>spore germination</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>9.07e-04</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cellular developmental process</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.96e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>2.96e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>anatomical structure morphogenesis</td>\n",
       "      <td>GO: biological_process</td>\n",
       "      <td>3.64e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>149</td>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>anchored component of membrane</td>\n",
       "      <td>GO: cellular_component</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>4</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       carbohydrate binding   \n",
       "1                                     polysaccharide binding   \n",
       "2                                       extracellular region   \n",
       "3                                          cellulose binding   \n",
       "4                                                sporulation   \n",
       "5     sporulation resulting in formation of a cellular spore   \n",
       "6   anatomical structure formation involved in morphogenesis   \n",
       "7                                          spore germination   \n",
       "8                             cellular developmental process   \n",
       "9                                       cell differentiation   \n",
       "10                        anatomical structure morphogenesis   \n",
       "11                            anchored component of membrane   \n",
       "\n",
       "                  Ontology       FDR  N in query  Set size  N in ref.  \n",
       "0   GO: molecular_function  1.63e-08          15        58         58  \n",
       "1   GO: molecular_function  5.22e-05           8        24         24  \n",
       "2   GO: cellular_component  5.54e-05          18       157        157  \n",
       "3   GO: molecular_function  4.62e-04           6        16         16  \n",
       "4   GO: biological_process  5.63e-04          11        75         75  \n",
       "5   GO: biological_process  5.63e-04          11        74         74  \n",
       "6   GO: biological_process  5.63e-04          11        74         74  \n",
       "7   GO: biological_process  9.07e-04           5        12         12  \n",
       "8   GO: biological_process  2.96e-03          14       145        145  \n",
       "9   GO: biological_process  2.96e-03          14       145        145  \n",
       "10  GO: biological_process  3.64e-03          14       149        149  \n",
       "11  GO: cellular_component  1.92e-01           4        23         23  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Genes sets, not including only slims\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=False,set_sizes=(5,500))\n",
    "for comparison in comparison_order:\n",
    "    query_names=list(abberant.query('Comparison == \"'+comparison+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=comparison,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "orange3venv",
   "language": "python",
   "name": "orange3venv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
