{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Enrichment for milestone genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from importlib import reload\n",
    "\n",
    "module_path = os.path.abspath(os.path.join('..'))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.insert(0,module_path)\n",
    "\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "pd.set_option('max_colwidth', 70)\n",
    "pd.set_option('display.max_rows', 500)\n",
    "import string\n",
    "import numpy as np\n",
    "\n",
    "from orangecontrib.bioinformatics.geneset.__init__ import list_all\n",
    "\n",
    "import jupyter_functions as jf\n",
    "reload(jf)\n",
    "import jupyter_functions as jf\n",
    "from networks.functionsDENet import loadPickle, savePickle\n",
    "import enrichment_library as enr\n",
    "reload(enr)\n",
    "import enrichment_library as enr\n",
    "from stages_DE.stages_library import PHENOTYPES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Path for expression data (mergedGenes_RPKUM.tsv)\n",
    "dataPath='/home/karin/Documents/timeTrajectories/data/RPKUM/combined/'\n",
    "path_de='/home/karin/Documents/timeTrajectories/data/deTime/stage_vs_other/'\n",
    "path_impulse='/home/karin/Documents/timeTrajectories/data/stages/DE_across_stages/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene sets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Uses gene sets that do not have less than 5 or more than 500 genes. Use all ontologies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all gene sets\n",
    "gene_sets=list(list_all(organism='44689'))\n",
    "GENE_SETS_ONTOLOGY=enr.get_gene_sets(gene_set_names=gene_sets, go_slims=True,set_sizes=(5,500))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For conversion of gene names to EID\n",
    "genes = pd.read_csv(dataPath + 'mergedGenes_RPKUM.tsv', sep='\\t', index_col=0)\n",
    "all_gene_names= genes[(genes != 0).any(axis=1)].index\n",
    "NAME_EID=enr.name_genes_entrez(gene_names=all_gene_names, key_entrez=False)\n",
    "ALL_GENE_NAMES_EID=enr.convert_EID(genes=all_gene_names, name_EID=NAME_EID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "def group_diff_enrichment(query_names,group:str,padj:float=0.25,min_overlap:int=None,\n",
    "                          use_annotated_genes:bool=False):\n",
    "    #Displays only gene sets that have overlap with query greater or equal to min_overlap \n",
    "    #For p value and padj calculation uses alll that have overlap >=1 } from gene_set_enrichment\n",
    "    \"\"\"\n",
    "    :param use_annotated_genes: if True use for reference and query  only genes that have at \n",
    "    least one gene set annotation\n",
    "    \"\"\"\n",
    "    query_EID=enr.convert_EID(genes=query_names, name_EID=NAME_EID)\n",
    "    print('***  '+group+' selected:',len(query_names),'with EID:',len(query_EID))\n",
    "    \n",
    "    reference_gene_eids=ALL_GENE_NAMES_EID.copy()\n",
    "    query_eids=query_EID.copy()\n",
    "    \n",
    "    if use_annotated_genes:\n",
    "        gene_sets_genes=set()\n",
    "        for gene_set_name, gene_sets in GENE_SETS_ONTOLOGY.items():\n",
    "            for gene_set in gene_sets:\n",
    "                gene_sets_genes.update(gene_set.genes)\n",
    "        reference_gene_eids=set(reference_gene_eids) & gene_sets_genes\n",
    "        query_eids=set(query_eids) & gene_sets_genes\n",
    "        \n",
    "        query_annotated_reatio='NA'\n",
    "        if len(query_EID)>0:\n",
    "            query_annotated_reatio=round(len(query_eids)/len(query_EID),2)\n",
    "        print('Ratio of genes annotated with a gene set in reference',\n",
    "              round(len(reference_gene_eids)/len(ALL_GENE_NAMES_EID),2),\n",
    "             'and query',query_annotated_reatio)\n",
    "    \n",
    "    query_in_enriched=set()\n",
    "    result=None\n",
    "    if len(query_eids) > 0:\n",
    "        enrichment=enr.gene_set_enrichment(query_eids, reference_EID=reference_gene_eids, \n",
    "                                                padj_threshold=padj,min_overlap=min_overlap,\n",
    "                                                gene_sets_ontology=GENE_SETS_ONTOLOGY)\n",
    "        if len(enrichment)>0:\n",
    "            enrichment_display=list()\n",
    "            enrichment= sorted(enrichment, key=lambda data: data.padj)\n",
    "            for enriched in enrichment:\n",
    "                query_in_enriched.update(enriched.gene_set.genes & query_eids)\n",
    "                enrichment_display.append({'Gene set':enriched.gene_set.name,'Ontology':enriched.ontology,\n",
    "                                           'FDR':\"{:.2e}\".format(enriched.padj),'N in query':enriched.in_query})\n",
    "            result=pd.DataFrame(enrichment_display)\n",
    "    print('Enrichment at FDR: '+str(padj)+' and min query - gene set overlap',str(min_overlap))\n",
    "    print('N query genes in displayed gene sets:',len(query_in_enriched),'out of', len(query_eids),\n",
    "          'query genes used for enrichment calculation.')\n",
    "    display(result)\n",
    "    print('\\n')\n",
    "    return result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enrichment report"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DESeq2\n",
    "Only positive log2 fold changes (lFC) are used to select only overexpressed genes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DESeq2 all strains"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### DESeq2 all strains lFC >= 2 and padj <= 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 216 with EID: 215\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 64 out of 82 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.33e-04</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.22e-03</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.38e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>homeostatic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.96e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>lysosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.83e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.83e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.05e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Lysine biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant cellular response to stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>increased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant cellular metabolic process</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>increased cell size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Porphyrin and chlorophyll metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>vacuole</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.35e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>lipid metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.35e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Gene set                  Ontology       FDR  \\\n",
       "0                    extracellular region  (GO, cellular_component)  4.33e-04   \n",
       "1                 oxidoreductase activity  (GO, molecular_function)  1.22e-03   \n",
       "2                      hydrolase activity  (GO, molecular_function)  1.38e-02   \n",
       "3                     homeostatic process  (GO, biological_process)  1.96e-02   \n",
       "4                                lysosome  (GO, cellular_component)  6.83e-02   \n",
       "5                      response to stress  (GO, biological_process)  6.83e-02   \n",
       "6            cytoskeletal protein binding  (GO, molecular_function)  1.05e-01   \n",
       "7                     Lysine biosynthesis          (KEGG, Pathways)  1.09e-01   \n",
       "8                      peptidase activity  (GO, molecular_function)  1.76e-01   \n",
       "9    aberrant cellular response to stress   (Dictybase, Phenotypes)  1.84e-01   \n",
       "10                        plasma membrane  (GO, cellular_component)  1.84e-01   \n",
       "11                      catabolic process  (GO, biological_process)  1.84e-01   \n",
       "12               increased aggregate size   (Dictybase, Phenotypes)  1.86e-01   \n",
       "13    aberrant cellular metabolic process   (Dictybase, Phenotypes)  1.86e-01   \n",
       "14  Biosynthesis of secondary metabolites          (KEGG, Pathways)  2.14e-01   \n",
       "15                    increased cell size   (Dictybase, Phenotypes)  2.26e-01   \n",
       "16   Porphyrin and chlorophyll metabolism          (KEGG, Pathways)  2.26e-01   \n",
       "17            Biosynthesis of amino acids          (KEGG, Pathways)  2.26e-01   \n",
       "18   cell wall organization or biogenesis  (GO, biological_process)  2.26e-01   \n",
       "19                                vacuole  (GO, cellular_component)  2.35e-01   \n",
       "20                lipid metabolic process  (GO, biological_process)  2.35e-01   \n",
       "\n",
       "    N in query  \n",
       "0           14  \n",
       "1           15  \n",
       "2            6  \n",
       "3            7  \n",
       "4            6  \n",
       "5           16  \n",
       "6            9  \n",
       "7            2  \n",
       "8            6  \n",
       "9            2  \n",
       "10          11  \n",
       "11          11  \n",
       "12           3  \n",
       "13           2  \n",
       "14          11  \n",
       "15           2  \n",
       "16           2  \n",
       "17           4  \n",
       "18           2  \n",
       "19           6  \n",
       "20           7  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 15 with EID: 15\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.13\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 2 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 17 with EID: 17\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.18\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 3 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 37 with EID: 37\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.11\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 2 out of 4 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene set                  Ontology       FDR  N in query\n",
       "0  kinase activity  (GO, molecular_function)  8.61e-02           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 48 with EID: 48\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.08\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 4 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.60e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.07e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                  Ontology       FDR  \\\n",
       "0              extracellular region  (GO, cellular_component)  7.60e-02   \n",
       "1  small molecule metabolic process  (GO, biological_process)  1.07e-01   \n",
       "2                Metabolic pathways          (KEGG, Pathways)  2.26e-01   \n",
       "\n",
       "   N in query  \n",
       "0           2  \n",
       "1           2  \n",
       "2           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 411 with EID: 410\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.18\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 43 out of 73 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.15e-14</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.57e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.57e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.48e-03</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.32e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant cell-cell adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.11e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.41e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.02e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.42e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Gene set                  Ontology       FDR  \\\n",
       "0                   extracellular region  (GO, cellular_component)  1.15e-14   \n",
       "1       external encapsulating structure  (GO, cellular_component)  2.57e-03   \n",
       "2                              cell wall  (GO, cellular_component)  2.57e-03   \n",
       "3       anatomical structure development  (GO, biological_process)  7.48e-03   \n",
       "4   cell wall organization or biogenesis  (GO, biological_process)  5.32e-02   \n",
       "5            aberrant cell-cell adhesion   (Dictybase, Phenotypes)  6.11e-02   \n",
       "6                   extracellular matrix  (GO, cellular_component)  1.41e-01   \n",
       "7         aberrant spore coat morphology   (Dictybase, Phenotypes)  2.02e-01   \n",
       "8              decreased spore viability   (Dictybase, Phenotypes)  2.13e-01   \n",
       "9                        plasma membrane  (GO, cellular_component)  2.13e-01   \n",
       "10                   cell-cell signaling  (GO, biological_process)  2.42e-01   \n",
       "\n",
       "    N in query  \n",
       "0           24  \n",
       "1            4  \n",
       "2            4  \n",
       "3           19  \n",
       "4            3  \n",
       "5            2  \n",
       "6            2  \n",
       "7            2  \n",
       "8            4  \n",
       "9           10  \n",
       "10           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 527 with EID: 526\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.27\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 89 out of 144 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.45e-10</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.24e-07</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.76e-07</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.07e-06</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.85e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.85e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.45e-06</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.81e-05</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.26e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.29e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.81e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.87e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.87e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.87e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>precocious spore germination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>thin standing slug</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>decreased autophagic vacuole formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.61e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.99e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.02e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>abolished stalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant spore morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>increased prestalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.46e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1                             aberrant spore coat morphology   \n",
       "2                       cell wall organization or biogenesis   \n",
       "3   anatomical structure formation involved in morphogenesis   \n",
       "4                           external encapsulating structure   \n",
       "5                                                  cell wall   \n",
       "6                                       cell differentiation   \n",
       "7                           anatomical structure development   \n",
       "8                              Starch and sucrose metabolism   \n",
       "9                                       extracellular matrix   \n",
       "10                          abolished cellulose biosynthesis   \n",
       "11                                 decreased spore viability   \n",
       "12                            carbohydrate metabolic process   \n",
       "13                                        hydrolase activity   \n",
       "14                               abolished cellulose binding   \n",
       "15                              precocious spore germination   \n",
       "16                                        thin standing slug   \n",
       "17                    decreased autophagic vacuole formation   \n",
       "18                                      aberrant culmination   \n",
       "19                                       cell-cell signaling   \n",
       "20                                                 wild type   \n",
       "21                                     abolished culmination   \n",
       "22                      abolished stalk cell differentiation   \n",
       "23                                 aberrant spore morphology   \n",
       "24                   increased prestalk cell differentiation   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  1.45e-10          28  \n",
       "1    (Dictybase, Phenotypes)  1.24e-07           8  \n",
       "2   (GO, biological_process)  9.76e-07           8  \n",
       "3   (GO, biological_process)  2.07e-06          15  \n",
       "4   (GO, cellular_component)  3.85e-06           7  \n",
       "5   (GO, cellular_component)  3.85e-06           7  \n",
       "6   (GO, biological_process)  5.45e-06          20  \n",
       "7   (GO, biological_process)  3.81e-05          36  \n",
       "8           (KEGG, Pathways)  2.26e-03           7  \n",
       "9   (GO, cellular_component)  2.29e-03           4  \n",
       "10   (Dictybase, Phenotypes)  8.81e-03           3  \n",
       "11   (Dictybase, Phenotypes)  1.87e-02           8  \n",
       "12  (GO, biological_process)  1.87e-02          10  \n",
       "13  (GO, molecular_function)  1.87e-02           7  \n",
       "14   (Dictybase, Phenotypes)  1.89e-02           2  \n",
       "15   (Dictybase, Phenotypes)  4.61e-02           2  \n",
       "16   (Dictybase, Phenotypes)  4.61e-02           2  \n",
       "17   (Dictybase, Phenotypes)  4.61e-02           2  \n",
       "18   (Dictybase, Phenotypes)  4.99e-02           9  \n",
       "19  (GO, biological_process)  2.02e-01           3  \n",
       "20   (Dictybase, Phenotypes)  2.05e-01          20  \n",
       "21   (Dictybase, Phenotypes)  2.29e-01           6  \n",
       "22   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "23   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "24   (Dictybase, Phenotypes)  2.46e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 933 with EID: 926\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 96 out of 192 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.44e-13</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.98e-06</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.05e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.05e-05</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.18e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>precocious spore germination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.12e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.89e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.89e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.89e-03</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.12e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.84e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.17e-02</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.17e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.28e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>development arrests at slug stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant stalk formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.44e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.44e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1                                       cell differentiation   \n",
       "2                             aberrant spore coat morphology   \n",
       "3   anatomical structure formation involved in morphogenesis   \n",
       "4                       cell wall organization or biogenesis   \n",
       "5                               precocious spore germination   \n",
       "6                           external encapsulating structure   \n",
       "7                                                  cell wall   \n",
       "8                           anatomical structure development   \n",
       "9                                       extracellular matrix   \n",
       "10                                 decreased spore viability   \n",
       "11                                   oxidoreductase activity   \n",
       "12                                        hydrolase activity   \n",
       "13                                 decreased gene expression   \n",
       "14                             Starch and sucrose metabolism   \n",
       "15                         development arrests at slug stage   \n",
       "16                                             cell adhesion   \n",
       "17                                  aberrant stalk formation   \n",
       "18                               secondary metabolic process   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  9.44e-13          35  \n",
       "1   (GO, biological_process)  7.98e-06          24  \n",
       "2    (Dictybase, Phenotypes)  2.05e-05           7  \n",
       "3   (GO, biological_process)  2.05e-05          16  \n",
       "4   (GO, biological_process)  1.18e-04           7  \n",
       "5    (Dictybase, Phenotypes)  4.12e-03           3  \n",
       "6   (GO, cellular_component)  5.89e-03           5  \n",
       "7   (GO, cellular_component)  5.89e-03           5  \n",
       "8   (GO, biological_process)  5.89e-03          38  \n",
       "9   (GO, cellular_component)  7.12e-03           4  \n",
       "10   (Dictybase, Phenotypes)  3.84e-02           9  \n",
       "11  (GO, molecular_function)  9.17e-02          19  \n",
       "12  (GO, molecular_function)  9.17e-02           7  \n",
       "13   (Dictybase, Phenotypes)  1.30e-01          10  \n",
       "14          (KEGG, Pathways)  2.28e-01           5  \n",
       "15   (Dictybase, Phenotypes)  2.38e-01           3  \n",
       "16  (GO, biological_process)  2.38e-01           9  \n",
       "17   (Dictybase, Phenotypes)  2.44e-01           2  \n",
       "18  (GO, biological_process)  2.44e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  yem selected: 1001 with EID: 995\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.32\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 149 out of 320 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.16e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.63e-03</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.85e-03</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.85e-03</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.15e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Cysteine and methionine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.15e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Phenylalanine</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.76e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Fatty acid biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.76e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.76e-02</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Phenylalanine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.85e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.85e-01</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.19e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                secondary metabolic process   \n",
       "1                                       cell differentiation   \n",
       "2                                         Metabolic pathways   \n",
       "3   anatomical structure formation involved in morphogenesis   \n",
       "4        Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "5                                  decreased gene expression   \n",
       "6                         Cysteine and methionine metabolism   \n",
       "7                                              Phenylalanine   \n",
       "8                                    Fatty acid biosynthesis   \n",
       "9                                    oxidoreductase activity   \n",
       "10                                  Phenylalanine metabolism   \n",
       "11                          anatomical structure development   \n",
       "12                 SNARE interactions in vesicular transport   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  1.16e-03           5  \n",
       "1   (GO, biological_process)  1.63e-03          28  \n",
       "2           (KEGG, Pathways)  6.85e-03          80  \n",
       "3   (GO, biological_process)  6.85e-03          17  \n",
       "4           (KEGG, Pathways)  4.15e-02           5  \n",
       "5    (Dictybase, Phenotypes)  6.15e-02          16  \n",
       "6           (KEGG, Pathways)  6.15e-02           8  \n",
       "7           (KEGG, Pathways)  7.76e-02           3  \n",
       "8           (KEGG, Pathways)  7.76e-02           4  \n",
       "9   (GO, molecular_function)  7.76e-02          29  \n",
       "10          (KEGG, Pathways)  1.85e-01           4  \n",
       "11  (GO, biological_process)  1.85e-01          50  \n",
       "12          (KEGG, Pathways)  2.19e-01           6  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 709 with EID: 705\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.22\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 65 out of 155 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.70e-09</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.12e-03</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.29e-03</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1                                      cell differentiation   \n",
       "2                                   oxidoreductase activity   \n",
       "3  anatomical structure formation involved in morphogenesis   \n",
       "4                               secondary metabolic process   \n",
       "5                                             cell adhesion   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  5.70e-09          27  \n",
       "1  (GO, biological_process)  4.12e-03          17  \n",
       "2  (GO, molecular_function)  9.29e-03          20  \n",
       "3  (GO, biological_process)  2.26e-02          10  \n",
       "4  (GO, biological_process)  2.26e-02           3  \n",
       "5  (GO, biological_process)  1.63e-01           9  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_de+'nobatchrep_combined.tsv',index_col=0,sep='\\t')\n",
    "for stage in [phenotype for phenotype in PHENOTYPES if phenotype in list(milestones['Stage'].unique())]:\n",
    "    query_names=list(milestones.query('log2FoldChange>=2 & padj <= 0.01 & Stage==\"'+stage+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)\n",
    "  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### DESeq2 all strains lFC >= 1 and padj <= 0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 1652 with EID: 1643\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.37\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 389 out of 600 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.24e-64</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.00e-59</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.37e-57</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.62e-44</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.33e-38</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.91e-09</td>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.64e-07</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.37e-06</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.16e-04</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.01e-03</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.69e-03</td>\n",
       "      <td>134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.84e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.84e-02</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.92e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>mitochondrion organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.35e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.35e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.75e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.41e-02</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Porphyrin and chlorophyll metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.28e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Biosynthesis of unsaturated fatty acids</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.10e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Gene set                  Ontology  \\\n",
       "0                                  Ribosome          (KEGG, Pathways)   \n",
       "1                                  ribosome  (GO, cellular_component)   \n",
       "2        structural constituent of ribosome  (GO, molecular_function)   \n",
       "3              structural molecule activity  (GO, molecular_function)   \n",
       "4                               translation  (GO, biological_process)   \n",
       "5                             mitochondrion  (GO, cellular_component)   \n",
       "6                      Steroid biosynthesis          (KEGG, Pathways)   \n",
       "7                              rRNA binding  (GO, molecular_function)   \n",
       "8                               RNA binding  (GO, molecular_function)   \n",
       "9     Biosynthesis of secondary metabolites          (KEGG, Pathways)   \n",
       "10                       Metabolic pathways          (KEGG, Pathways)   \n",
       "11                Oxidative phosphorylation          (KEGG, Pathways)   \n",
       "12                  oxidoreductase activity  (GO, molecular_function)   \n",
       "13              Aminoacyl-tRNA biosynthesis          (KEGG, Pathways)   \n",
       "14               mitochondrion organization  (GO, biological_process)   \n",
       "15               methyltransferase activity  (GO, molecular_function)   \n",
       "16                    Fatty acid elongation          (KEGG, Pathways)   \n",
       "17                     extracellular region  (GO, cellular_component)   \n",
       "18     Porphyrin and chlorophyll metabolism          (KEGG, Pathways)   \n",
       "19  Biosynthesis of unsaturated fatty acids          (KEGG, Pathways)   \n",
       "20                       hydrolase activity  (GO, molecular_function)   \n",
       "21                       isomerase activity  (GO, molecular_function)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   1.24e-64          90  \n",
       "1   1.00e-59          85  \n",
       "2   6.37e-57          79  \n",
       "3   8.62e-44          85  \n",
       "4   1.33e-38          99  \n",
       "5   2.91e-09          73  \n",
       "6   2.64e-07          14  \n",
       "7   2.37e-06          11  \n",
       "8   2.16e-04          48  \n",
       "9   1.01e-03          68  \n",
       "10  3.69e-03         134  \n",
       "11  1.84e-02          18  \n",
       "12  1.84e-02          49  \n",
       "13  1.92e-02          16  \n",
       "14  2.35e-02          10  \n",
       "15  2.35e-02          18  \n",
       "16  2.75e-02           7  \n",
       "17  3.41e-02          38  \n",
       "18  4.28e-02           7  \n",
       "19  2.10e-01           6  \n",
       "20  2.25e-01          13  \n",
       "21  2.25e-01          14  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 170 with EID: 167\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 30 out of 41 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.03e-05</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.72e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.36e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.68e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.68e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.68e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.68e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.68e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.84e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.08e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.95e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.35e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.39e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                  Ontology  \\\n",
       "0                       extracellular region  (GO, cellular_component)   \n",
       "1   development arrests at loose mound stage   (Dictybase, Phenotypes)   \n",
       "2                              cell adhesion  (GO, biological_process)   \n",
       "3                         aberrant streaming   (Dictybase, Phenotypes)   \n",
       "4                       aberrant aggregation   (Dictybase, Phenotypes)   \n",
       "5                   abolished slug migration   (Dictybase, Phenotypes)   \n",
       "6                            plasma membrane  (GO, cellular_component)   \n",
       "7                        signal transduction  (GO, biological_process)   \n",
       "8                       aberrant culmination   (Dictybase, Phenotypes)   \n",
       "9                     precocious aggregation   (Dictybase, Phenotypes)   \n",
       "10              aberrant cAMP wave formation   (Dictybase, Phenotypes)   \n",
       "11                   aberrant slug migration   (Dictybase, Phenotypes)   \n",
       "12                 abolished gene expression   (Dictybase, Phenotypes)   \n",
       "13                     increased growth rate   (Dictybase, Phenotypes)   \n",
       "14              decreased fruiting body size   (Dictybase, Phenotypes)   \n",
       "15                  decreased aggregate size   (Dictybase, Phenotypes)   \n",
       "16                   decreased cell motility   (Dictybase, Phenotypes)   \n",
       "17             aberrant F-actin organization   (Dictybase, Phenotypes)   \n",
       "18                      cell differentiation  (GO, biological_process)   \n",
       "19          anatomical structure development  (GO, biological_process)   \n",
       "20                      aberrant development   (Dictybase, Phenotypes)   \n",
       "21                     decreased sporulation   (Dictybase, Phenotypes)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   4.03e-05          11  \n",
       "1   2.72e-03           4  \n",
       "2   6.36e-02           5  \n",
       "3   8.68e-02           4  \n",
       "4   8.68e-02           4  \n",
       "5   8.68e-02           3  \n",
       "6   8.68e-02           8  \n",
       "7   8.68e-02           9  \n",
       "8   9.84e-02           4  \n",
       "9   1.08e-01           2  \n",
       "10  1.08e-01           4  \n",
       "11  1.09e-01           3  \n",
       "12  1.29e-01           2  \n",
       "13  1.48e-01           2  \n",
       "14  1.48e-01           3  \n",
       "15  1.48e-01           2  \n",
       "16  1.48e-01           3  \n",
       "17  1.80e-01           2  \n",
       "18  1.95e-01           4  \n",
       "19  2.14e-01           8  \n",
       "20  2.35e-01           2  \n",
       "21  2.39e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 193 with EID: 190\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.3\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 44 out of 57 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.59e-03</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.03e-03</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.91e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.92e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.92e-03</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.19e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.21e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.00e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.07e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.07e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>delayed tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.07e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>abolished cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.07e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.07e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>multiple tips in mound</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.57e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.66e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>AGE-RAGE signaling pathway in diabetic complications</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>GTPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>delayed culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.84e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.96e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.36e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Gene set  \\\n",
       "0                                   extracellular region   \n",
       "1                                        plasma membrane   \n",
       "2               development arrests at loose mound stage   \n",
       "3                               abolished slug migration   \n",
       "4                                    signal transduction   \n",
       "5                                   aberrant aggregation   \n",
       "6                             multiple tips in culminant   \n",
       "7                                     response to stress   \n",
       "8                                  decreased culmination   \n",
       "9                                    abolished streaming   \n",
       "10                                 delayed tip formation   \n",
       "11                         abolished cAMP wave formation   \n",
       "12                      anatomical structure development   \n",
       "13                                    aberrant streaming   \n",
       "14                                 abolished aggregation   \n",
       "15                                multiple tips in mound   \n",
       "16                                         cell adhesion   \n",
       "17                                precocious aggregation   \n",
       "18                    development arrests at mound stage   \n",
       "19                                  aberrant culmination   \n",
       "20                                  aberrant development   \n",
       "21                               aberrant slug migration   \n",
       "22                             abolished gene expression   \n",
       "23                                 abolished sporulation   \n",
       "24                          aberrant cAMP wave formation   \n",
       "25  AGE-RAGE signaling pathway in diabetic complications   \n",
       "26                                  cell differentiation   \n",
       "27                                       GTPase activity   \n",
       "28                                   delayed culmination   \n",
       "29                                 increased growth rate   \n",
       "30                              decreased aggregate size   \n",
       "31                          decreased fruiting body size   \n",
       "32                               decreased cell motility   \n",
       "33                         aberrant F-actin organization   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  1.59e-03          11  \n",
       "1   (GO, cellular_component)  6.03e-03          13  \n",
       "2    (Dictybase, Phenotypes)  7.91e-03           4  \n",
       "3    (Dictybase, Phenotypes)  9.92e-03           5  \n",
       "4   (GO, biological_process)  9.92e-03          14  \n",
       "5    (Dictybase, Phenotypes)  4.19e-02           5  \n",
       "6    (Dictybase, Phenotypes)  4.21e-02           2  \n",
       "7   (GO, biological_process)  8.00e-02          12  \n",
       "8    (Dictybase, Phenotypes)  1.07e-01           2  \n",
       "9    (Dictybase, Phenotypes)  1.07e-01           3  \n",
       "10   (Dictybase, Phenotypes)  1.07e-01           2  \n",
       "11   (Dictybase, Phenotypes)  1.07e-01           2  \n",
       "12  (GO, biological_process)  1.07e-01          13  \n",
       "13   (Dictybase, Phenotypes)  1.37e-01           4  \n",
       "14   (Dictybase, Phenotypes)  1.46e-01           4  \n",
       "15   (Dictybase, Phenotypes)  1.46e-01           2  \n",
       "16  (GO, biological_process)  1.53e-01           4  \n",
       "17   (Dictybase, Phenotypes)  1.57e-01           2  \n",
       "18   (Dictybase, Phenotypes)  1.66e-01           3  \n",
       "19   (Dictybase, Phenotypes)  1.66e-01           4  \n",
       "20   (Dictybase, Phenotypes)  1.66e-01           3  \n",
       "21   (Dictybase, Phenotypes)  1.74e-01           3  \n",
       "22   (Dictybase, Phenotypes)  1.74e-01           2  \n",
       "23   (Dictybase, Phenotypes)  1.74e-01           2  \n",
       "24   (Dictybase, Phenotypes)  1.74e-01           4  \n",
       "25          (KEGG, Pathways)  1.74e-01           2  \n",
       "26  (GO, biological_process)  1.75e-01           5  \n",
       "27  (GO, molecular_function)  1.75e-01           5  \n",
       "28   (Dictybase, Phenotypes)  1.84e-01           2  \n",
       "29   (Dictybase, Phenotypes)  1.96e-01           2  \n",
       "30   (Dictybase, Phenotypes)  2.04e-01           2  \n",
       "31   (Dictybase, Phenotypes)  2.34e-01           3  \n",
       "32   (Dictybase, Phenotypes)  2.34e-01           3  \n",
       "33   (Dictybase, Phenotypes)  2.36e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 384 with EID: 382\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.18\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 44 out of 70 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.50e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.94e-03</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.90e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.90e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.90e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>increased cell-substrate adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.04e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.04e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.04e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.00e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.40e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>delayed tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.65e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Gene set                  Ontology       FDR  \\\n",
       "0              chromosome segregation  (GO, biological_process)  1.50e-03   \n",
       "1                          cell cycle  (GO, biological_process)  1.94e-03   \n",
       "2            mitotic nuclear division  (GO, biological_process)  4.90e-03   \n",
       "3                     DNA replication          (KEGG, Pathways)  7.90e-03   \n",
       "4             chromosome organization  (GO, biological_process)  7.90e-03   \n",
       "5   increased cell-substrate adhesion   (Dictybase, Phenotypes)  3.04e-02   \n",
       "6                          chromosome  (GO, cellular_component)  3.04e-02   \n",
       "7                  mitotic cell cycle  (GO, biological_process)  3.04e-02   \n",
       "8                       cell division  (GO, biological_process)  6.00e-02   \n",
       "9                extracellular region  (GO, cellular_component)  8.40e-02   \n",
       "10          decreased spore viability   (Dictybase, Phenotypes)  1.33e-01   \n",
       "11              delayed tip formation   (Dictybase, Phenotypes)  1.33e-01   \n",
       "12   external encapsulating structure  (GO, cellular_component)  1.33e-01   \n",
       "13                          cell wall  (GO, cellular_component)  1.33e-01   \n",
       "14                    kinase activity  (GO, molecular_function)  1.65e-01   \n",
       "15                 nuclear chromosome  (GO, cellular_component)  1.87e-01   \n",
       "\n",
       "    N in query  \n",
       "0            5  \n",
       "1           12  \n",
       "2            4  \n",
       "3            5  \n",
       "4            7  \n",
       "5            4  \n",
       "6            6  \n",
       "7            7  \n",
       "8            7  \n",
       "9            8  \n",
       "10           4  \n",
       "11           2  \n",
       "12           2  \n",
       "13           2  \n",
       "14          11  \n",
       "15           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 616 with EID: 613\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.26\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 93 out of 157 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.37e-12</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.37e-12</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.25e-08</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.99e-07</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.65e-07</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.53e-05</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.42e-05</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.92e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.92e-04</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.42e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.03e-03</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.63e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.12e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.22e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.22e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>delayed tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.69e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.69e-02</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>abolished tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.92e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.82e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>abolished negative regulation of vegetative growth</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.86e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.05e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant myosin II organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>increased chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>increased macropinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>decreased cell-cell adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>decreased prestalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.67e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>aberrant culminant morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.16e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Gene set  \\\n",
       "0                                           chromosome   \n",
       "1                                      DNA replication   \n",
       "2                                           cell cycle   \n",
       "3                               chromosome segregation   \n",
       "4                              chromosome organization   \n",
       "5                             mitotic nuclear division   \n",
       "6                                        cell division   \n",
       "7                                   nuclear chromosome   \n",
       "8                                   mitotic cell cycle   \n",
       "9                                      Mismatch repair   \n",
       "10                                         DNA binding   \n",
       "11                          Nucleotide excision repair   \n",
       "12                                Base excision repair   \n",
       "13                               Pyrimidine metabolism   \n",
       "14                               DNA metabolic process   \n",
       "15                               delayed tip formation   \n",
       "16                                extracellular region   \n",
       "17                             abolished tip formation   \n",
       "18                       microtubule organizing center   \n",
       "19  abolished negative regulation of vegetative growth   \n",
       "20                      Ubiquitin mediated proteolysis   \n",
       "21                     aberrant myosin II organization   \n",
       "22                                   cell cycle arrest   \n",
       "23                      increased chemotaxis to folate   \n",
       "24                          increased macropinocytosis   \n",
       "25                              Glutathione metabolism   \n",
       "26                        decreased cell-cell adhesion   \n",
       "27             decreased prestalk cell differentiation   \n",
       "28                          precocious gene expression   \n",
       "29                                        cytoskeleton   \n",
       "30                       aberrant culminant morphology   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  1.37e-12          22  \n",
       "1           (KEGG, Pathways)  1.37e-12          16  \n",
       "2   (GO, biological_process)  1.25e-08          27  \n",
       "3   (GO, biological_process)  3.99e-07           9  \n",
       "4   (GO, biological_process)  7.65e-07          16  \n",
       "5   (GO, biological_process)  2.53e-05           7  \n",
       "6   (GO, biological_process)  6.42e-05          17  \n",
       "7   (GO, cellular_component)  7.92e-04           6  \n",
       "8   (GO, biological_process)  7.92e-04          14  \n",
       "9           (KEGG, Pathways)  1.42e-03           6  \n",
       "10  (GO, molecular_function)  8.03e-03          20  \n",
       "11          (KEGG, Pathways)  9.63e-03           7  \n",
       "12          (KEGG, Pathways)  1.12e-02           6  \n",
       "13          (KEGG, Pathways)  1.22e-02           6  \n",
       "14  (GO, biological_process)  1.22e-02          12  \n",
       "15   (Dictybase, Phenotypes)  1.69e-02           4  \n",
       "16  (GO, cellular_component)  1.69e-02          15  \n",
       "17   (Dictybase, Phenotypes)  1.92e-02           2  \n",
       "18  (GO, cellular_component)  3.82e-02           7  \n",
       "19   (Dictybase, Phenotypes)  9.86e-02           2  \n",
       "20          (KEGG, Pathways)  1.05e-01           7  \n",
       "21   (Dictybase, Phenotypes)  1.28e-01           2  \n",
       "22   (Dictybase, Phenotypes)  1.28e-01           2  \n",
       "23   (Dictybase, Phenotypes)  1.28e-01           2  \n",
       "24   (Dictybase, Phenotypes)  1.28e-01           2  \n",
       "25          (KEGG, Pathways)  1.35e-01           4  \n",
       "26   (Dictybase, Phenotypes)  1.62e-01           3  \n",
       "27   (Dictybase, Phenotypes)  1.67e-01           2  \n",
       "28   (Dictybase, Phenotypes)  1.86e-01           3  \n",
       "29  (GO, cellular_component)  1.86e-01          13  \n",
       "30   (Dictybase, Phenotypes)  2.16e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 1373 with EID: 1367\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.21\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 166 out of 292 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.18e-09</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.12e-05</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.22e-05</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.70e-05</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.70e-05</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.57e-04</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.98e-04</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.05e-04</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.49e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.22e-03</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.66e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.64e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.73e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant thermotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.44e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.44e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.44e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.44e-02</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.75e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>decreased prestalk A cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.75e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.86e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant phototaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.86e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>development arrests at slug stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.57e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>abolished DIF production</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>thin standing slug</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>decreased autophagic vacuole formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.69e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>increased stalk length</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.73e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.73e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1                           anatomical structure development   \n",
       "2                             aberrant spore coat morphology   \n",
       "3                           external encapsulating structure   \n",
       "4                                                  cell wall   \n",
       "5                                       cell differentiation   \n",
       "6                              Starch and sucrose metabolism   \n",
       "7   anatomical structure formation involved in morphogenesis   \n",
       "8                       cell wall organization or biogenesis   \n",
       "9                                        signal transduction   \n",
       "10                                      aberrant culmination   \n",
       "11                                 decreased spore viability   \n",
       "12                 DNA-binding transcription factor activity   \n",
       "13                                      aberrant thermotaxis   \n",
       "14                          abolished cellulose biosynthesis   \n",
       "15                                           spores inviable   \n",
       "16                                               DNA binding   \n",
       "17                               abolished cellulose binding   \n",
       "18                 decreased prestalk A cell differentiation   \n",
       "19                   increased prespore cell differentiation   \n",
       "20                                       aberrant phototaxis   \n",
       "21                                    Glutathione metabolism   \n",
       "22                         development arrests at slug stage   \n",
       "23                         aberrant fruiting body morphology   \n",
       "24                                     abolished culmination   \n",
       "25                                  abolished DIF production   \n",
       "26                                        thin standing slug   \n",
       "27                    decreased autophagic vacuole formation   \n",
       "28                                    increased stalk length   \n",
       "29                                      extracellular matrix   \n",
       "30                              aberrant cAMP wave formation   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  5.18e-09          38  \n",
       "1   (GO, biological_process)  2.12e-05          62  \n",
       "2    (Dictybase, Phenotypes)  3.22e-05           8  \n",
       "3   (GO, cellular_component)  4.70e-05           8  \n",
       "4   (GO, cellular_component)  4.70e-05           8  \n",
       "5   (GO, biological_process)  2.57e-04          27  \n",
       "6           (KEGG, Pathways)  2.98e-04          11  \n",
       "7   (GO, biological_process)  7.05e-04          17  \n",
       "8   (GO, biological_process)  1.49e-03           7  \n",
       "9   (GO, biological_process)  7.22e-03          44  \n",
       "10   (Dictybase, Phenotypes)  1.66e-02          16  \n",
       "11   (Dictybase, Phenotypes)  2.64e-02          12  \n",
       "12  (GO, molecular_function)  2.73e-02           9  \n",
       "13   (Dictybase, Phenotypes)  6.44e-02           3  \n",
       "14   (Dictybase, Phenotypes)  6.44e-02           3  \n",
       "15   (Dictybase, Phenotypes)  6.44e-02           3  \n",
       "16  (GO, molecular_function)  6.44e-02          28  \n",
       "17   (Dictybase, Phenotypes)  8.75e-02           2  \n",
       "18   (Dictybase, Phenotypes)  8.75e-02           2  \n",
       "19   (Dictybase, Phenotypes)  9.86e-02           3  \n",
       "20   (Dictybase, Phenotypes)  9.86e-02           3  \n",
       "21          (KEGG, Pathways)  1.16e-01           6  \n",
       "22   (Dictybase, Phenotypes)  1.20e-01           4  \n",
       "23   (Dictybase, Phenotypes)  1.57e-01          14  \n",
       "24   (Dictybase, Phenotypes)  1.69e-01          10  \n",
       "25   (Dictybase, Phenotypes)  1.69e-01           2  \n",
       "26   (Dictybase, Phenotypes)  1.69e-01           2  \n",
       "27   (Dictybase, Phenotypes)  1.69e-01           2  \n",
       "28   (Dictybase, Phenotypes)  1.73e-01           3  \n",
       "29  (GO, cellular_component)  1.73e-01           3  \n",
       "30   (Dictybase, Phenotypes)  2.31e-01          13  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 1871 with EID: 1864\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.3\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 337 out of 567 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.79e-07</td>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.46e-05</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.77e-05</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.76e-05</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.66e-04</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.91e-04</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.91e-04</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.03e-03</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.20e-03</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.28e-03</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.28e-03</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.24e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.25e-02</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.37e-02</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.37e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.84e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.84e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.37e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.37e-02</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant spore morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.37e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>thin standing slug</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.97e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>decreased autophagic vacuole formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.97e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant phototaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.66e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.68e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.38e-01</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>aberrant response to abiotic stimulus</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>increased sorus size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>decreased prestalk 0 cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>increased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.54e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.61e-01</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.97e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.97e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>aberrant protein phosphorylation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>increased stalk length</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>increased standing slug length</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>decreased prestalk A cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>abolished response to DIF</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>abolished tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.99e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>aberrant thermotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>precocious sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>abolished protein localization to nucleus</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>increased macropinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>aberrant autophagic cell death</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>Histidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>translucent sorus</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                           anatomical structure development   \n",
       "1   anatomical structure formation involved in morphogenesis   \n",
       "2                             aberrant spore coat morphology   \n",
       "3                                       cell differentiation   \n",
       "4                                       extracellular region   \n",
       "5                           external encapsulating structure   \n",
       "6                                                  cell wall   \n",
       "7                              Starch and sucrose metabolism   \n",
       "8                                  decreased spore viability   \n",
       "9                                        signal transduction   \n",
       "10                                           kinase activity   \n",
       "11                   increased prespore cell differentiation   \n",
       "12                 DNA-binding transcription factor activity   \n",
       "13                                      aberrant culmination   \n",
       "14                      cell wall organization or biogenesis   \n",
       "15                 development arrests at tipped mound stage   \n",
       "16                          abolished cellulose biosynthesis   \n",
       "17                                           spores inviable   \n",
       "18                                   aberrant slug migration   \n",
       "19                         aberrant fruiting body morphology   \n",
       "20                                 aberrant spore morphology   \n",
       "21                                        thin standing slug   \n",
       "22                    decreased autophagic vacuole formation   \n",
       "23                                       aberrant phototaxis   \n",
       "24                                     abolished culmination   \n",
       "25                                                 wild type   \n",
       "26                                               DNA binding   \n",
       "27                     aberrant response to abiotic stimulus   \n",
       "28                                      increased sorus size   \n",
       "29                 decreased prestalk 0 cell differentiation   \n",
       "30                                 increased gene expression   \n",
       "31                                      Galactose metabolism   \n",
       "32                            carbohydrate metabolic process   \n",
       "33                                          ABC transporters   \n",
       "34                                     decreased sporulation   \n",
       "35                                  abolished slug migration   \n",
       "36                               abolished cellulose binding   \n",
       "37                                 abolished gene expression   \n",
       "38                          aberrant protein phosphorylation   \n",
       "39                                    increased stalk length   \n",
       "40                            increased standing slug length   \n",
       "41                 decreased prestalk A cell differentiation   \n",
       "42                        decreased intracellular cAMP level   \n",
       "43                                 abolished response to DIF   \n",
       "44                                 decreased gene expression   \n",
       "45                                   abolished tip formation   \n",
       "46                                      extracellular matrix   \n",
       "47                                       cell-cell signaling   \n",
       "48                                      aberrant thermotaxis   \n",
       "49                                    precocious sporulation   \n",
       "50                 abolished protein localization to nucleus   \n",
       "51                                increased macropinocytosis   \n",
       "52                            aberrant autophagic cell death   \n",
       "53                                      Histidine metabolism   \n",
       "54                                         translucent sorus   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  9.79e-07         107  \n",
       "1   (GO, biological_process)  2.46e-05          29  \n",
       "2    (Dictybase, Phenotypes)  2.77e-05          10  \n",
       "3   (GO, biological_process)  4.76e-05          44  \n",
       "4   (GO, cellular_component)  1.66e-04          45  \n",
       "5   (GO, cellular_component)  6.91e-04           9  \n",
       "6   (GO, cellular_component)  6.91e-04           9  \n",
       "7           (KEGG, Pathways)  2.03e-03          14  \n",
       "8    (Dictybase, Phenotypes)  6.20e-03          20  \n",
       "9   (GO, biological_process)  9.28e-03          74  \n",
       "10  (GO, molecular_function)  9.28e-03          70  \n",
       "11   (Dictybase, Phenotypes)  1.24e-02           5  \n",
       "12  (GO, molecular_function)  1.25e-02          14  \n",
       "13   (Dictybase, Phenotypes)  1.37e-02          25  \n",
       "14  (GO, biological_process)  1.37e-02           8  \n",
       "15   (Dictybase, Phenotypes)  1.47e-02           7  \n",
       "16   (Dictybase, Phenotypes)  4.84e-02           4  \n",
       "17   (Dictybase, Phenotypes)  4.84e-02           4  \n",
       "18   (Dictybase, Phenotypes)  5.37e-02          17  \n",
       "19   (Dictybase, Phenotypes)  5.37e-02          25  \n",
       "20   (Dictybase, Phenotypes)  5.37e-02           5  \n",
       "21   (Dictybase, Phenotypes)  5.97e-02           3  \n",
       "22   (Dictybase, Phenotypes)  5.97e-02           3  \n",
       "23   (Dictybase, Phenotypes)  9.66e-02           4  \n",
       "24   (Dictybase, Phenotypes)  9.68e-02          17  \n",
       "25   (Dictybase, Phenotypes)  1.16e-01          65  \n",
       "26  (GO, molecular_function)  1.38e-01          45  \n",
       "27   (Dictybase, Phenotypes)  1.54e-01           3  \n",
       "28   (Dictybase, Phenotypes)  1.54e-01           3  \n",
       "29   (Dictybase, Phenotypes)  1.54e-01           3  \n",
       "30   (Dictybase, Phenotypes)  1.54e-01          13  \n",
       "31          (KEGG, Pathways)  1.54e-01           6  \n",
       "32  (GO, biological_process)  1.61e-01          21  \n",
       "33          (KEGG, Pathways)  1.86e-01           6  \n",
       "34   (Dictybase, Phenotypes)  1.97e-01          14  \n",
       "35   (Dictybase, Phenotypes)  1.97e-01          12  \n",
       "36   (Dictybase, Phenotypes)  1.99e-01           2  \n",
       "37   (Dictybase, Phenotypes)  1.99e-01           8  \n",
       "38   (Dictybase, Phenotypes)  1.99e-01           2  \n",
       "39   (Dictybase, Phenotypes)  1.99e-01           4  \n",
       "40   (Dictybase, Phenotypes)  1.99e-01           2  \n",
       "41   (Dictybase, Phenotypes)  1.99e-01           2  \n",
       "42   (Dictybase, Phenotypes)  1.99e-01           5  \n",
       "43   (Dictybase, Phenotypes)  1.99e-01           2  \n",
       "44   (Dictybase, Phenotypes)  1.99e-01          20  \n",
       "45   (Dictybase, Phenotypes)  1.99e-01           2  \n",
       "46  (GO, cellular_component)  1.99e-01           4  \n",
       "47  (GO, biological_process)  1.99e-01           6  \n",
       "48   (Dictybase, Phenotypes)  2.06e-01           3  \n",
       "49   (Dictybase, Phenotypes)  2.06e-01           3  \n",
       "50   (Dictybase, Phenotypes)  2.06e-01           3  \n",
       "51   (Dictybase, Phenotypes)  2.06e-01           3  \n",
       "52   (Dictybase, Phenotypes)  2.06e-01           3  \n",
       "53          (KEGG, Pathways)  2.47e-01           5  \n",
       "54   (Dictybase, Phenotypes)  2.47e-01           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 2344 with EID: 2328\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.27\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 315 out of 629 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.33e-11</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.01e-11</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.78e-09</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.45e-05</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.60e-05</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.46e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.46e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.46e-03</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.76e-03</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.70e-02</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased prestalk 0 cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.50e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.23e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.23e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.11e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.11e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.11e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>abolished DIF production</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.56e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>precocious spore germination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.56e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>thin standing slug</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.56e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant spore morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.56e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.56e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.21e-01</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.21e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.21e-01</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>abolished autophagic cell death</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.27e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>abolished autophagic vacuole formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.27e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>increased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.67e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.94e-01</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.98e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>increased sorus size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>aberrant prestalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.32e-01</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.50e-01</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0   anatomical structure formation involved in morphogenesis   \n",
       "1                                       cell differentiation   \n",
       "2                                       extracellular region   \n",
       "3                           anatomical structure development   \n",
       "4                             aberrant spore coat morphology   \n",
       "5                           external encapsulating structure   \n",
       "6                                                  cell wall   \n",
       "7                              Starch and sucrose metabolism   \n",
       "8                                  decreased spore viability   \n",
       "9                                       aberrant culmination   \n",
       "10                   increased prespore cell differentiation   \n",
       "11                 decreased prestalk 0 cell differentiation   \n",
       "12                 development arrests at tipped mound stage   \n",
       "13                      cell wall organization or biogenesis   \n",
       "14                                   aberrant slug migration   \n",
       "15                                           spores inviable   \n",
       "16                               secondary metabolic process   \n",
       "17                                  abolished DIF production   \n",
       "18                              precocious spore germination   \n",
       "19                                        thin standing slug   \n",
       "20                                 aberrant spore morphology   \n",
       "21                 DNA-binding transcription factor activity   \n",
       "22                                     abolished culmination   \n",
       "23                 SNARE interactions in vesicular transport   \n",
       "24                                           kinase activity   \n",
       "25                           abolished autophagic cell death   \n",
       "26                    abolished autophagic vacuole formation   \n",
       "27                                 increased gene expression   \n",
       "28                                           Golgi apparatus   \n",
       "29       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "30                                      increased sorus size   \n",
       "31                                 decreased gene expression   \n",
       "32                    aberrant prestalk cell differentiation   \n",
       "33                                     decreased sporulation   \n",
       "34                                      Galactose metabolism   \n",
       "35                         aberrant fruiting body morphology   \n",
       "36                                               DNA binding   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  1.33e-11          40  \n",
       "1   (GO, biological_process)  7.01e-11          59  \n",
       "2   (GO, cellular_component)  8.78e-09          58  \n",
       "3   (GO, biological_process)  2.45e-05         109  \n",
       "4    (Dictybase, Phenotypes)  4.60e-05          10  \n",
       "5   (GO, cellular_component)  1.46e-03           9  \n",
       "6   (GO, cellular_component)  1.46e-03           9  \n",
       "7           (KEGG, Pathways)  1.46e-03          15  \n",
       "8    (Dictybase, Phenotypes)  8.76e-03          21  \n",
       "9    (Dictybase, Phenotypes)  1.70e-02          27  \n",
       "10   (Dictybase, Phenotypes)  2.26e-02           5  \n",
       "11   (Dictybase, Phenotypes)  2.50e-02           4  \n",
       "12   (Dictybase, Phenotypes)  3.23e-02           7  \n",
       "13  (GO, biological_process)  3.23e-02           8  \n",
       "14   (Dictybase, Phenotypes)  8.11e-02          18  \n",
       "15   (Dictybase, Phenotypes)  8.11e-02           4  \n",
       "16  (GO, biological_process)  8.11e-02           4  \n",
       "17   (Dictybase, Phenotypes)  8.56e-02           3  \n",
       "18   (Dictybase, Phenotypes)  8.56e-02           3  \n",
       "19   (Dictybase, Phenotypes)  8.56e-02           3  \n",
       "20   (Dictybase, Phenotypes)  8.56e-02           5  \n",
       "21  (GO, molecular_function)  8.56e-02          13  \n",
       "22   (Dictybase, Phenotypes)  1.21e-01          18  \n",
       "23          (KEGG, Pathways)  1.21e-01           9  \n",
       "24  (GO, molecular_function)  1.21e-01          69  \n",
       "25   (Dictybase, Phenotypes)  1.27e-01           4  \n",
       "26   (Dictybase, Phenotypes)  1.27e-01           4  \n",
       "27   (Dictybase, Phenotypes)  1.67e-01          14  \n",
       "28  (GO, cellular_component)  1.94e-01          22  \n",
       "29          (KEGG, Pathways)  1.98e-01           5  \n",
       "30   (Dictybase, Phenotypes)  2.07e-01           3  \n",
       "31   (Dictybase, Phenotypes)  2.07e-01          22  \n",
       "32   (Dictybase, Phenotypes)  2.11e-01           4  \n",
       "33   (Dictybase, Phenotypes)  2.11e-01          15  \n",
       "34          (KEGG, Pathways)  2.11e-01           6  \n",
       "35   (Dictybase, Phenotypes)  2.32e-01          24  \n",
       "36  (GO, molecular_function)  2.50e-01          47  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  yem selected: 2231 with EID: 2218\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.34\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 56 out of 760 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Glycerolipid metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                               secondary metabolic process   \n",
       "1                                   Glycerolipid metabolism   \n",
       "2  anatomical structure formation involved in morphogenesis   \n",
       "3                 DNA-binding transcription factor activity   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, biological_process)  1.34e-01           5  \n",
       "1          (KEGG, Pathways)  1.45e-01          10  \n",
       "2  (GO, biological_process)  1.45e-01          26  \n",
       "3  (GO, molecular_function)  1.45e-01          16  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 1970 with EID: 1958\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 277 out of 494 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.97e-06</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.97e-06</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.88e-06</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.12e-04</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.81e-04</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.51e-02</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.82e-02</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.82e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.82e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.82e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.86e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.86e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>abolished DIF production</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.86e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.86e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.79e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>decreased prestalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.79e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.88e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>increased sorus size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>decreased prestalk 0 cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.97e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased prestalk A cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>abolished outer basal disc formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>abolished tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>aberrant autophagic cell death</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.11e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>delayed culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1                                       cell differentiation   \n",
       "2   anatomical structure formation involved in morphogenesis   \n",
       "3                           anatomical structure development   \n",
       "4                              Starch and sucrose metabolism   \n",
       "5                                       aberrant culmination   \n",
       "6                                                  wild type   \n",
       "7                                            spores inviable   \n",
       "8                  SNARE interactions in vesicular transport   \n",
       "9                                secondary metabolic process   \n",
       "10                 development arrests at tipped mound stage   \n",
       "11                                 decreased spore viability   \n",
       "12                                  abolished DIF production   \n",
       "13                            aberrant spore coat morphology   \n",
       "14                   increased prespore cell differentiation   \n",
       "15                   decreased prestalk cell differentiation   \n",
       "16                                     abolished culmination   \n",
       "17       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "18                                      increased sorus size   \n",
       "19                 decreased prestalk 0 cell differentiation   \n",
       "20                                           Golgi apparatus   \n",
       "21                                           kinase activity   \n",
       "22                                 decreased gene expression   \n",
       "23                                      extracellular matrix   \n",
       "24                        development arrests at mound stage   \n",
       "25                 decreased prestalk A cell differentiation   \n",
       "26                          abolished cellulose biosynthesis   \n",
       "27                      abolished outer basal disc formation   \n",
       "28                                   abolished tip formation   \n",
       "29                            aberrant autophagic cell death   \n",
       "30                                       signal transduction   \n",
       "31                 DNA-binding transcription factor activity   \n",
       "32                                   delayed gene expression   \n",
       "33                                       delayed culmination   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  1.97e-06          46  \n",
       "1   (GO, biological_process)  1.97e-06          44  \n",
       "2   (GO, biological_process)  2.88e-06          28  \n",
       "3   (GO, biological_process)  5.12e-04          86  \n",
       "4           (KEGG, Pathways)  5.81e-04          14  \n",
       "5    (Dictybase, Phenotypes)  2.51e-02          23  \n",
       "6    (Dictybase, Phenotypes)  4.82e-02          62  \n",
       "7    (Dictybase, Phenotypes)  4.82e-02           4  \n",
       "8           (KEGG, Pathways)  4.82e-02           9  \n",
       "9   (GO, biological_process)  4.82e-02           4  \n",
       "10   (Dictybase, Phenotypes)  5.86e-02           6  \n",
       "11   (Dictybase, Phenotypes)  5.86e-02          16  \n",
       "12   (Dictybase, Phenotypes)  5.86e-02           3  \n",
       "13   (Dictybase, Phenotypes)  5.86e-02           6  \n",
       "14   (Dictybase, Phenotypes)  7.79e-02           4  \n",
       "15   (Dictybase, Phenotypes)  7.79e-02           4  \n",
       "16   (Dictybase, Phenotypes)  7.88e-02          16  \n",
       "17          (KEGG, Pathways)  1.04e-01           5  \n",
       "18   (Dictybase, Phenotypes)  1.46e-01           3  \n",
       "19   (Dictybase, Phenotypes)  1.46e-01           3  \n",
       "20  (GO, cellular_component)  1.46e-01          19  \n",
       "21  (GO, molecular_function)  1.46e-01          56  \n",
       "22   (Dictybase, Phenotypes)  1.59e-01          19  \n",
       "23  (GO, cellular_component)  1.97e-01           4  \n",
       "24   (Dictybase, Phenotypes)  2.11e-01          13  \n",
       "25   (Dictybase, Phenotypes)  2.11e-01           2  \n",
       "26   (Dictybase, Phenotypes)  2.11e-01           3  \n",
       "27   (Dictybase, Phenotypes)  2.11e-01           2  \n",
       "28   (Dictybase, Phenotypes)  2.11e-01           2  \n",
       "29   (Dictybase, Phenotypes)  2.11e-01           3  \n",
       "30  (GO, biological_process)  2.11e-01          58  \n",
       "31  (GO, molecular_function)  2.11e-01          10  \n",
       "32   (Dictybase, Phenotypes)  2.29e-01           6  \n",
       "33   (Dictybase, Phenotypes)  2.29e-01           8  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_de+'nobatchrep_combined.tsv',index_col=0,sep='\\t')\n",
    "for stage in [phenotype for phenotype in PHENOTYPES if phenotype in list(milestones['Stage'].unique())]:\n",
    "    query_names=list(milestones.query('log2FoldChange>=1 & padj <= 0.05 & Stage==\"'+stage+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DESeq2 WT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### DESeq2 WT lFC >= 2 and padj <= 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 878 with EID: 873\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.39\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 260 out of 337 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.61e-19</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.12e-15</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.53e-15</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.53e-15</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.00e-10</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.79e-06</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.93e-04</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.16e-03</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.40e-03</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.05e-03</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>lipid metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.27e-03</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.85e-03</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Porphyrin and chlorophyll metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.20e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.96e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.22e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.51e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.26e-02</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.26e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Glycine</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.26e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cellular amino acid metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.78e-02</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.10e-02</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>tRNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.37e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.37e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Vitamin B6 metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.69e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Lysine biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.02e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.29e-01</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.36e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>Butanoate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>protein-macromolecule adaptor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.03e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>lysosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.41e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Gene set                  Ontology  \\\n",
       "0                                 Ribosome          (KEGG, Pathways)   \n",
       "1       structural constituent of ribosome  (GO, molecular_function)   \n",
       "2                                 ribosome  (GO, cellular_component)   \n",
       "3                              translation  (GO, biological_process)   \n",
       "4             structural molecule activity  (GO, molecular_function)   \n",
       "5                            mitochondrion  (GO, cellular_component)   \n",
       "6                     Steroid biosynthesis          (KEGG, Pathways)   \n",
       "7        Ribosome biogenesis in eukaryotes          (KEGG, Pathways)   \n",
       "8                  oxidoreductase activity  (GO, molecular_function)   \n",
       "9    Biosynthesis of secondary metabolites          (KEGG, Pathways)   \n",
       "10                 lipid metabolic process  (GO, biological_process)   \n",
       "11                               nucleolus  (GO, cellular_component)   \n",
       "12    Porphyrin and chlorophyll metabolism          (KEGG, Pathways)   \n",
       "13                      hydrolase activity  (GO, molecular_function)   \n",
       "14             Aminoacyl-tRNA biosynthesis          (KEGG, Pathways)   \n",
       "15              methyltransferase activity  (GO, molecular_function)   \n",
       "16                      Metabolic pathways          (KEGG, Pathways)   \n",
       "17                   Fatty acid elongation          (KEGG, Pathways)   \n",
       "18                                 Glycine          (KEGG, Pathways)   \n",
       "19   cellular amino acid metabolic process  (GO, biological_process)   \n",
       "20                             RNA binding  (GO, molecular_function)   \n",
       "21                  tRNA metabolic process  (GO, biological_process)   \n",
       "22                     ribosome biogenesis  (GO, biological_process)   \n",
       "23                   Vitamin B6 metabolism          (KEGG, Pathways)   \n",
       "24                     Lysine biosynthesis          (KEGG, Pathways)   \n",
       "25                    extracellular region  (GO, cellular_component)   \n",
       "26             Biosynthesis of amino acids          (KEGG, Pathways)   \n",
       "27                         ligase activity  (GO, molecular_function)   \n",
       "28                    Butanoate metabolism          (KEGG, Pathways)   \n",
       "29                      isomerase activity  (GO, molecular_function)   \n",
       "30  protein-macromolecule adaptor activity  (GO, molecular_function)   \n",
       "31                                lysosome  (GO, cellular_component)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   2.61e-19          43  \n",
       "1   2.12e-15          36  \n",
       "2   4.53e-15          37  \n",
       "3   4.53e-15          51  \n",
       "4   1.00e-10          36  \n",
       "5   1.79e-06          45  \n",
       "6   1.93e-04           9  \n",
       "7   1.16e-03          17  \n",
       "8   1.40e-03          35  \n",
       "9   2.05e-03          43  \n",
       "10  3.27e-03          28  \n",
       "11  5.85e-03          16  \n",
       "12  1.20e-02           6  \n",
       "13  1.96e-02          11  \n",
       "14  2.22e-02          11  \n",
       "15  3.51e-02          12  \n",
       "16  4.26e-02          76  \n",
       "17  4.26e-02           5  \n",
       "18  4.26e-02           7  \n",
       "19  4.78e-02          20  \n",
       "20  5.10e-02          25  \n",
       "21  5.37e-02          13  \n",
       "22  5.37e-02          12  \n",
       "23  5.69e-02           3  \n",
       "24  1.02e-01           3  \n",
       "25  1.29e-01          22  \n",
       "26  1.36e-01          11  \n",
       "27  1.53e-01          13  \n",
       "28  1.75e-01           4  \n",
       "29  2.00e-01           9  \n",
       "30  2.03e-01           3  \n",
       "31  2.41e-01          11  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 87 with EID: 86\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.17\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.38e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.38e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.38e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.23e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.23e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.06e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.06e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>GTPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.21e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Gene set                  Ontology  \\\n",
       "0                    precocious aggregation   (Dictybase, Phenotypes)   \n",
       "1  development arrests at loose mound stage   (Dictybase, Phenotypes)   \n",
       "2                             cell adhesion  (GO, biological_process)   \n",
       "3                     decreased sporulation   (Dictybase, Phenotypes)   \n",
       "4                           plasma membrane  (GO, cellular_component)   \n",
       "5              decreased fruiting body size   (Dictybase, Phenotypes)   \n",
       "6                   decreased cell motility   (Dictybase, Phenotypes)   \n",
       "7                      extracellular region  (GO, cellular_component)   \n",
       "8                           GTPase activity  (GO, molecular_function)   \n",
       "\n",
       "        FDR  N in query  \n",
       "0  4.38e-02           2  \n",
       "1  4.38e-02           2  \n",
       "2  4.38e-02           3  \n",
       "3  8.23e-02           2  \n",
       "4  8.23e-02           4  \n",
       "5  1.06e-01           2  \n",
       "6  1.06e-01           2  \n",
       "7  2.21e-01           2  \n",
       "8  2.21e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 70 with EID: 69\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.22\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 15 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.98e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.41e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.41e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.41e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.41e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.82e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>GTPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.31e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.80e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.04e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.34e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Gene set                  Ontology       FDR  \\\n",
       "0       abolished slug migration   (Dictybase, Phenotypes)  2.98e-02   \n",
       "1             aberrant streaming   (Dictybase, Phenotypes)  3.41e-02   \n",
       "2           aberrant culmination   (Dictybase, Phenotypes)  3.41e-02   \n",
       "3           aberrant aggregation   (Dictybase, Phenotypes)  3.41e-02   \n",
       "4                plasma membrane  (GO, cellular_component)  3.41e-02   \n",
       "5   aberrant cAMP wave formation   (Dictybase, Phenotypes)  3.82e-02   \n",
       "6                GTPase activity  (GO, molecular_function)  7.31e-02   \n",
       "7        aberrant slug migration   (Dictybase, Phenotypes)  7.80e-02   \n",
       "8                   reproduction  (GO, biological_process)  1.04e-01   \n",
       "9                  cell adhesion  (GO, biological_process)  1.09e-01   \n",
       "10           signal transduction  (GO, biological_process)  2.34e-01   \n",
       "\n",
       "    N in query  \n",
       "0            3  \n",
       "1            3  \n",
       "2            3  \n",
       "3            3  \n",
       "4            5  \n",
       "5            3  \n",
       "6            3  \n",
       "7            2  \n",
       "8            2  \n",
       "9            2  \n",
       "10           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 24 with EID: 24\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.17\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 4 out of 4 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.74e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.74e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.74e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.97e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Gene set                  Ontology       FDR  \\\n",
       "0  Biosynthesis of secondary metabolites          (KEGG, Pathways)  9.74e-02   \n",
       "1       small molecule metabolic process  (GO, biological_process)  9.74e-02   \n",
       "2                        kinase activity  (GO, molecular_function)  9.74e-02   \n",
       "3                     Metabolic pathways          (KEGG, Pathways)  1.97e-01   \n",
       "\n",
       "   N in query  \n",
       "0           2  \n",
       "1           2  \n",
       "2           2  \n",
       "3           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 60 with EID: 60\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.37\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 22 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.39e-08</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.46e-08</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.69e-08</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.24e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.66e-07</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.66e-07</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.25e-06</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.88e-03</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.96e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.57e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.03e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.07e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.67e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.67e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.90e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.48e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.48e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.48e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Gene set                  Ontology       FDR  \\\n",
       "0                         cell cycle  (GO, biological_process)  1.39e-08   \n",
       "1                         chromosome  (GO, cellular_component)  4.46e-08   \n",
       "2            chromosome organization  (GO, biological_process)  4.69e-08   \n",
       "3           mitotic nuclear division  (GO, biological_process)  1.24e-07   \n",
       "4             chromosome segregation  (GO, biological_process)  2.66e-07   \n",
       "5                 mitotic cell cycle  (GO, biological_process)  2.66e-07   \n",
       "6                    DNA replication          (KEGG, Pathways)  5.25e-06   \n",
       "7                  cell cycle arrest   (Dictybase, Phenotypes)  1.88e-03   \n",
       "8                      cell division  (GO, biological_process)  1.96e-03   \n",
       "9              Pyrimidine metabolism          (KEGG, Pathways)  2.57e-03   \n",
       "10                   ATPase activity  (GO, molecular_function)  9.03e-03   \n",
       "11     microtubule organizing center  (GO, cellular_component)  1.07e-02   \n",
       "12                   Mismatch repair          (KEGG, Pathways)  1.89e-02   \n",
       "13                      cytoskeleton  (GO, cellular_component)  5.67e-02   \n",
       "14        Nucleotide excision repair          (KEGG, Pathways)  5.67e-02   \n",
       "15             DNA metabolic process  (GO, biological_process)  5.90e-02   \n",
       "16                 helicase activity  (GO, molecular_function)  7.48e-02   \n",
       "17                       DNA binding  (GO, molecular_function)  7.48e-02   \n",
       "18  small molecule metabolic process  (GO, biological_process)  2.48e-01   \n",
       "\n",
       "    N in query  \n",
       "0           11  \n",
       "1            8  \n",
       "2            8  \n",
       "3            5  \n",
       "4            5  \n",
       "5            8  \n",
       "6            5  \n",
       "7            2  \n",
       "8            5  \n",
       "9            3  \n",
       "10           5  \n",
       "11           3  \n",
       "12           2  \n",
       "13           4  \n",
       "14           2  \n",
       "15           3  \n",
       "16           2  \n",
       "17           4  \n",
       "18           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 148 with EID: 148\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.14\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 11 out of 21 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.01e-05</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.88e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.88e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene set                  Ontology       FDR  \\\n",
       "0              extracellular region  (GO, cellular_component)  4.01e-05   \n",
       "1  external encapsulating structure  (GO, cellular_component)  3.88e-02   \n",
       "2                         cell wall  (GO, cellular_component)  3.88e-02   \n",
       "3            Glutathione metabolism          (KEGG, Pathways)  1.08e-01   \n",
       "\n",
       "   N in query  \n",
       "0           8  \n",
       "1           2  \n",
       "2           2  \n",
       "3           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 545 with EID: 544\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.24\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 63 out of 132 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.72e-07</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.45e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.83e-05</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.83e-05</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.73e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.21e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.68e-03</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.02e-02</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.31e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.72e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.77e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>precocious spore germination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.12e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>thin standing slug</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.12e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.12e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.37e-02</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1                             aberrant spore coat morphology   \n",
       "2                           external encapsulating structure   \n",
       "3                                                  cell wall   \n",
       "4                       cell wall organization or biogenesis   \n",
       "5                                       extracellular matrix   \n",
       "6   anatomical structure formation involved in morphogenesis   \n",
       "7                           anatomical structure development   \n",
       "8                                  decreased spore viability   \n",
       "9                                       cell differentiation   \n",
       "10                               abolished cellulose binding   \n",
       "11                              precocious spore germination   \n",
       "12                                        thin standing slug   \n",
       "13                             Starch and sucrose metabolism   \n",
       "14                                                 wild type   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  1.72e-07          23  \n",
       "1    (Dictybase, Phenotypes)  2.45e-06           7  \n",
       "2   (GO, cellular_component)  7.83e-05           6  \n",
       "3   (GO, cellular_component)  7.83e-05           6  \n",
       "4   (GO, biological_process)  1.73e-04           6  \n",
       "5   (GO, cellular_component)  2.21e-03           4  \n",
       "6   (GO, biological_process)  3.68e-03          10  \n",
       "7   (GO, biological_process)  1.02e-02          28  \n",
       "8    (Dictybase, Phenotypes)  1.31e-02           8  \n",
       "9   (GO, biological_process)  1.72e-02          13  \n",
       "10   (Dictybase, Phenotypes)  1.77e-02           2  \n",
       "11   (Dictybase, Phenotypes)  4.12e-02           2  \n",
       "12   (Dictybase, Phenotypes)  4.12e-02           2  \n",
       "13          (KEGG, Pathways)  4.12e-02           5  \n",
       "14   (Dictybase, Phenotypes)  9.37e-02          20  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 944 with EID: 936\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 94 out of 211 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.27e-10</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.40e-04</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.46e-04</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.79e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.70e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>precocious spore germination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.36e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.70e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.42e-02</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.80e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.52e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.05e-02</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.83e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.41e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       extracellular region   \n",
       "1   anatomical structure formation involved in morphogenesis   \n",
       "2                                       cell differentiation   \n",
       "3                             aberrant spore coat morphology   \n",
       "4                       cell wall organization or biogenesis   \n",
       "5                               precocious spore germination   \n",
       "6                                       extracellular matrix   \n",
       "7                           anatomical structure development   \n",
       "8                              Starch and sucrose metabolism   \n",
       "9                                secondary metabolic process   \n",
       "10                                                 wild type   \n",
       "11                                        hydrolase activity   \n",
       "12                                 decreased spore viability   \n",
       "13       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  7.27e-10          33  \n",
       "1   (GO, biological_process)  1.40e-04          16  \n",
       "2   (GO, biological_process)  1.46e-04          23  \n",
       "3    (Dictybase, Phenotypes)  9.79e-04           6  \n",
       "4   (GO, biological_process)  3.70e-03           6  \n",
       "5    (Dictybase, Phenotypes)  6.36e-03           3  \n",
       "6   (GO, cellular_component)  1.70e-02           4  \n",
       "7   (GO, biological_process)  2.42e-02          39  \n",
       "8           (KEGG, Pathways)  2.80e-02           7  \n",
       "9   (GO, biological_process)  3.52e-02           3  \n",
       "10   (Dictybase, Phenotypes)  4.05e-02          31  \n",
       "11  (GO, molecular_function)  4.83e-02           8  \n",
       "12   (Dictybase, Phenotypes)  2.41e-01           8  \n",
       "13          (KEGG, Pathways)  2.47e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 307 with EID: 303\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 21 out of 69 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.90e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.90e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Fructose and mannose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Pentose and glucuronate interconversions</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Gene set                  Ontology  \\\n",
       "0                      extracellular region  (GO, cellular_component)   \n",
       "1                   oxidoreductase activity  (GO, molecular_function)   \n",
       "2           Fructose and mannose metabolism          (KEGG, Pathways)   \n",
       "3  Pentose and glucuronate interconversions          (KEGG, Pathways)   \n",
       "\n",
       "        FDR  N in query  \n",
       "0  9.90e-02           9  \n",
       "1  9.90e-02          11  \n",
       "2  1.46e-01           3  \n",
       "3  1.46e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_de+'WT_batchrep_combined.tsv',index_col=0,sep='\\t')\n",
    "for stage in [phenotype for phenotype in PHENOTYPES if phenotype in list(milestones['Stage'].unique())]:\n",
    "    query_names=list(milestones.query('log2FoldChange>=2 & padj <= 0.01 & Stage==\"'+stage+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### DESeq2 WT lFC >= 1 and padj <= 0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 2578 with EID: 2560\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 640 out of 968 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.37e-52</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.25e-46</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.69e-45</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.11e-36</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.22e-30</td>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.34e-22</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.73e-19</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.65e-12</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.43e-11</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.13e-09</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>tRNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.03e-08</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.91e-08</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.56e-07</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.98e-07</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.24e-05</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.50e-05</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.05e-05</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.31e-04</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cellular amino acid metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.42e-04</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.42e-04</td>\n",
       "      <td>121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.25e-04</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.27e-02</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.42e-02</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>cofactor metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>mitochondrion organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.62e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Porphyrin and chlorophyll metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.85e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.94e-02</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.61e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.61e-02</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.61e-02</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Sulfur relay system</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.19e-01</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>Purine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.30e-01</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                  Ontology  \\\n",
       "0                                         Ribosome          (KEGG, Pathways)   \n",
       "1                                         ribosome  (GO, cellular_component)   \n",
       "2               structural constituent of ribosome  (GO, molecular_function)   \n",
       "3                                      translation  (GO, biological_process)   \n",
       "4                     structural molecule activity  (GO, molecular_function)   \n",
       "5                                    mitochondrion  (GO, cellular_component)   \n",
       "6                              ribosome biogenesis  (GO, biological_process)   \n",
       "7                                        nucleolus  (GO, cellular_component)   \n",
       "8                Ribosome biogenesis in eukaryotes          (KEGG, Pathways)   \n",
       "9                                      RNA binding  (GO, molecular_function)   \n",
       "10                          tRNA metabolic process  (GO, biological_process)   \n",
       "11                     Aminoacyl-tRNA biosynthesis          (KEGG, Pathways)   \n",
       "12                              Metabolic pathways          (KEGG, Pathways)   \n",
       "13                       Oxidative phosphorylation          (KEGG, Pathways)   \n",
       "14                                    rRNA binding  (GO, molecular_function)   \n",
       "15                      methyltransferase activity  (GO, molecular_function)   \n",
       "16                            Steroid biosynthesis          (KEGG, Pathways)   \n",
       "17                                 ligase activity  (GO, molecular_function)   \n",
       "18           cellular amino acid metabolic process  (GO, biological_process)   \n",
       "19                small molecule metabolic process  (GO, biological_process)   \n",
       "20           Biosynthesis of secondary metabolites          (KEGG, Pathways)   \n",
       "21  generation of precursor metabolites and energy  (GO, biological_process)   \n",
       "22                         oxidoreductase activity  (GO, molecular_function)   \n",
       "23                      cofactor metabolic process  (GO, biological_process)   \n",
       "24                      mitochondrion organization  (GO, biological_process)   \n",
       "25            Porphyrin and chlorophyll metabolism          (KEGG, Pathways)   \n",
       "26                        unfolded protein binding  (GO, molecular_function)   \n",
       "27                           Fatty acid elongation          (KEGG, Pathways)   \n",
       "28                           Pyrimidine metabolism          (KEGG, Pathways)   \n",
       "29                              isomerase activity  (GO, molecular_function)   \n",
       "30                             Sulfur relay system          (KEGG, Pathways)   \n",
       "31                                 protein folding  (GO, biological_process)   \n",
       "32                               Purine metabolism          (KEGG, Pathways)   \n",
       "33              transmembrane transporter activity  (GO, molecular_function)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   4.37e-52          94  \n",
       "1   3.25e-46          88  \n",
       "2   4.69e-45          82  \n",
       "3   1.11e-36         120  \n",
       "4   1.22e-30          89  \n",
       "5   6.34e-22         125  \n",
       "6   1.73e-19          50  \n",
       "7   1.65e-12          48  \n",
       "8   1.43e-11          45  \n",
       "9   5.13e-09          79  \n",
       "10  2.03e-08          41  \n",
       "11  2.91e-08          31  \n",
       "12  2.56e-07         223  \n",
       "13  2.98e-07          33  \n",
       "14  1.24e-05          12  \n",
       "15  6.50e-05          30  \n",
       "16  7.05e-05          14  \n",
       "17  1.31e-04          38  \n",
       "18  2.42e-04          52  \n",
       "19  2.42e-04         121  \n",
       "20  4.25e-04          99  \n",
       "21  1.27e-02          32  \n",
       "22  1.42e-02          71  \n",
       "23  1.61e-02          30  \n",
       "24  1.62e-02          13  \n",
       "25  2.85e-02           9  \n",
       "26  2.94e-02          14  \n",
       "27  6.61e-02           8  \n",
       "28  6.61e-02          14  \n",
       "29  6.61e-02          21  \n",
       "30  1.16e-01           6  \n",
       "31  1.19e-01          16  \n",
       "32  1.34e-01          20  \n",
       "33  2.30e-01          48  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 557 with EID: 553\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.35\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 88 out of 194 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.43e-40</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.46e-09</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.26e-07</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.01e-05</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.53e-05</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.21e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Protein export</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.18e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.85e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>curly fingers</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.85e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased contractile vacuole discharge</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.85e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.85e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.31e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Gene set                  Ontology  \\\n",
       "0                                    Proteasome          (KEGG, Pathways)   \n",
       "1                            peptidase activity  (GO, molecular_function)   \n",
       "2                             catabolic process  (GO, biological_process)   \n",
       "3   Protein processing in endoplasmic reticulum          (KEGG, Pathways)   \n",
       "4                         endoplasmic reticulum  (GO, cellular_component)   \n",
       "5        Various types of N-glycan biosynthesis          (KEGG, Pathways)   \n",
       "6                                Protein export          (KEGG, Pathways)   \n",
       "7                            aberrant streaming   (Dictybase, Phenotypes)   \n",
       "8                                 curly fingers   (Dictybase, Phenotypes)   \n",
       "9       decreased contractile vacuole discharge   (Dictybase, Phenotypes)   \n",
       "10     development arrests at loose mound stage   (Dictybase, Phenotypes)   \n",
       "11                        N-Glycan biosynthesis          (KEGG, Pathways)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   3.43e-40          34  \n",
       "1   3.46e-09          23  \n",
       "2   6.26e-07          35  \n",
       "3   1.01e-05          18  \n",
       "4   3.53e-05          20  \n",
       "5   2.21e-02           6  \n",
       "6   5.18e-02           5  \n",
       "7   5.85e-02          10  \n",
       "8   5.85e-02           2  \n",
       "9   5.85e-02           2  \n",
       "10  5.85e-02           5  \n",
       "11  6.31e-02           6  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 509 with EID: 505\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.36\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 136 out of 183 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.39e-21</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.82e-03</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.82e-03</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.90e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.97e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>curly fingers</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased contractile vacuole discharge</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.73e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.49e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.59e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant contractile vacuole morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.59e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.67e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.03e-02</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.03e-02</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.08e-02</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.08e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.08e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>aberrant binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.08e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.08e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>decreased pinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>precocious culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Other glycan degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.13e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.20e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>aberrant positive regulation of chemorepellant activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.60e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.72e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>decreased phagocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.96e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>increased protein level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.48e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                                Proteasome   \n",
       "1                                           plasma membrane   \n",
       "2                                        peptidase activity   \n",
       "3                                        aberrant streaming   \n",
       "4                                  decreased aggregate size   \n",
       "5                                  aberrant cell morphology   \n",
       "6                                      aberrant cytokinesis   \n",
       "7                                             curly fingers   \n",
       "8                   decreased contractile vacuole discharge   \n",
       "9                  development arrests at loose mound stage   \n",
       "10                                    endoplasmic reticulum   \n",
       "11                   Various types of N-glycan biosynthesis   \n",
       "12                                               locomotion   \n",
       "13                                      signal transduction   \n",
       "14                                        catabolic process   \n",
       "15                                cytoskeleton organization   \n",
       "16                             cytoskeletal protein binding   \n",
       "17                                       cell morphogenesis   \n",
       "18                             aberrant cAMP wave formation   \n",
       "19                  aberrant contractile vacuole morphology   \n",
       "20                                             cytoskeleton   \n",
       "21                                     extracellular region   \n",
       "22                         anatomical structure development   \n",
       "23                                                wild type   \n",
       "24                                    decreased culmination   \n",
       "25                            increased basal F-actin level   \n",
       "26                                         aberrant binding   \n",
       "27                                            cell motility   \n",
       "28                                    decreased pinocytosis   \n",
       "29                                   precocious culmination   \n",
       "30                                 Other glycan degradation   \n",
       "31                                      delayed development   \n",
       "32                                            cell adhesion   \n",
       "33                               multiple tips in culminant   \n",
       "34                                     aberrant culmination   \n",
       "35  aberrant positive regulation of chemorepellant activity   \n",
       "36              Protein processing in endoplasmic reticulum   \n",
       "37                                     aberrant aggregation   \n",
       "38                             decreased fruiting body size   \n",
       "39                                    decreased growth rate   \n",
       "40                                   decreased phagocytosis   \n",
       "41                            aberrant F-actin organization   \n",
       "42                                         ABC transporters   \n",
       "43                abolished activation of adenylate cyclase   \n",
       "44                       development arrests at mound stage   \n",
       "45                                  increased protein level   \n",
       "46                                 increased slug migration   \n",
       "47                                 abolished slug migration   \n",
       "48                                    N-Glycan biosynthesis   \n",
       "49                                       mitotic cell cycle   \n",
       "50                                    increased growth rate   \n",
       "51                                enzyme regulator activity   \n",
       "52                                  decreased cell motility   \n",
       "53                                   precocious aggregation   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0           (KEGG, Pathways)  1.39e-21          24  \n",
       "1   (GO, cellular_component)  7.82e-03          27  \n",
       "2   (GO, molecular_function)  7.82e-03          14  \n",
       "3    (Dictybase, Phenotypes)  2.90e-02          11  \n",
       "4    (Dictybase, Phenotypes)  2.97e-02           7  \n",
       "5    (Dictybase, Phenotypes)  4.73e-02           7  \n",
       "6    (Dictybase, Phenotypes)  4.73e-02          12  \n",
       "7    (Dictybase, Phenotypes)  4.73e-02           2  \n",
       "8    (Dictybase, Phenotypes)  4.73e-02           2  \n",
       "9    (Dictybase, Phenotypes)  4.73e-02           5  \n",
       "10  (GO, cellular_component)  4.73e-02          13  \n",
       "11          (KEGG, Pathways)  4.73e-02           5  \n",
       "12  (GO, biological_process)  4.73e-02          18  \n",
       "13  (GO, biological_process)  4.73e-02          28  \n",
       "14  (GO, biological_process)  4.73e-02          23  \n",
       "15  (GO, biological_process)  4.73e-02          19  \n",
       "16  (GO, molecular_function)  4.73e-02          16  \n",
       "17  (GO, biological_process)  5.49e-02           5  \n",
       "18   (Dictybase, Phenotypes)  6.59e-02          11  \n",
       "19   (Dictybase, Phenotypes)  6.59e-02           3  \n",
       "20  (GO, cellular_component)  6.67e-02          17  \n",
       "21  (GO, cellular_component)  7.03e-02          15  \n",
       "22  (GO, biological_process)  7.03e-02          32  \n",
       "23   (Dictybase, Phenotypes)  7.08e-02          26  \n",
       "24   (Dictybase, Phenotypes)  7.08e-02           3  \n",
       "25   (Dictybase, Phenotypes)  7.08e-02           4  \n",
       "26   (Dictybase, Phenotypes)  7.08e-02           2  \n",
       "27  (GO, biological_process)  1.08e-01           8  \n",
       "28   (Dictybase, Phenotypes)  1.11e-01           5  \n",
       "29   (Dictybase, Phenotypes)  1.13e-01           2  \n",
       "30          (KEGG, Pathways)  1.13e-01           3  \n",
       "31   (Dictybase, Phenotypes)  1.20e-01           8  \n",
       "32  (GO, biological_process)  1.20e-01           9  \n",
       "33   (Dictybase, Phenotypes)  1.53e-01           2  \n",
       "34   (Dictybase, Phenotypes)  1.53e-01           9  \n",
       "35   (Dictybase, Phenotypes)  1.53e-01           2  \n",
       "36          (KEGG, Pathways)  1.53e-01           9  \n",
       "37   (Dictybase, Phenotypes)  1.60e-01           7  \n",
       "38   (Dictybase, Phenotypes)  1.60e-01           8  \n",
       "39   (Dictybase, Phenotypes)  1.72e-01          12  \n",
       "40   (Dictybase, Phenotypes)  1.77e-01           7  \n",
       "41   (Dictybase, Phenotypes)  1.80e-01           5  \n",
       "42          (KEGG, Pathways)  1.96e-01           3  \n",
       "43   (Dictybase, Phenotypes)  2.08e-01           2  \n",
       "44   (Dictybase, Phenotypes)  2.08e-01           6  \n",
       "45   (Dictybase, Phenotypes)  2.08e-01           2  \n",
       "46   (Dictybase, Phenotypes)  2.08e-01           2  \n",
       "47   (Dictybase, Phenotypes)  2.08e-01           5  \n",
       "48          (KEGG, Pathways)  2.08e-01           4  \n",
       "49  (GO, biological_process)  2.08e-01           9  \n",
       "50   (Dictybase, Phenotypes)  2.17e-01           4  \n",
       "51  (GO, molecular_function)  2.17e-01           9  \n",
       "52   (Dictybase, Phenotypes)  2.45e-01           7  \n",
       "53   (Dictybase, Phenotypes)  2.48e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 305 with EID: 305\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.36\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 93 out of 109 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.25e-29</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.50e-26</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.93e-19</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.37e-16</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.07e-16</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.46e-15</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.45e-14</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.47e-11</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.48e-10</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.06e-10</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.88e-10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.95e-09</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.11e-07</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.54e-06</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.19e-04</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.61e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.85e-03</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.81e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.54e-03</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.06e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.63e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.76e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.45e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.87e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.87e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>increased cell-cell adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.24e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Gene set                  Ontology       FDR  \\\n",
       "0                  DNA replication          (KEGG, Pathways)  1.25e-29   \n",
       "1                       chromosome  (GO, cellular_component)  1.50e-26   \n",
       "2                       cell cycle  (GO, biological_process)  5.93e-19   \n",
       "3          chromosome organization  (GO, biological_process)  2.37e-16   \n",
       "4            DNA metabolic process  (GO, biological_process)  3.07e-16   \n",
       "5                  Mismatch repair          (KEGG, Pathways)  1.46e-15   \n",
       "6           chromosome segregation  (GO, biological_process)  3.45e-14   \n",
       "7       Nucleotide excision repair          (KEGG, Pathways)  9.47e-11   \n",
       "8                    cell division  (GO, biological_process)  2.48e-10   \n",
       "9         mitotic nuclear division  (GO, biological_process)  4.06e-10   \n",
       "10            Base excision repair          (KEGG, Pathways)  7.88e-10   \n",
       "11                     DNA binding  (GO, molecular_function)  6.95e-09   \n",
       "12              mitotic cell cycle  (GO, biological_process)  6.11e-07   \n",
       "13              nuclear chromosome  (GO, cellular_component)  1.54e-06   \n",
       "14                 ATPase activity  (GO, molecular_function)  4.19e-04   \n",
       "15   microtubule organizing center  (GO, cellular_component)  4.61e-04   \n",
       "16  Ubiquitin mediated proteolysis          (KEGG, Pathways)  1.85e-03   \n",
       "17        Homologous recombination          (KEGG, Pathways)  3.81e-03   \n",
       "18              response to stress  (GO, biological_process)  4.54e-03   \n",
       "19                    cytoskeleton  (GO, cellular_component)  1.06e-02   \n",
       "20               cell cycle arrest   (Dictybase, Phenotypes)  4.63e-02   \n",
       "21           Pyrimidine metabolism          (KEGG, Pathways)  4.76e-02   \n",
       "22               helicase activity  (GO, molecular_function)  7.45e-02   \n",
       "23      Non-homologous end-joining          (KEGG, Pathways)  7.87e-02   \n",
       "24                 histone binding  (GO, molecular_function)  7.87e-02   \n",
       "25    increased cell-cell adhesion   (Dictybase, Phenotypes)  2.24e-01   \n",
       "\n",
       "    N in query  \n",
       "0           24  \n",
       "1           29  \n",
       "2           33  \n",
       "3           22  \n",
       "4           25  \n",
       "5           13  \n",
       "6           12  \n",
       "7           13  \n",
       "8           20  \n",
       "9            9  \n",
       "10          11  \n",
       "11          25  \n",
       "12          15  \n",
       "13           7  \n",
       "14          15  \n",
       "15           8  \n",
       "16           8  \n",
       "17           5  \n",
       "18          21  \n",
       "19          13  \n",
       "20           2  \n",
       "21           4  \n",
       "22           5  \n",
       "23           2  \n",
       "24           2  \n",
       "25           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 309 with EID: 308\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.39\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 103 out of 119 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.09e-33</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.65e-29</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.72e-23</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.55e-21</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.91e-19</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.36e-17</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.62e-15</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.23e-13</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.40e-12</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.47e-12</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.12e-12</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.73e-11</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.51e-08</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.11e-08</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.29e-05</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.36e-04</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.13e-04</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.20e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.38e-03</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.52e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.89e-03</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.12e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant nucleus morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.21e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.24e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.05e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>nuclear envelope</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.77e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Gene set                  Ontology       FDR  \\\n",
       "0                       chromosome  (GO, cellular_component)  8.09e-33   \n",
       "1                  DNA replication          (KEGG, Pathways)  5.65e-29   \n",
       "2          chromosome organization  (GO, biological_process)  2.72e-23   \n",
       "3                       cell cycle  (GO, biological_process)  5.55e-21   \n",
       "4            DNA metabolic process  (GO, biological_process)  9.91e-19   \n",
       "5           chromosome segregation  (GO, biological_process)  1.36e-17   \n",
       "6                  Mismatch repair          (KEGG, Pathways)  3.62e-15   \n",
       "7         mitotic nuclear division  (GO, biological_process)  2.23e-13   \n",
       "8                      DNA binding  (GO, molecular_function)  1.40e-12   \n",
       "9                    cell division  (GO, biological_process)  1.47e-12   \n",
       "10      Nucleotide excision repair          (KEGG, Pathways)  9.12e-12   \n",
       "11            Base excision repair          (KEGG, Pathways)  6.73e-11   \n",
       "12              mitotic cell cycle  (GO, biological_process)  3.51e-08   \n",
       "13              nuclear chromosome  (GO, cellular_component)  9.11e-08   \n",
       "14   microtubule organizing center  (GO, cellular_component)  1.29e-05   \n",
       "15  Ubiquitin mediated proteolysis          (KEGG, Pathways)  5.36e-04   \n",
       "16                 ATPase activity  (GO, molecular_function)  9.13e-04   \n",
       "17           Pyrimidine metabolism          (KEGG, Pathways)  1.20e-03   \n",
       "18                    cytoskeleton  (GO, cellular_component)  2.38e-03   \n",
       "19        Homologous recombination          (KEGG, Pathways)  4.52e-03   \n",
       "20                 histone binding  (GO, molecular_function)  4.89e-03   \n",
       "21                     nucleoplasm  (GO, cellular_component)  3.12e-02   \n",
       "22               cell cycle arrest   (Dictybase, Phenotypes)  4.24e-02   \n",
       "23     aberrant nucleus morphology   (Dictybase, Phenotypes)  4.24e-02   \n",
       "24      Non-homologous end-joining          (KEGG, Pathways)  8.21e-02   \n",
       "25               helicase activity  (GO, molecular_function)  8.24e-02   \n",
       "26              response to stress  (GO, biological_process)  9.05e-02   \n",
       "27                nuclear envelope  (GO, cellular_component)  1.77e-01   \n",
       "28      methyltransferase activity  (GO, molecular_function)  2.38e-01   \n",
       "\n",
       "    N in query  \n",
       "0           34  \n",
       "1           24  \n",
       "2           28  \n",
       "3           36  \n",
       "4           28  \n",
       "5           14  \n",
       "6           13  \n",
       "7           11  \n",
       "8           31  \n",
       "9           23  \n",
       "10          14  \n",
       "11          12  \n",
       "12          17  \n",
       "13           8  \n",
       "14          10  \n",
       "15           9  \n",
       "16          15  \n",
       "17           6  \n",
       "18          15  \n",
       "19           5  \n",
       "20           3  \n",
       "21           8  \n",
       "22           2  \n",
       "23           2  \n",
       "24           2  \n",
       "25           5  \n",
       "26          18  \n",
       "27           3  \n",
       "28           4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 757 with EID: 753\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.24\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 107 out of 182 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.32e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.75e-04</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.75e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.75e-04</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.75e-04</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.05e-03</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.33e-03</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.29e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.97e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.97e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.06e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.41e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.59e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>7.59e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>8.14e-02</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.14e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.14e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.14e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.14e-02</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.14e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.40e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.40e-02</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.05e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>abolished negative regulation of vegetative growth</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.25e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>aberrant myosin II organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>increased sorting to prestalk region</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>increased prestalk 0 cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>increased macropinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.07e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.10e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>extracellular space</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.14e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                     chromosome segregation   \n",
       "1                                                 chromosome   \n",
       "2                           external encapsulating structure   \n",
       "3                                                  cell wall   \n",
       "4                                            DNA replication   \n",
       "5                                    chromosome organization   \n",
       "6                                                 cell cycle   \n",
       "7                                   mitotic nuclear division   \n",
       "8                                         nuclear chromosome   \n",
       "9                                     Glutathione metabolism   \n",
       "10                                 decreased spore viability   \n",
       "11                            aberrant spore coat morphology   \n",
       "12                                          ABC transporters   \n",
       "13                                           ATPase activity   \n",
       "14                                      extracellular region   \n",
       "15  anatomical structure formation involved in morphogenesis   \n",
       "16                                             cell division   \n",
       "17                                        mitotic cell cycle   \n",
       "18                                               DNA binding   \n",
       "19                 DNA-binding transcription factor activity   \n",
       "20                                      Base excision repair   \n",
       "21                        transmembrane transporter activity   \n",
       "22                                     Pyrimidine metabolism   \n",
       "23        abolished negative regulation of vegetative growth   \n",
       "24                                      cell differentiation   \n",
       "25                           aberrant myosin II organization   \n",
       "26                      increased sorting to prestalk region   \n",
       "27                                         cell cycle arrest   \n",
       "28                 increased prestalk 0 cell differentiation   \n",
       "29                                increased macropinocytosis   \n",
       "30                                   transmembrane transport   \n",
       "31                             microtubule organizing center   \n",
       "32                                Nucleotide excision repair   \n",
       "33                   increased prespore cell differentiation   \n",
       "34                      cell wall organization or biogenesis   \n",
       "35                                       extracellular space   \n",
       "36                          anatomical structure development   \n",
       "37                                           histone binding   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  1.32e-04           8  \n",
       "1   (GO, cellular_component)  6.75e-04          13  \n",
       "2   (GO, cellular_component)  6.75e-04           6  \n",
       "3   (GO, cellular_component)  6.75e-04           6  \n",
       "4           (KEGG, Pathways)  6.75e-04           9  \n",
       "5   (GO, biological_process)  1.05e-03          13  \n",
       "6   (GO, biological_process)  2.33e-03          20  \n",
       "7   (GO, biological_process)  1.29e-02           5  \n",
       "8   (GO, cellular_component)  1.97e-02           5  \n",
       "9           (KEGG, Pathways)  1.97e-02           6  \n",
       "10   (Dictybase, Phenotypes)  3.06e-02           9  \n",
       "11   (Dictybase, Phenotypes)  3.41e-02           4  \n",
       "12          (KEGG, Pathways)  7.59e-02           4  \n",
       "13  (GO, molecular_function)  7.59e-02          16  \n",
       "14  (GO, cellular_component)  8.14e-02          15  \n",
       "15  (GO, biological_process)  8.14e-02           9  \n",
       "16  (GO, biological_process)  8.14e-02          12  \n",
       "17  (GO, biological_process)  8.14e-02          11  \n",
       "18  (GO, molecular_function)  8.14e-02          19  \n",
       "19  (GO, molecular_function)  8.14e-02           6  \n",
       "20          (KEGG, Pathways)  9.40e-02           5  \n",
       "21  (GO, molecular_function)  9.40e-02          14  \n",
       "22          (KEGG, Pathways)  1.05e-01           5  \n",
       "23   (Dictybase, Phenotypes)  1.25e-01           2  \n",
       "24  (GO, biological_process)  1.56e-01          13  \n",
       "25   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "26   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "27   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "28   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "29   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "30  (GO, biological_process)  1.56e-01          14  \n",
       "31  (GO, cellular_component)  2.00e-01           6  \n",
       "32          (KEGG, Pathways)  2.00e-01           5  \n",
       "33   (Dictybase, Phenotypes)  2.07e-01           2  \n",
       "34  (GO, biological_process)  2.10e-01           3  \n",
       "35  (GO, cellular_component)  2.14e-01           2  \n",
       "36  (GO, biological_process)  2.14e-01          28  \n",
       "37  (GO, molecular_function)  2.14e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 1615 with EID: 1609\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.29\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 219 out of 468 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.04e-04</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.82e-03</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.82e-03</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.80e-03</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.52e-03</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.15e-03</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.24e-03</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.40e-02</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.89e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.89e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased autophagic vacuole formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.60e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.53e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.53e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.55e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Alanine</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.29e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.29e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Phenylalanine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.29e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>increased catalytic activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.45e-01</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant thermotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>delayed culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>decreased prestalk A cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>aberrant stalk formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>abolished tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>aberrant spore morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Tyrosine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Taurine and hypotaurine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.86e-01</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>abolished catalytic activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                             aberrant spore coat morphology   \n",
       "1   anatomical structure formation involved in morphogenesis   \n",
       "2                                       cell differentiation   \n",
       "3                                       extracellular region   \n",
       "4                           anatomical structure development   \n",
       "5                                  decreased spore viability   \n",
       "6                              Starch and sucrose metabolism   \n",
       "7                                            Golgi apparatus   \n",
       "8                           abolished cellulose biosynthesis   \n",
       "9                       cell wall organization or biogenesis   \n",
       "10                    decreased autophagic vacuole formation   \n",
       "11                          external encapsulating structure   \n",
       "12                                                 cell wall   \n",
       "13                   increased prespore cell differentiation   \n",
       "14                                                   Alanine   \n",
       "15                                      Galactose metabolism   \n",
       "16                                  Phenylalanine metabolism   \n",
       "17                              increased catalytic activity   \n",
       "18                            carbohydrate metabolic process   \n",
       "19                               abolished cellulose binding   \n",
       "20                                                 wild type   \n",
       "21                                      aberrant thermotaxis   \n",
       "22                                       delayed culmination   \n",
       "23                 decreased prestalk A cell differentiation   \n",
       "24                                      aberrant culmination   \n",
       "25                                  aberrant stalk formation   \n",
       "26                                           spores inviable   \n",
       "27                                   abolished tip formation   \n",
       "28                                 aberrant spore morphology   \n",
       "29                                      extracellular matrix   \n",
       "30                                       Tyrosine metabolism   \n",
       "31                        Taurine and hypotaurine metabolism   \n",
       "32                                       signal transduction   \n",
       "33                                   aberrant slug migration   \n",
       "34                              abolished catalytic activity   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0    (Dictybase, Phenotypes)  3.04e-04           9  \n",
       "1   (GO, biological_process)  3.82e-03          22  \n",
       "2   (GO, biological_process)  3.82e-03          35  \n",
       "3   (GO, cellular_component)  5.80e-03          36  \n",
       "4   (GO, biological_process)  6.52e-03          78  \n",
       "5    (Dictybase, Phenotypes)  7.15e-03          18  \n",
       "6           (KEGG, Pathways)  7.24e-03          12  \n",
       "7   (GO, cellular_component)  1.40e-02          22  \n",
       "8    (Dictybase, Phenotypes)  3.89e-02           4  \n",
       "9   (GO, biological_process)  3.89e-02           7  \n",
       "10   (Dictybase, Phenotypes)  6.60e-02           3  \n",
       "11  (GO, cellular_component)  7.53e-02           6  \n",
       "12  (GO, cellular_component)  7.53e-02           6  \n",
       "13   (Dictybase, Phenotypes)  7.55e-02           4  \n",
       "14          (KEGG, Pathways)  9.29e-02           8  \n",
       "15          (KEGG, Pathways)  9.29e-02           6  \n",
       "16          (KEGG, Pathways)  9.29e-02           5  \n",
       "17   (Dictybase, Phenotypes)  1.45e-01           3  \n",
       "18  (GO, biological_process)  1.45e-01          19  \n",
       "19   (Dictybase, Phenotypes)  1.86e-01           2  \n",
       "20   (Dictybase, Phenotypes)  1.86e-01          54  \n",
       "21   (Dictybase, Phenotypes)  1.86e-01           3  \n",
       "22   (Dictybase, Phenotypes)  1.86e-01           8  \n",
       "23   (Dictybase, Phenotypes)  1.86e-01           2  \n",
       "24   (Dictybase, Phenotypes)  1.86e-01          18  \n",
       "25   (Dictybase, Phenotypes)  1.86e-01           3  \n",
       "26   (Dictybase, Phenotypes)  1.86e-01           3  \n",
       "27   (Dictybase, Phenotypes)  1.86e-01           2  \n",
       "28   (Dictybase, Phenotypes)  1.86e-01           4  \n",
       "29  (GO, cellular_component)  1.86e-01           4  \n",
       "30          (KEGG, Pathways)  1.86e-01           5  \n",
       "31          (KEGG, Pathways)  1.86e-01           3  \n",
       "32  (GO, biological_process)  1.86e-01          56  \n",
       "33   (Dictybase, Phenotypes)  2.09e-01          13  \n",
       "34   (Dictybase, Phenotypes)  2.31e-01           5  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 2289 with EID: 2274\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 373 out of 631 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.55e-08</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.55e-08</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.66e-05</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.66e-05</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant spore coat morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.21e-05</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.22e-03</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.16e-02</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.47e-02</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.04e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased prestalk 0 cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.04e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.04e-02</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>cell wall organization or biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.99e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.55e-02</td>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.95e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.13e-02</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>abolished DIF production</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>precocious spore germination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>thin standing slug</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>increased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>decreased autophagic vacuole formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>abolished cellulose biosynthesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant spore morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Galactose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.74e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>abolished autophagic cell death</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.19e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>abolished autophagic vacuole formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.19e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>aberrant phototaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.19e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Pentose and glucuronate interconversions</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.19e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>delayed culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.36e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>external encapsulating structure</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>cell wall</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>Phenylalanine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>increased catalytic activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>increased sorus size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>autophagy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>aberrant prestalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.81e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.92e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>aberrant fruiting body morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>abolished cellulose binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>aberrant response to valproic acid</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>aberrant protein phosphorylation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>decreased protein tyrosine phosphorylation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>decreased pseudopod retraction</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>increased standing slug length</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>decreased prestalk A cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>aberrant protein localization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>abolished outer basal disc formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>aberrant response to lithium</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>abolished tip formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>aberrant negative chemotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>extracellular matrix</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>Nitrogen metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>aberrant gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>aberrant thermotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>precocious sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>aberrant stalk formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>spores inviable</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>aberrant autophagic cell death</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>Arginine and proline metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>cell-cell signaling</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.38e-01</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>68</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.47e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0   anatomical structure formation involved in morphogenesis   \n",
       "1                                       cell differentiation   \n",
       "2                              Starch and sucrose metabolism   \n",
       "3                           anatomical structure development   \n",
       "4                             aberrant spore coat morphology   \n",
       "5                                       extracellular region   \n",
       "6                  DNA-binding transcription factor activity   \n",
       "7                                       aberrant culmination   \n",
       "8                    increased prespore cell differentiation   \n",
       "9                  decreased prestalk 0 cell differentiation   \n",
       "10                                 decreased gene expression   \n",
       "11                      cell wall organization or biogenesis   \n",
       "12                                           kinase activity   \n",
       "13                 SNARE interactions in vesicular transport   \n",
       "14                                       signal transduction   \n",
       "15                                   aberrant slug migration   \n",
       "16                                 decreased spore viability   \n",
       "17                                  abolished DIF production   \n",
       "18                              precocious spore germination   \n",
       "19                                        thin standing slug   \n",
       "20                                 increased gene expression   \n",
       "21                    decreased autophagic vacuole formation   \n",
       "22                          abolished cellulose biosynthesis   \n",
       "23                                 aberrant spore morphology   \n",
       "24                                           Golgi apparatus   \n",
       "25                                      Galactose metabolism   \n",
       "26                               secondary metabolic process   \n",
       "27                 development arrests at tipped mound stage   \n",
       "28                           abolished autophagic cell death   \n",
       "29                    abolished autophagic vacuole formation   \n",
       "30                                       aberrant phototaxis   \n",
       "31                  Pentose and glucuronate interconversions   \n",
       "32                                       delayed culmination   \n",
       "33                                                 wild type   \n",
       "34                          external encapsulating structure   \n",
       "35                                                 cell wall   \n",
       "36       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "37                                  Phenylalanine metabolism   \n",
       "38                              increased catalytic activity   \n",
       "39                                      increased sorus size   \n",
       "40                                                 autophagy   \n",
       "41                                     abolished culmination   \n",
       "42                    aberrant prestalk cell differentiation   \n",
       "43                                   delayed gene expression   \n",
       "44                         aberrant fruiting body morphology   \n",
       "45                               abolished cellulose binding   \n",
       "46                        aberrant response to valproic acid   \n",
       "47                          aberrant protein phosphorylation   \n",
       "48                decreased protein tyrosine phosphorylation   \n",
       "49                            decreased pseudopod retraction   \n",
       "50                            increased standing slug length   \n",
       "51                 decreased prestalk A cell differentiation   \n",
       "52                             aberrant protein localization   \n",
       "53                      abolished outer basal disc formation   \n",
       "54                              aberrant response to lithium   \n",
       "55                                   abolished tip formation   \n",
       "56                              aberrant negative chemotaxis   \n",
       "57                                      extracellular matrix   \n",
       "58                                       Nitrogen metabolism   \n",
       "59                                  aberrant gene expression   \n",
       "60                                      aberrant thermotaxis   \n",
       "61                                    precocious sporulation   \n",
       "62                                  aberrant stalk formation   \n",
       "63                                           spores inviable   \n",
       "64                            aberrant autophagic cell death   \n",
       "65                           Arginine and proline metabolism   \n",
       "66                                       cell-cell signaling   \n",
       "67                                               DNA binding   \n",
       "68                                  abolished slug migration   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  5.55e-08          35  \n",
       "1   (GO, biological_process)  5.55e-08          54  \n",
       "2           (KEGG, Pathways)  1.66e-05          18  \n",
       "3   (GO, biological_process)  1.66e-05         110  \n",
       "4    (Dictybase, Phenotypes)  5.21e-05          10  \n",
       "5   (GO, cellular_component)  1.22e-03          46  \n",
       "6   (GO, molecular_function)  2.16e-02          15  \n",
       "7    (Dictybase, Phenotypes)  2.47e-02          27  \n",
       "8    (Dictybase, Phenotypes)  3.04e-02           5  \n",
       "9    (Dictybase, Phenotypes)  3.04e-02           4  \n",
       "10   (Dictybase, Phenotypes)  3.04e-02          26  \n",
       "11  (GO, biological_process)  3.99e-02           8  \n",
       "12  (GO, molecular_function)  4.55e-02          73  \n",
       "13          (KEGG, Pathways)  5.95e-02          10  \n",
       "14  (GO, biological_process)  7.13e-02          76  \n",
       "15   (Dictybase, Phenotypes)  7.74e-02          18  \n",
       "16   (Dictybase, Phenotypes)  7.74e-02          18  \n",
       "17   (Dictybase, Phenotypes)  7.74e-02           3  \n",
       "18   (Dictybase, Phenotypes)  7.74e-02           3  \n",
       "19   (Dictybase, Phenotypes)  7.74e-02           3  \n",
       "20   (Dictybase, Phenotypes)  7.74e-02          15  \n",
       "21   (Dictybase, Phenotypes)  7.74e-02           3  \n",
       "22   (Dictybase, Phenotypes)  7.74e-02           4  \n",
       "23   (Dictybase, Phenotypes)  7.74e-02           5  \n",
       "24  (GO, cellular_component)  7.74e-02          24  \n",
       "25          (KEGG, Pathways)  7.74e-02           7  \n",
       "26  (GO, biological_process)  7.74e-02           4  \n",
       "27   (Dictybase, Phenotypes)  1.17e-01           6  \n",
       "28   (Dictybase, Phenotypes)  1.19e-01           4  \n",
       "29   (Dictybase, Phenotypes)  1.19e-01           4  \n",
       "30   (Dictybase, Phenotypes)  1.19e-01           4  \n",
       "31          (KEGG, Pathways)  1.19e-01           4  \n",
       "32   (Dictybase, Phenotypes)  1.36e-01          10  \n",
       "33   (Dictybase, Phenotypes)  1.46e-01          70  \n",
       "34  (GO, cellular_component)  1.46e-01           6  \n",
       "35  (GO, cellular_component)  1.46e-01           6  \n",
       "36          (KEGG, Pathways)  1.74e-01           5  \n",
       "37          (KEGG, Pathways)  1.74e-01           5  \n",
       "38   (Dictybase, Phenotypes)  1.80e-01           3  \n",
       "39   (Dictybase, Phenotypes)  1.80e-01           3  \n",
       "40  (GO, biological_process)  1.80e-01          11  \n",
       "41   (Dictybase, Phenotypes)  1.81e-01          17  \n",
       "42   (Dictybase, Phenotypes)  1.81e-01           4  \n",
       "43   (Dictybase, Phenotypes)  1.92e-01           7  \n",
       "44   (Dictybase, Phenotypes)  2.12e-01          24  \n",
       "45   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "46   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "47   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "48   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "49   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "50   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "51   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "52   (Dictybase, Phenotypes)  2.29e-01          10  \n",
       "53   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "54   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "55   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "56   (Dictybase, Phenotypes)  2.29e-01           2  \n",
       "57  (GO, cellular_component)  2.31e-01           4  \n",
       "58          (KEGG, Pathways)  2.31e-01           4  \n",
       "59   (Dictybase, Phenotypes)  2.38e-01           8  \n",
       "60   (Dictybase, Phenotypes)  2.38e-01           3  \n",
       "61   (Dictybase, Phenotypes)  2.38e-01           3  \n",
       "62   (Dictybase, Phenotypes)  2.38e-01           3  \n",
       "63   (Dictybase, Phenotypes)  2.38e-01           3  \n",
       "64   (Dictybase, Phenotypes)  2.38e-01           3  \n",
       "65          (KEGG, Pathways)  2.38e-01           7  \n",
       "66  (GO, biological_process)  2.38e-01           6  \n",
       "67  (GO, molecular_function)  2.38e-01          46  \n",
       "68   (Dictybase, Phenotypes)  2.47e-01          12  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 1096 with EID: 1085\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 148 out of 304 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.34e-02</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.04e-02</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>DNA-binding transcription factor activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.30e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.03e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>decreased prestalk 0 cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.03e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.03e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.03e-02</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.03e-02</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.37e-01</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant protein phosphorylation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased pseudopod retraction</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>decreased prestalk A cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>abolished outer basal disc formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.48e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>increased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>decreased slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant prestalk cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>autophagy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>precocious spore germination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>aberrant spore germination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Arginine and proline metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.43e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Gene set  \\\n",
       "0                                       cell differentiation   \n",
       "1                           anatomical structure development   \n",
       "2                  DNA-binding transcription factor activity   \n",
       "3                                   aberrant gene expression   \n",
       "4                  decreased prestalk 0 cell differentiation   \n",
       "5                  SNARE interactions in vesicular transport   \n",
       "6   anatomical structure formation involved in morphogenesis   \n",
       "7                                                DNA binding   \n",
       "8                                            kinase activity   \n",
       "9                           aberrant protein phosphorylation   \n",
       "10                            decreased pseudopod retraction   \n",
       "11                 decreased prestalk A cell differentiation   \n",
       "12                      abolished outer basal disc formation   \n",
       "13                                 decreased gene expression   \n",
       "14                               secondary metabolic process   \n",
       "15                   increased prespore cell differentiation   \n",
       "16                                      aberrant culmination   \n",
       "17                                  decreased slug migration   \n",
       "18                 development arrests at tipped mound stage   \n",
       "19                                   aberrant slug migration   \n",
       "20                    aberrant prestalk cell differentiation   \n",
       "21                                                 autophagy   \n",
       "22                                                 wild type   \n",
       "23                              precocious spore germination   \n",
       "24                                aberrant spore germination   \n",
       "25                           Arginine and proline metabolism   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, biological_process)  3.34e-02          25  \n",
       "1   (GO, biological_process)  4.04e-02          54  \n",
       "2   (GO, molecular_function)  4.30e-02          10  \n",
       "3    (Dictybase, Phenotypes)  8.03e-02           7  \n",
       "4    (Dictybase, Phenotypes)  8.03e-02           3  \n",
       "5           (KEGG, Pathways)  8.03e-02           7  \n",
       "6   (GO, biological_process)  8.03e-02          14  \n",
       "7   (GO, molecular_function)  8.03e-02          30  \n",
       "8   (GO, molecular_function)  1.37e-01          39  \n",
       "9    (Dictybase, Phenotypes)  1.48e-01           2  \n",
       "10   (Dictybase, Phenotypes)  1.48e-01           2  \n",
       "11   (Dictybase, Phenotypes)  1.48e-01           2  \n",
       "12   (Dictybase, Phenotypes)  1.48e-01           2  \n",
       "13   (Dictybase, Phenotypes)  1.48e-01          14  \n",
       "14  (GO, biological_process)  1.48e-01           3  \n",
       "15   (Dictybase, Phenotypes)  1.59e-01           3  \n",
       "16   (Dictybase, Phenotypes)  1.59e-01          14  \n",
       "17   (Dictybase, Phenotypes)  1.59e-01           4  \n",
       "18   (Dictybase, Phenotypes)  2.06e-01           4  \n",
       "19   (Dictybase, Phenotypes)  2.25e-01          10  \n",
       "20   (Dictybase, Phenotypes)  2.25e-01           3  \n",
       "21  (GO, biological_process)  2.27e-01           7  \n",
       "22   (Dictybase, Phenotypes)  2.43e-01          37  \n",
       "23   (Dictybase, Phenotypes)  2.43e-01           2  \n",
       "24   (Dictybase, Phenotypes)  2.43e-01           2  \n",
       "25          (KEGG, Pathways)  2.43e-01           5  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_de+'WT_batchrep_combined.tsv',index_col=0,sep='\\t')\n",
    "for stage in [phenotype for phenotype in PHENOTYPES if phenotype in list(milestones['Stage'].unique())]:\n",
    "    query_names=list(milestones.query('log2FoldChange>=1 & padj <= 0.05 & Stage==\"'+stage+'\"').index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ImpulseDE2\n",
    "Milestone genes are selected as those that are at up-level in a stage and at down level at non neighbouring stages (to account for imperfect annotation of up/down). no_agg is not considered a neighbour of stream and vice versa."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### ImpulseDE2 all strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 1199 with EID: 1193\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 344 out of 479 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.40e-38</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.54e-30</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.02e-30</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.96e-23</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.78e-20</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.78e-08</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.02e-05</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.02e-05</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.62e-05</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.36e-03</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.87e-03</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.90e-03</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.31e-03</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cellular amino acid metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.44e-02</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.86e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>lysosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.00e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Glycine</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.23e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.06e-02</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.48e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.97e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cofactor metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.49e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Pentose phosphate pathway</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.57e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.57e-02</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.57e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Vitamin B6 metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.64e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>Sulfur relay system</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>beta-Alanine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>vacuole</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>Phagosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.82e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.82e-01</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.00e-01</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>Lysine biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.36e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                  Ontology  \\\n",
       "0                                         Ribosome          (KEGG, Pathways)   \n",
       "1                                         ribosome  (GO, cellular_component)   \n",
       "2               structural constituent of ribosome  (GO, molecular_function)   \n",
       "3                     structural molecule activity  (GO, molecular_function)   \n",
       "4                                      translation  (GO, biological_process)   \n",
       "5                                    mitochondrion  (GO, cellular_component)   \n",
       "6                               Metabolic pathways          (KEGG, Pathways)   \n",
       "7                          oxidoreductase activity  (GO, molecular_function)   \n",
       "8                        Oxidative phosphorylation          (KEGG, Pathways)   \n",
       "9                      Biosynthesis of amino acids          (KEGG, Pathways)   \n",
       "10           Biosynthesis of secondary metabolites          (KEGG, Pathways)   \n",
       "11                        unfolded protein binding  (GO, molecular_function)   \n",
       "12                small molecule metabolic process  (GO, biological_process)   \n",
       "13           cellular amino acid metabolic process  (GO, biological_process)   \n",
       "14                                 protein folding  (GO, biological_process)   \n",
       "15                                        lysosome  (GO, cellular_component)   \n",
       "16                                         Glycine          (KEGG, Pathways)   \n",
       "17              transmembrane transporter activity  (GO, molecular_function)   \n",
       "18                            Steroid biosynthesis          (KEGG, Pathways)   \n",
       "19  generation of precursor metabolites and energy  (GO, biological_process)   \n",
       "20                      cofactor metabolic process  (GO, biological_process)   \n",
       "21                       Pentose phosphate pathway          (KEGG, Pathways)   \n",
       "22                              peptidase activity  (GO, molecular_function)   \n",
       "23                              isomerase activity  (GO, molecular_function)   \n",
       "24                           Fatty acid elongation          (KEGG, Pathways)   \n",
       "25                           Vitamin B6 metabolism          (KEGG, Pathways)   \n",
       "26                             Sulfur relay system          (KEGG, Pathways)   \n",
       "27                         beta-Alanine metabolism          (KEGG, Pathways)   \n",
       "28                                         vacuole  (GO, cellular_component)   \n",
       "29                                       Phagosome          (KEGG, Pathways)   \n",
       "30                               Carbon metabolism          (KEGG, Pathways)   \n",
       "31                            extracellular region  (GO, cellular_component)   \n",
       "32                             Lysine biosynthesis          (KEGG, Pathways)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   7.40e-38          67  \n",
       "1   3.54e-30          59  \n",
       "2   5.02e-30          56  \n",
       "3   4.96e-23          59  \n",
       "4   1.78e-20          69  \n",
       "5   7.78e-08          60  \n",
       "6   6.02e-05         119  \n",
       "7   6.02e-05          49  \n",
       "8   6.62e-05          20  \n",
       "9   2.36e-03          19  \n",
       "10  3.87e-03          55  \n",
       "11  3.90e-03          11  \n",
       "12  9.31e-03          64  \n",
       "13  1.44e-02          28  \n",
       "14  1.86e-02          12  \n",
       "15  2.00e-02          18  \n",
       "16  2.23e-02           9  \n",
       "17  5.06e-02          30  \n",
       "18  5.48e-02           7  \n",
       "19  5.97e-02          18  \n",
       "20  6.49e-02          17  \n",
       "21  6.57e-02           7  \n",
       "22  6.57e-02          20  \n",
       "23  6.57e-02          13  \n",
       "24  1.64e-01           5  \n",
       "25  1.64e-01           3  \n",
       "26  1.70e-01           4  \n",
       "27  1.70e-01           7  \n",
       "28  1.70e-01          24  \n",
       "29  1.82e-01          12  \n",
       "30  1.82e-01          20  \n",
       "31  2.00e-01          28  \n",
       "32  2.36e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 956 with EID: 941\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 270 out of 377 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.91e-08</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.91e-08</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.16e-06</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.19e-06</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.81e-06</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.81e-06</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.81e-06</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.40e-05</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.71e-05</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.53e-04</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.05e-04</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.52e-03</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased activation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.52e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.15e-03</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.15e-03</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.09e-03</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Protein export</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.09e-03</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.09e-03</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>abolished chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.17e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.88e-03</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.46e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.09e-03</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.98e-03</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.01e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.01e-02</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.07e-02</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.07e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.07e-02</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.19e-02</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>abolished chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.26e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>abolished adenylate cyclase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.26e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.26e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.36e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>aberrant chemotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.38e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>decreased protein kinase activation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.61e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.83e-02</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.83e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.00e-02</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.03e-02</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.04e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>abolished protein phosphorylation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.41e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>aberrant protein localization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.48e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.87e-02</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>aberrant adaptation of signaling pathway</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>aberrant localization to plasma membrane</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.23e-02</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>aberrant phagocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.86e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>increased intracellular cAMP level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.86e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>decreased growth rate in suspension</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.86e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>GTPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.39e-02</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>decreased cAMP-induced cAMP pulse</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.61e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>decreased macropinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.73e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>aberrant chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.73e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.73e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>decreased lateral pseudopod suppression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.73e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>aberrant pseudopod formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.73e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>vesicle-mediated transport</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.73e-02</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.87e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.30e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.38e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>increased period of cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.60e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>aberrant chemotaxis to cAMP in response to DIF-2</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.60e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.07e-02</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>decreased cell polarity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.37e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.37e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>aberrant slug formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.39e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.77e-02</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>68</td>\n",
       "      <td>protein targeting</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>69</td>\n",
       "      <td>abolished cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>70</td>\n",
       "      <td>enzyme binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>71</td>\n",
       "      <td>decreased myosin II assembly</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73</td>\n",
       "      <td>decreased pinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>76</td>\n",
       "      <td>decreased exocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>77</td>\n",
       "      <td>decreased chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78</td>\n",
       "      <td>abolished activation of GTPase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.31e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>79</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.35e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>81</td>\n",
       "      <td>aberrant adaptation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.39e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>82</td>\n",
       "      <td>decreased secretion of lysosomal enzyme</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.39e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>decreased number of aggregates</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.39e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>84</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>85</td>\n",
       "      <td>Glycosylphosphatidylinositol (GPI)-anchor biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>86</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.67e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>87</td>\n",
       "      <td>homeostatic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.67e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>88</td>\n",
       "      <td>aberrant stalk morphogenesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.73e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>89</td>\n",
       "      <td>decreased phagocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.75e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>90</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.82e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>91</td>\n",
       "      <td>abolished positive regulation of chemorepellant activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>92</td>\n",
       "      <td>aberrant cell separation during cytokinesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>93</td>\n",
       "      <td>aberrant activation of adenylate cyclase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>94</td>\n",
       "      <td>decreased cAMP-induced cGMP pulse</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>95</td>\n",
       "      <td>aberrant taxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>96</td>\n",
       "      <td>aberrant cellular response to fluid shear stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>97</td>\n",
       "      <td>abolished cell polarity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>98</td>\n",
       "      <td>decreased stalk width</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>99</td>\n",
       "      <td>decreased accumulation of cGMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>100</td>\n",
       "      <td>decreased protein localization to plasma membrane</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>101</td>\n",
       "      <td>precocious culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>102</td>\n",
       "      <td>multiple tips</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>103</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>104</td>\n",
       "      <td>transmembrane transporter activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>105</td>\n",
       "      <td>increased lateral pseudopod formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>106</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     Gene set  \\\n",
       "0                                          aberrant streaming   \n",
       "1                                        aberrant cytokinesis   \n",
       "2                                                  locomotion   \n",
       "3                                               cell division   \n",
       "4                                                  Proteasome   \n",
       "5                                         signal transduction   \n",
       "6                                          mitotic cell cycle   \n",
       "7                                   cytoskeleton organization   \n",
       "8                                             plasma membrane   \n",
       "9                               aberrant F-actin organization   \n",
       "10                                                 cell cycle   \n",
       "11                                      abolished aggregation   \n",
       "12                  decreased activation of adenylate cyclase   \n",
       "13                                   aberrant cell morphology   \n",
       "14                               aberrant cAMP wave formation   \n",
       "15                                               cytoskeleton   \n",
       "16                                             Protein export   \n",
       "17                                cellular component assembly   \n",
       "18                               abolished chemotaxis to cAMP   \n",
       "19                Protein processing in endoplasmic reticulum   \n",
       "20                  abolished activation of adenylate cyclase   \n",
       "21                         development arrests at mound stage   \n",
       "22                                       aberrant aggregation   \n",
       "23                                   decreased aggregate size   \n",
       "24                                      endoplasmic reticulum   \n",
       "25                               decreased chemotaxis to cAMP   \n",
       "26                                    decreased cell motility   \n",
       "27                           anatomical structure development   \n",
       "28                                  enzyme regulator activity   \n",
       "29                             abolished chemotaxis to folate   \n",
       "30                       abolished adenylate cyclase activity   \n",
       "31                                         cell morphogenesis   \n",
       "32                                        delayed aggregation   \n",
       "33                                               reproduction   \n",
       "34                                        aberrant chemotaxis   \n",
       "35                        decreased protein kinase activation   \n",
       "36                                                  wild type   \n",
       "37                                  decreased chemotaxis rate   \n",
       "38                                      abolished culmination   \n",
       "39                                        delayed development   \n",
       "40                                   abolished slug migration   \n",
       "41                          abolished protein phosphorylation   \n",
       "42                              aberrant protein localization   \n",
       "43                                                    cytosol   \n",
       "44                   aberrant adaptation of signaling pathway   \n",
       "45                   aberrant localization to plasma membrane   \n",
       "46                        protein-containing complex assembly   \n",
       "47                                      aberrant phagocytosis   \n",
       "48                         increased intracellular cAMP level   \n",
       "49                        decreased growth rate in suspension   \n",
       "50                                            GTPase activity   \n",
       "51                          decreased cAMP-induced cAMP pulse   \n",
       "52                                 decreased macropinocytosis   \n",
       "53                                aberrant chemotaxis to cAMP   \n",
       "54                                      increased growth rate   \n",
       "55                    decreased lateral pseudopod suppression   \n",
       "56                               aberrant pseudopod formation   \n",
       "57                                 vesicle-mediated transport   \n",
       "58                                    aberrant slug migration   \n",
       "59                                       aberrant culmination   \n",
       "60                                              cell motility   \n",
       "61                    increased period of cAMP wave formation   \n",
       "62           aberrant chemotaxis to cAMP in response to DIF-2   \n",
       "63                                      decreased growth rate   \n",
       "64                                    decreased cell polarity   \n",
       "65                                        abolished streaming   \n",
       "66                                    aberrant slug formation   \n",
       "67                                         response to stress   \n",
       "68                                          protein targeting   \n",
       "69                              abolished cAMP wave formation   \n",
       "70                                             enzyme binding   \n",
       "71                               decreased myosin II assembly   \n",
       "72                              increased basal F-actin level   \n",
       "73                                      decreased pinocytosis   \n",
       "74                                       aberrant development   \n",
       "75                                      N-Glycan biosynthesis   \n",
       "76                                       decreased exocytosis   \n",
       "77                             decreased chemotaxis to folate   \n",
       "78                    abolished activation of GTPase activity   \n",
       "79                                 precocious gene expression   \n",
       "80                               decreased fruiting body size   \n",
       "81                   aberrant adaptation of adenylate cyclase   \n",
       "82                    decreased secretion of lysosomal enzyme   \n",
       "83                             decreased number of aggregates   \n",
       "84                                            kinase activity   \n",
       "85     Glycosylphosphatidylinositol (GPI)-anchor biosynthesis   \n",
       "86                                  decreased gene expression   \n",
       "87                                        homeostatic process   \n",
       "88                               aberrant stalk morphogenesis   \n",
       "89                                     decreased phagocytosis   \n",
       "90                                     precocious aggregation   \n",
       "91   abolished positive regulation of chemorepellant activity   \n",
       "92                aberrant cell separation during cytokinesis   \n",
       "93          aberrant activation of adenylate cyclase activity   \n",
       "94                          decreased cAMP-induced cGMP pulse   \n",
       "95                                             aberrant taxis   \n",
       "96           aberrant cellular response to fluid shear stress   \n",
       "97                                    abolished cell polarity   \n",
       "98                                      decreased stalk width   \n",
       "99                             decreased accumulation of cGMP   \n",
       "100         decreased protein localization to plasma membrane   \n",
       "101                                    precocious culmination   \n",
       "102                                             multiple tips   \n",
       "103                             microtubule organizing center   \n",
       "104                        transmembrane transporter activity   \n",
       "105                     increased lateral pseudopod formation   \n",
       "106                                    precocious development   \n",
       "\n",
       "                     Ontology       FDR  N in query  \n",
       "0     (Dictybase, Phenotypes)  2.91e-08          27  \n",
       "1     (Dictybase, Phenotypes)  2.91e-08          32  \n",
       "2    (GO, biological_process)  4.16e-06          43  \n",
       "3    (GO, biological_process)  5.19e-06          31  \n",
       "4            (KEGG, Pathways)  5.81e-06          16  \n",
       "5    (GO, biological_process)  5.81e-06          64  \n",
       "6    (GO, biological_process)  5.81e-06          28  \n",
       "7    (GO, biological_process)  1.40e-05          42  \n",
       "8    (GO, cellular_component)  7.71e-05          50  \n",
       "9     (Dictybase, Phenotypes)  1.53e-04          14  \n",
       "10   (GO, biological_process)  5.05e-04          34  \n",
       "11    (Dictybase, Phenotypes)  1.52e-03          20  \n",
       "12    (Dictybase, Phenotypes)  1.52e-03           5  \n",
       "13    (Dictybase, Phenotypes)  2.15e-03          13  \n",
       "14    (Dictybase, Phenotypes)  2.15e-03          22  \n",
       "15   (GO, cellular_component)  3.09e-03          34  \n",
       "16           (KEGG, Pathways)  3.09e-03           8  \n",
       "17   (GO, biological_process)  3.09e-03          36  \n",
       "18    (Dictybase, Phenotypes)  3.17e-03           7  \n",
       "19           (KEGG, Pathways)  3.88e-03          20  \n",
       "20    (Dictybase, Phenotypes)  7.46e-03           5  \n",
       "21    (Dictybase, Phenotypes)  8.09e-03          14  \n",
       "22    (Dictybase, Phenotypes)  8.98e-03          15  \n",
       "23    (Dictybase, Phenotypes)  1.01e-02          10  \n",
       "24   (GO, cellular_component)  1.01e-02          23  \n",
       "25    (Dictybase, Phenotypes)  1.07e-02          15  \n",
       "26    (Dictybase, Phenotypes)  1.07e-02          17  \n",
       "27   (GO, biological_process)  1.07e-02          62  \n",
       "28   (GO, molecular_function)  1.19e-02          21  \n",
       "29    (Dictybase, Phenotypes)  1.26e-02           3  \n",
       "30    (Dictybase, Phenotypes)  1.26e-02           3  \n",
       "31   (GO, biological_process)  1.26e-02           8  \n",
       "32    (Dictybase, Phenotypes)  1.36e-02          16  \n",
       "33   (GO, biological_process)  1.36e-02          17  \n",
       "34    (Dictybase, Phenotypes)  1.38e-02           4  \n",
       "35    (Dictybase, Phenotypes)  1.61e-02           7  \n",
       "36    (Dictybase, Phenotypes)  1.83e-02          49  \n",
       "37    (Dictybase, Phenotypes)  1.83e-02          10  \n",
       "38    (Dictybase, Phenotypes)  2.00e-02          14  \n",
       "39    (Dictybase, Phenotypes)  2.03e-02          15  \n",
       "40    (Dictybase, Phenotypes)  2.04e-02          11  \n",
       "41    (Dictybase, Phenotypes)  2.41e-02           5  \n",
       "42    (Dictybase, Phenotypes)  2.48e-02           9  \n",
       "43   (GO, cellular_component)  2.87e-02          32  \n",
       "44    (Dictybase, Phenotypes)  3.20e-02           3  \n",
       "45    (Dictybase, Phenotypes)  3.20e-02           3  \n",
       "46   (GO, biological_process)  3.23e-02          23  \n",
       "47    (Dictybase, Phenotypes)  3.86e-02           4  \n",
       "48    (Dictybase, Phenotypes)  3.86e-02           4  \n",
       "49    (Dictybase, Phenotypes)  3.86e-02           4  \n",
       "50   (GO, molecular_function)  4.39e-02          24  \n",
       "51    (Dictybase, Phenotypes)  4.61e-02           5  \n",
       "52    (Dictybase, Phenotypes)  5.73e-02           6  \n",
       "53    (Dictybase, Phenotypes)  5.73e-02           8  \n",
       "54    (Dictybase, Phenotypes)  5.73e-02           8  \n",
       "55    (Dictybase, Phenotypes)  5.73e-02           4  \n",
       "56    (Dictybase, Phenotypes)  5.73e-02           4  \n",
       "57   (GO, biological_process)  5.73e-02          34  \n",
       "58    (Dictybase, Phenotypes)  5.87e-02          12  \n",
       "59    (Dictybase, Phenotypes)  6.30e-02          16  \n",
       "60   (GO, biological_process)  6.38e-02          13  \n",
       "61    (Dictybase, Phenotypes)  6.60e-02           2  \n",
       "62    (Dictybase, Phenotypes)  6.60e-02           2  \n",
       "63    (Dictybase, Phenotypes)  7.07e-02          22  \n",
       "64    (Dictybase, Phenotypes)  7.37e-02           8  \n",
       "65    (Dictybase, Phenotypes)  7.37e-02           8  \n",
       "66    (Dictybase, Phenotypes)  7.39e-02           4  \n",
       "67   (GO, biological_process)  8.77e-02          47  \n",
       "68   (GO, biological_process)  1.01e-01           8  \n",
       "69    (Dictybase, Phenotypes)  1.01e-01           4  \n",
       "70   (GO, molecular_function)  1.01e-01          18  \n",
       "71    (Dictybase, Phenotypes)  1.15e-01           5  \n",
       "72    (Dictybase, Phenotypes)  1.15e-01           5  \n",
       "73    (Dictybase, Phenotypes)  1.23e-01           7  \n",
       "74    (Dictybase, Phenotypes)  1.23e-01          10  \n",
       "75           (KEGG, Pathways)  1.23e-01           7  \n",
       "76    (Dictybase, Phenotypes)  1.28e-01           4  \n",
       "77    (Dictybase, Phenotypes)  1.28e-01           4  \n",
       "78    (Dictybase, Phenotypes)  1.31e-01           3  \n",
       "79    (Dictybase, Phenotypes)  1.34e-01           5  \n",
       "80    (Dictybase, Phenotypes)  1.35e-01          13  \n",
       "81    (Dictybase, Phenotypes)  1.39e-01           2  \n",
       "82    (Dictybase, Phenotypes)  1.39e-01           2  \n",
       "83    (Dictybase, Phenotypes)  1.39e-01           2  \n",
       "84   (GO, molecular_function)  1.40e-01          42  \n",
       "85           (KEGG, Pathways)  1.56e-01           5  \n",
       "86    (Dictybase, Phenotypes)  1.67e-01          14  \n",
       "87   (GO, biological_process)  1.67e-01          12  \n",
       "88    (Dictybase, Phenotypes)  1.73e-01           3  \n",
       "89    (Dictybase, Phenotypes)  1.75e-01          11  \n",
       "90    (Dictybase, Phenotypes)  1.82e-01           5  \n",
       "91    (Dictybase, Phenotypes)  1.87e-01           4  \n",
       "92    (Dictybase, Phenotypes)  1.87e-01           4  \n",
       "93    (Dictybase, Phenotypes)  2.09e-01           2  \n",
       "94    (Dictybase, Phenotypes)  2.09e-01           3  \n",
       "95    (Dictybase, Phenotypes)  2.09e-01           2  \n",
       "96    (Dictybase, Phenotypes)  2.09e-01           2  \n",
       "97    (Dictybase, Phenotypes)  2.09e-01           3  \n",
       "98    (Dictybase, Phenotypes)  2.09e-01           3  \n",
       "99    (Dictybase, Phenotypes)  2.09e-01           2  \n",
       "100   (Dictybase, Phenotypes)  2.09e-01           2  \n",
       "101   (Dictybase, Phenotypes)  2.09e-01           2  \n",
       "102   (Dictybase, Phenotypes)  2.09e-01           4  \n",
       "103  (GO, cellular_component)  2.09e-01           9  \n",
       "104  (GO, molecular_function)  2.09e-01          21  \n",
       "105   (Dictybase, Phenotypes)  2.25e-01           5  \n",
       "106   (Dictybase, Phenotypes)  2.25e-01           6  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 1190 with EID: 1175\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.39\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 332 out of 461 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.50e-14</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.30e-10</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.04e-08</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.25e-06</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.94e-06</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.42e-06</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.43e-06</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.14e-05</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>enzyme binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.51e-05</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.55e-05</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.19e-04</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.69e-04</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.72e-04</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.95e-04</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.70e-04</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>decreased lateral pseudopod suppression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.70e-04</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.76e-04</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.76e-04</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.76e-04</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>decreased cAMP-induced cAMP pulse</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.13e-04</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.54e-04</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.26e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-03</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.04e-03</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased macropinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.25e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased protein kinase activation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.25e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>aberrant pseudopod formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.78e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.96e-03</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>decreased myosin II assembly</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.20e-03</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>abolished positive regulation of chemorepellant activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.70e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Protein export</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.87e-03</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased cell-substrate adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.89e-03</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.58e-03</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>abolished protein phosphorylation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.58e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.58e-03</td>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.08e-02</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.51e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>increased filopodium formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.53e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.53e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.53e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>increased F-actin polymerization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.53e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>abolished chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.73e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>decreased lateral pseudopod formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.73e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>decreased 3' 5'-cyclic-AMP phosphodiesterase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.73e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>abolished filopodium formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.73e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.82e-02</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.01e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>increased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.01e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>decreased taxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.15e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>aberrant chemotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.15e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>decreased activation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.15e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.15e-02</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>increased lateral pseudopod formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.20e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>aberrant protein localization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-02</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.49e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.50e-02</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>abolished chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.68e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.68e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>cytoskeletal protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.52e-02</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>aberrant cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.82e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>decreased positive regulation of GTPase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.21e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>increased cell polarity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.23e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>aberrant cellular response to fluid shear stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.23e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>increased cAMP-induced cAMP pulse</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.23e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>increased cell-substrate adhesion</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.23e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.40e-02</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>68</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.47e-02</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>69</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.27e-02</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>70</td>\n",
       "      <td>aberrant stalk morphogenesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.81e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>71</td>\n",
       "      <td>increased intracellular cAMP level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.81e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>decreased growth rate in suspension</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.81e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.81e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.47e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.91e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>76</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.08e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>77</td>\n",
       "      <td>increased period of cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.83e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.83e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>79</td>\n",
       "      <td>aberrant signal transduction</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.83e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80</td>\n",
       "      <td>abolished negative chemotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.83e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>81</td>\n",
       "      <td>decreased substrate adhesion-dependent cell spreading</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.83e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>82</td>\n",
       "      <td>increased Ras guanyl-nucleotide exchange factor activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.83e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>decreased activation of GTPase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.83e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>84</td>\n",
       "      <td>aberrant development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.83e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>85</td>\n",
       "      <td>decreased cAMP-induced cGMP pulse</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.99e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>86</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.99e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>87</td>\n",
       "      <td>decreased stalk width</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.99e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>88</td>\n",
       "      <td>decreased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.06e-02</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>89</td>\n",
       "      <td>GTPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.75e-02</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>90</td>\n",
       "      <td>N-Glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.01e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>91</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>92</td>\n",
       "      <td>aberrant slug formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>93</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.19e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>94</td>\n",
       "      <td>increased activation of GTPase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>95</td>\n",
       "      <td>decreased protein phosphorylation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>96</td>\n",
       "      <td>abolished protein localization to cell cortex</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.22e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>97</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>98</td>\n",
       "      <td>decreased phagocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.38e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>99</td>\n",
       "      <td>decreased F-actin polymerization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>100</td>\n",
       "      <td>aberrant activation of GTPase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>101</td>\n",
       "      <td>abolished cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>102</td>\n",
       "      <td>Other glycan degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>103</td>\n",
       "      <td>phosphatase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>104</td>\n",
       "      <td>increased phagocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>105</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.53e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>106</td>\n",
       "      <td>decreased secretion of lysosomal enzyme</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>107</td>\n",
       "      <td>aberrant socially cooperative development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>108</td>\n",
       "      <td>abolished adenylate cyclase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>109</td>\n",
       "      <td>decreased number of aggregates</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>110</td>\n",
       "      <td>aberrant cytokinesis  contractile ring contraction</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>111</td>\n",
       "      <td>aberrant binding</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>112</td>\n",
       "      <td>aberrant response to cation stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>113</td>\n",
       "      <td>aberrant autophagy</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>114</td>\n",
       "      <td>decreased engulfment during phagocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>115</td>\n",
       "      <td>aberrant production of siRNA involved in RNA interference</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>116</td>\n",
       "      <td>aberrant sorting to prestalk region</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>117</td>\n",
       "      <td>Various types of N-glycan biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>118</td>\n",
       "      <td>abolished activation of GTPase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>119</td>\n",
       "      <td>decreased exocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>120</td>\n",
       "      <td>decreased chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>121</td>\n",
       "      <td>increased myosin II assembly</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.70e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>122</td>\n",
       "      <td>protein targeting</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.79e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>123</td>\n",
       "      <td>precocious development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.79e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>124</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.83e-01</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>125</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.87e-01</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>126</td>\n",
       "      <td>aberrant vegetative growth</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>127</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>128</td>\n",
       "      <td>increased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.02e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>129</td>\n",
       "      <td>aberrant phagocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>130</td>\n",
       "      <td>aberrant response to hyperosmotic stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>131</td>\n",
       "      <td>increased infection rate by a gram-negative pathogenic bacterium</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>132</td>\n",
       "      <td>increased protein kinase activation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>133</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>134</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>135</td>\n",
       "      <td>growth</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.04e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>136</td>\n",
       "      <td>cell death</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>137</td>\n",
       "      <td>aberrant activation of adenylate cyclase activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>138</td>\n",
       "      <td>aberrant adaptation of signaling pathway</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>139</td>\n",
       "      <td>increased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>140</td>\n",
       "      <td>increased F-actin polymerization at the leading edge</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>141</td>\n",
       "      <td>abolished F-actin polymerization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>142</td>\n",
       "      <td>increased membrane ruffling</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>143</td>\n",
       "      <td>abolished positive chemotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>144</td>\n",
       "      <td>increased chemotaxis rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>145</td>\n",
       "      <td>increased cytosolic Ca2+ response</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>146</td>\n",
       "      <td>aberrant localization to plasma membrane</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>147</td>\n",
       "      <td>decreased accumulation of cGMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>148</td>\n",
       "      <td>precocious culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>149</td>\n",
       "      <td>abolished transitive RNA interference</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.27e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>150</td>\n",
       "      <td>decreased prespore cell differentiation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>151</td>\n",
       "      <td>abolished catalytic activity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.29e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>152</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>153</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             Gene set  \\\n",
       "0                                                 signal transduction   \n",
       "1                                                          locomotion   \n",
       "2                                                aberrant cytokinesis   \n",
       "3                                                  aberrant streaming   \n",
       "4                                           cytoskeleton organization   \n",
       "5                                                     kinase activity   \n",
       "6                                               abolished aggregation   \n",
       "7                                       aberrant F-actin organization   \n",
       "8                                                      enzyme binding   \n",
       "9                                            aberrant cell morphology   \n",
       "10                                            decreased cell motility   \n",
       "11                        Protein processing in endoplasmic reticulum   \n",
       "12                                          decreased chemotaxis rate   \n",
       "13                                                    plasma membrane   \n",
       "14                                                          wild type   \n",
       "15                            decreased lateral pseudopod suppression   \n",
       "16                                       decreased chemotaxis to cAMP   \n",
       "17                                              increased growth rate   \n",
       "18                                              endoplasmic reticulum   \n",
       "19                                  decreased cAMP-induced cAMP pulse   \n",
       "20                                        cellular component assembly   \n",
       "21                          abolished activation of adenylate cyclase   \n",
       "22                                                delayed development   \n",
       "23                                        aberrant chemotaxis to cAMP   \n",
       "24                                         decreased macropinocytosis   \n",
       "25                                decreased protein kinase activation   \n",
       "26                                       aberrant pseudopod formation   \n",
       "27                                                       cytoskeleton   \n",
       "28                                       decreased myosin II assembly   \n",
       "29           abolished positive regulation of chemorepellant activity   \n",
       "30                                                     Protein export   \n",
       "31                                  decreased cell-substrate adhesion   \n",
       "32                                           decreased aggregate size   \n",
       "33                                  abolished protein phosphorylation   \n",
       "34                                   anatomical structure development   \n",
       "35                                          enzyme regulator activity   \n",
       "36                                                      cell motility   \n",
       "37                                     increased filopodium formation   \n",
       "38                                                delayed aggregation   \n",
       "39                                               aberrant aggregation   \n",
       "40                                   increased F-actin polymerization   \n",
       "41                                     abolished chemotaxis to folate   \n",
       "42                              decreased lateral pseudopod formation   \n",
       "43              decreased 3' 5'-cyclic-AMP phosphodiesterase activity   \n",
       "44                                     abolished filopodium formation   \n",
       "45                                          decreased gene expression   \n",
       "46                                       decreased fruiting body size   \n",
       "47                                            increased cell motility   \n",
       "48                                                    decreased taxis   \n",
       "49                                                aberrant chemotaxis   \n",
       "50                          decreased activation of adenylate cyclase   \n",
       "51                                                 mitotic cell cycle   \n",
       "52                              increased lateral pseudopod formation   \n",
       "53                                      aberrant protein localization   \n",
       "54                                                         Proteasome   \n",
       "55                                protein-containing complex assembly   \n",
       "56                                           abolished slug migration   \n",
       "57                                       aberrant cAMP wave formation   \n",
       "58                                       abolished chemotaxis to cAMP   \n",
       "59                                                 cell morphogenesis   \n",
       "60                                       cytoskeletal protein binding   \n",
       "61                                             aberrant cell motility   \n",
       "62                   decreased positive regulation of GTPase activity   \n",
       "63                                            increased cell polarity   \n",
       "64                   aberrant cellular response to fluid shear stress   \n",
       "65                                  increased cAMP-induced cAMP pulse   \n",
       "66                                  increased cell-substrate adhesion   \n",
       "67                                                            cytosol   \n",
       "68                                                      cell division   \n",
       "69                                                 response to stress   \n",
       "70                                       aberrant stalk morphogenesis   \n",
       "71                                 increased intracellular cAMP level   \n",
       "72                                decreased growth rate in suspension   \n",
       "73                                                       reproduction   \n",
       "74                                      increased basal F-actin level   \n",
       "75                                      microtubule organizing center   \n",
       "76                                                abolished streaming   \n",
       "77                            increased period of cAMP wave formation   \n",
       "78                                         multiple tips in culminant   \n",
       "79                                       aberrant signal transduction   \n",
       "80                                      abolished negative chemotaxis   \n",
       "81              decreased substrate adhesion-dependent cell spreading   \n",
       "82           increased Ras guanyl-nucleotide exchange factor activity   \n",
       "83                            decreased activation of GTPase activity   \n",
       "84                                               aberrant development   \n",
       "85                                  decreased cAMP-induced cGMP pulse   \n",
       "86                                              decreased culmination   \n",
       "87                                              decreased stalk width   \n",
       "88                                              decreased growth rate   \n",
       "89                                                    GTPase activity   \n",
       "90                                              N-Glycan biosynthesis   \n",
       "91                                          abolished gene expression   \n",
       "92                                            aberrant slug formation   \n",
       "93                           development arrests at loose mound stage   \n",
       "94                            increased activation of GTPase activity   \n",
       "95                                  decreased protein phosphorylation   \n",
       "96                      abolished protein localization to cell cortex   \n",
       "97                                              abolished culmination   \n",
       "98                                             decreased phagocytosis   \n",
       "99                                   decreased F-actin polymerization   \n",
       "100                            aberrant activation of GTPase activity   \n",
       "101                                     abolished cAMP wave formation   \n",
       "102                                          Other glycan degradation   \n",
       "103                                              phosphatase activity   \n",
       "104                                            increased phagocytosis   \n",
       "105                                           aberrant slug migration   \n",
       "106                           decreased secretion of lysosomal enzyme   \n",
       "107                         aberrant socially cooperative development   \n",
       "108                              abolished adenylate cyclase activity   \n",
       "109                                    decreased number of aggregates   \n",
       "110                aberrant cytokinesis  contractile ring contraction   \n",
       "111                                                  aberrant binding   \n",
       "112                                aberrant response to cation stress   \n",
       "113                                                aberrant autophagy   \n",
       "114                          decreased engulfment during phagocytosis   \n",
       "115         aberrant production of siRNA involved in RNA interference   \n",
       "116                               aberrant sorting to prestalk region   \n",
       "117                            Various types of N-glycan biosynthesis   \n",
       "118                           abolished activation of GTPase activity   \n",
       "119                                              decreased exocytosis   \n",
       "120                                    decreased chemotaxis to folate   \n",
       "121                                      increased myosin II assembly   \n",
       "122                                                 protein targeting   \n",
       "123                                            precocious development   \n",
       "124                                              aberrant culmination   \n",
       "125                                                        cell cycle   \n",
       "126                                        aberrant vegetative growth   \n",
       "127                                        precocious gene expression   \n",
       "128                                          increased aggregate size   \n",
       "129                                             aberrant phagocytosis   \n",
       "130                          aberrant response to hyperosmotic stress   \n",
       "131  increased infection rate by a gram-negative pathogenic bacterium   \n",
       "132                               increased protein kinase activation   \n",
       "133                                          increased slug migration   \n",
       "134                                                     cell adhesion   \n",
       "135                                                            growth   \n",
       "136                                                        cell death   \n",
       "137                 aberrant activation of adenylate cyclase activity   \n",
       "138                          aberrant adaptation of signaling pathway   \n",
       "139                                      increased fruiting body size   \n",
       "140              increased F-actin polymerization at the leading edge   \n",
       "141                                  abolished F-actin polymerization   \n",
       "142                                       increased membrane ruffling   \n",
       "143                                     abolished positive chemotaxis   \n",
       "144                                         increased chemotaxis rate   \n",
       "145                                 increased cytosolic Ca2+ response   \n",
       "146                          aberrant localization to plasma membrane   \n",
       "147                                    decreased accumulation of cGMP   \n",
       "148                                            precocious culmination   \n",
       "149                             abolished transitive RNA interference   \n",
       "150                           decreased prespore cell differentiation   \n",
       "151                                      abolished catalytic activity   \n",
       "152                                            precocious aggregation   \n",
       "153                                              cell differentiation   \n",
       "\n",
       "                     Ontology       FDR  N in query  \n",
       "0    (GO, biological_process)  2.50e-14          95  \n",
       "1    (GO, biological_process)  1.30e-10          58  \n",
       "2     (Dictybase, Phenotypes)  1.04e-08          36  \n",
       "3     (Dictybase, Phenotypes)  1.25e-06          27  \n",
       "4    (GO, biological_process)  2.94e-06          50  \n",
       "5    (GO, molecular_function)  9.42e-06          70  \n",
       "6     (Dictybase, Phenotypes)  9.43e-06          27  \n",
       "7     (Dictybase, Phenotypes)  1.14e-05          17  \n",
       "8    (GO, molecular_function)  2.51e-05          32  \n",
       "9     (Dictybase, Phenotypes)  5.55e-05          17  \n",
       "10    (Dictybase, Phenotypes)  1.19e-04          24  \n",
       "11           (KEGG, Pathways)  1.69e-04          26  \n",
       "12    (Dictybase, Phenotypes)  1.72e-04          15  \n",
       "13   (GO, cellular_component)  1.95e-04          56  \n",
       "14    (Dictybase, Phenotypes)  2.70e-04          66  \n",
       "15    (Dictybase, Phenotypes)  2.70e-04           7  \n",
       "16    (Dictybase, Phenotypes)  4.76e-04          20  \n",
       "17    (Dictybase, Phenotypes)  4.76e-04          13  \n",
       "18   (GO, cellular_component)  4.76e-04          30  \n",
       "19    (Dictybase, Phenotypes)  6.13e-04           8  \n",
       "20   (GO, biological_process)  9.54e-04          43  \n",
       "21    (Dictybase, Phenotypes)  1.26e-03           6  \n",
       "22    (Dictybase, Phenotypes)  1.56e-03          20  \n",
       "23    (Dictybase, Phenotypes)  2.04e-03          12  \n",
       "24    (Dictybase, Phenotypes)  2.25e-03           9  \n",
       "25    (Dictybase, Phenotypes)  2.25e-03           9  \n",
       "26    (Dictybase, Phenotypes)  2.78e-03           6  \n",
       "27   (GO, cellular_component)  4.96e-03          38  \n",
       "28    (Dictybase, Phenotypes)  5.20e-03           8  \n",
       "29    (Dictybase, Phenotypes)  7.70e-03           7  \n",
       "30           (KEGG, Pathways)  7.87e-03           8  \n",
       "31    (Dictybase, Phenotypes)  8.89e-03          10  \n",
       "32    (Dictybase, Phenotypes)  9.58e-03          11  \n",
       "33    (Dictybase, Phenotypes)  9.58e-03           6  \n",
       "34   (GO, biological_process)  9.58e-03          73  \n",
       "35   (GO, molecular_function)  1.08e-02          24  \n",
       "36   (GO, biological_process)  1.51e-02          17  \n",
       "37    (Dictybase, Phenotypes)  1.53e-02           7  \n",
       "38    (Dictybase, Phenotypes)  1.53e-02          18  \n",
       "39    (Dictybase, Phenotypes)  1.53e-02          16  \n",
       "40    (Dictybase, Phenotypes)  1.53e-02           6  \n",
       "41    (Dictybase, Phenotypes)  1.73e-02           3  \n",
       "42    (Dictybase, Phenotypes)  1.73e-02           3  \n",
       "43    (Dictybase, Phenotypes)  1.73e-02           3  \n",
       "44    (Dictybase, Phenotypes)  1.73e-02           3  \n",
       "45    (Dictybase, Phenotypes)  1.82e-02          20  \n",
       "46    (Dictybase, Phenotypes)  2.01e-02          18  \n",
       "47    (Dictybase, Phenotypes)  2.01e-02           7  \n",
       "48    (Dictybase, Phenotypes)  2.15e-02           4  \n",
       "49    (Dictybase, Phenotypes)  2.15e-02           4  \n",
       "50    (Dictybase, Phenotypes)  2.15e-02           4  \n",
       "51   (GO, biological_process)  2.15e-02          22  \n",
       "52    (Dictybase, Phenotypes)  2.20e-02           8  \n",
       "53    (Dictybase, Phenotypes)  2.26e-02          10  \n",
       "54           (KEGG, Pathways)  2.26e-02          11  \n",
       "55   (GO, biological_process)  2.26e-02          27  \n",
       "56    (Dictybase, Phenotypes)  2.49e-02          12  \n",
       "57    (Dictybase, Phenotypes)  2.50e-02          21  \n",
       "58    (Dictybase, Phenotypes)  2.68e-02           6  \n",
       "59   (GO, biological_process)  2.68e-02           8  \n",
       "60   (GO, molecular_function)  3.52e-02          30  \n",
       "61    (Dictybase, Phenotypes)  3.82e-02           4  \n",
       "62    (Dictybase, Phenotypes)  4.21e-02           5  \n",
       "63    (Dictybase, Phenotypes)  4.23e-02           3  \n",
       "64    (Dictybase, Phenotypes)  4.23e-02           3  \n",
       "65    (Dictybase, Phenotypes)  4.23e-02           3  \n",
       "66    (Dictybase, Phenotypes)  4.23e-02           9  \n",
       "67   (GO, cellular_component)  4.40e-02          36  \n",
       "68   (GO, biological_process)  4.47e-02          23  \n",
       "69   (GO, biological_process)  5.27e-02          57  \n",
       "70    (Dictybase, Phenotypes)  5.81e-02           4  \n",
       "71    (Dictybase, Phenotypes)  5.81e-02           4  \n",
       "72    (Dictybase, Phenotypes)  5.81e-02           4  \n",
       "73   (GO, biological_process)  5.81e-02          17  \n",
       "74    (Dictybase, Phenotypes)  6.47e-02           6  \n",
       "75   (GO, cellular_component)  6.91e-02          12  \n",
       "76    (Dictybase, Phenotypes)  7.08e-02           9  \n",
       "77    (Dictybase, Phenotypes)  7.83e-02           2  \n",
       "78    (Dictybase, Phenotypes)  7.83e-02           3  \n",
       "79    (Dictybase, Phenotypes)  7.83e-02           3  \n",
       "80    (Dictybase, Phenotypes)  7.83e-02           2  \n",
       "81    (Dictybase, Phenotypes)  7.83e-02           2  \n",
       "82    (Dictybase, Phenotypes)  7.83e-02           2  \n",
       "83    (Dictybase, Phenotypes)  7.83e-02           3  \n",
       "84    (Dictybase, Phenotypes)  7.83e-02          12  \n",
       "85    (Dictybase, Phenotypes)  7.99e-02           4  \n",
       "86    (Dictybase, Phenotypes)  7.99e-02           4  \n",
       "87    (Dictybase, Phenotypes)  7.99e-02           4  \n",
       "88    (Dictybase, Phenotypes)  8.06e-02          25  \n",
       "89   (GO, molecular_function)  8.75e-02          26  \n",
       "90           (KEGG, Pathways)  1.01e-01           8  \n",
       "91    (Dictybase, Phenotypes)  1.12e-01           7  \n",
       "92    (Dictybase, Phenotypes)  1.15e-01           4  \n",
       "93    (Dictybase, Phenotypes)  1.19e-01           6  \n",
       "94    (Dictybase, Phenotypes)  1.22e-01           3  \n",
       "95    (Dictybase, Phenotypes)  1.22e-01           3  \n",
       "96    (Dictybase, Phenotypes)  1.22e-01           3  \n",
       "97    (Dictybase, Phenotypes)  1.23e-01          13  \n",
       "98    (Dictybase, Phenotypes)  1.38e-01          13  \n",
       "99    (Dictybase, Phenotypes)  1.47e-01           5  \n",
       "100   (Dictybase, Phenotypes)  1.47e-01           4  \n",
       "101   (Dictybase, Phenotypes)  1.47e-01           4  \n",
       "102          (KEGG, Pathways)  1.47e-01           4  \n",
       "103  (GO, molecular_function)  1.47e-01          11  \n",
       "104   (Dictybase, Phenotypes)  1.47e-01           7  \n",
       "105   (Dictybase, Phenotypes)  1.53e-01          12  \n",
       "106   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "107   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "108   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "109   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "110   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "111   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "112   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "113   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "114   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "115   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "116   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "117          (KEGG, Pathways)  1.56e-01           6  \n",
       "118   (Dictybase, Phenotypes)  1.59e-01           3  \n",
       "119   (Dictybase, Phenotypes)  1.70e-01           4  \n",
       "120   (Dictybase, Phenotypes)  1.70e-01           4  \n",
       "121   (Dictybase, Phenotypes)  1.70e-01           4  \n",
       "122  (GO, biological_process)  1.79e-01           8  \n",
       "123   (Dictybase, Phenotypes)  1.79e-01           7  \n",
       "124   (Dictybase, Phenotypes)  1.83e-01          16  \n",
       "125  (GO, biological_process)  1.87e-01          27  \n",
       "126   (Dictybase, Phenotypes)  1.93e-01           9  \n",
       "127   (Dictybase, Phenotypes)  1.93e-01           5  \n",
       "128   (Dictybase, Phenotypes)  2.02e-01           7  \n",
       "129   (Dictybase, Phenotypes)  2.04e-01           3  \n",
       "130   (Dictybase, Phenotypes)  2.04e-01           4  \n",
       "131   (Dictybase, Phenotypes)  2.04e-01           4  \n",
       "132   (Dictybase, Phenotypes)  2.04e-01           4  \n",
       "133   (Dictybase, Phenotypes)  2.04e-01           3  \n",
       "134  (GO, biological_process)  2.04e-01          15  \n",
       "135  (GO, biological_process)  2.04e-01           3  \n",
       "136  (GO, biological_process)  2.26e-01           6  \n",
       "137   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "138   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "139   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "140   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "141   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "142   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "143   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "144   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "145   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "146   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "147   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "148   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "149   (Dictybase, Phenotypes)  2.27e-01           2  \n",
       "150   (Dictybase, Phenotypes)  2.29e-01           4  \n",
       "151   (Dictybase, Phenotypes)  2.29e-01           4  \n",
       "152   (Dictybase, Phenotypes)  2.40e-01           5  \n",
       "153  (GO, biological_process)  2.45e-01          23  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 191 with EID: 191\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 62 out of 76 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.87e-25</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.42e-23</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.98e-14</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.78e-12</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.17e-10</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.03e-10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.28e-09</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.25e-09</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.32e-08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.37e-07</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>4.49e-06</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.21e-05</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.14e-04</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.14e-04</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.27e-04</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.17e-03</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.81e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.24e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.51e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>nuclear envelope</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>7.94e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.94e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.94e-02</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>nuclease activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.67e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.82e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.32e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Gene set                  Ontology       FDR  \\\n",
       "0                 DNA replication          (KEGG, Pathways)  1.87e-25   \n",
       "1                      chromosome  (GO, cellular_component)  1.42e-23   \n",
       "2         chromosome organization  (GO, biological_process)  1.98e-14   \n",
       "3                 Mismatch repair          (KEGG, Pathways)  8.78e-12   \n",
       "4          chromosome segregation  (GO, biological_process)  2.17e-10   \n",
       "5      Nucleotide excision repair          (KEGG, Pathways)  7.03e-10   \n",
       "6                      cell cycle  (GO, biological_process)  2.28e-09   \n",
       "7           DNA metabolic process  (GO, biological_process)  9.25e-09   \n",
       "8        mitotic nuclear division  (GO, biological_process)  5.32e-08   \n",
       "9            Base excision repair          (KEGG, Pathways)  3.37e-07   \n",
       "10             nuclear chromosome  (GO, cellular_component)  4.49e-06   \n",
       "11             mitotic cell cycle  (GO, biological_process)  2.21e-05   \n",
       "12                ATPase activity  (GO, molecular_function)  1.14e-04   \n",
       "13                    DNA binding  (GO, molecular_function)  1.14e-04   \n",
       "14                  cell division  (GO, biological_process)  4.27e-04   \n",
       "15                    nucleoplasm  (GO, cellular_component)  2.17e-03   \n",
       "16              helicase activity  (GO, molecular_function)  1.81e-02   \n",
       "17     Non-homologous end-joining          (KEGG, Pathways)  4.24e-02   \n",
       "18                histone binding  (GO, molecular_function)  4.24e-02   \n",
       "19       Homologous recombination          (KEGG, Pathways)  6.51e-02   \n",
       "20               nuclear envelope  (GO, cellular_component)  7.94e-02   \n",
       "21          Pyrimidine metabolism          (KEGG, Pathways)  7.94e-02   \n",
       "22             response to stress  (GO, biological_process)  7.94e-02   \n",
       "23              nuclease activity  (GO, molecular_function)  1.67e-01   \n",
       "24                      nucleolus  (GO, cellular_component)  1.82e-01   \n",
       "25                   cytoskeleton  (GO, cellular_component)  2.17e-01   \n",
       "26  microtubule organizing center  (GO, cellular_component)  2.32e-01   \n",
       "\n",
       "    N in query  \n",
       "0           20  \n",
       "1           24  \n",
       "2           18  \n",
       "3           10  \n",
       "4            9  \n",
       "5           11  \n",
       "6           19  \n",
       "7           15  \n",
       "8            7  \n",
       "9            8  \n",
       "10           6  \n",
       "11          11  \n",
       "12          13  \n",
       "13          15  \n",
       "14          10  \n",
       "15           8  \n",
       "16           5  \n",
       "17           2  \n",
       "18           2  \n",
       "19           3  \n",
       "20           3  \n",
       "21           3  \n",
       "22          13  \n",
       "23           3  \n",
       "24           4  \n",
       "25           7  \n",
       "26           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 261 with EID: 260\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.38\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 73 out of 99 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.23e-27</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.97e-23</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.11e-13</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.63e-13</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.41e-12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.38e-11</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.08e-09</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.03e-09</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.49e-08</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.91e-07</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.30e-07</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.67e-06</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.36e-06</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.85e-06</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.12e-03</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.40e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.07e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.32e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.41e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.85e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.26e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.80e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.80e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.16e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>nucleobase-containing compound catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                                         chromosome   \n",
       "1                                    DNA replication   \n",
       "2                            chromosome organization   \n",
       "3                                         cell cycle   \n",
       "4                                    Mismatch repair   \n",
       "5                         Nucleotide excision repair   \n",
       "6                             chromosome segregation   \n",
       "7                              DNA metabolic process   \n",
       "8                                 nuclear chromosome   \n",
       "9                                      cell division   \n",
       "10                          mitotic nuclear division   \n",
       "11                              Base excision repair   \n",
       "12                                       DNA binding   \n",
       "13                                mitotic cell cycle   \n",
       "14                                   ATPase activity   \n",
       "15                             Pyrimidine metabolism   \n",
       "16                     microtubule organizing center   \n",
       "17                                       nucleoplasm   \n",
       "18                                      cytoskeleton   \n",
       "19                                 cell cycle arrest   \n",
       "20                                 helicase activity   \n",
       "21                        Non-homologous end-joining   \n",
       "22                                   histone binding   \n",
       "23                          Homologous recombination   \n",
       "24  nucleobase-containing compound catabolic process   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0   (GO, cellular_component)  1.23e-27          29  \n",
       "1           (KEGG, Pathways)  3.97e-23          20  \n",
       "2   (GO, biological_process)  2.11e-13          19  \n",
       "3   (GO, biological_process)  3.63e-13          26  \n",
       "4           (KEGG, Pathways)  2.41e-12          11  \n",
       "5           (KEGG, Pathways)  3.38e-11          13  \n",
       "6   (GO, biological_process)  2.08e-09           9  \n",
       "7   (GO, biological_process)  7.03e-09          17  \n",
       "8   (GO, cellular_component)  3.49e-08           8  \n",
       "9   (GO, biological_process)  1.91e-07          16  \n",
       "10  (GO, biological_process)  3.30e-07           7  \n",
       "11          (KEGG, Pathways)  2.67e-06           8  \n",
       "12  (GO, molecular_function)  3.36e-06          20  \n",
       "13  (GO, biological_process)  7.85e-06          13  \n",
       "14  (GO, molecular_function)  2.12e-03          13  \n",
       "15          (KEGG, Pathways)  5.40e-03           5  \n",
       "16  (GO, cellular_component)  1.07e-02           6  \n",
       "17  (GO, cellular_component)  1.32e-02           8  \n",
       "18  (GO, cellular_component)  1.41e-02          12  \n",
       "19   (Dictybase, Phenotypes)  3.85e-02           2  \n",
       "20  (GO, molecular_function)  5.26e-02           5  \n",
       "21          (KEGG, Pathways)  6.80e-02           2  \n",
       "22  (GO, molecular_function)  6.80e-02           2  \n",
       "23          (KEGG, Pathways)  1.16e-01           3  \n",
       "24  (GO, biological_process)  1.17e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 76 with EID: 76\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 5 out of 21 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.99e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene set                  Ontology       FDR  N in query\n",
       "0      cell cycle arrest   (Dictybase, Phenotypes)  1.99e-02           2\n",
       "1  Pyrimidine metabolism          (KEGG, Pathways)  2.17e-01           2\n",
       "2          cell division  (GO, biological_process)  2.17e-01           3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 8 with EID: 8\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.0\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 0 out of 0 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 565 with EID: 559\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.28\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 32 out of 156 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Golgi apparatus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>9.22e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.18e-01</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                  Ontology  \\\n",
       "0                            Golgi apparatus  (GO, cellular_component)   \n",
       "1  SNARE interactions in vesicular transport          (KEGG, Pathways)   \n",
       "2                    oxidoreductase activity  (GO, molecular_function)   \n",
       "\n",
       "        FDR  N in query  \n",
       "0  9.22e-02          11  \n",
       "1  1.18e-01           5  \n",
       "2  1.18e-01          18  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 555 with EID: 546\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.23\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 34 out of 123 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.78e-05</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.16e-03</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.23e-03</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.60e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.10e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.80e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0  anatomical structure formation involved in morphogenesis   \n",
       "1                                      extracellular region   \n",
       "2                                      cell differentiation   \n",
       "3                 SNARE interactions in vesicular transport   \n",
       "4                               secondary metabolic process   \n",
       "5       Ubiquinone and other terpenoid-quinone biosynthesis   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, biological_process)  5.78e-05          13  \n",
       "1  (GO, cellular_component)  2.16e-03          16  \n",
       "2  (GO, biological_process)  2.23e-03          15  \n",
       "3          (KEGG, Pathways)  2.60e-03           6  \n",
       "4  (GO, biological_process)  1.10e-02           3  \n",
       "5          (KEGG, Pathways)  9.80e-02           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_impulse+'DEacrossStages_summary_noyem.tsv',index_col=0,sep='\\t')\n",
    "stages=np.array([phenotype for phenotype in PHENOTYPES if phenotype in list(milestones.columns)])\n",
    "for stage_idx,stage in enumerate(stages):\n",
    "    \n",
    "    #Nonneighbouring stages\n",
    "    neigh_previous=1\n",
    "    neigh_next=1\n",
    "    #Assumes phenotypes order\n",
    "    if stage_idx==1:\n",
    "        neigh_previous=0\n",
    "    if stage_idx==0:\n",
    "        neigh_next=0\n",
    "    non_neighbours = stages[[(idx <stage_idx-neigh_previous or idx>stage_idx+neigh_next) \n",
    "                 for idx in range(len(stages)) ]]\n",
    "    #print(stage,':',non_neighbours)\n",
    "    \n",
    "    #Find genes up at stage and down at nonenighbouring\n",
    "    query_condition=stage+'==\"up\"'\n",
    "    for non_neigh in non_neighbours:\n",
    "        query_condition=query_condition+' & '+non_neigh+' ==\"down\"'\n",
    "    #print(query_condition)\n",
    "    \n",
    "    query_names=list(milestones.query(query_condition).index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### ImpulseDE2 WT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***  no_agg selected: 2330 with EID: 2310\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.4\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 568 out of 925 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Ribosome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.31e-28</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>translation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.33e-25</td>\n",
       "      <td>104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.06e-23</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.58e-22</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>mitochondrion</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.55e-19</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>ribosome biogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.12e-17</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>nucleolus</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.82e-16</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.63e-15</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Ribosome biogenesis in eukaryotes</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.93e-14</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>tRNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.01e-12</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.52e-09</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.87e-08</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>RNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.77e-07</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Steroid biosynthesis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.85e-05</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>ligase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.92e-05</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>methyltransferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>8.98e-05</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.29e-04</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>cellular amino acid metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.80e-04</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.23e-03</td>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>cofactor metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.44e-03</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.83e-03</td>\n",
       "      <td>193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.95e-03</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.96e-03</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>unfolded protein binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.10e-02</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Porphyrin and chlorophyll metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.13e-02</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>protein folding</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.36e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>mitochondrion organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.90e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.33e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>RNA polymerase</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.71e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>increased sensitivity to hydrogen peroxide</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.74e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>isomerase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Fatty acid elongation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>Butanoate metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>sulfur compound metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.05e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                  Ontology  \\\n",
       "0                                         Ribosome          (KEGG, Pathways)   \n",
       "1                                      translation  (GO, biological_process)   \n",
       "2               structural constituent of ribosome  (GO, molecular_function)   \n",
       "3                                         ribosome  (GO, cellular_component)   \n",
       "4                                    mitochondrion  (GO, cellular_component)   \n",
       "5                              ribosome biogenesis  (GO, biological_process)   \n",
       "6                                        nucleolus  (GO, cellular_component)   \n",
       "7                     structural molecule activity  (GO, molecular_function)   \n",
       "8                Ribosome biogenesis in eukaryotes          (KEGG, Pathways)   \n",
       "9                           tRNA metabolic process  (GO, biological_process)   \n",
       "10                     Aminoacyl-tRNA biosynthesis          (KEGG, Pathways)   \n",
       "11                       Oxidative phosphorylation          (KEGG, Pathways)   \n",
       "12                                     RNA binding  (GO, molecular_function)   \n",
       "13                            Steroid biosynthesis          (KEGG, Pathways)   \n",
       "14                                 ligase activity  (GO, molecular_function)   \n",
       "15                      methyltransferase activity  (GO, molecular_function)   \n",
       "16  generation of precursor metabolites and energy  (GO, biological_process)   \n",
       "17           cellular amino acid metabolic process  (GO, biological_process)   \n",
       "18                         oxidoreductase activity  (GO, molecular_function)   \n",
       "19                      cofactor metabolic process  (GO, biological_process)   \n",
       "20                              Metabolic pathways          (KEGG, Pathways)   \n",
       "21           Biosynthesis of secondary metabolites          (KEGG, Pathways)   \n",
       "22                small molecule metabolic process  (GO, biological_process)   \n",
       "23                        unfolded protein binding  (GO, molecular_function)   \n",
       "24            Porphyrin and chlorophyll metabolism          (KEGG, Pathways)   \n",
       "25                                 protein folding  (GO, biological_process)   \n",
       "26                      mitochondrion organization  (GO, biological_process)   \n",
       "27                                    rRNA binding  (GO, molecular_function)   \n",
       "28                                  RNA polymerase          (KEGG, Pathways)   \n",
       "29      increased sensitivity to hydrogen peroxide   (Dictybase, Phenotypes)   \n",
       "30                              isomerase activity  (GO, molecular_function)   \n",
       "31                           Fatty acid elongation          (KEGG, Pathways)   \n",
       "32                            Butanoate metabolism          (KEGG, Pathways)   \n",
       "33               sulfur compound metabolic process  (GO, biological_process)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   3.31e-28          76  \n",
       "1   6.33e-25         104  \n",
       "2   3.06e-23          65  \n",
       "3   1.58e-22          68  \n",
       "4   1.55e-19         118  \n",
       "5   5.12e-17          47  \n",
       "6   6.82e-16          51  \n",
       "7   3.63e-15          69  \n",
       "8   4.93e-14          47  \n",
       "9   5.01e-12          45  \n",
       "10  9.52e-09          31  \n",
       "11  1.87e-08          34  \n",
       "12  4.77e-07          72  \n",
       "13  4.85e-05          14  \n",
       "14  4.92e-05          38  \n",
       "15  8.98e-05          29  \n",
       "16  1.29e-04          36  \n",
       "17  8.80e-04          49  \n",
       "18  1.23e-03          73  \n",
       "19  3.44e-03          31  \n",
       "20  3.83e-03         193  \n",
       "21  4.95e-03          91  \n",
       "22  5.96e-03         110  \n",
       "23  2.10e-02          14  \n",
       "24  2.13e-02           9  \n",
       "25  3.36e-02          17  \n",
       "26  3.90e-02          12  \n",
       "27  5.33e-02           8  \n",
       "28  8.71e-02          12  \n",
       "29  1.74e-01           3  \n",
       "30  1.93e-01          19  \n",
       "31  2.05e-01           7  \n",
       "32  2.05e-01           7  \n",
       "33  2.05e-01          14  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  stream selected: 409 with EID: 403\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.41\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 131 out of 166 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.95e-06</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.29e-04</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.02e-03</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.21e-03</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.26e-03</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.06e-03</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.44e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.44e-03</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.03e-03</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.03e-03</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.18e-03</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.78e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.78e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>decreased exocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.43e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.43e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>aberrant chemotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.47e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.47e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.74e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>curly fingers</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.77e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant chemotaxis to cAMP in response to DIF-2</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.77e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.14e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.14e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.84e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>increased filopodium formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.87e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>abolished chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>decreased secretion of lysosomal enzyme</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>decreased engulfment during phagocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>abolished filopodium formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>4.89e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.39e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.39e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>aberrant cytoskeleton organisation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>decreased macropinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>aberrant slug formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.92e-02</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>decreased pinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.95e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>6.13e-02</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.13e-02</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.24e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>abolished protein phosphorylation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.59e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.59e-02</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>abolished F-actin polymerization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.68e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>precocious culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.68e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.37e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>decreased cAMP-induced cAMP pulse</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.66e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>aberrant response to hyperosmotic stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.66e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.98e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>enzyme binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.00e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>abolished chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.12e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>nucleobase-containing compound catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.28e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>aberrant chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>decreased activation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.33e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.44e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>increased protein level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>increased intracellular cAMP level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>decreased growth rate in suspension</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>peptidase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.47e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>decreased protein kinase activation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.50e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.59e-01</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>68</td>\n",
       "      <td>abolished growth</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.62e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>69</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>70</td>\n",
       "      <td>abolished cell polarity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>71</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>decreased lateral pseudopod suppression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73</td>\n",
       "      <td>aberrant pseudopod formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>76</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.63e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>77</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.80e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>79</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.90e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>81</td>\n",
       "      <td>cell morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.08e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>82</td>\n",
       "      <td>AGE-RAGE signaling pathway in diabetic complications</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>Other glycan degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>84</td>\n",
       "      <td>decreased chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>85</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>86</td>\n",
       "      <td>abolished gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>87</td>\n",
       "      <td>abolished sporulation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>88</td>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>89</td>\n",
       "      <td>increased myosin II assembly</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>90</td>\n",
       "      <td>development arrests at slug stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>91</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.13e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Gene set  \\\n",
       "0                                             Proteasome   \n",
       "1            Protein processing in endoplasmic reticulum   \n",
       "2                                             locomotion   \n",
       "3                                    signal transduction   \n",
       "4                                     aberrant streaming   \n",
       "5                                      catabolic process   \n",
       "6              abolished activation of adenylate cyclase   \n",
       "7                                   aberrant aggregation   \n",
       "8                               decreased aggregate size   \n",
       "9                                decreased cell motility   \n",
       "10                                             wild type   \n",
       "11                                  aberrant cytokinesis   \n",
       "12                         aberrant F-actin organization   \n",
       "13                                  decreased exocytosis   \n",
       "14              development arrests at loose mound stage   \n",
       "15                                   aberrant chemotaxis   \n",
       "16                             cytoskeleton organization   \n",
       "17                                          reproduction   \n",
       "18                                         curly fingers   \n",
       "19      aberrant chemotaxis to cAMP in response to DIF-2   \n",
       "20                                   abolished streaming   \n",
       "21                          aberrant cAMP wave formation   \n",
       "22                                  aberrant culmination   \n",
       "23                        increased filopodium formation   \n",
       "24                          decreased chemotaxis to cAMP   \n",
       "25                                 decreased culmination   \n",
       "26                         increased basal F-actin level   \n",
       "27                        abolished chemotaxis to folate   \n",
       "28               decreased secretion of lysosomal enzyme   \n",
       "29              decreased engulfment during phagocytosis   \n",
       "30                        abolished filopodium formation   \n",
       "31                           cellular component assembly   \n",
       "32                             enzyme regulator activity   \n",
       "33                            precocious gene expression   \n",
       "34                                         cell motility   \n",
       "35                    aberrant cytoskeleton organisation   \n",
       "36                            decreased macropinocytosis   \n",
       "37                               aberrant slug formation   \n",
       "38                                    response to stress   \n",
       "39                                 decreased pinocytosis   \n",
       "40                                       plasma membrane   \n",
       "41                      anatomical structure development   \n",
       "42                              abolished slug migration   \n",
       "43                     abolished protein phosphorylation   \n",
       "44                                       kinase activity   \n",
       "45                      abolished F-actin polymerization   \n",
       "46                                precocious culmination   \n",
       "47                          decreased fruiting body size   \n",
       "48                     decreased cAMP-induced cAMP pulse   \n",
       "49              aberrant response to hyperosmotic stress   \n",
       "50                            multiple tips in culminant   \n",
       "51                                        enzyme binding   \n",
       "52                          abolished chemotaxis to cAMP   \n",
       "53      nucleobase-containing compound catabolic process   \n",
       "54                         aberrant chemotaxis to folate   \n",
       "55             decreased activation of adenylate cyclase   \n",
       "56                    development arrests at mound stage   \n",
       "57                                   delayed development   \n",
       "58                                 endoplasmic reticulum   \n",
       "59                              aberrant cell morphology   \n",
       "60                               aberrant slug migration   \n",
       "61                               increased protein level   \n",
       "62                                 increased growth rate   \n",
       "63                    increased intracellular cAMP level   \n",
       "64                   decreased growth rate in suspension   \n",
       "65                                    peptidase activity   \n",
       "66                   decreased protein kinase activation   \n",
       "67                                          cytoskeleton   \n",
       "68                                      abolished growth   \n",
       "69                                precocious aggregation   \n",
       "70                               abolished cell polarity   \n",
       "71                                 abolished aggregation   \n",
       "72               decreased lateral pseudopod suppression   \n",
       "73                          aberrant pseudopod formation   \n",
       "74                   protein-containing complex assembly   \n",
       "75                                         cell division   \n",
       "76                                       ATPase activity   \n",
       "77                                    mitotic cell cycle   \n",
       "78                                         cell adhesion   \n",
       "79                             decreased chemotaxis rate   \n",
       "80                             decreased gene expression   \n",
       "81                                    cell morphogenesis   \n",
       "82  AGE-RAGE signaling pathway in diabetic complications   \n",
       "83                              Other glycan degradation   \n",
       "84                        decreased chemotaxis to folate   \n",
       "85             development arrests at tipped mound stage   \n",
       "86                             abolished gene expression   \n",
       "87                                 abolished sporulation   \n",
       "88                    decreased intracellular cAMP level   \n",
       "89                          increased myosin II assembly   \n",
       "90                     development arrests at slug stage   \n",
       "91                              Homologous recombination   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0           (KEGG, Pathways)  4.95e-06          12  \n",
       "1           (KEGG, Pathways)  4.29e-04          15  \n",
       "2   (GO, biological_process)  1.02e-03          22  \n",
       "3   (GO, biological_process)  1.21e-03          32  \n",
       "4    (Dictybase, Phenotypes)  2.26e-03          12  \n",
       "5   (GO, biological_process)  5.06e-03          24  \n",
       "6    (Dictybase, Phenotypes)  7.44e-03           4  \n",
       "7    (Dictybase, Phenotypes)  7.44e-03          10  \n",
       "8    (Dictybase, Phenotypes)  9.03e-03           7  \n",
       "9    (Dictybase, Phenotypes)  9.03e-03          11  \n",
       "10   (Dictybase, Phenotypes)  9.18e-03          28  \n",
       "11   (Dictybase, Phenotypes)  1.78e-02          12  \n",
       "12   (Dictybase, Phenotypes)  1.78e-02           7  \n",
       "13   (Dictybase, Phenotypes)  2.43e-02           4  \n",
       "14   (Dictybase, Phenotypes)  2.43e-02           5  \n",
       "15   (Dictybase, Phenotypes)  2.47e-02           3  \n",
       "16  (GO, biological_process)  2.47e-02          18  \n",
       "17  (GO, biological_process)  2.74e-02          10  \n",
       "18   (Dictybase, Phenotypes)  2.77e-02           2  \n",
       "19   (Dictybase, Phenotypes)  2.77e-02           2  \n",
       "20   (Dictybase, Phenotypes)  3.14e-02           6  \n",
       "21   (Dictybase, Phenotypes)  3.14e-02          11  \n",
       "22   (Dictybase, Phenotypes)  4.84e-02          10  \n",
       "23   (Dictybase, Phenotypes)  4.87e-02           4  \n",
       "24   (Dictybase, Phenotypes)  4.89e-02           8  \n",
       "25   (Dictybase, Phenotypes)  4.89e-02           3  \n",
       "26   (Dictybase, Phenotypes)  4.89e-02           4  \n",
       "27   (Dictybase, Phenotypes)  4.89e-02           2  \n",
       "28   (Dictybase, Phenotypes)  4.89e-02           2  \n",
       "29   (Dictybase, Phenotypes)  4.89e-02           2  \n",
       "30   (Dictybase, Phenotypes)  4.89e-02           2  \n",
       "31  (GO, biological_process)  4.89e-02          17  \n",
       "32  (GO, molecular_function)  4.89e-02          11  \n",
       "33   (Dictybase, Phenotypes)  5.39e-02           4  \n",
       "34  (GO, biological_process)  5.39e-02           8  \n",
       "35   (Dictybase, Phenotypes)  5.92e-02           3  \n",
       "36   (Dictybase, Phenotypes)  5.92e-02           4  \n",
       "37   (Dictybase, Phenotypes)  5.92e-02           3  \n",
       "38  (GO, biological_process)  5.92e-02          25  \n",
       "39   (Dictybase, Phenotypes)  5.95e-02           5  \n",
       "40  (GO, cellular_component)  6.13e-02          20  \n",
       "41  (GO, biological_process)  6.13e-02          29  \n",
       "42   (Dictybase, Phenotypes)  6.24e-02           6  \n",
       "43   (Dictybase, Phenotypes)  6.59e-02           3  \n",
       "44  (GO, molecular_function)  6.59e-02          23  \n",
       "45   (Dictybase, Phenotypes)  6.68e-02           2  \n",
       "46   (Dictybase, Phenotypes)  6.68e-02           2  \n",
       "47   (Dictybase, Phenotypes)  8.37e-02           8  \n",
       "48   (Dictybase, Phenotypes)  9.66e-02           3  \n",
       "49   (Dictybase, Phenotypes)  9.66e-02           3  \n",
       "50   (Dictybase, Phenotypes)  9.98e-02           2  \n",
       "51  (GO, molecular_function)  1.00e-01          10  \n",
       "52   (Dictybase, Phenotypes)  1.12e-01           3  \n",
       "53  (GO, biological_process)  1.28e-01           4  \n",
       "54   (Dictybase, Phenotypes)  1.33e-01           2  \n",
       "55   (Dictybase, Phenotypes)  1.33e-01           2  \n",
       "56   (Dictybase, Phenotypes)  1.33e-01           6  \n",
       "57   (Dictybase, Phenotypes)  1.33e-01           7  \n",
       "58  (GO, cellular_component)  1.33e-01          10  \n",
       "59   (Dictybase, Phenotypes)  1.44e-01           5  \n",
       "60   (Dictybase, Phenotypes)  1.47e-01           6  \n",
       "61   (Dictybase, Phenotypes)  1.47e-01           2  \n",
       "62   (Dictybase, Phenotypes)  1.47e-01           4  \n",
       "63   (Dictybase, Phenotypes)  1.47e-01           2  \n",
       "64   (Dictybase, Phenotypes)  1.47e-01           2  \n",
       "65  (GO, molecular_function)  1.47e-01           8  \n",
       "66   (Dictybase, Phenotypes)  1.50e-01           3  \n",
       "67  (GO, cellular_component)  1.59e-01          13  \n",
       "68   (Dictybase, Phenotypes)  1.62e-01           4  \n",
       "69   (Dictybase, Phenotypes)  1.63e-01           3  \n",
       "70   (Dictybase, Phenotypes)  1.63e-01           2  \n",
       "71   (Dictybase, Phenotypes)  1.63e-01           7  \n",
       "72   (Dictybase, Phenotypes)  1.63e-01           2  \n",
       "73   (Dictybase, Phenotypes)  1.63e-01           2  \n",
       "74  (GO, biological_process)  1.63e-01          10  \n",
       "75  (GO, biological_process)  1.63e-01           9  \n",
       "76  (GO, molecular_function)  1.63e-01          12  \n",
       "77  (GO, biological_process)  1.80e-01           8  \n",
       "78  (GO, biological_process)  1.80e-01           7  \n",
       "79   (Dictybase, Phenotypes)  1.90e-01           4  \n",
       "80   (Dictybase, Phenotypes)  2.06e-01           7  \n",
       "81  (GO, biological_process)  2.08e-01           3  \n",
       "82          (KEGG, Pathways)  2.12e-01           3  \n",
       "83          (KEGG, Pathways)  2.12e-01           2  \n",
       "84   (Dictybase, Phenotypes)  2.13e-01           2  \n",
       "85   (Dictybase, Phenotypes)  2.13e-01           2  \n",
       "86   (Dictybase, Phenotypes)  2.13e-01           3  \n",
       "87   (Dictybase, Phenotypes)  2.13e-01           3  \n",
       "88   (Dictybase, Phenotypes)  2.13e-01           2  \n",
       "89   (Dictybase, Phenotypes)  2.13e-01           2  \n",
       "90   (Dictybase, Phenotypes)  2.13e-01           2  \n",
       "91          (KEGG, Pathways)  2.13e-01           3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  lag selected: 429 with EID: 423\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.42\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 135 out of 176 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Proteasome</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.91e-06</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>signal transduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.48e-05</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>locomotion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.26e-04</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>aberrant aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.07e-04</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Protein processing in endoplasmic reticulum</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.73e-03</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>aberrant streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.97e-03</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased aggregate size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.97e-03</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>abolished activation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>9.46e-03</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>abolished streaming</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.51e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>decreased cell motility</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.51e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>abolished slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.51e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.86e-02</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>aberrant F-actin organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.59e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>aberrant cytokinesis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.83e-02</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>decreased exocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.85e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>development arrests at loose mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.85e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>aberrant chemotaxis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.85e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.85e-02</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>curly fingers</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>aberrant chemotaxis to cAMP in response to DIF-2</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.20e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>cytoskeleton organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.88e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>aberrant cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>4.90e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>wild type</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.35e-02</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.65e-02</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>increased filopodium formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.94e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>increased basal F-actin level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.05e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>abolished chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.05e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>decreased secretion of lysosomal enzyme</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.05e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>aberrant culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.05e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>decreased engulfment during phagocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.05e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>abolished filopodium formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.05e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>decreased culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.09e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>decreased chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>6.27e-02</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.31e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.06e-02</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>aberrant cytoskeleton organisation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.48e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>decreased macropinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>7.69e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>development arrests at mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.00e-02</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>decreased pinocytosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.00e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>precocious aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.42e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>abolished F-actin polymerization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.42e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>abolished protein phosphorylation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.42e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>precocious culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.42e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.07e-01</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>decreased cAMP-induced cAMP pulse</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>multiple tips in culminant</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>aberrant response to hyperosmotic stress</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>decreased fruiting body size</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>anatomical structure development</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.17e-01</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>abolished chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.40e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>reproduction</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>aberrant chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>decreased activation of adenylate cyclase</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>cell motility</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.56e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>aberrant cell morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>aberrant slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>delayed gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>increased protein level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>aberrant chemotaxis to cAMP</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>increased growth rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>increased intracellular cAMP level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>decreased growth rate in suspension</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>delayed development</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>increased slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>precocious gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>cell adhesion</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>67</td>\n",
       "      <td>decreased protein kinase activation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.76e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>68</td>\n",
       "      <td>abolished culmination</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>69</td>\n",
       "      <td>abolished cell polarity</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>70</td>\n",
       "      <td>decreased lateral pseudopod suppression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>71</td>\n",
       "      <td>abolished growth</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>aberrant pseudopod formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73</td>\n",
       "      <td>kinase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74</td>\n",
       "      <td>enzyme binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.89e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75</td>\n",
       "      <td>abolished aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.09e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>76</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.17e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>77</td>\n",
       "      <td>aberrant slug formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.20e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.20e-01</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>79</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.20e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80</td>\n",
       "      <td>protein-containing complex assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.22e-01</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>81</td>\n",
       "      <td>decreased chemotaxis rate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>82</td>\n",
       "      <td>decreased slug migration</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>abolished cAMP wave formation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>84</td>\n",
       "      <td>Other glycan degradation</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>85</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.25e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>86</td>\n",
       "      <td>decreased gene expression</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.26e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>87</td>\n",
       "      <td>decreased chemotaxis to folate</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>88</td>\n",
       "      <td>development arrests at tipped mound stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>89</td>\n",
       "      <td>decreased intracellular cAMP level</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>90</td>\n",
       "      <td>increased myosin II assembly</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>91</td>\n",
       "      <td>development arrests at slug stage</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>92</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.40e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>93</td>\n",
       "      <td>delayed aggregation</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.45e-01</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Gene set  \\\n",
       "0                                         Proteasome   \n",
       "1                                signal transduction   \n",
       "2                                         locomotion   \n",
       "3                               aberrant aggregation   \n",
       "4        Protein processing in endoplasmic reticulum   \n",
       "5                                 aberrant streaming   \n",
       "6                           decreased aggregate size   \n",
       "7          abolished activation of adenylate cyclase   \n",
       "8                                abolished streaming   \n",
       "9                            decreased cell motility   \n",
       "10                          abolished slug migration   \n",
       "11                                   plasma membrane   \n",
       "12                     aberrant F-actin organization   \n",
       "13                              aberrant cytokinesis   \n",
       "14                              decreased exocytosis   \n",
       "15          development arrests at loose mound stage   \n",
       "16                               aberrant chemotaxis   \n",
       "17                                 catabolic process   \n",
       "18                                     curly fingers   \n",
       "19  aberrant chemotaxis to cAMP in response to DIF-2   \n",
       "20                         cytoskeleton organization   \n",
       "21                      aberrant cAMP wave formation   \n",
       "22                                         wild type   \n",
       "23                                response to stress   \n",
       "24                    increased filopodium formation   \n",
       "25                     increased basal F-actin level   \n",
       "26                    abolished chemotaxis to folate   \n",
       "27           decreased secretion of lysosomal enzyme   \n",
       "28                              aberrant culmination   \n",
       "29          decreased engulfment during phagocytosis   \n",
       "30                    abolished filopodium formation   \n",
       "31                             decreased culmination   \n",
       "32                      decreased chemotaxis to cAMP   \n",
       "33                         enzyme regulator activity   \n",
       "34                       cellular component assembly   \n",
       "35                aberrant cytoskeleton organisation   \n",
       "36                        decreased macropinocytosis   \n",
       "37                development arrests at mound stage   \n",
       "38                             decreased pinocytosis   \n",
       "39                            precocious aggregation   \n",
       "40                  abolished F-actin polymerization   \n",
       "41                 abolished protein phosphorylation   \n",
       "42                            precocious culmination   \n",
       "43                             endoplasmic reticulum   \n",
       "44                 decreased cAMP-induced cAMP pulse   \n",
       "45                        multiple tips in culminant   \n",
       "46          aberrant response to hyperosmotic stress   \n",
       "47                      decreased fruiting body size   \n",
       "48                                      cytoskeleton   \n",
       "49                  anatomical structure development   \n",
       "50                      abolished chemotaxis to cAMP   \n",
       "51                                      reproduction   \n",
       "52                     aberrant chemotaxis to folate   \n",
       "53         decreased activation of adenylate cyclase   \n",
       "54                                     cell motility   \n",
       "55                          aberrant cell morphology   \n",
       "56                           aberrant slug migration   \n",
       "57                           delayed gene expression   \n",
       "58                           increased protein level   \n",
       "59                       aberrant chemotaxis to cAMP   \n",
       "60                             increased growth rate   \n",
       "61                increased intracellular cAMP level   \n",
       "62               decreased growth rate in suspension   \n",
       "63                               delayed development   \n",
       "64                          increased slug migration   \n",
       "65                        precocious gene expression   \n",
       "66                                     cell adhesion   \n",
       "67               decreased protein kinase activation   \n",
       "68                             abolished culmination   \n",
       "69                           abolished cell polarity   \n",
       "70           decreased lateral pseudopod suppression   \n",
       "71                                  abolished growth   \n",
       "72                      aberrant pseudopod formation   \n",
       "73                                   kinase activity   \n",
       "74                                    enzyme binding   \n",
       "75                             abolished aggregation   \n",
       "76                                           cytosol   \n",
       "77                           aberrant slug formation   \n",
       "78                                     cell division   \n",
       "79                                   ATPase activity   \n",
       "80               protein-containing complex assembly   \n",
       "81                         decreased chemotaxis rate   \n",
       "82                          decreased slug migration   \n",
       "83                     abolished cAMP wave formation   \n",
       "84                          Other glycan degradation   \n",
       "85                                mitotic cell cycle   \n",
       "86                         decreased gene expression   \n",
       "87                    decreased chemotaxis to folate   \n",
       "88         development arrests at tipped mound stage   \n",
       "89                decreased intracellular cAMP level   \n",
       "90                      increased myosin II assembly   \n",
       "91                 development arrests at slug stage   \n",
       "92                          Homologous recombination   \n",
       "93                               delayed aggregation   \n",
       "\n",
       "                    Ontology       FDR  N in query  \n",
       "0           (KEGG, Pathways)  9.91e-06          12  \n",
       "1   (GO, biological_process)  5.48e-05          37  \n",
       "2   (GO, biological_process)  8.26e-04          23  \n",
       "3    (Dictybase, Phenotypes)  9.07e-04          12  \n",
       "4           (KEGG, Pathways)  1.73e-03          14  \n",
       "5    (Dictybase, Phenotypes)  2.97e-03          12  \n",
       "6    (Dictybase, Phenotypes)  2.97e-03           8  \n",
       "7    (Dictybase, Phenotypes)  9.46e-03           4  \n",
       "8    (Dictybase, Phenotypes)  1.51e-02           7  \n",
       "9    (Dictybase, Phenotypes)  1.51e-02          11  \n",
       "10   (Dictybase, Phenotypes)  1.51e-02           8  \n",
       "11  (GO, cellular_component)  1.86e-02          24  \n",
       "12   (Dictybase, Phenotypes)  2.59e-02           7  \n",
       "13   (Dictybase, Phenotypes)  2.83e-02          12  \n",
       "14   (Dictybase, Phenotypes)  2.85e-02           4  \n",
       "15   (Dictybase, Phenotypes)  2.85e-02           5  \n",
       "16   (Dictybase, Phenotypes)  2.85e-02           3  \n",
       "17  (GO, biological_process)  2.85e-02          22  \n",
       "18   (Dictybase, Phenotypes)  3.20e-02           2  \n",
       "19   (Dictybase, Phenotypes)  3.20e-02           2  \n",
       "20  (GO, biological_process)  3.88e-02          18  \n",
       "21   (Dictybase, Phenotypes)  4.90e-02          11  \n",
       "22   (Dictybase, Phenotypes)  5.35e-02          26  \n",
       "23  (GO, biological_process)  5.65e-02          27  \n",
       "24   (Dictybase, Phenotypes)  5.94e-02           4  \n",
       "25   (Dictybase, Phenotypes)  6.05e-02           4  \n",
       "26   (Dictybase, Phenotypes)  6.05e-02           2  \n",
       "27   (Dictybase, Phenotypes)  6.05e-02           2  \n",
       "28   (Dictybase, Phenotypes)  6.05e-02          10  \n",
       "29   (Dictybase, Phenotypes)  6.05e-02           2  \n",
       "30   (Dictybase, Phenotypes)  6.05e-02           2  \n",
       "31   (Dictybase, Phenotypes)  6.09e-02           3  \n",
       "32   (Dictybase, Phenotypes)  6.27e-02           8  \n",
       "33  (GO, molecular_function)  6.31e-02          11  \n",
       "34  (GO, biological_process)  7.06e-02          17  \n",
       "35   (Dictybase, Phenotypes)  7.48e-02           3  \n",
       "36   (Dictybase, Phenotypes)  7.69e-02           4  \n",
       "37   (Dictybase, Phenotypes)  8.00e-02           7  \n",
       "38   (Dictybase, Phenotypes)  8.00e-02           5  \n",
       "39   (Dictybase, Phenotypes)  8.42e-02           4  \n",
       "40   (Dictybase, Phenotypes)  8.42e-02           2  \n",
       "41   (Dictybase, Phenotypes)  8.42e-02           3  \n",
       "42   (Dictybase, Phenotypes)  8.42e-02           2  \n",
       "43  (GO, cellular_component)  1.07e-01          11  \n",
       "44   (Dictybase, Phenotypes)  1.17e-01           3  \n",
       "45   (Dictybase, Phenotypes)  1.17e-01           2  \n",
       "46   (Dictybase, Phenotypes)  1.17e-01           3  \n",
       "47   (Dictybase, Phenotypes)  1.17e-01           8  \n",
       "48  (GO, cellular_component)  1.17e-01          15  \n",
       "49  (GO, biological_process)  1.17e-01          29  \n",
       "50   (Dictybase, Phenotypes)  1.40e-01           3  \n",
       "51  (GO, biological_process)  1.46e-01           8  \n",
       "52   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "53   (Dictybase, Phenotypes)  1.56e-01           2  \n",
       "54  (GO, biological_process)  1.56e-01           7  \n",
       "55   (Dictybase, Phenotypes)  1.68e-01           5  \n",
       "56   (Dictybase, Phenotypes)  1.68e-01           6  \n",
       "57   (Dictybase, Phenotypes)  1.68e-01           3  \n",
       "58   (Dictybase, Phenotypes)  1.68e-01           2  \n",
       "59   (Dictybase, Phenotypes)  1.68e-01           4  \n",
       "60   (Dictybase, Phenotypes)  1.68e-01           4  \n",
       "61   (Dictybase, Phenotypes)  1.68e-01           2  \n",
       "62   (Dictybase, Phenotypes)  1.68e-01           2  \n",
       "63   (Dictybase, Phenotypes)  1.68e-01           7  \n",
       "64   (Dictybase, Phenotypes)  1.68e-01           2  \n",
       "65   (Dictybase, Phenotypes)  1.68e-01           3  \n",
       "66  (GO, biological_process)  1.68e-01           8  \n",
       "67   (Dictybase, Phenotypes)  1.76e-01           3  \n",
       "68   (Dictybase, Phenotypes)  1.89e-01           6  \n",
       "69   (Dictybase, Phenotypes)  1.89e-01           2  \n",
       "70   (Dictybase, Phenotypes)  1.89e-01           2  \n",
       "71   (Dictybase, Phenotypes)  1.89e-01           4  \n",
       "72   (Dictybase, Phenotypes)  1.89e-01           2  \n",
       "73  (GO, molecular_function)  1.89e-01          21  \n",
       "74  (GO, molecular_function)  1.89e-01           9  \n",
       "75   (Dictybase, Phenotypes)  2.09e-01           7  \n",
       "76  (GO, cellular_component)  2.17e-01          14  \n",
       "77   (Dictybase, Phenotypes)  2.20e-01           2  \n",
       "78  (GO, biological_process)  2.20e-01           9  \n",
       "79  (GO, molecular_function)  2.20e-01          12  \n",
       "80  (GO, biological_process)  2.22e-01          10  \n",
       "81   (Dictybase, Phenotypes)  2.25e-01           4  \n",
       "82   (Dictybase, Phenotypes)  2.25e-01           2  \n",
       "83   (Dictybase, Phenotypes)  2.25e-01           2  \n",
       "84          (KEGG, Pathways)  2.25e-01           2  \n",
       "85  (GO, biological_process)  2.25e-01           8  \n",
       "86   (Dictybase, Phenotypes)  2.26e-01           7  \n",
       "87   (Dictybase, Phenotypes)  2.40e-01           2  \n",
       "88   (Dictybase, Phenotypes)  2.40e-01           2  \n",
       "89   (Dictybase, Phenotypes)  2.40e-01           2  \n",
       "90   (Dictybase, Phenotypes)  2.40e-01           2  \n",
       "91   (Dictybase, Phenotypes)  2.40e-01           2  \n",
       "92          (KEGG, Pathways)  2.40e-01           3  \n",
       "93   (Dictybase, Phenotypes)  2.45e-01           6  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tag selected: 273 with EID: 272\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.35\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 71 out of 95 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.55e-16</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.55e-16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.67e-14</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>8.21e-11</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.62e-10</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.75e-09</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.84e-08</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.93e-08</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.02e-07</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.38e-06</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>6.58e-06</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.04e-04</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.04e-04</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>5.91e-04</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.02e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.20e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.64e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>6.01e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>6.01e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.71e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.34e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>GTPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>2.16e-01</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Gene set                  Ontology       FDR  \\\n",
       "0                       chromosome  (GO, cellular_component)  1.55e-16   \n",
       "1                  DNA replication          (KEGG, Pathways)  1.55e-16   \n",
       "2            DNA metabolic process  (GO, biological_process)  3.67e-14   \n",
       "3                  Mismatch repair          (KEGG, Pathways)  8.21e-11   \n",
       "4          chromosome organization  (GO, biological_process)  1.62e-10   \n",
       "5                      DNA binding  (GO, molecular_function)  2.75e-09   \n",
       "6                       cell cycle  (GO, biological_process)  1.84e-08   \n",
       "7           chromosome segregation  (GO, biological_process)  3.93e-08   \n",
       "8             Base excision repair          (KEGG, Pathways)  1.02e-07   \n",
       "9       Nucleotide excision repair          (KEGG, Pathways)  1.38e-06   \n",
       "10        mitotic nuclear division  (GO, biological_process)  6.58e-06   \n",
       "11              nuclear chromosome  (GO, cellular_component)  3.04e-04   \n",
       "12                 ATPase activity  (GO, molecular_function)  3.04e-04   \n",
       "13                   cell division  (GO, biological_process)  5.91e-04   \n",
       "14              mitotic cell cycle  (GO, biological_process)  4.02e-03   \n",
       "15              response to stress  (GO, biological_process)  1.20e-02   \n",
       "16        Homologous recombination          (KEGG, Pathways)  1.64e-02   \n",
       "17      Non-homologous end-joining          (KEGG, Pathways)  6.01e-02   \n",
       "18                 histone binding  (GO, molecular_function)  6.01e-02   \n",
       "19  Ubiquitin mediated proteolysis          (KEGG, Pathways)  7.71e-02   \n",
       "20           Pyrimidine metabolism          (KEGG, Pathways)  1.34e-01   \n",
       "21                 GTPase activity  (GO, molecular_function)  2.16e-01   \n",
       "\n",
       "    N in query  \n",
       "0           21  \n",
       "1           16  \n",
       "2           22  \n",
       "3           10  \n",
       "4           16  \n",
       "5           24  \n",
       "6           20  \n",
       "7            8  \n",
       "8            9  \n",
       "9            9  \n",
       "10           6  \n",
       "11           5  \n",
       "12          14  \n",
       "13          11  \n",
       "14           9  \n",
       "15          18  \n",
       "16           4  \n",
       "17           2  \n",
       "18           2  \n",
       "19           5  \n",
       "20           3  \n",
       "21           7  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  tip selected: 486 with EID: 485\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.34\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 119 out of 166 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.11e-31</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.66e-27</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.12e-21</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.05e-20</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.26e-16</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.26e-16</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chromosome segregation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.96e-15</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>mitotic nuclear division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.27e-11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>3.79e-11</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Base excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.87e-10</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.64e-10</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.13e-09</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>mitotic cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.60e-08</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>nuclear chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.68e-06</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>microtubule organizing center</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.06e-05</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>ATPase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>5.06e-05</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Homologous recombination</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.34e-03</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>cytoskeleton</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>5.77e-03</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Ubiquitin mediated proteolysis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.41e-03</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>increased cell cycle length</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.33e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>aberrant centromere complex assembly</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.33e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>histone binding</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.64e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>3.12e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>aberrant mitosis</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.26e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>aberrant centrosome localization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>3.26e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>3.98e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>response to stress</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>4.32e-02</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>aberrant spindle organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.66e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>aberrant centrosome organization</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.03e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.03e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>aberrant cell cycle</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.03e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>aberrant nucleus morphology</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>8.03e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>helicase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>9.20e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>nuclear envelope</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>Non-homologous end-joining</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.46e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>increased sensitivity to DNA-damaging agents</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>1.55e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Gene set                  Ontology  \\\n",
       "0                                     chromosome  (GO, cellular_component)   \n",
       "1                                DNA replication          (KEGG, Pathways)   \n",
       "2                        chromosome organization  (GO, biological_process)   \n",
       "3                                     cell cycle  (GO, biological_process)   \n",
       "4                                Mismatch repair          (KEGG, Pathways)   \n",
       "5                          DNA metabolic process  (GO, biological_process)   \n",
       "6                         chromosome segregation  (GO, biological_process)   \n",
       "7                       mitotic nuclear division  (GO, biological_process)   \n",
       "8                                    DNA binding  (GO, molecular_function)   \n",
       "9                           Base excision repair          (KEGG, Pathways)   \n",
       "10                                 cell division  (GO, biological_process)   \n",
       "11                    Nucleotide excision repair          (KEGG, Pathways)   \n",
       "12                            mitotic cell cycle  (GO, biological_process)   \n",
       "13                            nuclear chromosome  (GO, cellular_component)   \n",
       "14                 microtubule organizing center  (GO, cellular_component)   \n",
       "15                               ATPase activity  (GO, molecular_function)   \n",
       "16                      Homologous recombination          (KEGG, Pathways)   \n",
       "17                                  cytoskeleton  (GO, cellular_component)   \n",
       "18                Ubiquitin mediated proteolysis          (KEGG, Pathways)   \n",
       "19                   increased cell cycle length   (Dictybase, Phenotypes)   \n",
       "20          aberrant centromere complex assembly   (Dictybase, Phenotypes)   \n",
       "21                               histone binding  (GO, molecular_function)   \n",
       "22                                   nucleoplasm  (GO, cellular_component)   \n",
       "23                              aberrant mitosis   (Dictybase, Phenotypes)   \n",
       "24              aberrant centrosome localization   (Dictybase, Phenotypes)   \n",
       "25                         Pyrimidine metabolism          (KEGG, Pathways)   \n",
       "26                            response to stress  (GO, biological_process)   \n",
       "27                 aberrant spindle organization   (Dictybase, Phenotypes)   \n",
       "28              aberrant centrosome organization   (Dictybase, Phenotypes)   \n",
       "29                             cell cycle arrest   (Dictybase, Phenotypes)   \n",
       "30                           aberrant cell cycle   (Dictybase, Phenotypes)   \n",
       "31                   aberrant nucleus morphology   (Dictybase, Phenotypes)   \n",
       "32                             helicase activity  (GO, molecular_function)   \n",
       "33                              nuclear envelope  (GO, cellular_component)   \n",
       "34                    Non-homologous end-joining          (KEGG, Pathways)   \n",
       "35  increased sensitivity to DNA-damaging agents   (Dictybase, Phenotypes)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   1.11e-31          37  \n",
       "1   5.66e-27          25  \n",
       "2   2.12e-21          30  \n",
       "3   3.05e-20          41  \n",
       "4   1.26e-16          15  \n",
       "5   1.26e-16          30  \n",
       "6   1.96e-15          14  \n",
       "7   1.27e-11          11  \n",
       "8   3.79e-11          35  \n",
       "9   2.87e-10          13  \n",
       "10  3.64e-10          24  \n",
       "11  1.13e-09          14  \n",
       "12  3.60e-08          20  \n",
       "13  1.68e-06           8  \n",
       "14  5.06e-05          11  \n",
       "15  5.06e-05          21  \n",
       "16  4.34e-03           6  \n",
       "17  5.77e-03          18  \n",
       "18  7.41e-03           9  \n",
       "19  1.33e-02           2  \n",
       "20  1.33e-02           2  \n",
       "21  1.64e-02           3  \n",
       "22  3.12e-02          10  \n",
       "23  3.26e-02           2  \n",
       "24  3.26e-02           2  \n",
       "25  3.98e-02           5  \n",
       "26  4.32e-02          25  \n",
       "27  5.66e-02           2  \n",
       "28  8.03e-02           2  \n",
       "29  8.03e-02           2  \n",
       "30  8.03e-02           2  \n",
       "31  8.03e-02           2  \n",
       "32  9.20e-02           6  \n",
       "33  1.46e-01           4  \n",
       "34  1.46e-01           2  \n",
       "35  1.55e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  slug selected: 117 with EID: 116\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.22\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 13 out of 25 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chromosome organization</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.73e-03</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nucleoplasm</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.11e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chromosome</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>RNA polymerase</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell cycle</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DNA metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cell death</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>enzyme regulator activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.15e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>cell division</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.23e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>transferase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.39e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene set                  Ontology       FDR  N in query\n",
       "0    chromosome organization  (GO, biological_process)  7.73e-03           5\n",
       "1                nucleoplasm  (GO, cellular_component)  1.11e-01           4\n",
       "2                 chromosome  (GO, cellular_component)  1.15e-01           3\n",
       "3             RNA polymerase          (KEGG, Pathways)  1.15e-01           2\n",
       "4                 cell cycle  (GO, biological_process)  1.15e-01           4\n",
       "5      DNA metabolic process  (GO, biological_process)  1.15e-01           3\n",
       "6                 cell death  (GO, biological_process)  1.15e-01           2\n",
       "7  enzyme regulator activity  (GO, molecular_function)  1.15e-01           3\n",
       "8              cell division  (GO, biological_process)  1.23e-01           3\n",
       "9       transferase activity  (GO, molecular_function)  1.39e-01           2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  mhat selected: 28 with EID: 28\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.5\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 9 out of 14 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.86e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Amino sugar and nucleotide sugar metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>7.86e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>carbohydrate metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.86e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>generation of precursor metabolites and energy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>7.86e-02</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>catabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>8.11e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>small molecule metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.26e-01</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "      <td>(GO, molecular_function)</td>\n",
       "      <td>1.30e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.68e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Biosynthesis of secondary metabolites</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>1.93e-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>cellular component assembly</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.35e-01</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Gene set                  Ontology  \\\n",
       "0                    Starch and sucrose metabolism          (KEGG, Pathways)   \n",
       "1      Amino sugar and nucleotide sugar metabolism          (KEGG, Pathways)   \n",
       "2                   carbohydrate metabolic process  (GO, biological_process)   \n",
       "3   generation of precursor metabolites and energy  (GO, biological_process)   \n",
       "4                                catabolic process  (GO, biological_process)   \n",
       "5                 small molecule metabolic process  (GO, biological_process)   \n",
       "6                          oxidoreductase activity  (GO, molecular_function)   \n",
       "7                                Carbon metabolism          (KEGG, Pathways)   \n",
       "8                               Metabolic pathways          (KEGG, Pathways)   \n",
       "9            Biosynthesis of secondary metabolites          (KEGG, Pathways)   \n",
       "10                     cellular component assembly  (GO, biological_process)   \n",
       "\n",
       "         FDR  N in query  \n",
       "0   7.86e-02           2  \n",
       "1   7.86e-02           2  \n",
       "2   7.86e-02           3  \n",
       "3   7.86e-02           3  \n",
       "4   8.11e-02           4  \n",
       "5   1.26e-01           4  \n",
       "6   1.30e-01           3  \n",
       "7   1.68e-01           2  \n",
       "8   1.93e-01           5  \n",
       "9   1.93e-01           3  \n",
       "10  2.35e-01           2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  cul selected: 1412 with EID: 1403\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.31\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 76 out of 431 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>4.17e-03</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>secondary metabolic process</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.57e-02</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>autophagy</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>9.57e-02</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Endocytosis</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>9.93e-02</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>1.49e-01</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Autophagy - other</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>2.12e-01</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>decreased spore viability</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>2.31e-01</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Gene set                  Ontology  \\\n",
       "0  SNARE interactions in vesicular transport          (KEGG, Pathways)   \n",
       "1                secondary metabolic process  (GO, biological_process)   \n",
       "2                                  autophagy  (GO, biological_process)   \n",
       "3                                Endocytosis          (KEGG, Pathways)   \n",
       "4                       cell differentiation  (GO, biological_process)   \n",
       "5                          Autophagy - other          (KEGG, Pathways)   \n",
       "6                  decreased spore viability   (Dictybase, Phenotypes)   \n",
       "\n",
       "        FDR  N in query  \n",
       "0  4.17e-03          11  \n",
       "1  9.57e-02           4  \n",
       "2  9.57e-02          11  \n",
       "3  9.93e-02          18  \n",
       "4  1.49e-01          28  \n",
       "5  2.12e-01           8  \n",
       "6  2.31e-01          14  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "***  FB selected: 564 with EID: 560\n",
      "Ratio of genes annotated with a gene set in reference 0.32 and query 0.25\n",
      "Enrichment at FDR: 0.25 and min query - gene set overlap 2\n",
      "N query genes in displayed gene sets: 35 out of 141 query genes used for enrichment calculation.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene set</th>\n",
       "      <th>Ontology</th>\n",
       "      <th>FDR</th>\n",
       "      <th>N in query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>extracellular region</td>\n",
       "      <td>(GO, cellular_component)</td>\n",
       "      <td>2.87e-02</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>anatomical structure formation involved in morphogenesis</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>3.00e-02</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Starch and sucrose metabolism</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.23e-02</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>decreased pseudopod retraction</td>\n",
       "      <td>(Dictybase, Phenotypes)</td>\n",
       "      <td>5.99e-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>SNARE interactions in vesicular transport</td>\n",
       "      <td>(KEGG, Pathways)</td>\n",
       "      <td>5.99e-02</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>cell differentiation</td>\n",
       "      <td>(GO, biological_process)</td>\n",
       "      <td>2.06e-01</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Gene set  \\\n",
       "0                                      extracellular region   \n",
       "1  anatomical structure formation involved in morphogenesis   \n",
       "2                             Starch and sucrose metabolism   \n",
       "3                            decreased pseudopod retraction   \n",
       "4                 SNARE interactions in vesicular transport   \n",
       "5                                      cell differentiation   \n",
       "\n",
       "                   Ontology       FDR  N in query  \n",
       "0  (GO, cellular_component)  2.87e-02          16  \n",
       "1  (GO, biological_process)  3.00e-02          10  \n",
       "2          (KEGG, Pathways)  5.23e-02           6  \n",
       "3   (Dictybase, Phenotypes)  5.99e-02           2  \n",
       "4          (KEGG, Pathways)  5.99e-02           5  \n",
       "5  (GO, biological_process)  2.06e-01          12  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "milestones=pd.read_table(path_impulse+'DEacrossStages_summary_WT_noyem_batchrep.tsv',index_col=0,sep='\\t')\n",
    "stages=np.array([phenotype for phenotype in PHENOTYPES if phenotype in list(milestones.columns)])\n",
    "for stage_idx,stage in enumerate(stages):\n",
    "    \n",
    "    #Nonneighbouring stages\n",
    "    neigh_previous=1\n",
    "    neigh_next=1\n",
    "    #Assumes phenotypes order\n",
    "    if stage_idx==1:\n",
    "        neigh_previous=0\n",
    "    if stage_idx==0:\n",
    "        neigh_next=0\n",
    "    non_neighbours = stages[[(idx <stage_idx-neigh_previous or idx>stage_idx+neigh_next) \n",
    "                 for idx in range(len(stages)) ]]\n",
    "    #print(stage,':',non_neighbours)\n",
    "    \n",
    "    #Find genes up at stage and down at nonenighbouring\n",
    "    query_condition=stage+'==\"up\"'\n",
    "    for non_neigh in non_neighbours:\n",
    "        query_condition=query_condition+' & '+non_neigh+' ==\"down\"'\n",
    "    #print(query_condition)\n",
    "    \n",
    "    query_names=list(milestones.query(query_condition).index)\n",
    "    enriched=group_diff_enrichment(query_names=query_names,group=stage,min_overlap=2,use_annotated_genes=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
